---
title: "BMDM_bio_figure_generation_script"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r package loading}
library(dplyr)
library(ggplot2)
library(reshape2)
library(stringr)
library(patchwork)
```

```{r function}
PCA_PSEA <- function(PCA_weights, GeneProteinMappings, GeneSetDB){
GeneSetDB <- GeneSetDB %>% filter(!is.na(GO_term_name))
GO_terms <- unique(GeneSetDB$GO_term_name)

WRS_test_out <- data.frame(Process = character(),numDBentries = numeric(),numberMatchingEntries = numeric(),fracDB = numeric(),pVal = numeric(),EffectSize = numeric(),PC = numeric(),Association = numeric(), stringsAsFactors = FALSE)
WRS_test_out_full <- data.frame(Process = character(),numDBentries = numeric(),numberMatchingEntries = character(),fracDB = numeric(),pVal = numeric(),EffectSize = numeric(),PC = numeric(),Association = numeric(), stringsAsFactors = FALSE)

ReqNumEntry <- 5
fracDB <- .1
maxNum <- 200

GO_in <- PCA_weights %>% left_join(GeneProteinMappings, by = c("Proteins" = "UNIPROTKB"))

for(i in 1:(ncol(GO_in)-2)){
  GeneInd <- which(colnames(GO_in) == "GENES")
  GO_in_lim <- GO_in[,c(i,GeneInd)]
  
for(j in 1:(length(GO_terms))){

    GO_term <- GeneSetDB %>% dplyr::filter(GO_term_name == GO_terms[[j]])
    GO_term <- GO_term %>% group_by(Gene) %>% slice(1)
    
    GO_in_lim$Lab <- ifelse(GO_in_lim$GENES %in% GO_term$Gene,"Target","Background")
    colnames(GO_in_lim) <- c("Loadings","GENES","Lab")
    WRS_test_out[j,1] <- GO_terms[[j]]
    WRS_test_out[j,2] <- length(unique(GO_term$Gene))
    WRS_test_out[j,3] <- sum(GO_in_lim$Lab == "Target")
    WRS_test_out[j,4] <- WRS_test_out[j,3]/WRS_test_out[j,2]
    WRS_test_out[j,7] <- i
 
    
    if((sum(GO_in_lim$Lab == "Target") >= ReqNumEntry)&(WRS_test_out[j,4] >= fracDB)&(WRS_test_out[j,2] <= maxNum)){
      GO_in_lim2 <- GO_in_lim
      GO_in_lim2$zScore <- (GO_in_lim2$Loadings-mean(GO_in_lim2$Loadings, na.rm=TRUE))/sd(GO_in_lim2$Loadings, na.rm=TRUE)
      
      TargGroup <- GO_in_lim2 %>% dplyr::filter(Lab == "Target")
      BackgroundGroup <- GO_in_lim2 %>% dplyr::filter(Lab != "Target")
      
      WRSTest <- wilcox.test(TargGroup$Loadings, BackgroundGroup$Loadings, alternative = "two.sided", paired = FALSE, exact = FALSE)
      WRS_test_out[j,5] <- WRSTest$p.value
      WRS_test_out[j,6] <- median(GO_in_lim2[GO_in_lim2$Lab == "Target",4], na.rm = TRUE)/median(GO_in_lim2[GO_in_lim2$Lab != "Target",4], na.rm = TRUE)
      WRS_test_out[j,8] <- median(GO_in_lim2[GO_in_lim2$Lab == "Target",4], na.rm = TRUE)

    }
    if((sum(GO_in_lim$Lab == "Target") < ReqNumEntry)|(WRS_test_out[j,4] < fracDB)|(WRS_test_out[j,2] > maxNum)){
      WRS_test_out[j,5] <- NA
      WRS_test_out[j,6] <- NA
      WRS_test_out[j,8] <- NA
    }
    }
  WRS_test_out_full <- rbind(WRS_test_out_full,WRS_test_out)
}
return(WRS_test_out_full)
}

# Column and row normalization
cr_norm_log<-function(dat){
  
  for(k in 1:ncol(dat)){
    
    dat[,k]<-dat[,k]-median(dat[,k], na.rm = T)
    
    
  }
  
  
  for(k in 1:nrow(dat)){
    
    dat[k,]<-dat[k,]-mean(dat[k,], na.rm = T)
    
  }
  
  return(dat)
}

flattenCorrMatrix <- function(cormat, pmat) {
  ut <- upper.tri(cormat)
  data.frame(
    row = rownames(cormat)[row(cormat)[ut]],
    column = rownames(cormat)[col(cormat)[ut]],
    cor  =(cormat)[ut],
    p = pmat[ut]
    )
}

# Get z-scored median abundance for protein set
GO_for_PCA <- function(GOvec,Unimp_in){
  
  for (i in 1:length(GOvec)){
  GO_db_lim <- GO_db %>% filter(GO_term_name %in% GOvec[i])
  unimp_lim <- Unimp_in %>% filter(GENES %in% GO_db_lim$Gene)
  unimp_lim_sum <- unimp_lim %>% dplyr::group_by(Var2) %>% dplyr::summarize(medAbund = median(newVar, na.rm = TRUE)) 
  unimp_lim_sum$medAbund <- (unimp_lim_sum$medAbund-mean(unimp_lim_sum$medAbund, na.rm=TRUE))/sd(unimp_lim_sum$medAbund, na.rm=TRUE)
  unimp_lim_sum$medAbund <- ifelse(unimp_lim_sum$medAbund >= 2,2,ifelse(unimp_lim_sum$medAbund <= -2,-2,unimp_lim_sum$medAbund))
  colnames(unimp_lim_sum) <- c("id",as.character(GOvec[[i]]))
  if(i == 1){
    GSEA_PC_out <- unimp_lim_sum
  }
  if(i > 1){
    GSEA_PC_out <- GSEA_PC_out %>% left_join(unimp_lim_sum, by = "id")
  }
  }
  return(GSEA_PC_out)
}

# Permutation function for MEROPS analysis
CorPermutation <- function(df,CodingDoc,dbString){
    df_lim <- df[which(colnames(df) %in% c(MEROPSvec,CodingDoc$Entry))]
    corObs <- as.data.frame(cor(df_lim, method = "pearson", use="pairwise.complete.obs"))
    corObs$MEROPS <- row.names(corObs)
    corObs.m <- melt(corObs)
    corObs.m.f <- corObs.m %>% dplyr::filter((MEROPS %in% MEROPSvec)&(!(variable %in% MEROPSvec)))
    corObs.m.f <- corObs.m.f %>% dplyr::left_join(CodingDoc, by = c("variable" = "Entry"))
    corObs.m.f$Coding <- ifelse(corObs.m.f$Type == "Inflammatory","Pro","Anti")
    corObs_sum <- corObs.m.f %>% dplyr::group_by(MEROPS) %>% dplyr::summarize(corPro = median(value[Coding == "Pro"],na.rm = TRUE),corAnti = median(value[Coding == "Anti"],na.rm = TRUE),CorDist = abs(corPro - corAnti))
    
    corObs_sum_lim <- corObs_sum %>% dplyr::select(MEROPS,CorDist)
    
    colInds_MEROPS <- which((colnames(df_lim))%in% MEROPSvec) 
    
    for(i in 1:10000){
      sampNames <- sample(which(!(colnames(df_lim))%in% MEROPSvec),length(which(!(colnames(df_lim)%in% MEROPSvec))),replace = FALSE)
      df_rearranged <- df_lim
      colnames(df_rearranged) <- c(colnames(df_lim)[sampNames],colnames(df_lim)[colInds_MEROPS])
      corSim <- as.data.frame(cor(df_rearranged, method = "pearson", use="pairwise.complete.obs"))
      corSim$MEROPS <- row.names(corSim)
      corSim.m <- melt(corSim)
      corSim.m.f <- corSim.m %>% dplyr::filter((MEROPS %in% MEROPSvec)&(!(variable %in% MEROPSvec)))
      corSim.m.f <- corSim.m.f %>% dplyr::left_join(CodingDoc, by = c("variable" = "Entry"))
      corSim.m.f$Coding <- ifelse(corSim.m.f$Type == "Inflammatory","Pro","Anti")
      corSim_sum <- corSim.m.f %>% dplyr::group_by(MEROPS) %>% dplyr::summarize(corPro = median(value[Coding == "Pro"],na.rm = TRUE),corAnti = median(value[Coding == "Anti"],na.rm = TRUE),CorDist = abs(corPro - corAnti))
      if(i ==1){
        CorSim_sumOut <- corSim_sum
      }
      if(i > 1){
        CorSim_sumOut <- rbind(CorSim_sumOut,corSim_sum)
      }
    }
    
    CorSim_sumOut_Comp <- CorSim_sumOut %>% dplyr::left_join(corObs_sum_lim, by = "MEROPS")
    CorSim_sumOut_Comp$DistBool <- CorSim_sumOut_Comp$CorDist.x > CorSim_sumOut_Comp$CorDist.y 
  
    CS_permTest <- CorSim_sumOut_Comp %>% dplyr::group_by(MEROPS) %>% dplyr::summarize(fraction = sum(DistBool,na.rm = TRUE)/n())
    CS_permTest$qVal <- p.adjust(CS_permTest$fraction, method = "fdr")
    CS_permTest$qVal2 <- ifelse(CS_permTest$qVal == 0,10^-5,CS_permTest$qVal)
    CS_permTest$lab <- dbString
    CS_permTest2 <- CS_permTest %>% left_join(corObs_sum, by = "MEROPS")
    return(CS_permTest2)
    }

```

```{r data import and export paths}

#orig datPath <- "C:/My_Drive/MS/SCoPE/t-SCoPE/code/pSCoPE_pubFigs/dat/"
datPath <- "C:/My_Drive/MS/SCoPE/t-SCoPE/Pub/Data_Share/FigureGen/biofigs_BMDM/"

#orig prePCA <- read.csv(paste0(datPath,"limmaCorrected_normed_prePCA_Priori40_DART.csv"),row.names = "X")
prePCA <- read.csv(paste0(datPath,"limmaCorrected_normed_prePCA_Priori_mrri02_PIF50_DART_1pFDR.csv"),row.names = "X")

#orig CG <- read.delim(paste0(datPath, "ChanKey_BMDM_Priori.txt"))
#orig GeneMappings_single <- read.csv("C:/My_Drive/MS/SCoPE/t-SCoPE/code/pSCoPE_pubFigs/BatchCorrection_examination/GeneMappings.csv")

CG <- read.delim(paste0(datPath, "ChanKey_BMDM_Priori_mrri02_PIF50_DART_FDR1p.txt"))
GeneMappings_single <- read.csv(paste0(datPath,"GeneMappings.csv"))

#orig ProtMissing <- read.delim(paste0(datPath,"Prot_missing_BMDM.txt"))
ProtMissing <- read.delim(paste0(datPath,"Prot_missing_BMDM_priori_mrri02_PIF50_DART_1pFDR.txt"))

GO_DB_Path <- paste0(datPath,"GOA_db.txt")
GO_db <- read.delim(GO_DB_Path, sep = "", header = TRUE)
GO_db.2 <- GO_db %>% filter(Letter != "C")

# Dextran-Uptake Data
Dextran_DIA_LPS <- read.delim(paste0(datPath,"DexLPS_LowOverHigh__eGH_358_eGH_360_Data_for_volcanoplot.txt"))
Dextran_DIA_NT <- read.delim(paste0(datPath,"DexNT_LowOverHigh__eGH_362_eGH_364_Data_for_volcanoplot.txt"))

# Marker Databases for MEROPS
MarkerDB_SCoPE2 <- read.delim(paste0(datPath,"SCoPE2_markers.txt"))
SCoPE2_Human2Mouse <- read.delim(paste0(datPath,"SCoPE2_Markers_Dict.tab"))
BulkDDARatioSet <- read.delim(paste0(datPath,"BulkRatioEvidence.txt"))

# pre-digestion labeled bulk sample for MEROPS analysis
DIA_Bulk <- read.delim(paste0(datPath,"GHRT_eGH692_Report.xls"))


#**********
# Enter your preferred figure and data output paths
#**********
figOut <- "~/TEST_SCRIPT_OUT/"
datOut <- "~/TEST_SCRIPT_OUT/"
#**********

```

```{r PCA coordinate and protein weight extraction}
# Xcondition comparison
PCAout_both <- prcomp(t(prePCA))
PCAout_both_coords <- PCAout_both$x
PCAout_both_weights <- as.data.frame(PCAout_both$rotation[,1:5])
PCAout_both_weights$Proteins <- row.names(PCAout_both_weights)

# LPS 
CG_LPS <- CG %>% filter(celltype == "LPS", cvm < .4)
prePCA_LPS <- prePCA %>% select(CG_LPS$id) 
prePCA_LPS_norm <- cr_norm_log(as.matrix(prePCA_LPS))
PCAout_LPS <- prcomp(t(prePCA_LPS_norm))
PCAout_LPS_coords <- PCAout_LPS$x
PCAout_LPS_weights <- as.data.frame(PCAout_LPS$rotation[,1:5])
PCAout_LPS_weights$Proteins <- row.names(PCAout_LPS_weights)

# Untreated
CG_NT <- CG %>% filter(celltype != "LPS", cvm < .4)
prePCA_NT <- prePCA %>% select(CG_NT$id) 
prePCA_NT_norm <- cr_norm_log(as.matrix(prePCA_NT))
PCAout_NT <- prcomp(t(prePCA_NT_norm))
PCAout_NT_coords <- PCAout_NT$x
PCAout_NT_weights <- as.data.frame(PCAout_NT$rotation[,1:5])
PCAout_NT_weights$Proteins <- row.names(PCAout_NT_weights)
```

```{r Gene Set prep}
GeneMappings_single$GENES <- gsub(" .*","",GeneMappings_single$GENES)
GeneMappings_single$GENES <- toupper(GeneMappings_single$GENES)
GeneMappings_single2 <- GeneMappings_single %>% select(-X)
```

```{r x_cond PSEA}
#************
# Uncomment if you would like to re-run the PC-weight-based PSEA instead of loading saved versions of the report
#************

# Run PC-weight-based PSEA
#xcond_PSEA <- PCA_PSEA(PCAout_both_weights,GeneMappings_single2,GO_db.2)
#LPS_PSEA <- PCA_PSEA(PCAout_LPS_weights,GeneMappings_single2,GO_db.2)
#NT_PSEA <- PCA_PSEA(PCAout_NT_weights,GeneMappings_single2,GO_db.2)
#************


#Uncomment to save the results of the re-run analyses
#write.table(xcond_PSEA,paste0(datOut,"xCond2.txt"))
#write.table(LPS_PSEA,paste0(datOut,"LPS_only_PCAPSEA.txt"))
#write.table(NT_PSEA,paste0(datOut,"NT_only_PCAPSEA.txt"))

# Load PC-weight-based PSEA results
xcond_PSEA <- read.table(paste0(datPath,"xCond2.txt"))
LPS_PSEA <- read.table(paste0(datPath,"LPS_only_PCAPSEA.txt"))
NT_PSEA <- read.table(paste0(datPath,"NT_only_PCAPSEA.txt"))
```


```{r PC-weight-based PCA result filtration}
# Removing redundant Protein Sets

# xCond
XC_excludeList <- c("viral infectious cycle","viral transcription","melanosome","proton-transporting ATPase activity, rotational mechanism","Proton transport","ribosomal small subunit biogenesis","ATP hydrolysis coupled proton transport","cellular iron ion homeostasis")
Cc_excludeList <- c("viral infectious cycle","viral transcription","melanosome","proton-transporting ATPase activity, rotational mechanism","translation initiation factor activity")


# q value correction and FDR filtration of PSEA results
xcond.f <- xcond_PSEA %>% filter(!is.na(pVal))
xcond.f$qVal <- p.adjust(xcond.f$pVal, method = "fdr")
xcond.f_fdr <- xcond.f %>% filter((qVal <= .05)& !(Process %in% XC_excludeList))

# LPS
# Note: Association multiplied by negative 1, since eigenvector sign is arbitrary 
LPS_PSEA.f <- LPS_PSEA %>% filter(!is.na(pVal))
LPS_PSEA.f$qVal <- p.adjust(LPS_PSEA.f$pVal, method = "fdr")
LPS_PSEA.f_fdr <- LPS_PSEA.f %>% filter(qVal <= .05 & !(Process %in% Cc_excludeList)) %>% mutate(Condition = "LPS")
LPS_PSEA.f_fdr2 <- LPS_PSEA.f_fdr
LPS_PSEA.f_fdr2$Association <- ifelse(LPS_PSEA.f_fdr2$PC == 1,LPS_PSEA.f_fdr2$Association*-1,ifelse(LPS_PSEA.f_fdr2$PC == 2,LPS_PSEA.f_fdr2$Association*-1,LPS_PSEA.f_fdr2$Association))

# NT
NT_PSEA.f <- NT_PSEA %>% filter(!is.na(pVal))
NT_PSEA.f$qVal <- p.adjust(NT_PSEA.f$pVal, method = "fdr")
NT_PSEA.f_fdr <- NT_PSEA.f %>% filter(qVal <= .05 & !(Process %in% Cc_excludeList)) %>% mutate(Condition = "Untreated")
```

```{r Fig2A PCA accompaniment data wrangling}
# Batch corrected, imputed cells x proteins matrix
BC_imp_PxC2 <- prePCA
BC_imp_PxC2$Proteins <- row.names(BC_imp_PxC2)
BC_imp_PxC2.m <- melt(BC_imp_PxC2, id.vars = "Proteins")

# Removing imputed protein measurements, by cross-referencing pre-imputation cells x proteins matrix
ProtMissing.m <- melt(ProtMissing)
BC_imp_PxC3.m <- BC_imp_PxC2.m %>% left_join(ProtMissing.m, by = c("Proteins", "variable"))
BC_imp_PxC3.m$newVar <- ifelse(is.na(BC_imp_PxC3.m$value.y),NA,BC_imp_PxC3.m$value.x)
BC_unImp_PxC3.m2 <- BC_imp_PxC3.m %>% select(Proteins,variable,newVar)
BC_unImp_PxC3 <- dcast(BC_unImp_PxC3.m2, Proteins~variable)
row.names(BC_unImp_PxC3) <- BC_unImp_PxC3$Proteins
BC_unImp_PxC4 <- BC_unImp_PxC3 %>% select(-Proteins)

BC_unImp_CxP <- as.data.frame(t(BC_unImp_PxC4))
BC_unImp_CxP$id <- row.names(BC_unImp_CxP)

BC_unImp_PxC3.m3 <- BC_unImp_PxC3.m2 %>% left_join(GeneMappings_single2, by = c("Proteins"="UNIPROTKB"))
```

```{r Fig2A color-coded PCA input}
GO_Set_for_xCond <- c("type I interferon-mediated signaling pathway","phagosome maturation")

for (i in 1:length(GO_Set_for_xCond)){
  GO_db_lim <- GO_db %>% filter(GO_term_name %in% GO_Set_for_xCond[i])
  BC_unImp_PxC3.m3_lim <- BC_unImp_PxC3.m3 %>% filter(GENES %in% GO_db_lim$Gene)
  BC_unImp_PxC3.m3_lim_sum <- BC_unImp_PxC3.m3_lim %>% dplyr::group_by(variable) %>% dplyr::summarize(medAbund = median(newVar, na.rm = TRUE)) 
  colnames(BC_unImp_PxC3.m3_lim_sum) <- c("id",as.character(GO_Set_for_xCond[[i]]))
  if(i == 1){
    GSEA_PC_out <- BC_unImp_PxC3.m3_lim_sum
  }
  if(i > 1){
    GSEA_PC_out <- GSEA_PC_out %>% left_join(BC_unImp_PxC3.m3_lim_sum, by = "id")
  }
}

PCAout_both_coords2 <- as.data.frame(PCAout_both_coords[,1:2])
PCAout_both_coords2$id <- row.names(PCAout_both_coords2)
PCAout_both_coords2 <- PCAout_both_coords2 %>% left_join(GSEA_PC_out, by = "id")
PCAout_both_coords2 <- PCAout_both_coords2 %>% left_join(CG, by = "id")
```

```{r}
# treatment labeled PCA without accompanying bulk measurements (unused in publication)
celltype_PCA <- PCAout_both_coords2 %>% ggplot(aes(x = PC1, y = PC2, color = celltype)) + geom_point(size = 3, alpha = 0.7, shape = 16) + theme_light() + theme(axis.title = element_text(size = 24), axis.text = element_text(size = 22), legend.position = "null") + ggtitle("") + annotate("text", x = 5.1, y = 6,size = 7, label = "LPS-treated", color = "#F8766D") + annotate("text", x = -5, y = 6,size = 7, label = "Untreated", color = "#00BFC4")


PCAout_both_coords2$PhagosomeBin2 <- ((PCAout_both_coords2$`phagosome maturation`)-mean(PCAout_both_coords2$`phagosome maturation`,na.rm = TRUE))/sd(PCAout_both_coords2$`phagosome maturation`, na.rm = TRUE)

PCAout_both_coords2$PhagosomeBin3 <- ifelse(PCAout_both_coords2$PhagosomeBin2 >= 2,2,ifelse(PCAout_both_coords2$PhagosomeBin2 <= -2,-2,PCAout_both_coords2$PhagosomeBin2))

PCAout_both_coords2$IFBin2 <- ((PCAout_both_coords2$`type I interferon-mediated signaling pathway`)-mean(PCAout_both_coords2$`type I interferon-mediated signaling pathway`,na.rm = TRUE))/sd(PCAout_both_coords2$`type I interferon-mediated signaling pathway`, na.rm = TRUE)

PCAout_both_coords2$IFBin3 <- ifelse(PCAout_both_coords2$IFBin2 >= 2,2,ifelse(PCAout_both_coords2$IFBin2 <= -2,-2,PCAout_both_coords2$IFBin2))

# IFN signaling PCA (Fig 2A)
IFN_PCA <- PCAout_both_coords2 %>% ggplot(aes(x = PC1, y = PC2, color = IFBin3)) + geom_point(size = 2, alpha = 0.7, shape = 16) + theme_light() + theme(axis.title = element_text(size = 16), axis.text = element_blank(), title = element_text(size = 16),legend.title = element_text(size = 22), legend.text = element_text(size = 22), plot.margin = unit(c(0,0.3,0,0),"cm")) + scale_color_gradient2(limits = c(-2,2),low = "blue", mid = "white", high = "red", midpoint =0, breaks = c(-2,0,2), labels = c("-2","0","2")) + ggtitle("Type I IFN\nsignaling") + labs(x ="", y = "", color = "Abundance\nz-score")

# Phagosome maturation PCA (Fig 2A)
PM_PCA <- PCAout_both_coords2 %>% ggplot(aes(x = PC1, y = PC2, color = PhagosomeBin3)) + geom_point(size = 2, alpha = 0.7, shape = 16) + theme_light() + theme(axis.title = element_text(size = 16), axis.text = element_blank(), title = element_text(size = 16),legend.title = element_text(size = 22), legend.text = element_text(size = 22), plot.margin = unit(c(0,0,0,0.3),"cm")) + scale_color_gradient2(limits = c(-2,2),low = "blue", mid = "white", high = "red", midpoint =0, breaks = c(-2,0,2), labels = c("-2","0","2")) + ggtitle("Phagosome\nmaturation") + labs(x ="", y = "", color = "Abundance\nz-score") 

AbundancePCAs <- (IFN_PCA + PM_PCA) + plot_layout(guides = "collect") & theme(legend.position = "bottom")

AbundancePCAs & theme(plot.margin =  margin(0, 0, 0, 0, "cm"))
ggsave(paste0(figOut,"xCond_PSEA_PCs_horiz_v3.pdf"), width = 6, height = 4.5, units = "in")
```

```{r Fig 2A: treatment-condition PCA with bulk}

# PCA of original BMDM single-cell data
PCAout_both <- prcomp(t(prePCA))
PCAout_both_coords_for_bulkPCA <- as.data.frame(PCAout_both$x[,1:2])
PCAout_both_coords_for_bulkPCA$id <- row.names(PCAout_both_coords_for_bulkPCA)
PCAout_both_coords_for_bulkPCA <- PCAout_both_coords_for_bulkPCA %>% left_join(CG, by = "id")

# Projection of bulk samples into single-cell low-dimensional space
# Note: bulk samples had previously been incorporated into the scp pipeline for normalization,
#       imputation, and batch correction with the original single-cell BMDM measurements
bulkBC <- read.delim(paste0(datPath,"imputed_BC_BMDM_priori_mrri02_PIF50_DART1pFDR_withBulk.txt"))
bulkBC_imp.mat <- bulkBC[,which(colnames(bulkBC) %in% c("i700","i701"))]
bulkProjection <- predict((PCAout_both),t(bulkBC_imp.mat))
bulkProjection_coord <- as.data.frame(bulkProjection[,1:2])
bulkProjection_coord$id <- row.names(bulkProjection_coord)
bulkProjection_coord$Raw.file <- c("XwGH215","XwGH215")  
bulkProjection_coord$celltype <- c("LPS","untreated")  
bulkProjection_coord$cvm <- c(0.2,0.2) 
bulkProjection_coord$RI <- c("Reporter.intensity.corrected.3","Reporter.intensity.corrected.4")
sc_and_bulkprojection_comb <- rbind(PCAout_both_coords_for_bulkPCA,bulkProjection_coord)
sc_and_bulkprojection_comb$Source <- ifelse(sc_and_bulkprojection_comb$Raw.file %in% c("XwGH215","XwGH215","XwGH216","XwGH216"),"Bulk","Single cell")


sc_and_bulkprojection_comb %>% ggplot(aes(x = PC1, y = PC2, color = celltype, shape = Source, size = Source,alpha = Source))  + geom_point(stroke = 2) + scale_size_discrete(range = c(15,3)) + scale_shape_manual(values = c(23,16)) +   
  geom_point(data = subset(sc_and_bulkprojection_comb, Source == 'Bulk'),aes(x = PC1, y = PC2),size = 14, color = "grey50", shape = 23,stroke = 2)+   geom_point(data = subset(sc_and_bulkprojection_comb, Source == 'Bulk'),aes(x = PC1, y = PC2),size = 16, color = "grey50", shape = 23,stroke = 2)+
  geom_point(data = subset(sc_and_bulkprojection_comb, Source == 'Bulk'),aes(x = PC1, y = PC2, color = celltype), shape = 23,stroke = 2) +   
  theme_light() + scale_alpha_discrete(range = c(1,.6)) +
theme(axis.title = element_text(size = 24), axis.text = element_text(size = 22),legend.position = c(.24,.15),legend.background = element_rect(color = "transparent", fill = "transparent"),legend.text = element_text(size = 17)) + ggtitle("") + 
  annotate("text", x = 5.1, y = 6,size = 7, label = "LPS-treated", color = "#F8766D") + 
  annotate("text", x = -5, y = 6,size = 7, label = "Untreated", color = "#00BFC4") + 
  guides(color = "none",fill = "none", size = "none",alpha = "none") + labs(shape = "",x="PC1",y = "PC2") 
ggsave(paste0(figOut,"celltypePlusBulk.pdf"), width = 5, height = 5, units = "in")
```

```{r Fig2B: PC-weight-based PCA result clustering and fig-gen}
#Shortening Process Names

xcond.f_fdr$Process <- ifelse(xcond.f_fdr$Process=="nuclear-transcribed mRNA catabolic process, nonsense-mediated decay","nuclear-transcribed mRNA catabolic process, NMD",ifelse(xcond.f_fdr$Process == "SRP-dependent cotranslational protein targeting to membrane","SRP-dependent protein targeting to membrane",ifelse(xcond.f_fdr$Process == "positive regulation of NF-kappaB transcription factor activity","positive regulation of NF-kappaB TF activity",xcond.f_fdr$Process))) 

#clustering 
xcond.f_fdr_sel <- xcond.f_fdr %>% select(Process,PC,Association)
xcond.f_fdr_clustIn <- (dcast(xcond.f_fdr_sel,Process~PC))
row.names(xcond.f_fdr_clustIn) <- xcond.f_fdr_clustIn$Process
xcond.f_fdr_clustIn <- xcond.f_fdr_clustIn %>% select(-Process)
xcond.f_fdr_clustIn <- as.matrix(xcond.f_fdr_clustIn)
xcond.f_fdr_clustIn[is.na(xcond.f_fdr_clustIn)] <- 0
xcond_dist1 <- dist(xcond.f_fdr_clustIn)
xcond_dist2 <- dist(t(xcond.f_fdr_clustIn))
xcond_clust1 <- hclust(t(xcond_dist1), method = "ward.D")
xcond_clust2 <- hclust(t(xcond_dist2), method = "ward.D")

ProcOrder <- xcond_clust1$labels[xcond_clust1$order]
xcond.f_fdr$negLogQ <- -log10(xcond.f_fdr$qVal)
xcond.f_fdr$binnedNegLogQ <- ifelse(xcond.f_fdr$negLogQ >= 3, 3,xcond.f_fdr$negLogQ)



xcond.f_fdr %>% ggplot(aes(x = PC, y = factor(str_wrap(Process, width = 30), levels = str_wrap(ProcOrder, width = 30)),fill = Association, size = binnedNegLogQ)) + geom_point(shape = 21, stroke = 1) + scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0) + theme_light() + labs(fill = "Loading\nz-score",x = "Principal component",y = "", size = bquote("-log"[10]~"(q val)")) + theme(axis.text.y = element_text(size = 18),axis.text.x = element_text(size = 26),axis.title = element_text(size = 30), legend.title = element_text(size = 20), legend.text = element_text(size = 20), legend.position = "bottom", legend.box="vertical", legend.margin=margin()) +
  scale_size_continuous(breaks = seq(2,3,1),range=c(1,20),
                  limits = c(0, 4), labels = c("2",">3")) +  scale_x_discrete(limits = c("1",
     "2","3","4","5"), expand=c(0.1, 0)) 
ggsave(paste0(figOut,"xCond_PC_PSEA_legendBottom.pdf"), width = 7.75, height = 10, units = "in")
```


```{r Fig3a: Joint LPS and untreated PSEA fig}

Joint_PC_PSEA <- rbind(LPS_PSEA.f_fdr2,NT_PSEA.f_fdr)

Joint_PC_PSEA$Process <- ifelse(Joint_PC_PSEA$Process == "antigen processing and presentation of exogenous peptide antigen via MHC class II","antigen processing and presentation via MHC class II", ifelse(Joint_PC_PSEA$Process=="nuclear-transcribed mRNA catabolic process, nonsense-mediated decay","nuclear-transcribed mRNA catabolic process, NMD",ifelse(Joint_PC_PSEA$Process=="SRP-dependent cotranslational protein targeting to membrane","SRP-dependent protein targeting to membrane",Joint_PC_PSEA$Process))) 



Ccond.f_fdr_sel <- NT_PSEA.f_fdr %>% select(Process,PC,Association)
Ccond.f_fdr_clustIn <- (dcast(Ccond.f_fdr_sel,Process~PC))
row.names(Ccond.f_fdr_clustIn) <- Ccond.f_fdr_clustIn$Process
Ccond.f_fdr_clustIn <- Ccond.f_fdr_clustIn %>% select(-Process)
Ccond.f_fdr_clustIn <- as.matrix(Ccond.f_fdr_clustIn)
Ccond.f_fdr_clustIn[is.na(Ccond.f_fdr_clustIn)] <- 0
Ccond_dist1 <- dist(Ccond.f_fdr_clustIn)
Ccond_dist2 <- dist(t(Ccond.f_fdr_clustIn))
Ccond_clust1 <- hclust(t(Ccond_dist1), method = "ward.D")
Ccond_clust2 <- hclust(t(Ccond_dist2), method = "ward.D")

Joint_PC_PSEA_sum <- Joint_PC_PSEA %>% dplyr::group_by(Process) %>% dplyr::summarize(numCond = length(unique(Condition))) %>% dplyr::arrange(numCond)
Joint_PC_PSEA_sum1 <- Joint_PC_PSEA_sum %>% dplyr::filter(numCond ==1)


ProcOrderCc <- (Ccond_clust1$labels[Ccond_clust1$order])
ProcOrderCc_filt <- ProcOrderCc[!(ProcOrderCc)%in%Joint_PC_PSEA_sum1$Process]
interProts <- intersect(levels(factor(Joint_PC_PSEA$Process)),ProcOrderCc_filt) 
ProcOrderCc2 <- c(ProcOrderCc_filt,levels(factor(Joint_PC_PSEA$Process))[(!levels(factor(Joint_PC_PSEA$Process)) %in% interProts)])             
Joint_PC_PSEA$negLog10q <- -log10(Joint_PC_PSEA$qVal)              
Joint_PC_PSEA$nLog_bin <- ifelse(Joint_PC_PSEA$negLog10q >= 3,3,Joint_PC_PSEA$negLog10q)

Joint_PC_PSEA_termSum <- Joint_PC_PSEA %>% dplyr::group_by(Process) %>% dplyr::summarize(Process_order = n(), sumPC = sum(PC)) %>% dplyr::arrange(Process_order, sumPC)
Joint_PC_PSEA_termSum$Ind <- seq(1,nrow(Joint_PC_PSEA_termSum)) 
Joint_PC_PSEA <- Joint_PC_PSEA %>% dplyr::left_join(Joint_PC_PSEA_termSum, by = "Process")
Joint_PC_PSEA$PC_lab <- paste0("PC",Joint_PC_PSEA$PC)
Joint_PC_PSEA$Condition2 <- ifelse(Joint_PC_PSEA$Condition == "LPS","LPS-treated","Untreated")

Joint_PC_PSEA %>% dplyr::filter(PC != 6) %>% ggplot(aes(x = factor(Condition2, levels = c("Untreated","LPS-treated")), y =reorder(str_wrap(Process, width = 30), -Ind),fill = (Association), size = nLog_bin)) + geom_point(shape = 21, stroke = 1) + scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0) + theme_light() + labs(fill = "Loading\nz-score", y = "",x = "",size = bquote("-log"[10]~"(q val)")) + theme(axis.text.y = element_text(size = 20),axis.text.x = element_text(angle = 45, hjust = 1, size = 16), legend.title =element_text(size = 28), legend.text = element_text(size = 20, hjust = 0.4),strip.background = element_rect(fill = "white", color = "grey70"), strip.text = element_text(size = 22, color = "grey20", face = "bold"), axis.title.x = element_text(size = 18), legend.position = "bottom", legend.box="vertical", legend.margin=margin(),plot.margin=unit(c(0,1,0,0),"cm")) + facet_wrap(~PC_lab,nrow = 1) + scale_color_manual(values = c("red","blue")) +
  scale_size_continuous(breaks = seq(2,3,1),range=c(3,15),
                  limits = c(1, 4), labels = c("2","> 3"))
ggsave(paste0(figOut,"CondComp_PC_PSEA_v3.pdf"), width = 9, height = 9, units = "in")

```



```{r Fig 3a: color-coded PCA (Proton transport)}
PCAout_NT_coords.df <- as.data.frame(PCAout_NT_coords[,1:5])
PCAout_LPS_coords.df <- as.data.frame(PCAout_LPS_coords[,1:5])

PCAout_NT_coords.df$id <- row.names(PCAout_NT_coords.df)
PCAout_LPS_coords.df$id <- row.names(PCAout_LPS_coords.df)

prePCA_LPS_norm.m <- melt(prePCA_LPS_norm)
prePCA_NT_norm.m <- melt(prePCA_NT_norm)

#LPS cells (unimputed)
BC_imp_LPS_PxC3.m <- prePCA_LPS_norm.m %>% left_join(ProtMissing.m, by = c("Var1"="Proteins", "Var2"="variable"))
BC_imp_LPS_PxC3.m$newVar <- ifelse(is.na(BC_imp_LPS_PxC3.m$value.y),NA,BC_imp_LPS_PxC3.m$value.x)
BC_unimp_LPS_PxC3.m <- BC_imp_LPS_PxC3.m %>% select(Var1,Var2,newVar)

#Untreated cells (unimputed)
BC_imp_NT_PxC3.m <- prePCA_NT_norm.m %>% left_join(ProtMissing.m, by = c("Var1"="Proteins", "Var2"="variable"))
BC_imp_NT_PxC3.m$newVar <- ifelse(is.na(BC_imp_NT_PxC3.m$value.y),NA,BC_imp_NT_PxC3.m$value.x)
BC_unimp_NT_PxC3.m <- BC_imp_NT_PxC3.m %>% select(Var1,Var2,newVar)

#untreated sets
NT_sets <- c("proton transport")

#LPS sets
LPS_sets <- c("proton transport")

BC_unimp_NT_GOinput <- BC_unimp_NT_PxC3.m %>% left_join(GeneMappings_single2, by = c("Var1"="UNIPROTKB"))
BC_unimp_LPS_GOinput <- BC_unimp_LPS_PxC3.m %>% left_join(GeneMappings_single2, by = c("Var1"="UNIPROTKB"))


GOout_NT.df <- GO_for_PCA(NT_sets,BC_unimp_NT_GOinput)
GOout_LPS.df<- GO_for_PCA(LPS_sets,BC_unimp_LPS_GOinput)

PCAout_NT_coords.df2 <- GOout_NT.df %>% left_join(PCAout_NT_coords.df, by = "id")
PCAout_NT_coords.df2$Condition <- "Untreated"
PCAout_LPS_coords.df2 <- GOout_LPS.df %>% left_join(PCAout_LPS_coords.df, by = "id")
PCAout_LPS_coords.df2$Condition <- "LPS-treated"

ProtonNT <- PCAout_NT_coords.df2 %>% ggplot(aes(x = -PC1, y = PC2, color =`proton transport`)) + geom_point(size = 3, alpha = 0.7, shape = 16) + scale_color_gradient2(low = "blue", mid = "white", high = "red", midpoint=0) + theme_light() + theme(plot.title = element_text(size = 28), legend.position = "null", axis.text = element_blank(), axis.title = element_text(size = 28), strip.text = element_text(size = 24),strip.background = element_rect(color = "grey40",fill = "grey40"))  + labs(x = "PC1", color = "")+ theme(legend.position = "null") + ggtitle("Proton Transport") + facet_wrap(~Condition, strip.position="top")

ProtonLPS <- PCAout_LPS_coords.df2 %>% ggplot(aes(x = PC1, y = PC2, color =`proton transport`)) + geom_point(size = 3, alpha = 0.7, shape = 16) + scale_color_gradient2(low = "blue", mid = "white", high = "red", midpoint=0) + theme_light() +theme(plot.title = element_text(size = 28), legend.position = "null", axis.text = element_blank(), axis.title = element_text(size = 28), strip.text = element_text(size = 24),strip.background = element_rect(color = "grey40",fill = "grey40")) + labs(color = "")+ theme(legend.position = "null") + ggtitle("Proton Transport")+ facet_wrap(~Condition, strip.position="top")

(ProtonNT + ProtonLPS) + plot_layout(guides = 'collect') & theme(legend.position = "null")
ggsave(paste0(figOut,"Proton_v2.pdf"), width = 8, height = 4, units = "in")
```

```{r Protein names from original FASTA file}
library(seqinr)
myfasta<- read.fasta(file = paste0(datPath,"musmusculus_SPonly_MEROPS_012221.fasta"), seqtype = "AA",as.string = TRUE, set.attributes = FALSE, whole.header = TRUE)
FASTA_in_proteins <- as.data.frame(names(myfasta))
FASTA_in_proteins$Prot <- gsub("sp\\|","",FASTA_in_proteins$`names(myfasta)`)
FASTA_in_proteins$Prot <- gsub("\\|.*","",FASTA_in_proteins$Prot)
FASTA_in_proteins$names <- gsub(".*MOUSE","",FASTA_in_proteins$`names(myfasta)`)
FASTA_in_proteins$names <- gsub("OS.*","",FASTA_in_proteins$names)
FASTA_in_proteins$genes <- gsub("sp\\|","",FASTA_in_proteins$`names(myfasta)`)
FASTA_in_proteins$genes <- gsub(".*\\|","",FASTA_in_proteins$genes)
FASTA_in_proteins$genes <- gsub("_MOUSE.*","",FASTA_in_proteins$genes)

ProtRef_df <- as.data.frame(FASTA_in_proteins[,c(2,3)])
ProtRef_df_gene <- as.data.frame(FASTA_in_proteins[,c(2,4)])
detach("package:seqinr", unload = TRUE)
```

#---------------
# Endocytosis Figure Components
#---------------

```{r Dextran data import}

Dextran_DIA_LPS_filt <- Dextran_DIA_LPS %>% filter(direction != "Not Significant",abs(median_prot_FC) >= 1,!grepl("keratin",PG.ProteinDescriptions))
Dextran_DIA_NT_filt <- Dextran_DIA_NT %>% filter(direction != "Not Significant",abs(median_prot_FC) >= 1,!grepl("keratin",PG.ProteinDescriptions))

# Extracting sets of antiendocytic and proendocytic proteins
Dextran_DIA_LPS_filt_AntiEndocytic <- Dextran_DIA_LPS %>% filter(direction != "Not Significant",median_prot_FC >= 1,n>1,FC_qval <= .01)
Dextran_DIA_LPS_filt_Endocytic <- Dextran_DIA_LPS %>% filter(direction != "Not Significant",median_prot_FC <= -1,n>1,FC_qval <= .01)
Dextran_DIA_NT_filt_AntiEndocytic <- Dextran_DIA_NT %>% filter(direction != "Not Significant",median_prot_FC >= 1,n>1,FC_qval <= .01)
Dextran_DIA_NT_filt_Endocytic <- Dextran_DIA_NT %>% filter(direction != "Not Significant",median_prot_FC <= -1,n>1,FC_qval <= .01)

Dextran_DIA_LPS_filt_AntiEndocytic_lim <- Dextran_DIA_LPS_filt_AntiEndocytic %>% select(PG.ProteinAccessions)
Dextran_DIA_LPS_filt_Endocytic_lim <- Dextran_DIA_LPS_filt_Endocytic %>% select(PG.ProteinAccessions)
Dextran_DIA_NT_filt_AntiEndocytic_lim <- Dextran_DIA_NT_filt_AntiEndocytic %>% select(PG.ProteinAccessions)
Dextran_DIA_NT_filt_Endocytic_lim <- Dextran_DIA_NT_filt_Endocytic %>% select(PG.ProteinAccessions)

Dextran_dict_LPS <- Dextran_DIA_LPS %>% select(PG.ProteinAccessions,PG.ProteinDescriptions) %>% group_by(PG.ProteinAccessions) %>% slice(1)

Endocytosis_union <- as.data.frame(intersect(Dextran_DIA_LPS_filt_Endocytic_lim$PG.ProteinAccessions,Dextran_DIA_NT_filt_Endocytic_lim$PG.ProteinAccessions))
colnames(Endocytosis_union) <- "PG.ProteinAccessions"
Endocytosis_LPS_outer <- Dextran_DIA_LPS_filt_Endocytic_lim %>% filter(!(PG.ProteinAccessions %in% Endocytosis_union$PG.ProteinAccessions))
Endocytosis_NT_outer <- Dextran_DIA_NT_filt_Endocytic_lim %>% filter(!(PG.ProteinAccessions %in% Endocytosis_union$PG.ProteinAccessions))

Endocytosis_union <- Endocytosis_union %>% left_join(Dextran_dict_LPS, by = "PG.ProteinAccessions")
```

```{r LPS Dextran-Uptake Volcano Plot}

require("ggrepel")
Dextran_DIA_LPS2 <- Dextran_DIA_LPS

Dextran_DIA_LPS2$Labs <- ifelse(Dextran_DIA_LPS2$direction == "Decreased","High uptake",ifelse(Dextran_DIA_LPS2$direction == "Increased","Low uptake","Not significant"))
Dextran_DIA_LPS2$GeneLab2 <- ifelse(Dextran_DIA_LPS2$median_prot_FC < -3,Dextran_DIA_LPS2$Gene.names...primary..,ifelse(Dextran_DIA_LPS2$median_prot_FC > 3,Dextran_DIA_LPS2$Gene.names...primary..,""))
Dextran_DIA_LPS2$GeneLab3 <- ifelse(Dextran_DIA_LPS2$Gene.names...primary.. %in% c("Mrc1","Stab1","Snx17"),Dextran_DIA_LPS2$Gene.names...primary..,"")

Dextran_DIA_LPS2$negLog_q <- -log10(Dextran_DIA_LPS2$FC_qval)
Dextran_DIA_LPS2$negLog_q_bin <- ifelse(Dextran_DIA_LPS2$negLog_q >= 30, 30,Dextran_DIA_LPS2$negLog_q)

Dextran_DIA_LPS2 %>% ggplot(aes(x = -median_prot_FC, y = negLog_q_bin, color = factor(Labs, levels = c("High uptake","Low uptake","Not significant")))) + geom_point(shape = 16,size = 5,alpha = 0.7) + theme_light() + labs(x = bquote("High / low uptake, log"[2]), y = bquote("-log"[10]~"(q val)"), color = "") + scale_color_manual(values = c("#009E73", "#CC79A7","#888888"))+ theme(axis.text = element_text(size = 24), axis.title = element_text(size = 32), axis.title.y = element_text(margin = margin(t = 0, r = 15, b = 0, l = 0)), legend.text = element_text(size = 28), legend.title = element_blank(), legend.position="null",plot.margin = unit(c(1,0,0,0),"cm")) + geom_text_repel(aes(label = GeneLab2),size = 8, color = "black",min.segment.length = unit(0.33, "lines")) + coord_cartesian(xlim = c(-6.5,6.5), ylim = c(0,33))
ggsave(paste0(figOut,"LPS_DexVolcano_wide_v5_SelectLabs2.pdf"), width =8, height = 4, units = "in")
```

```{r Dextran Uptake Sorting histogram}
library(tidyr)
library(scales)

LPS_uptake1 <- read.delim(paste0(datPath,"dextran_graph_LPS1.txt"))
LPS_uptake2 <- read.delim(paste0(datPath,"dextran_graph_LPS2.txt"))

LPS_uptake1_obs <- LPS_uptake1 %>% uncount(weights = Events)
LPS_uptake1_obs.f <- LPS_uptake1_obs %>% filter((LPS_uptake1_obs$MFI > 1000)&(LPS_uptake1_obs$MFI < 50000))


LPS_uptake1_obs.f$log10MFI <- log10(LPS_uptake1_obs.f$MFI)

LPS_quants <- (quantile((LPS_uptake1_obs.f$MFI),c(.1,.5,.9)))

ggplot(LPS_uptake1_obs.f, aes(x = (MFI), y = ..count.., fill = ..x..)) +
  geom_histogram(binwidth = .05, pad = TRUE, color = "grey70") +
  scale_x_log10(breaks = trans_breaks("log10", function(x) 10^x),
              labels = trans_format("log10", math_format(.x))) + 
  scale_fill_gradient2(low = "#ad0060",high = "#008f67", midpoint = log10(LPS_quants[2]))  + theme_light() +labs(y = "Cells", x = bquote("Dextran Intensity, log"[10])) + theme(axis.text = element_text(size = 24), axis.title = element_text(size = 32), legend.text = element_text(size = 26), legend.position="null",plot.margin =unit(c(1,0,0,0),"cm")) + geom_vline(xintercept = LPS_quants[[1]],linetype = "dashed", color = "grey60", size = 2) + geom_vline(xintercept = LPS_quants[[3]],linetype = "dashed",color = "grey60", size = 2) + 
  annotate("text",x = (LPS_quants[[1]] - 3750) , y = 1550,size = 12, label = "Low", color = "#D1509D")+
  annotate("text",x = (LPS_quants[[1]] - 3750) , y = 1275,size = 12, label = "Uptake", color = "#D1509D")+
  annotate("text",x = (LPS_quants[[1]] - 3750) , y = 1000,size = 8.5, label = "10th Percentile", color = "#D1509D")+
  annotate("text",x = (LPS_quants[[3]] + 44000) , y = 1550,size = 12 ,label = "High", color = "#02875E") +
  annotate("text",x = (LPS_quants[[3]] + 44000) , y = 1275,size = 12 ,label = "Uptake", color = "#02875E") +
  annotate("text",x = (LPS_quants[[3]] + 44000) , y = 1000,size = 8.5 ,label = "90th Percentile", color = "#02875E") +coord_cartesian(xlim = c(10^3,11^5))
ggsave(paste0(figOut,"LPS_DextranSort_wide.pdf"), width = 8.15, height = 4.2, units = "in")

detach("package:tidyr", unload = TRUE)
#detach("package:scales", unload = TRUE)
```

```{r Dextran Uptake Annotated PCA}
#PCAout_NT_coords.df2 <- GOout_NT.df %>% left_join(PCAout_NT_coords.df, by = "id")
#PCAout_NT_coords.df2$Condition <- "Untreated"
Dextran_DIA_LPS_filt_Endocytic_lim$PG.ProteinAccessions
Dextran_DIA_LPS_filt_AntiEndocytic_lim$PG.ProteinAccessions

BC_unimp_LPS_PxC3.m_ProEndo <- BC_unimp_LPS_PxC3.m %>% 
    dplyr::filter(Var1 %in% Dextran_DIA_LPS_filt_Endocytic_lim$PG.ProteinAccessions) %>%
    dplyr::group_by(Var2) %>% dplyr::summarize(ProEndocytic = median(newVar,na.rm = TRUE))
BC_unimp_LPS_PxC3.m_AntiEndo <- BC_unimp_LPS_PxC3.m %>% 
    dplyr::filter(Var1 %in% Dextran_DIA_LPS_filt_AntiEndocytic_lim$PG.ProteinAccessions) %>%
    dplyr::group_by(Var2) %>% dplyr::summarize(AntiEndocytic = median(newVar,na.rm = TRUE))

#z-score abundances
BC_unimp_LPS_PxC3.m_ProEndo$ProEndocytic <- (BC_unimp_LPS_PxC3.m_ProEndo$ProEndocytic - mean(BC_unimp_LPS_PxC3.m_ProEndo$ProEndocytic, na.rm = TRUE))/sd(BC_unimp_LPS_PxC3.m_ProEndo$ProEndocytic, na.rm = TRUE)
BC_unimp_LPS_PxC3.m_AntiEndo$AntiEndocytic <- (BC_unimp_LPS_PxC3.m_AntiEndo$AntiEndocytic - mean(BC_unimp_LPS_PxC3.m_AntiEndo$AntiEndocytic, na.rm = TRUE))/sd(BC_unimp_LPS_PxC3.m_AntiEndo$AntiEndocytic, na.rm = TRUE)

BC_unimp_LPS_PxC3.m_ProEndo$ProEndocytic <- ifelse(BC_unimp_LPS_PxC3.m_ProEndo$ProEndocytic >= 2, 2, ifelse(BC_unimp_LPS_PxC3.m_ProEndo$ProEndocytic <= -2,-2,BC_unimp_LPS_PxC3.m_ProEndo$ProEndocytic))
BC_unimp_LPS_PxC3.m_AntiEndo$AntiEndocytic <- ifelse(BC_unimp_LPS_PxC3.m_AntiEndo$AntiEndocytic >= 2, 2,ifelse(BC_unimp_LPS_PxC3.m_AntiEndo$AntiEndocytic <= -2,-2,BC_unimp_LPS_PxC3.m_AntiEndo$AntiEndocytic))

#PCAout_LPS_coords.df2 <- GOout_LPS.df %>% left_join(PCAout_LPS_coords.df, by = "id")
#PCAout_LPS_coords.df2$Condition <- "LPS-treated"

PCAout_LPS_coords.df_endo <- PCAout_LPS_coords.df
PCAout_LPS_coords.df_endo <- PCAout_LPS_coords.df_endo %>% left_join(BC_unimp_LPS_PxC3.m_ProEndo, by = c("id"="Var2"))
PCAout_LPS_coords.df_endo <- PCAout_LPS_coords.df_endo %>% left_join(BC_unimp_LPS_PxC3.m_AntiEndo, by = c("id"="Var2"))

LPS_Anti_scatter <- PCAout_LPS_coords.df_endo %>% ggplot(aes(x = PC1,y = PC2, color = AntiEndocytic)) + geom_point(shape = 16, size = 5, alpha = 0.75) + scale_color_gradient2(low = "blue",mid = "white", high = "red", midpoint = 0) + theme_light() + labs(x = "PC1", y = "PC2")  + theme(plot.title = element_text(size = 32, color = "#D1509D"), legend.position = "null", axis.text = element_blank(), axis.title = element_text(size = 32)) + ggtitle("Low Uptake\nProteins")

LPS_Pro_scatter <- PCAout_LPS_coords.df_endo %>% ggplot(aes(x = PC1,y = PC2, color = ProEndocytic)) + geom_point(shape = 16, size = 5, alpha = 0.75) + scale_color_gradient2(low = "blue",mid = "white", high = "red", midpoint = 0) + theme_light() + labs(x = "PC1", y = "PC2")  + theme(plot.title = element_text(size = 32, color = "#02875E"), legend.position = "null", axis.text = element_blank(), axis.title = element_text(size = 32)) + ggtitle("High Uptake\nProteins")

LPS_Anti_scatter + plot_spacer() + LPS_Pro_scatter + plot_layout(widths = c(5,0.75,5))
ggsave(paste0(figOut,"EndoPCA_v3_forSCoPE_BluetoRed_wide_colorCoded.pdf"), width = 9, height = 5, units = "in")
```

#---------------
# MEROPS Figure Components
#---------------

```{r MEROPS analysis section}
#LPS Channel = Reporter.intensity.corrected.3
#Untreated Channel = Reporter.intensity.corrected.4

BulkDDARatioSet_1 <- BulkDDARatioSet %>% filter(Raw.file == "wGH215", PEP <= .02, PIF > 0.8, !grepl("REV",Leading.razor.protein),!grepl("CON",Leading.razor.protein)) %>% mutate(SeqCharge = paste0(Modified.sequence,Charge)) %>% group_by(SeqCharge) %>% top_n(-1,PEP) %>% slice(1) %>% select(Leading.razor.protein,SeqCharge,Reporter.intensity.corrected.3,Reporter.intensity.corrected.4)

BulkDDARatioSet_1_Seq_Prot <- BulkDDARatioSet_1 %>% select(SeqCharge,Leading.razor.protein)


# Column and row normalize intensities
# Then compare the distribution of intensities for precursors that map to the same Protein
bulk_int <- BulkDDARatioSet_1 %>% select(SeqCharge,Reporter.intensity.corrected.3,Reporter.intensity.corrected.4)                           
bulk_int <- as.data.frame(bulk_int)
row.names(bulk_int) <- bulk_int$SeqCharge                                         
bulk_int_preMat <- bulk_int %>% select(Reporter.intensity.corrected.3,Reporter.intensity.corrected.4)  

colnames(bulk_int_preMat) <- c("LPS","NT")

bulk_int.mat <- as.matrix(bulk_int_preMat)
bulk_int.mat_colMean <- matrixStats::colMeans2(bulk_int.mat)
bim_diag <- diag(1/bulk_int.mat_colMean)
bim_CN <- bulk_int.mat %*% bim_diag
bim_RowMean <- matrixStats::rowMeans2(bim_CN)
bim_RMdiag <- diag(1/bim_RowMean)
bim_CNRN <- bim_RMdiag %*% bim_CN
bim_CNRN.df <- as.data.frame(bim_CNRN)
colnames(bim_CNRN.df) <-  c("LPS","NT")
row.names(bim_CNRN.df) <- row.names(bulk_int)
bim_CNRN.df$SeqCharge <- row.names(bim_CNRN.df)
bim_CNRN.df <- bim_CNRN.df %>% left_join(BulkDDARatioSet_1_Seq_Prot, by = "SeqCharge")

Prot_Comp_out <- data.frame(Protein = character(), Ratio = numeric(), p.value = numeric(),numSeq = numeric(), stringsAsFactors = FALSE)

for(i in 1:length(unique(bim_CNRN.df$Leading.razor.protein))){
  Prot_single <- bim_CNRN.df %>% filter(Leading.razor.protein == unique(bim_CNRN.df$Leading.razor.protein)[i])
  Prot_Comp_out[i,1] <- unique(bim_CNRN.df$Leading.razor.protein)[i]
  if(nrow(Prot_single) > 2){
    Prot_Comp_out[i,2] <- median(Prot_single$LPS,na.rm = TRUE)/median(Prot_single$NT,na.rm = TRUE)
    wt <- wilcox.test(Prot_single$LPS,Prot_single$NT, alternative = "two.sided")
    Prot_Comp_out[i,3] <- wt$p.value
    Prot_Comp_out[i,4] <- nrow(Prot_single)
  }
}

Prot_Comp_out.f <- Prot_Comp_out %>% filter(!is.na(p.value))
Prot_Comp_out.f$qVal <- p.adjust(Prot_Comp_out.f$p.value)
Prot_Comp_out.f2 <- Prot_Comp_out.f %>% filter(qVal <= .01)
Prot_Comp_out.f2$Type <- ifelse(Prot_Comp_out.f2$Ratio > 1, "Inflammatory","Control")

Prot_Comp_out.f3_Control <- Prot_Comp_out.f2 %>% top_n(-20,Ratio)
Prot_Comp_out.f3_Inf <- Prot_Comp_out.f2 %>% top_n(20,Ratio)
MarkerDB_DDA <- rbind(Prot_Comp_out.f3_Control,Prot_Comp_out.f3_Inf)
MarkerDB_DDA.f <- MarkerDB_DDA %>% select(Protein,Type)
colnames(MarkerDB_DDA.f) <- c("Entry","Type") 

```

```{r External marker db prep}
MarkerDB_SCoPE2$Gene.Name <- str_sub(MarkerDB_SCoPE2$Gene.Name, start = 2)
MarkerDB_SCoPE2 <- MarkerDB_SCoPE2 %>% select(Gene.Name,Gene.Symbol,M1,M2,M1.M2.Ratio) %>% arrange(-M1.M2.Ratio)
MarkerDB_SCoPE2$Type <- ifelse(MarkerDB_SCoPE2$M1.M2.Ratio > 0, "Inflammatory","AntiInflammatory")
MarkerDB_SCoPE2 <- MarkerDB_SCoPE2 %>% filter(!is.na(Type))
MarkerDB_SCoPE2.join <- MarkerDB_SCoPE2 %>% left_join(SCoPE2_Human2Mouse,by = c("Gene.Symbol"="Gene.Entry"))
MarkerDB_SCoPE2.2 <- MarkerDB_SCoPE2.join %>% select(Entry, Type)
```

```{r Fig 4b: correlation analysis for MEROPS peptides, message = FALSE, echo = FALSE, warning = FALSE}
# MEROPS peptides
MEROPSvec <- c("P60710_DIG_10362102",	"Q9CZU6_DIG_55521","P60710_DIG_1036223","P60710_DIG_1036222")

#************
# Uncomment these lines to perform correlation analysis
#************
# Condition-specific db
#Ratio_markOut <- CorPermutation(BC_unImp_CxP,MarkerDB_DDA.f,"Internal")
# External db
#SCoPE2_markOut <- CorPermutation(BC_unImp_CxP,MarkerDB_SCoPE2.2,"SCoPE2")
#************

#Uncomment these lines to store your results
#write.table(Ratio_markOut,paste0(figOut,"Ratio_markOut.txt"))
#write.table(SCoPE2_markOut,paste0(figOut,"SCoPE2_markOut.txt"))

#Read in previously generated correlation results
Ratio_markOut <- read.table(paste0(datPath,"Ratio_markOut.txt"))
SCoPE2_markOut <- read.table(paste0(datPath,"SCoPE2_markOut.txt"))


# combining correlation analysis results
db_bind <- rbind(Ratio_markOut,SCoPE2_markOut)

db_bind2 <- db_bind %>% select(-CorDist)
db_bind.m <- melt(db_bind2, id.vars = c("MEROPS","fraction","qVal","qVal2","lab"))
db_bind.m$xLab <- ifelse((db_bind.m$variable == "corPro")&(db_bind.m$lab == "Internal"),"LPS-treated",ifelse((db_bind.m$variable == "corAnti")&(db_bind.m$lab == "Internal"),"Untreated",ifelse((db_bind.m$variable == "corPro")&(db_bind.m$lab == "SCoPE2"),"M1","M2")))

db_bind.m <- db_bind.m %>% left_join(ProtRef_df, by = c("MEROPS"="Prot"))

db_bind.m$EditedName <- ifelse(grepl("_25_",db_bind.m$names),"Citrate synthase: H25, F2, cathepsin E",ifelse(grepl("_104_",db_bind.m$names),"Actin: L104, F1, cathepsin D",ifelse(grepl("_299__Half1",db_bind.m$names),"Actin: L299, F1, cathepsin D","Actin: L299, F2, cathepsin D")))
levelVec <- c("Actin: L104, F1, cathepsin D","Citrate synthase: H25, F2, cathepsin E","Actin: L299, F2, cathepsin D","Actin: L299, F1, cathepsin D")

db_bind.m$refName <- ifelse(db_bind.m$lab == "Internal","Bulk","Martinez, et al.")

db_bind.m$nLogQVal <- -log10(db_bind.m$qVal2)
db_bind.m$nLogQVal_bin <- ifelse(db_bind.m$nLogQVal >= 3, 3,db_bind.m$nLogQVal)

# Figure 4b
db_bind.m %>% ggplot(aes(x = factor(xLab,levels = c("Untreated","LPS-treated","M2","M1")), y = factor(EditedName,levels = rev(levelVec)), size = nLogQVal_bin, fill = value)) + geom_point(shape = 21, stroke = 1) + theme_light() + facet_wrap(~lab) + scale_fill_gradient2(low = "blue", mid = "white", high = "red",midpoint = 0,breaks=c(-.3,0,0.2)) + theme(axis.text.y = element_text(size = 18),axis.text.x = element_text(angle = 45, hjust = 1, size = 22), legend.title =element_text(size = 23.5), legend.text = element_text(size = 20,hjust =1),strip.background = element_rect(fill = "white", color = "grey70"), strip.text = element_text(size = 24, color = "grey20", face = "bold"), axis.title.x = element_text(size = 34)) + labs(x = "Protein Sets",y="",fill = "Correlation",size = expression("-log"[10]~"(q val)")) + facet_wrap(~refName, scales = "free_x") +
  scale_size_continuous(breaks = seq(2,3,1),range=c(1,15),
                  limits = c(0, 3), labels = c("2",">3")) + scale_y_discrete(labels = c("Actin: L104, F1, cathepsin D" = "Fragment 1", "Citrate synthase: H25, F2, cathepsin E" =  "Fragment 2", "Actin: L299, F2, cathepsin D" = "Fragment 2", "Actin: L299, F1, cathepsin D" = "Fragment 1") )
ggsave(paste0(figOut,"Internal_and_SCoPE2_DB_MEROPS_v8.pdf"), width = 9, height = 5.25)

```

```{r Fig 4a: Agreement between bulk and single-cell measurements}
#-------------------------------
# Bulk data preprocessing
#-------------------------------

DIA_Bulk$SeqCharge <- paste0(DIA_Bulk$EG.ModifiedSequence,DIA_Bulk$FG.Charge)
DIA_Bulk_LPS <- DIA_Bulk %>% filter(R.FileName == "eGH694", EG.PEP <= .01)
DIA_Bulk_NT <- DIA_Bulk %>% filter(R.FileName == "eGH693", EG.PEP <= .01) 

DIA_bulk_LPS.sel <- DIA_Bulk_LPS %>% select(SeqCharge,EG.TargetQuantity..Settings.) %>% group_by(SeqCharge) %>% slice(1)
DIA_bulk_NT.sel <- DIA_Bulk_NT %>% select(SeqCharge,EG.TargetQuantity..Settings.) %>% group_by(SeqCharge) %>% slice(1)
DIA.j <- DIA_bulk_LPS.sel %>% inner_join(DIA_bulk_NT.sel, by = "SeqCharge")
DIA.j.df <- as.data.frame(DIA.j)
row.names(DIA.j.df) <- DIA.j.df$SeqCharge

# Column and Row normalizing Precursors
# Taking condition-specific ratios
DIA.j.df.sel <- DIA.j.df %>% select(-SeqCharge) 
DIA.j.mat <- as.matrix(DIA.j.df.sel)
colMed_dex <- matrixStats::colMedians(DIA.j.mat)
colMed_diag <- diag(1/colMed_dex)
DIA.j.mat.CN <- DIA.j.mat %*% colMed_diag
rowMean_dex <- matrixStats::rowMeans2(DIA.j.mat.CN)
DIA.j.mat.CN.df <- as.data.frame(DIA.j.mat.CN)
DIA.j.mat.CN.df$RowmeanFrac <- (1/rowMean_dex)
DIA.j.mat.CN.df[,c(1,2)] <- DIA.j.mat.CN.df[,c(1,2)]*DIA.j.mat.CN.df[,3]
DIA.j.mat.CNRN.df <- DIA.j.mat.CN.df %>% select(V1,V2)
colnames(DIA.j.mat.CNRN.df) <- c("LPS","NT")
SeqProtLib <- DIA_Bulk %>% select(SeqCharge,PG.ProteinAccessions,PG.ProteinDescriptions) %>% group_by(SeqCharge) %>% slice(1)
DIA.j.mat.CNRN.df$SeqCharge <- row.names(DIA.j.mat.CNRN.df)
DIA.j.mat.CNRN.df <- DIA.j.mat.CNRN.df %>% left_join(SeqProtLib, by = "SeqCharge")
DIA.j.mat.CNRN.df$Ratio <- DIA.j.mat.CNRN.df$LPS/DIA.j.mat.CNRN.df$NT

#-------------------------------
# Single-cell data preprocessing
#-------------------------------
CxP.m <- melt(BC_unImp_CxP)
CxP.m <- CxP.m %>% left_join(CG, by = "id")
CxP.m_sum <- CxP.m %>% dplyr::group_by(celltype,variable) %>% dplyr::summarize(medianInt = median(value, na.rm = TRUE))
CxP.m_sum2 <- CxP.m_sum %>% dplyr::group_by(variable) %>% dplyr::summarize(ratio = medianInt[celltype=="LPS"]-medianInt[celltype == "untreated"])
CxP.m_sum3 <- CxP.m_sum2 %>% left_join(ProtRef_df, by = c("variable" = "Prot"))
CxP.m_sum3$MatchID <- ifelse(CxP.m_sum3$variable == "P60710_DIG_10362102","P60710_DIG_10362103",CxP.m_sum3$variable)
CxP.m_sum3.lim <- CxP.m_sum3 %>% filter(variable %in% c("P60710_DIG_10362102",	
"P60710_DIG_1036222","P60710_DIG_1036223","Q9CZU6_DIG_55521"))

CxP.m_sum3.lim2 <- CxP.m_sum3.lim %>% dplyr::select(variable,names,ratio)
colnames(CxP.m_sum3.lim2) <- c("UniProt","Desc","log2Ratio")
CxP.m_sum3.lim2$Lab <- "Single Cell"
CxP.m_sum3.lim2$Desc <- str_sub(CxP.m_sum3.lim2$Desc, start =2)
```

```{r Fig 4a: Plot output}
DIA.j.mat.CNRN.df_sel <- DIA.j.mat.CNRN.df %>% filter(SeqCharge %in% c("_[+C15+H25+N3+O3]LTEAPLNPK_2","_[+C15+H25+N3+O3]ASASSTNLK_2","_[+C15+H25+N3+O3]SGGTTM[Oxidation (M)]YPGIADR_3"))

DIA.j.mat.CNRN.df_sel$joinDef <- gsub(";.*","",DIA.j.mat.CNRN.df_sel$PG.ProteinDescriptions)
DIA.j.mat.CNRN.df_sel$joinDef <- gsub("Half.*","",DIA.j.mat.CNRN.df_sel$joinDef)

DIA.j.mat.CNRN.df_sel$logRat <- log2(DIA.j.mat.CNRN.df_sel$Ratio)
DIA.j.mat.CNRN.df_sel.lim <- DIA.j.mat.CNRN.df_sel %>% dplyr::select(PG.ProteinAccessions,PG.ProteinDescriptions,logRat)
colnames(DIA.j.mat.CNRN.df_sel.lim) <- c("UniProt","Desc","log2Ratio")
DIA.j.mat.CNRN.df_sel.lim$Lab <- "Discovery\nBulk"
Bulk_and_SC <- rbind(DIA.j.mat.CNRN.df_sel.lim,CxP.m_sum3.lim2)
Bulk_and_SC$Desc2 <- gsub("__Half.*","",Bulk_and_SC$Desc)
Bulk_and_SC$Desc3 <- ifelse(Bulk_and_SC$Desc2 == "Actin, cytoplasmic 1 _cathepsin D_104","Actin: L104\ncathepsin D",ifelse(Bulk_and_SC$Desc2 == "Actin, cytoplasmic 1 _cathepsin D_299","Actin: L299\ncathepsin D","Citrate synthase: H25\ncathepsin E"))

Bulk_and_SC2 <- Bulk_and_SC
Bulk_and_SC2 <- Bulk_and_SC %>% dplyr::group_by(Desc3,Lab) %>% dplyr::summarize(log2Ratio2 = mean(log2Ratio, na.rm = TRUE))

FacetOrder <- c("Actin: L104\ncathepsin D","Citrate synthase: H25\ncathepsin E","Actin: L299\ncathepsin D")

#Bulk_and_SC2 %>% ggplot(aes(y = 1,x = Lab, fill = (log2Ratio2))) + geom_tile() + scale_fill_gradient2(limits = c(-.5,.5),low = "blue", mid = "white", high = "red", midpoint = 0) + labs(y = "", x = "") + facet_wrap(~factor(Desc3, levels = FacetOrder), ncol = 1) + labs(fill = expression("Log"[2]~"(LPS:Untreated)")) + theme_light() + theme(axis.text = element_text(size = 18), axis.text.y = element_blank(), strip.background = element_rect(fill = "white", color = "grey70"), strip.text = element_text(size = 22, color = "grey20", face = "bold"), legend.position = "bottom", legend.text = element_text(size = 14, angle = 45, hjust = 1), legend.title = element_text(size = 16))
#ggsave(paste0(figOut,"BulkAndSC_Val_v2.png"), width = 6, height = 8)

Top2 <- Bulk_and_SC2 %>% filter(Desc3 %in% c("Actin: L104\ncathepsin D","Citrate synthase: H25\ncathepsin E")) %>% ggplot(aes(y = 1,x = Lab, fill = (log2Ratio2))) + geom_tile() + scale_fill_gradient2(limits = c(-.5,.5),low = "blue", mid = "white", high = "red", midpoint = 0) + labs(y = "", x = "") + facet_wrap(~factor(Desc3, levels = FacetOrder), ncol = 1) + labs(fill = expression("Log"[2]~"(LPS:Untreated)")) + theme_light() + theme( axis.text = element_blank(), strip.background = element_rect(fill = "white", color = "grey70"), strip.text = element_text(size = 18, color = "grey20", face = "bold"), legend.position = "bottom", legend.text = element_text(size = 14, angle = 45, hjust = 1),panel.spacing.y=unit(0, "lines"), legend.title = element_text(size = 16))

BottomFacet <- Bulk_and_SC2 %>% filter(!(Desc3 %in% c("Actin: L104\ncathepsin D","Citrate synthase: H25\ncathepsin E"))) %>% ggplot(aes(y = 1,x = Lab, fill = (log2Ratio2))) + geom_tile() + scale_fill_gradient2(limits = c(-.5,.5),low = "blue", mid = "white", high = "red", midpoint = 0) + labs(y = "", x = "") + facet_wrap(~factor(Desc3, levels = FacetOrder), ncol = 1) + labs(fill = expression("Log"[2]~"(LPS:Untreated)")) + theme_light() + theme(axis.text = element_text(size = 16), axis.text.y = element_blank(), strip.background = element_rect(fill = "white", color = "grey70"), strip.text = element_text(size = 18, color = "grey20", face = "bold"), legend.position = "bottom", legend.text = element_text(size = 14, angle = 45, hjust = 1), legend.title = element_text(size = 16),plot.margin = unit(c(0.0,0,0.3,0),"cm"))

Top2 / BottomFacet + plot_layout(guides = "collect",heights = c(3,1.5)) & theme(legend.position = "bottom",plot.margin =  margin(0, 0, 0, 0, "cm")) 
ggsave(paste0(figOut,"BulkAndSC_Val_v3_2part_v4.pdf"), width = 4, height = 5.25)
```

```{r fig4c: PCA color-coded by MEROPS peptide abundances}

#PC_both <- prcomp(t(BC_imp_PxC))
PC_both <- prcomp(t(prePCA))
PC_both_coords <- as.data.frame(PC_both$x[,1:2])
PC_both_coords$id <- row.names(PC_both_coords)
BC_unImp_CxP_DIG <- BC_unImp_CxP %>% dplyr::select(contains("_DIG_"))
BC_unImp_CxP_DIG$id <- row.names(BC_unImp_CxP_DIG)
PC_both_coords <- PC_both_coords %>% left_join(BC_unImp_CxP_DIG, by = "id")

#"Actin: L104, F1, cathepsin D","Citrate synthase: H25, F2, cathepsin E"
#"P60710_DIG_10362102",	"P60710_DIG_1036222","P60710_DIG_1036223","Q9CZU6_DIG_55521"
PC_both_coords_Actin_L299_F1 <- PC_both_coords %>% dplyr::select(id,PC1,PC2,P60710_DIG_1036222)
PC_both_coords_Actin_L299_F2 <- PC_both_coords %>% dplyr::select(id,PC1,PC2,P60710_DIG_1036223)
PC_both_coords_Actin_L104_F1 <- PC_both_coords %>% dplyr::select(id,PC1,PC2,P60710_DIG_10362102)
PC_both_coords_CS_H25_F1 <- PC_both_coords %>% dplyr::select(id,PC1,PC2,Q9CZU6_DIG_55521)

PC_both_coords_Actin_L104_F1$AlphVal <- ifelse(is.na(PC_both_coords_Actin_L104_F1$P60710_DIG_10362102),"Low","High")
PC_both_coords_Actin_L299_F1$AlphVal <- ifelse(is.na(PC_both_coords_Actin_L299_F1$P60710_DIG_1036222),"Low","High")
PC_both_coords_Actin_L299_F2$AlphVal <- ifelse(is.na(PC_both_coords_Actin_L299_F2$P60710_DIG_1036223),"Low","High")
PC_both_coords_CS_H25_F1$AlphVal <- ifelse(is.na(PC_both_coords_CS_H25_F1$Q9CZU6_DIG_55521),"Low","High")

# z transformed intensities
PC_both_coords_Actin_L104_F1$zInt <- (PC_both_coords_Actin_L104_F1$P60710_DIG_10362102 - mean(PC_both_coords_Actin_L104_F1$P60710_DIG_10362102, na.rm = TRUE))/sd(PC_both_coords_Actin_L104_F1$P60710_DIG_10362102,na.rm = TRUE)
PC_both_coords_Actin_L299_F1$zInt <- (PC_both_coords_Actin_L299_F1$P60710_DIG_1036222 - mean(PC_both_coords_Actin_L299_F1$P60710_DIG_1036222, na.rm = TRUE))/sd(PC_both_coords_Actin_L299_F1$P60710_DIG_1036222,na.rm = TRUE)
PC_both_coords_Actin_L299_F2$zInt <- (PC_both_coords_Actin_L299_F2$P60710_DIG_1036223 - mean(PC_both_coords_Actin_L299_F2$P60710_DIG_1036223, na.rm = TRUE))/sd(PC_both_coords_Actin_L299_F2$P60710_DIG_1036223,na.rm = TRUE)
PC_both_coords_CS_H25_F1$zInt <- (PC_both_coords_CS_H25_F1$Q9CZU6_DIG_55521 - mean(PC_both_coords_CS_H25_F1$Q9CZU6_DIG_55521, na.rm = TRUE))/sd(PC_both_coords_CS_H25_F1$Q9CZU6_DIG_55521,na.rm = TRUE)

PC_both_coords_Actin_L104_F1$zInt_bin <- ifelse(PC_both_coords_Actin_L104_F1$zInt >= 2, 2, ifelse(PC_both_coords_Actin_L104_F1$zInt <= -2,-2,PC_both_coords_Actin_L104_F1$zInt))
PC_both_coords_Actin_L299_F1$zInt_bin <- ifelse(PC_both_coords_Actin_L299_F1$zInt >= 2, 2, ifelse(PC_both_coords_Actin_L299_F1$zInt <= -2,-2,PC_both_coords_Actin_L299_F1$zInt))
PC_both_coords_Actin_L299_F2$zInt_bin <- ifelse(PC_both_coords_Actin_L299_F2$zInt >= 2, 2, ifelse(PC_both_coords_Actin_L299_F2$zInt <= -2,-2,PC_both_coords_Actin_L299_F2$zInt))
PC_both_coords_CS_H25_F1$zInt_bin <- ifelse(PC_both_coords_CS_H25_F1$zInt >= 2, 2, ifelse(PC_both_coords_CS_H25_F1$zInt <= -2,-2,PC_both_coords_CS_H25_F1$zInt))

sizeVal <- 2

#-----
# Missing data shown as point
#-----
ActL299_F1_v2 <- PC_both_coords_Actin_L299_F1 %>% ggplot(aes(x = PC1, y = PC2, color = zInt_bin,alpha = AlphVal,shape = AlphVal, size = AlphVal)) + 
  geom_point() +
  scale_color_gradient2(low = "blue", mid = "white", high = "red", midpoint = 0, limits = c(-2,2),breaks = c(-2,0,2), labels = c("-2","0","2"),na.value="black") + 
  theme_light() + 
  guides(alpha = "none",shape = "none") + 
  ggtitle("Actin, L299-F1") + 
  theme(axis.text = element_blank(),axis.title = element_text(size = 20),title = element_text(size = 24),legend.title = element_text(size = 20), legend.text = element_text(size = 14), legend.position = "right") +
  scale_alpha_discrete(range = c(0.75, 0.10)) + 
  scale_shape_manual(values = c(16,16))+ 
  scale_size_manual(values = c(6,1), labels = c("Quantified","Missing")) + 
  labs(color = "Abundance\nz-score", size = "")

ActL104_v2 <- PC_both_coords_Actin_L104_F1 %>% ggplot(aes(x = PC1, y = PC2, color = zInt_bin,alpha = AlphVal,shape = AlphVal, size = AlphVal)) + 
  geom_point() +
  scale_color_gradient2(low = "blue", mid = "white", high = "red", midpoint = 0, limits = c(-2,2),breaks = c(-2,0,2), labels = c("-2","0","2"),na.value="black") + 
  theme_light() + guides(alpha = "none",shape = "none") + 
  ggtitle("Actin, L104-F1") + 
  theme(axis.text = element_blank(),axis.title = element_text(size = 20),title = element_text(size = 24),legend.title = element_text(size = 20), legend.text = element_text(size = 14), legend.position = "right") +
  scale_alpha_discrete(range = c(0.75, 0.10)) + 
  scale_shape_manual(values = c(16,16)) + 
  scale_size_manual(values = c(6,1), labels = c("Quantified","Missing")) + 
  labs(color = "Abundance\nz-score", size = "")

((ActL299_F1_v2) + plot_spacer() + (ActL104_v2)) + plot_layout(widths = c(1,0.3,1), guides = "collect") & theme(legend.position = "null")
ggsave(paste0(figOut,"MEROPS_PCA_Fig2E_v2.pdf"), width = 10, height = 4.25, units = "in")

((ActL299_F1_v2) + plot_spacer() + (ActL104_v2)) + plot_layout(widths = c(1,0.5,1), guides = "collect") & theme(legend.position = "right")
ggsave(paste0(figOut,"MEROPS_PCA_Fig2E_v2_legend.pdf"), width = 10, height = 4.25, units = "in")
```
#----------
# Supplemental figures
#----------
```{r unimputed analysis}
# column and row normalizing unimputed data for PCA
centered_unimp_data <- scale(BC_unImp_PxC4, center = TRUE, scale = FALSE)
corrCells <- cor(centered_unimp_data, method = "pearson", use = "pairwise.complete.obs")

sampXfeature <- scale(as.matrix(t(BC_unImp_PxC4)),center = TRUE, scale = FALSE)
#sampXfeature <- (as.matrix(t(BC_unImp_PCin)))
corrFeature <- cor(sampXfeature, use = "pairwise.complete.obs")
corrFeature.df <- (as.data.frame(corrFeature))

corrFeature.f <- corrFeature[!is.na(colSums(corrFeature)),!is.na(colSums(corrFeature))]

eigDecomp.feature <- eigen(corrFeature.f)
eigDecomp.feature.vec <- as.data.frame(eigDecomp.feature$vectors[,1:5])
row.names(eigDecomp.feature.vec) <- row.names(corrFeature.f)
colnames(eigDecomp.feature.vec) <- c("PC1","PC2","PC3","PC4","PC5")

eigDecomp.cells <- eigen(corrCells)
eigDecomp.cells.vec <- as.data.frame(eigDecomp.cells$vectors[,1:5])
row.names(eigDecomp.cells.vec) <- row.names(corrCells)
eigDecomp.cells.vec$id <- row.names(eigDecomp.cells.vec)
eigDecomp.cells.vec <- eigDecomp.cells.vec %>% left_join(CG, by = "id")


featureXsamp_lim <- centered_unimp_data[row.names(centered_unimp_data) %in% row.names(eigDecomp.feature.vec),]
featureXsamp_lim.zeroed <- featureXsamp_lim 
featureXsamp_lim.zeroed[is.na(featureXsamp_lim.zeroed)] <- 0
PC_coords <- as.data.frame(t(as.matrix(featureXsamp_lim.zeroed)) %*% as.matrix(eigDecomp.feature.vec))
PC_coords$id <- row.names(PC_coords)
PC_coords <- PC_coords %>% left_join(CG, by = "id")

BC_unimp_LPS_PxC3.m %>% left_join(GeneMappings_single2, by = c("Var1"="UNIPROTKB"))
unimp_xcond_PS_vec <- c("phagosome maturation","type I interferon-mediated signaling pathway")
BC_unImp_PxC3.m3_GO_in <- BC_unImp_PxC3.m3
BC_unImp_PxC3.m3_GO_in$Var2 <- BC_unImp_PxC3.m3_GO_in$variable
unimp_xcond_PS_abund <- GO_for_PCA(unimp_xcond_PS_vec,BC_unImp_PxC3.m3_GO_in)

PC_coords_PSabund <- PC_coords %>% left_join(unimp_xcond_PS_abund, by = "id")

celltype_unImp <- PC_coords_PSabund %>% ggplot(aes(x = -PC1, y = -PC2, color = celltype)) + geom_point(size = 3,alpha = 0.75, shape = 16) + theme_light()  + labs(color = "") + annotate("text", x = 3, y = 4.5,size = 8, label = "LPS-treated", color = "#F8766D") + annotate("text", x = -2.5, y = 4.5,size = 8, label = "Untreated", color = "#00BFC4") + theme(legend.position = "null", axis.text = element_text(size = 16),axis.title = element_text(size = 20)) + labs(x = "PC1", y = "PC2")

typeIIFN <- PC_coords_PSabund %>% ggplot(aes(x = -PC1, y = -PC2, color = `type I interferon-mediated signaling pathway`)) + geom_point(size = 2) + theme_light() + ggtitle("Type I IFN signaling") + scale_color_gradient2(low = "blue",mid = "white",high = "red",midpoint = 0) + labs(color = "Abundance\nz-score") + theme(legend.title = element_text(size = 12),plot.title = element_text(size = 14)) + labs(x = "PC1", y = "PC2")

PhagosomeMat <- PC_coords_PSabund %>% ggplot(aes(x = -PC1, y = -PC2, color = `phagosome maturation`)) + geom_point(size = 2) + theme_light() + ggtitle("Phagosome Maturation") + scale_color_gradient2(low = "blue",mid = "white",high = "red",midpoint = 0) + labs(color = "Abundance\nz-score") + theme(legend.title = element_text(size = 12),plot.title = element_text(size = 14)) + labs(x = "PC1", y = "PC2")

celltype_unImp 
ggsave(paste0(figOut,"unimp_PCA_celltype_header.pdf"), width = 5.5, height = 5.5, units = "in")
  
(typeIIFN + PhagosomeMat) + plot_layout(guides = "collect", widths = c(1,1)) & theme(legend.position = "bottom", legend.text = element_text(size = 14),legend.title = element_text(size = 12)) 
ggsave(paste0(figOut,"unimp_PCA_PSEA_footer_select.pdf"), width = 5.5, height = 3.75, units = "in")
```

```{r Missing data PCA}

t0.m <- melt(t0_mat, id.vars = "SeqCharge")
t0.m_sum <- t0.m %>% group_by(variable) %>% summarize(fracMD = sum(is.na(value))/n())

PCAout_both_coords_MDsupp <- as.data.frame(PCAout_both_coords[,1:2])
PCAout_both_coords_MDsupp$id <- row.names(PCAout_both_coords_MDsupp)

t0_pca <- PCAout_both_coords_MDsupp %>% left_join(t0.m_sum, by = c("id" = "variable"))
t0_pca$Completeness <- (1-t0_pca$fracMD)*100

MD_pca <- t0_pca %>% ggplot(aes(x = PC1, y = PC2, color = Completeness)) + geom_point( size = 5, alpha = 0.7,shape = 16) + scale_color_gradient2(low = "blue", mid = "white", high = "red", midpoint = 25) + theme_light() + labs(color = "Precursor-level\ndata completeness, %")  + theme(legend.position = "bottom")  + theme(legend.position = "bottom",legend.title = element_text(size = 14),legend.text = element_text(size = 14), axis.text = element_text(size = 18),axis.title = element_text(size = 24)) + labs(x = "PC1", y = "PC2")
ggsave(paste0(figOut,"DataCompleteness_PCA_supplemental.pdf"), width = 7, height = 7, units = "in")

```
