---
title: "pSCoPE_FigGen_Code"
output: html_notebook
---

```{r libraries}
#uncomment and install any packages, if necessary
#install.packages(dplyr)
#install.packages(ggplot2)
#install.packages(reshape2)
#install.packages(stringr)
#install.packages(patchwork)
#install.packages(matrixStats)
#install.packages(ggrepel)
#install.packages(seqinr)
#install.packages(ggpubr)
#install.packages(scales)
#install.packages(Hmisc)

library(dplyr)
library(ggplot2)
library(reshape2)
library(stringr)
library(patchwork)
```

```{r functions}

Empirical_FDR_PEP <- function(ev){
  exps <- levels(factor(ev$Raw.file))
  ev$Score_naOmit <- ifelse(is.na(ev$PEP),100,ev$PEP)
  for(i in 1:length(exps)){
    
    ev_single <- ev %>% filter(Raw.file == levels(factor(ev$Raw.file))[i])
    ev_rank <- ev_single %>% arrange(Score_naOmit) 
    ev_rank$Ind <- seq(1:nrow(ev_rank)) 
    ev_rank$revCount <- ifelse(ev_rank$Reverse=="+",1,0)
    ev_rank <- ev_rank %>% mutate(RevHits = cumsum(revCount),qVal = RevHits/Ind)
                       
    if(i== 1){
      ev_rank_out <- ev_rank
    }
    if(i > 1){
      ev_rank_out <- rbind(ev_rank_out,ev_rank)
    }
  }
  return(ev_rank_out)
}

Empirical_FP_FDR_PEP <- function(ev){
  exps <- levels(factor(ev$Raw.file))
  ev$Score_naOmit <- ifelse(is.na(ev$PEP),100,ev$PEP)
  for(i in 1:length(exps)){
    
    ev_single <- ev %>% filter(Raw.file == levels(factor(ev$Raw.file))[i])
    ev_rank <- ev_single %>% arrange(Score_naOmit) 
    ev_rank$Ind <- seq(1:nrow(ev_rank)) 
    ev_rank$revCount <- ifelse(ev_rank$Reverse=="+",1,0)
    ev_rank <- ev_rank %>% mutate(RevHits = cumsum(revCount),qVal = RevHits/Ind)
                       
    if(i== 1){
      ev_rank_out <- ev_rank
    }
    if(i > 1){
      ev_rank_out <- rbind(ev_rank_out,ev_rank)
    }
  }
  return(ev_rank_out)
}

Empirical_FDR_msms <- function(ev){
  exps <- levels(factor(ev$Raw.file))
  ev <- ev %>% dplyr::filter(Length != 0)
  ev$Score_naOmit <- ifelse(is.na(ev$PEP),1,ev$PEP)
  for(i in 1:length(exps)){
    
    ev_single <- ev %>% filter(Raw.file == levels(factor(ev$Raw.file))[i])
    ev_rank <- ev_single %>% arrange(Score_naOmit) 
    ev_rank$Ind <- seq(1:nrow(ev_rank)) 
    ev_rank$revCount <- ifelse(ev_rank$Reverse=="+",1,0)
    ev_rank <- ev_rank %>% mutate(RevHits = cumsum(revCount),qVal = RevHits/Ind)
                       
    if(i== 1){
      ev_rank_out <- ev_rank
    }
    if(i > 1){
      ev_rank_out <- rbind(ev_rank_out,ev_rank)
    }
  }
  return(ev_rank_out)
}


# MQ.live log parsing functions
getMS1 <- function(log){
  MS1 <- log %>% filter(grepl("PEPTIDE_FOUND_MS1",Message))
  
  MS1$id <- sub(".*id: *(.*?) *DetectedInMs1:.*", "\\1", MS1$Message)
  MS1$DetectedInMs1 <- sub(".*DetectedInMs1: *(.*?) *DetectedInMs2:.*", "\\1", MS1$Message)
  MS1$DetectedInMs2 <- sub(".*DetectedInMs2: *(.*?) *Disabled:.*", "\\1", MS1$Message)
  MS1$Disabled <- sub(".*Disabled: *(.*?) *mz:.*", "\\1", MS1$Message)
  MS1$mz <- as.numeric(sub(".*mz: *(.*?) *mz_deviation:.*", "\\1", MS1$Message))
  MS1$mz_deviation <- as.numeric(sub(".*mz_deviation: *(.*?) *rt_expected:.*", "\\1", MS1$Message))
  MS1$rt_expected <- as.numeric(sub(".*rt_expected: *(.*?) *rt_deviation:.*", "\\1", MS1$Message))
  MS1$rt_deviation <- as.numeric(sub(".*rt_deviation: *(.*?) *intensity:.*", "\\1", MS1$Message))
  MS1$intensity <- as.numeric(sub(".*intensity: *(.*?) *int_ratio:.*", "\\1", MS1$Message))
  MS1$int_ratio <- as.numeric(sub(".*int_ratio: *(.*?) *int_max_ratio:.*", "\\1", MS1$Message))
  MS1$int_max_ratio <- as.numeric(sub(".*int_max_ratio: *(.*?) *mod_seq:.*", "\\1", MS1$Message))
  MS1$mod_seq <- (sub(".*mod_seq: *(.*?) *m:.*", "\\1", MS1$Message))
  MS1$m <- as.numeric(sub(".*m: *(.*?) *z:.*", "\\1", MS1$Message))
  MS1$z <- as.numeric(sub(".*z: *(.*?) *ApexMzDiff:.*", "\\1", MS1$Message))
  MS1$ApexMzDiff <- as.numeric(sub(".*ApexMzDiff: *(.*?) *ApexRtDiff:.*", "\\1", MS1$Message))
  MS1$ApexRtDiff <- as.numeric(sub(".*ApexRtDiff: *(.*?) *ApexRt:.*", "\\1", MS1$Message))
  MS1$ApexRt <- as.numeric(sub(".*ApexRt: *(.*?) *ApexIntRatio:.*", "\\1", MS1$Message))
  MS1$ApexIntRatio <- as.numeric(sub(".*ApexIntRatio: *(.*?) *mz_correction:.*", "\\1", MS1$Message))
  MS1$mz_correction <- as.numeric(sub(".*mz_correction: *(.*?) *mz_tol:.*", "\\1", MS1$Message))
  MS1$mz_tol <- as.numeric(sub(".*mz_tol: *(.*?) *rt_correction:.*", "\\1", MS1$Message))
  MS1$rt_correction <- as.numeric(sub(".*rt_correction: *(.*?) *rt_tol:.*", "\\1", MS1$Message))
  MS1$rt_window <- as.numeric(sub(".*rt_tol: *(.*?) *int_correction:.*", "\\1", MS1$Message))
  MS1$int_correction <- as.numeric(sub(".*int_correction: *(.*?)", "\\1", MS1$Message))

  return(MS1)
  }

# Get MS2
getMS2 <- function(log){
  MS2 <- log %>% filter(grepl("Found_fragment_ions",Message))
  MS2$id <- gsub("Found_fragment_ions_in_scan 2","",MS2$Message)
  MS2$id  <- as.numeric(str_remove(MS2$id, "^0+"))
  return(MS2)
}

# Generating correction plots
getCorrection <- function(log){
  log_corrections <- log %>% filter(grepl("UPDATED_CORRECTION", Message)) %>% dplyr::select(Date, Time, Scan_protocol, RT, Message)
  log_corrections$mz_correction <- as.numeric(sub(".*mz_correction: *(.*?) *mz_window:.*", "\\1", log_corrections$Message))
  log_corrections$mz_window <- as.numeric(sub(".*mz_window: *(.*?) *rt_correction:.*", "\\1", log_corrections$Message))
  log_corrections$rt_correction <- as.numeric(sub(".*rt_correction: *(.*?) *rt_window:.*", "\\1", log_corrections$Message))
  log_corrections$rt_window <- as.numeric(sub(".*rt_window: *(.*?) *int_correction:.*", "\\1", log_corrections$Message))
  log_corrections$int_correction <- as.numeric(sub(".*int_correction: *(.*?) *num_of_corr_peptides:.*", "\\1", log_corrections$Message))
  log_corrections$num_of_corr_peptides <- as.numeric(sub(".*num_of_corr_peptides: *(.*?) *used_corr_peptides:.*", "\\1", log_corrections$Message))
  log_corrections$used_corr_peptides <- as.numeric(sub(".*used_corr_peptides: *(.*?) *first_corr_peptide:.*", "\\1", log_corrections$Message))
  log_corrections$first_corr_peptide <- as.numeric(sub(".*first_corr_peptide: *(.*?) *last_corr_peptide:.*", "\\1", log_corrections$Message))
  log_corrections$last_corr_peptide <- as.numeric(sub(".*last_corr_peptide: *(.*?) *only_ms2_detected:.*", "\\1", log_corrections$Message))
  log_corrections$only_ms2_detected <- (sub(".*only_ms2_detected: *(.*?)", "\\1", log_corrections$Message))
  return(log_corrections)
}

PCA_PSEA_Rebuttal <- function(PCA_weights, GeneProteinMappings, GeneSetDB){
GeneSetDB <- GeneSetDB %>% filter(!is.na(GO_term_name))
GO_terms <- unique(GeneSetDB$GO_term_name)

WRS_test_out <- data.frame(Process = character(),numDBentries = numeric(),numberMatchingEntries = numeric(),fracDB = numeric(),pVal = numeric(),EffectSize = numeric(),PC = numeric(),Association = numeric(), stringsAsFactors = FALSE)
WRS_test_out_full <- data.frame(Process = character(),numDBentries = numeric(),numberMatchingEntries = character(),fracDB = numeric(),pVal = numeric(),EffectSize = numeric(),PC = numeric(),Association = numeric(), stringsAsFactors = FALSE)

ReqNumEntry <- 3
fracDB <- .1
maxNum <- 100

GO_in <- PCA_weights %>% left_join(GeneProteinMappings, by = c("Proteins" = "UNIPROTKB"))

for(i in 1:(ncol(GO_in)-2)){
  GeneInd <- which(colnames(GO_in) == "GENES")
  GO_in_lim <- GO_in[,c(i,GeneInd)]
  
for(j in 1:(length(GO_terms))){

    GO_term <- GeneSetDB %>% dplyr::filter(GO_term_name == GO_terms[[j]])
    GO_term <- GO_term %>% group_by(Gene) %>% slice(1)
    
    GO_in_lim$Lab <- ifelse(GO_in_lim$GENES %in% GO_term$Gene,"Target","Background")
    colnames(GO_in_lim) <- c("Loadings","GENES","Lab")
    WRS_test_out[j,1] <- GO_terms[[j]]
    WRS_test_out[j,2] <- length(unique(GO_term$Gene))
    WRS_test_out[j,3] <- sum(GO_in_lim$Lab == "Target")
    WRS_test_out[j,4] <- WRS_test_out[j,3]/WRS_test_out[j,2]
    WRS_test_out[j,7] <- i
 
    
    if((sum(GO_in_lim$Lab == "Target") >= ReqNumEntry)&(WRS_test_out[j,4] >= fracDB)&(WRS_test_out[j,2] <= maxNum)){
      GO_in_lim2 <- GO_in_lim
      GO_in_lim2$zScore <- (GO_in_lim2$Loadings-mean(GO_in_lim2$Loadings, na.rm=TRUE))/sd(GO_in_lim2$Loadings, na.rm=TRUE)
      
      TargGroup <- GO_in_lim2 %>% dplyr::filter(Lab == "Target")
      BackgroundGroup <- GO_in_lim2 %>% dplyr::filter(Lab != "Target")
      
      WRSTest <- wilcox.test(TargGroup$Loadings, BackgroundGroup$Loadings, alternative = "two.sided", paired = FALSE, exact = FALSE)
      WRS_test_out[j,5] <- WRSTest$p.value
      WRS_test_out[j,6] <- median(GO_in_lim2[GO_in_lim2$Lab == "Target",4], na.rm = TRUE)/median(GO_in_lim2[GO_in_lim2$Lab != "Target",4], na.rm = TRUE)
      WRS_test_out[j,8] <- median(GO_in_lim2[GO_in_lim2$Lab == "Target",4], na.rm = TRUE)

    }
    if((sum(GO_in_lim$Lab == "Target") < ReqNumEntry)|(WRS_test_out[j,4] < fracDB)|(WRS_test_out[j,2] > maxNum)){
      WRS_test_out[j,5] <- NA
      WRS_test_out[j,6] <- NA
      WRS_test_out[j,8] <- NA
    }
    }
  WRS_test_out_full <- rbind(WRS_test_out_full,WRS_test_out)
}
return(WRS_test_out_full)
}


PCA_PSEA_Rebuttal_prot <- function(PCA_weights, GeneSetDB){
GeneSetDB <- GeneSetDB %>% filter(!is.na(GO_term_name))
GO_terms <- unique(GeneSetDB$GO_term_name)

WRS_test_out <- data.frame(Process = character(),numDBentries = numeric(),numberMatchingEntries = numeric(),fracDB = numeric(),pVal = numeric(),EffectSize = numeric(),PC = numeric(),Association = numeric(), stringsAsFactors = FALSE)
WRS_test_out_full <- data.frame(Process = character(),numDBentries = numeric(),numberMatchingEntries = character(),fracDB = numeric(),pVal = numeric(),EffectSize = numeric(),PC = numeric(),Association = numeric(), stringsAsFactors = FALSE)

ReqNumEntry <- 15
fracDB <- .1
maxNum <- 200

GO_in <- PCA_weights 

for(i in 1:(ncol(GO_in)-1)){
  GeneInd <- which(colnames(GO_in) == "Proteins")
  GO_in_lim <- GO_in[,c(i,GeneInd)]
  
for(j in 1:(length(GO_terms))){

    GO_term <- GeneSetDB %>% dplyr::filter(GO_term_name == GO_terms[[j]])
    GO_term <- GO_term %>% group_by(Uniprot) %>% slice(1)
    
    GO_in_lim$Lab <- ifelse(GO_in_lim$Proteins %in% GO_term$Uniprot,"Target","Background")
    colnames(GO_in_lim) <- c("Loadings","Proteins","Lab")
    WRS_test_out[j,1] <- GO_terms[[j]]
    WRS_test_out[j,2] <- length(unique(GO_term$Gene))
    WRS_test_out[j,3] <- sum(GO_in_lim$Lab == "Target")
    WRS_test_out[j,4] <- WRS_test_out[j,3]/WRS_test_out[j,2]
    WRS_test_out[j,7] <- i
 
    
    if((sum(GO_in_lim$Lab == "Target") >= ReqNumEntry)&(WRS_test_out[j,4] >= fracDB)&(WRS_test_out[j,2] <= maxNum)){
      GO_in_lim2 <- GO_in_lim
      GO_in_lim2$zScore <- (GO_in_lim2$Loadings-mean(GO_in_lim2$Loadings, na.rm=TRUE))/sd(GO_in_lim2$Loadings, na.rm=TRUE)
      
      TargGroup <- GO_in_lim2 %>% dplyr::filter(Lab == "Target")
      BackgroundGroup <- GO_in_lim2 %>% dplyr::filter(Lab != "Target")
      
      WRSTest <- wilcox.test(TargGroup$Loadings, BackgroundGroup$Loadings, alternative = "two.sided", paired = FALSE, exact = FALSE)
      WRS_test_out[j,5] <- WRSTest$p.value
      WRS_test_out[j,6] <- median(GO_in_lim2[GO_in_lim2$Lab == "Target",4], na.rm = TRUE)/median(GO_in_lim2[GO_in_lim2$Lab != "Target",4], na.rm = TRUE)
      WRS_test_out[j,8] <- median(GO_in_lim2[GO_in_lim2$Lab == "Target",4], na.rm = TRUE)

    }
    if((sum(GO_in_lim$Lab == "Target") < ReqNumEntry)|(WRS_test_out[j,4] < fracDB)|(WRS_test_out[j,2] > maxNum)){
      WRS_test_out[j,5] <- NA
      WRS_test_out[j,6] <- NA
      WRS_test_out[j,8] <- NA
    }
    }
  WRS_test_out_full <- rbind(WRS_test_out_full,WRS_test_out)
}
return(WRS_test_out_full)
}


PCA_PSEA_Rebuttal_BMDM_prot <- function(PCA_weights, GeneSetDB, PC_cutoff){
GeneSetDB <- GeneSetDB %>% filter(!is.na(GO_term_name))
GO_terms <- unique(GeneSetDB$GO_term_name)

WRS_test_out <- data.frame(Process = character(),numDBentries = numeric(),numberMatchingEntries = numeric(),fracDB = numeric(),pVal = numeric(),EffectSize = numeric(),PC = numeric(),Association = numeric(), stringsAsFactors = FALSE)
WRS_test_out_full <- data.frame(Process = character(),numDBentries = numeric(),numberMatchingEntries = character(),fracDB = numeric(),pVal = numeric(),EffectSize = numeric(),PC = numeric(),Association = numeric(), stringsAsFactors = FALSE)

ReqNumEntry <- 5
fracDB <- .1
maxNum <- 200

indRef <- ifelse(PC_cutoff == 5, 1,ifelse(PC_cutoff == 4,2,1))

GO_in <- PCA_weights 

for(i in 1:(ncol(GO_in)-indRef)){
  GeneInd <- which(colnames(GO_in) == "Proteins")
  GO_in_lim <- GO_in[,c(i,GeneInd)]
  
for(j in 1:(length(GO_terms))){

    GO_term <- GeneSetDB %>% dplyr::filter(GO_term_name == GO_terms[[j]])
    GO_term.sliced <- GO_term %>% group_by(Uniprot) %>% slice(1)
    
    GO_in_lim$Lab <- ifelse(GO_in_lim$Proteins %in% GO_term$Uniprot,"Target","Background")
    colnames(GO_in_lim) <- c("Loadings","Proteins","Lab")
    WRS_test_out[j,1] <- GO_terms[[j]]
    WRS_test_out[j,2] <- length(unique(GO_term$Gene))
    WRS_test_out[j,3] <- sum(GO_in_lim$Lab == "Target")
    WRS_test_out[j,4] <- WRS_test_out[j,3]/WRS_test_out[j,2]
    WRS_test_out[j,7] <- i
 
    
    if((sum(GO_in_lim$Lab == "Target") >= ReqNumEntry)&(WRS_test_out[j,4] >= fracDB)&(WRS_test_out[j,2] <= maxNum)){
      GO_in_lim2 <- GO_in_lim
      GO_in_lim2$zScore <- (GO_in_lim2$Loadings-mean(GO_in_lim2$Loadings, na.rm=TRUE))/sd(GO_in_lim2$Loadings, na.rm=TRUE)
      
      TargGroup <- GO_in_lim2 %>% dplyr::filter(Lab == "Target")
      BackgroundGroup <- GO_in_lim2 %>% dplyr::filter(Lab != "Target")
      
      WRSTest <- wilcox.test(TargGroup$Loadings, BackgroundGroup$Loadings, alternative = "two.sided", paired = FALSE, exact = FALSE)
      WRS_test_out[j,5] <- WRSTest$p.value
      WRS_test_out[j,6] <- median(GO_in_lim2[GO_in_lim2$Lab == "Target",4], na.rm = TRUE)/median(GO_in_lim2[GO_in_lim2$Lab != "Target",4], na.rm = TRUE)
      WRS_test_out[j,8] <- median(GO_in_lim2[GO_in_lim2$Lab == "Target",4], na.rm = TRUE)

    }
    if((sum(GO_in_lim$Lab == "Target") < ReqNumEntry)|(WRS_test_out[j,4] < fracDB)|(WRS_test_out[j,2] > maxNum)){
      WRS_test_out[j,5] <- NA
      WRS_test_out[j,6] <- NA
      WRS_test_out[j,8] <- NA
    }
    }
  WRS_test_out_full <- rbind(WRS_test_out_full,WRS_test_out)
}
return(WRS_test_out_full)
}

# Borrowed from SCoPE2
#Specht, H. et al. Single-cell proteomic and transcriptomic analysis of macrophage hetero-
#geneity using SCoPE2. Genome Biology 22. doi:10 . 1186 / s13059 - 021 - 02267 - 5
#(2021).

cr_norm_log<-function(dat){
  
  for(k in 1:ncol(dat)){
    
    dat[,k]<-dat[,k]-median(dat[,k], na.rm = T)
    
    
  }
  
  
  for(k in 1:nrow(dat)){
    
    dat[k,]<-dat[k,]-mean(dat[k,], na.rm = T)
    
  }
  
  return(dat)
}

flattenCorrMatrix <- function(cormat, pmat) {
  ut <- upper.tri(cormat)

    data.frame(
    row = rownames(cormat)[row(cormat)[ut]],
    column = rownames(cormat)[col(cormat)[ut]],
    cor  =(cormat)[ut],
    p = pmat[ut]
    )
}

PCA_PSEA <- function(PCA_weights, GeneProteinMappings, GeneSetDB){
    GeneSetDB <- GeneSetDB %>% filter(!is.na(GO_term_name))
    GO_terms <- unique(GeneSetDB$GO_term_name)
    WRS_test_out <- data.frame(Process = character(),numDBentries = numeric(),numberMatchingEntries =     
                                 numeric(),fracDB = numeric(),pVal = numeric(),EffectSize = numeric(),PC = 
                                 numeric(),Association = numeric(), stringsAsFactors = FALSE)

    WRS_test_out_full <- data.frame(Process = character(),numDBentries = numeric(),numberMatchingEntries = 
                                      character(),fracDB = numeric(),pVal = numeric(),EffectSize = numeric(),PC =
                                      numeric(),Association = numeric(), stringsAsFactors = FALSE)

    ReqNumEntry <- 5
    fracDB <- .1
    maxNum <- 200
    GO_in <- PCA_weights %>% left_join(GeneProteinMappings, by = c("Proteins" = "UNIPROTKB"))
       
        for(i in 1:(ncol(GO_in)-2)){
          GeneInd <- which(colnames(GO_in) == "GENES")
          GO_in_lim <- GO_in[,c(i,GeneInd)]
  
          for(j in 1:(length(GO_terms))){
             GO_term <- GeneSetDB %>% dplyr::filter(GO_term_name == GO_terms[[j]])
             GO_term <- GO_term %>% group_by(Gene) %>% slice(1)
    
             GO_in_lim$Lab <- ifelse(GO_in_lim$GENES %in% GO_term$Gene,"Target","Background")
             colnames(GO_in_lim) <- c("Loadings","GENES","Lab")
             WRS_test_out[j,1] <- GO_terms[[j]]
             WRS_test_out[j,2] <- length(unique(GO_term$Gene))
             WRS_test_out[j,3] <- sum(GO_in_lim$Lab == "Target")
             WRS_test_out[j,4] <- WRS_test_out[j,3]/WRS_test_out[j,2]
             WRS_test_out[j,7] <- i
 
    
             if((sum(GO_in_lim$Lab == "Target") >= ReqNumEntry)&(WRS_test_out[j,4] >= fracDB)&(WRS_test_out[j,2]              <= maxNum)){
                GO_in_lim2 <- GO_in_lim
                GO_in_lim2$zScore <- (GO_in_lim2$Loadings-mean(GO_in_lim2$Loadings,        
                                      na.rm=TRUE))/sd(GO_in_lim2$Loadings, na.rm=TRUE)
      
                TargGroup <- GO_in_lim2 %>% dplyr::filter(Lab == "Target")
                BackgroundGroup <- GO_in_lim2 %>% dplyr::filter(Lab != "Target")
      
                WRSTest <- wilcox.test(TargGroup$Loadings, BackgroundGroup$Loadings, alternative = "two.sided", 
                                       paired = FALSE, exact = FALSE)
                WRS_test_out[j,5] <- WRSTest$p.value
                WRS_test_out[j,6] <- median(GO_in_lim2[GO_in_lim2$Lab == "Target",4], na.rm = 
                                             TRUE)/median(GO_in_lim2[GO_in_lim2$Lab != "Target",4], na.rm = TRUE)
                WRS_test_out[j,8] <- median(GO_in_lim2[GO_in_lim2$Lab == "Target",4], na.rm = TRUE)
              }
            if((sum(GO_in_lim$Lab == "Target") < ReqNumEntry)|(WRS_test_out[j,4] < fracDB)|(WRS_test_out[j,2] >                maxNum)){
                WRS_test_out[j,5] <- NA
                WRS_test_out[j,6] <- NA
                WRS_test_out[j,8] <- NA
            }
           }
           WRS_test_out_full <- rbind(WRS_test_out_full,WRS_test_out)
        }
        return(WRS_test_out_full)
}

# Get z-scored median abundance for protein set
GO_for_PCA <- function(GOvec,Unimp_in){
  
  for (i in 1:length(GOvec)){
  GO_db_lim <- GO_db %>% filter(GO_term_name %in% GOvec[i])
  unimp_lim <- Unimp_in %>% filter(GENES %in% GO_db_lim$Gene)
  unimp_lim_sum <- unimp_lim %>% dplyr::group_by(Var2) %>% dplyr::summarize(medAbund = median(newVar, na.rm = TRUE)) 
  unimp_lim_sum$medAbund <- (unimp_lim_sum$medAbund-mean(unimp_lim_sum$medAbund, na.rm=TRUE))/sd(unimp_lim_sum$medAbund, na.rm=TRUE)
  unimp_lim_sum$medAbund <- ifelse(unimp_lim_sum$medAbund >= 2,2,ifelse(unimp_lim_sum$medAbund <= -2,-2,unimp_lim_sum$medAbund))
  colnames(unimp_lim_sum) <- c("id",as.character(GOvec[[i]]))
  if(i == 1){
    GSEA_PC_out <- unimp_lim_sum
  }
  if(i > 1){
    GSEA_PC_out <- GSEA_PC_out %>% left_join(unimp_lim_sum, by = "id")
  }
  }
  return(GSEA_PC_out)
}

GO_for_PCA_BMDM <- function(GOvec,Unimp_in){
  
  for (i in 1:length(GOvec)){
  GO_db_lim <- GO_db.3 %>% filter(GO_term_name %in% GOvec[i])
  unimp_lim <- Unimp_in %>% filter(GENES %in% GO_db_lim$Gene)
  unimp_lim_sum <- unimp_lim %>% dplyr::group_by(Var2) %>% dplyr::summarize(medAbund = median(newVar, na.rm = TRUE)) 
  unimp_lim_sum$medAbund <- (unimp_lim_sum$medAbund-mean(unimp_lim_sum$medAbund, na.rm=TRUE))/sd(unimp_lim_sum$medAbund, na.rm=TRUE)
  unimp_lim_sum$medAbund <- ifelse(unimp_lim_sum$medAbund >= 2,2,ifelse(unimp_lim_sum$medAbund <= -2,-2,unimp_lim_sum$medAbund))
  colnames(unimp_lim_sum) <- c("id",as.character(GOvec[[i]]))
  if(i == 1){
    GSEA_PC_out <- unimp_lim_sum
  }
  if(i > 1){
    GSEA_PC_out <- GSEA_PC_out %>% left_join(unimp_lim_sum, by = "id")
  }
  }
  return(GSEA_PC_out)
}

# Permutation function for MEROPS analysis
CorPermutation <- function(df,CodingDoc,dbString){
    df_lim <- df[which(colnames(df) %in% c(MEROPSvec,CodingDoc$Entry))]
    corObs <- as.data.frame(cor(df_lim, method = "pearson", use="pairwise.complete.obs"))
    corObs$MEROPS <- row.names(corObs)
    corObs.m <- melt(corObs)
    corObs.m.f <- corObs.m %>% dplyr::filter((MEROPS %in% MEROPSvec)&(!(variable %in% MEROPSvec)))
    corObs.m.f <- corObs.m.f %>% dplyr::left_join(CodingDoc, by = c("variable" = "Entry"))
    corObs.m.f$Coding <- ifelse(corObs.m.f$Type == "Inflammatory","Pro","Anti")
    corObs_sum <- corObs.m.f %>% dplyr::group_by(MEROPS) %>% dplyr::summarize(corPro = median(value[Coding == "Pro"],na.rm = TRUE),corAnti = median(value[Coding == "Anti"],na.rm = TRUE),CorDist = abs(corPro - corAnti))
    
    corObs_sum_lim <- corObs_sum %>% dplyr::select(MEROPS,CorDist)
    
    colInds_MEROPS <- which((colnames(df_lim))%in% MEROPSvec) 
    
    for(i in 1:10000){
      sampNames <- sample(which(!(colnames(df_lim))%in% MEROPSvec),length(which(!(colnames(df_lim)%in% MEROPSvec))),replace = FALSE)
      df_rearranged <- df_lim
      colnames(df_rearranged) <- c(colnames(df_lim)[sampNames],colnames(df_lim)[colInds_MEROPS])
      corSim <- as.data.frame(cor(df_rearranged, method = "pearson", use="pairwise.complete.obs"))
      corSim$MEROPS <- row.names(corSim)
      corSim.m <- melt(corSim)
      corSim.m.f <- corSim.m %>% dplyr::filter((MEROPS %in% MEROPSvec)&(!(variable %in% MEROPSvec)))
      corSim.m.f <- corSim.m.f %>% dplyr::left_join(CodingDoc, by = c("variable" = "Entry"))
      corSim.m.f$Coding <- ifelse(corSim.m.f$Type == "Inflammatory","Pro","Anti")
      corSim_sum <- corSim.m.f %>% dplyr::group_by(MEROPS) %>% dplyr::summarize(corPro = median(value[Coding == "Pro"],na.rm = TRUE),corAnti = median(value[Coding == "Anti"],na.rm = TRUE),CorDist = abs(corPro - corAnti))
      if(i ==1){
        CorSim_sumOut <- corSim_sum
      }
      if(i > 1){
        CorSim_sumOut <- rbind(CorSim_sumOut,corSim_sum)
      }
    }
    
    CorSim_sumOut_Comp <- CorSim_sumOut %>% dplyr::left_join(corObs_sum_lim, by = "MEROPS")
    CorSim_sumOut_Comp$DistBool <- CorSim_sumOut_Comp$CorDist.x > CorSim_sumOut_Comp$CorDist.y 
  
    CS_permTest <- CorSim_sumOut_Comp %>% dplyr::group_by(MEROPS) %>% dplyr::summarize(fraction = sum(DistBool,na.rm = TRUE)/n())
    CS_permTest$qVal <- p.adjust(CS_permTest$fraction, method = "fdr")
    CS_permTest$qVal2 <- ifelse(CS_permTest$qVal == 0,10^-5,CS_permTest$qVal)
    CS_permTest$lab <- dbString
    CS_permTest2 <- CS_permTest %>% left_join(corObs_sum, by = "MEROPS")
    return(CS_permTest2)
    }
```

```{r plot themes}
theme_box <- function() { 
    theme_light() %+replace% 
        theme(
          axis.text.x = element_blank(), 
          axis.text.y = element_text(size = 20), 
          axis.title.y = element_text(size = 22, angle = 90), 
          plot.title = element_text(size =24, vjust = 2),
          legend.position = "null", 
          legend.text = element_text(size = 18),
          strip.text = element_text(size = 18, face= "bold", color = "black",margin = margin(0.1,0,0.1,0, "cm")),
          strip.background = element_rect( fill = "grey95", color = "grey70")
        )
}

theme_box_MSMS <- function() { 
    theme_light() %+replace% 
        theme(
          axis.text = element_text(size = 20),
          axis.title.y = element_text(size = 22, angle = 90),
          axis.title.x = element_blank(),
          plot.title = element_text(size =24, vjust = 2),
          legend.position = "null", 
          legend.text = element_text(size = 18),
          strip.text = element_text(size = 18, face= "bold", color = "black",margin = margin(0.1,0,0.1,0, "cm")),
          strip.background = element_rect( fill = "grey95", color = "grey70")
        )
}

theme_heatmap <- function() { 
    theme_light() %+replace% 
        theme(
          axis.text = element_blank(), 
          axis.title = element_text(size = 22), 
          legend.position = "null",
          strip.text = element_text(size = 18, face= "bold", color = "black",margin = margin(0.1,0,0.1,0, "cm")), 
          strip.background = element_rect( fill = "grey95", color = "grey70")
        )
}

theme_missingData_tickChart <- function() {
  theme_light() %+replace% 
    theme(
      axis.text.y = element_blank(), 
      legend.position = "null",
      axis.title = element_text(size = 22), 
      axis.text.x = element_text(size = 20), 
      strip.text = element_text(color ="black", face = "bold",size = 18,margin = margin(0.1,0,0.1,0, "cm")), 
      strip.background = element_rect(fill = "grey95", color = "grey70"),
      plot.title = element_text(size = 24,hjust = 0.5, vjust = 2)
    )
}
```

```{r download figure generating data from Zenodo}
#Creating directories
dir.create("pSCoPE_data")
dir.create("figOut")
dir.create("datOut")

download.file("https://zenodo.org/record/7455405/files/pSCoPE_figureGeneratingData.zip?raw=TRUE", destfile="pSCoPE_data/pSCoPE_figureGeneratingData.zip",mode = "wb")

unzip("pSCoPE_data/pSCoPE_figureGeneratingData.zip")

# Figure output path, here:
FigOut <- "figOut/"
# Intermediate data path, here:
datOut <- "datOut/"
# Unzipped data path, here:
datIn <- "datIn/"
```

# %%%%%%%%%%%%%%%%
# Main Figures
# %%%%%%%%%%%%%%%%

# Figure 1: Benchmarking performance of MaxQuant.Live with and without prioritization

```{r Technical Figure 1b}
TargContrast <- read.delim(paste0(datIn,"Fig1/","MQL_contrast_evidence.txt"))

MQLSets <- c("wGH0727","wGH0729","wGH0731","wGH0733","wGH0735","wGH0737")
prioriSets <- c("wGH0728","wGH0730","wGH0732","wGH0734","wGH0736","wGH0738")


TargContrast <- TargContrast %>% dplyr::filter(Raw.file %in% c(prioriSets,MQLSets))


TargContrast <- Empirical_FDR_PEP(TargContrast)
TargContrast <- TargContrast %>% filter(qVal <= .01)

TargList <- read.delim(paste0(datIn,"Fig1/","Rebuttal_aSQC_MQL_contrast_v2.txt"))

TargList.lim <- TargList %>% dplyr::select(SeqCharge,Priority,FT)
TargList.lim$SeqCharge <- gsub("\\(UniMod:35\\)","\\(Oxidation \\(M\\)\\)",TargList.lim$SeqCharge)


# SeqCharge
TargContrast$SeqCharge <- paste0(TargContrast$Modified.sequence,TargContrast$Charge)

TargContrast$Platform <- ifelse(TargContrast$Raw.file %in% MQLSets,"MQL",ifelse(TargContrast$Raw.file %in% prioriSets,"Prioritized","Omit"))

TargContrast.join <- TargContrast %>% left_join(TargList.lim, by = "SeqCharge")
TargContrastSingle <- TargContrast %>% group_by(SeqCharge,Raw.file) %>% top_n(-1,PEP)

TargContrast_sum <- TargContrastSingle %>% group_by(SeqCharge, Platform) %>% summarize(numExps = n())
TargContrast_sum_p <- TargContrast_sum %>% filter(Platform == "Prioritized")
TargContrast_sum_m <- TargContrast_sum %>% filter(Platform == "MQL")
TargContrast_sum.join_p <- TargList.lim %>% left_join(TargContrast_sum_p, by = "SeqCharge")
TargContrast_sum.join_m <- TargList.lim %>% left_join(TargContrast_sum_m, by = "SeqCharge")

TargContrast_sum.join_p$Platform <- "Prioritized"
TargContrast_sum.join_m$Platform <- "Not Prioritized"
TargContrast_sum.join <- rbind(TargContrast_sum.join_p,TargContrast_sum.join_m)

TargContrast_sum.join$numExps <- ifelse(is.na(TargContrast_sum.join$numExps),0,TargContrast_sum.join$numExps)
TargContrast_sum.join$Frac <- TargContrast_sum.join$numExps/6

# Boxplot showing Missing at Random

TargList.nums <- TargList.lim %>% group_by(Priority) %>% summarize(count = n())

TargContrast_sum.join$PriorityLab <- ifelse(TargContrast_sum.join$Priority == 3, "High Priority",ifelse(TargContrast_sum.join$Priority == 2, "Medium",ifelse(TargContrast_sum.join$Priority == 1, "Low","RT Only")))

TargContrast_filt.inter <- TargContrast %>% left_join(TargList.lim, by = "SeqCharge")
TargContrast_filt.inter_sum <- TargContrast_filt.inter %>% group_by(Raw.file, Priority) %>% summarize(count = n())

TargContrast_filt.inter_sum$Platform <- ifelse(TargContrast_filt.inter_sum$Raw.file %in% MQLSets, "Base",ifelse(TargContrast_filt.inter_sum$Raw.file %in% prioriSets,"Prioritization","Omit"))

TargContrast_filt.inter_sumMed <- TargContrast_filt.inter_sum %>% group_by(Platform,Priority) %>% summarize(medianVal = median(count, na.rm = TRUE), sd = sd(count, na.rm = TRUE))

TargContrast_filt.inter_sumMed$PriorityVal <- ifelse(TargContrast_filt.inter_sumMed$Priority ==3, "High Priority",ifelse(TargContrast_filt.inter_sumMed$Priority == 2, "Medium",ifelse(TargContrast_filt.inter_sumMed$Priority == 1, "Low","RT_only")))

TargContrast_IDsum <- TargContrast %>% filter(SeqCharge %in% TargList.lim$SeqCharge)%>% dplyr::group_by(Raw.file) %>% dplyr::summarize(numFDR1 = sum(qVal <= .01),all = n())


TargNums <- data.frame(PriorityVal = character(), count = numeric(), stringsAsFactors = FALSE)
TargNums[1:6,1] <- c("High Priority","Medium","Low","High Priority","Medium","Low")
TargNums[1:6,2] <- c(4000,4000,3679,4000,4000,3679)


TargContrast_filt.inter_sumMed <- TargContrast_filt.inter_sumMed %>% left_join(TargNums, by = "PriorityVal")


TargContrast_sum.join2 <- TargContrast_sum.join %>% filter(PriorityLab == "High Priority") %>% group_by(Platform,Frac,PriorityLab) %>% summarize(FracCount = n())

TargContrast_sum.join2$Percent <- TargContrast_sum.join2$Frac*100

ID_dists.top2 <- TargContrast_sum.join2 %>% ggplot(aes(x = Percent, y = FracCount, fill = factor(Platform))) + geom_col(position = "dodge") + coord_flip() + facet_wrap(~Platform) + theme_light() + facet_wrap(~factor(PriorityLab,levels = c("High Priority","Medium","Low")), nrow = 1, scales = "free_x") + scale_fill_manual(values = c("red","grey30"),breaks=c("Prioritized","Not Prioritized")) + theme(axis.text = element_text(size = 20),axis.title = element_text(size = 22), legend.position  = c(0.57, 0.125),legend.background =  element_rect(fill = "transparent", colour = "transparent"), legend.key = element_rect(fill = "transparent", colour = "transparent"), legend.title = element_blank(),legend.text = element_text(face= "bold", size = 16), axis.text.x = element_text(angle = 45, hjust = 1), strip.text = element_text(size = 18, color = "black", face= "bold"), strip.background = element_rect( fill = "grey95", color = "grey70")) + labs(x = "Success Rate, %", y = "Precursors", fill = "") + guides(fill = guide_legend(label.position = "left", label.hjust = 1))
#ggsave("Technical2_7c_consistency_coverage_MQLcontrast.pdf", width = 7, height = 5, units = "in")

TargContrast_prot <- TargContrast %>% group_by(Raw.file) %>% summarize(numProts = length(unique(Leading.razor.protein))) %>% ungroup()

TargProt_in <- TargContrast %>% group_by(Raw.file, Modified.sequence) %>% top_n(-1, PEP) %>% slice(1)

TargContrast_pepsPerProt <- TargProt_in %>% group_by(Raw.file, Leading.razor.protein) %>% summarize(pepsPerProt = n())
TargContrast_pepsPerProt$Platform <- ifelse(TargContrast_pepsPerProt$Raw.file %in% MQLSets,"Not Prioritized",ifelse(TargContrast_pepsPerProt$Raw.file %in% prioriSets,"Prioritized","Omit"))

TargContrast_prot$Platform <- ifelse(TargContrast_prot$Raw.file %in% MQLSets,"Not Prioritized",ifelse(TargContrast_prot$Raw.file %in% prioriSets,"Prioritized","Omit"))

TargContrast_prot$FacetHeading <- "All Priority Levels"

TargContrast_prot_sum <- TargContrast_prot %>% group_by(Platform) %>% summarize(medProt = median(numProts, na.rm = TRUE))

ProtsBox <- TargContrast_prot %>% filter(Platform!= "Omit") %>% ggplot(aes(x = numProts,y = Platform, color = Platform )) + geom_boxplot(lwd = 1) + geom_point(position=position_jitterdodge(), size = 2, shape = 21, fill = "grey95") + coord_flip(xlim = c(1400,1850)) + theme_light() + theme(axis.text = element_text(size = 20),axis.title = element_text(size = 22), legend.position = "null", axis.text.x = element_blank(), strip.text = element_text(size = 18, color = "black", face= "bold"), strip.background = element_rect( fill = "grey95", color = "grey70")) + labs(x = "Proteins per Run,", color = "", y = "") + scale_color_manual(values = c("grey30","red")) + facet_wrap(~FacetHeading)

 
ID_dists.top2 + plot_spacer() + ProtsBox + plot_layout(widths = c(4, 0.3 ,4))
ggsave(paste0(FigOut,"Technical2_7c_consistency_coverage_MQLcontrast.pdf"), width = 7, height = 5, units = "in")

#TargContrast_prot_sum <- TargContrast_prot %>% group_by(Platform) %>% summarize(medVal = median(numProts, na.rm = TRUE))
```

```{r Technical Figure 1c: MaxQuant.Live log file }
# Rebuttal coverage targList and path

Rebuttal_coverage_targList <- paste0(datIn,"Fig1/","Rebuttal_aSQC_MQL_contrast_v2.txt")

Rebuttal_log_path <- paste0(datIn,"Fig1/Fig1c_logs/MQL_base/")
Rebuttal_log_path.p <- paste0(datIn,"Fig1/Fig1c_logs/MQL_prioritized/")

#re-coding priority b/c we had both RT-only and targetable peptides assigned priority 0
tList_instance <- read.delim(Rebuttal_coverage_targList)
tList_instance$id <- seq(0,nrow(tList_instance)-1,1)
tList_instance$Orig_Priority <- tList_instance$Priority
tList_instance$Priority <- ifelse(tList_instance$TargBool == FALSE, 0, tList_instance$Priority + 1)

tList_instance$id <- seq(0,nrow(tList_instance)-1,1)

logs <- list.files(Rebuttal_log_path)
log_filt <- logs[!(logs %in% c("desktop.ini"))]
numLogs <- length(log_filt)

logs.p <- list.files(Rebuttal_log_path.p)
log.p_filt <- logs.p[!(logs.p %in% c("desktop.ini"))]
numLogs.p <- length(log.p_filt)

# MaxQuant.live base
for(i in 1:numLogs){
  log_instance <- read.delim(paste0(Rebuttal_log_path,log_filt[i]), header = FALSE)
  colnames(log_instance) <-  c("Date","Time","Scan_protocol","RT","Message")

  MS1 <- getMS1(log_instance)
  MS1_unique <- unique(MS1$id)
  MS2 <- getMS2(log_instance)
  tList_instance$seen <- ifelse(tList_instance$id %in% MS1_unique,TRUE,FALSE)
  tList_instance$MS2d <- ifelse(tList_instance$id %in% MS2$id,TRUE,FALSE)
  tList_sum <- tList_instance %>% dplyr::group_by(Priority) %>% 
    dplyr::summarize(seenSum = sum(seen),seenFrac = seenSum/n(), MS2Sum = sum(MS2d),MS2Frac = MS2Sum/n())
  #tList_sum <- tList_instance %>% dplyr::group_by(PriorityTop) %>% 
    #dplyr::summarize(seenSum = sum(seen),seenFrac = seenSum/n(), MS2Sum = sum(MS2d),MS2Frac = MS2Sum/n())
  #tList_sum3 <- tList_instance %>% dplyr::group_by(PriorityTop3) %>% 
    #dplyr::summarize(seenSum = sum(seen),seenFrac = seenSum/n(), MS2Sum = sum(MS2d),MS2Frac = MS2Sum/n())
  tList_sum$file_ind <- i
  if(i ==1){
    tList_sum_out <-tList_sum
    #tList_sum_out3 <-tList_sum3
  }
  if(i > 1){
    tList_sum_out <- rbind(tList_sum_out,tList_sum)
    #tList_sum_out3 <- rbind(tList_sum_out3,tList_sum3)
  }
}

# MaxQuant.live with prioritization
for(i in 1:numLogs.p){
  log_instance <- read.delim(paste0(Rebuttal_log_path.p,log.p_filt[i]), header = FALSE)
  colnames(log_instance) <-  c("Date","Time","Scan_protocol","RT","Message")

  MS1 <- getMS1(log_instance)
  MS1_unique <- unique(MS1$id)
  MS2 <- getMS2(log_instance)
  tList_instance$seen <- ifelse(tList_instance$id %in% MS1_unique,TRUE,FALSE)
  tList_instance$MS2d <- ifelse(tList_instance$id %in% MS2$id,TRUE,FALSE)
  tList_sum <- tList_instance %>% dplyr::group_by(Priority) %>% 
    dplyr::summarize(seenSum = sum(seen),seenFrac = seenSum/n(), MS2Sum = sum(MS2d),MS2Frac = MS2Sum/n())
  #tList_sum <- tList_instance %>% dplyr::group_by(PriorityTop) %>% 
    #dplyr::summarize(seenSum = sum(seen),seenFrac = seenSum/n(), MS2Sum = sum(MS2d),MS2Frac = MS2Sum/n())
  #tList_sum3 <- tList_instance %>% dplyr::group_by(PriorityTop3) %>% 
    #dplyr::summarize(seenSum = sum(seen),seenFrac = seenSum/n(), MS2Sum = sum(MS2d),MS2Frac = MS2Sum/n())
  tList_sum$file_ind <- i
  if(i ==1){
    tList_sum_out.p <-tList_sum
    #tList_sum_out3 <-tList_sum3
  }
  if(i > 1){
    tList_sum_out.p <- rbind(tList_sum_out.p,tList_sum)
    #tList_sum_out3 <- rbind(tList_sum_out3,tList_sum3)
  }
}

targList_Contrast_sum <- tList_instance %>% dplyr::group_by(Priority) %>% dplyr::summarize(TargCount = n())

Priority_dict.contrast <- data.frame(Priority = numeric(), Title = character(), stringsAsFactors = FALSE)
Priority_dict.contrast[1:4,1] <- c(4,3,2,0)
Priority_dict.contrast[1:4,2] <- c("High","Medium","Low","RT only")

tList_sum_out.contrast.j <- tList_sum_out %>% left_join(Priority_dict.contrast, by = "Priority")
tList_sum_out.contrast.j.f <- tList_sum_out.contrast.j %>% select(seenFrac,MS2Frac,Priority,file_ind,Title)
tList_sum_out.contrast.j.f.m <- melt(tList_sum_out.contrast.j.f, id.vars = c("file_ind","Priority","Title"))

tList_sum_out.contrast.p.j <- tList_sum_out.p %>% left_join(Priority_dict.contrast, by = "Priority")
tList_sum_out.contrast.p.j.f <- tList_sum_out.contrast.p.j %>% select(seenFrac,MS2Frac,Priority,file_ind,Title)
tList_sum_out.contrast.p.j.f.m <- melt(tList_sum_out.contrast.p.j.f, id.vars = c("file_ind","Priority","Title"))

tList_sum_out.contrast.j.f.m <- tList_sum_out.contrast.j.f.m %>% left_join(targList_Contrast_sum, by = "Priority")
tList_sum_out.contrast.p.j.f.m <- tList_sum_out.contrast.p.j.f.m %>% left_join(targList_Contrast_sum, by = "Priority")

tList_sum_out.contrast.j.f.m$Type <- "NoPrioritization"
tList_sum_out.contrast.p.j.f.m$Type <- "Prioritization"

CollectedContrast <- rbind(tList_sum_out.contrast.j.f.m,tList_sum_out.contrast.p.j.f.m)

CollectedContrast$legendTitle <- ifelse(CollectedContrast$variable == "seenFrac","Detected at MS1","Sent for MS2")


library(scales)


CollectedContrast %>% filter(Type == "Prioritization") %>% ggplot(aes(value*100,y = legendTitle ,color = legendTitle)) + geom_boxplot(lwd = 1) + 
   geom_point(position=position_jitterdodge(), size = 2, shape = 21, fill = "grey95") + 
  #ggtitle("Technical Performance: Coverage") + 
  coord_flip() + 
  theme_light() + 
  theme(axis.text.x =  element_blank(), 
        strip.background = element_rect( fill = "grey95", color = "grey70"), 
        strip.text = element_text(color= "black", size = 18,face = "bold"),
        axis.text.y = element_text(size =20),
        axis.title.y = element_text(size =22),
        plot.title = element_text(size = 18),
        legend.position = "bottom") + 
  labs(color = "", x = "Prioritized Peptides, %",y = "") + 
  facet_wrap(~factor(Title,levels = c("High","Medium","Low","RT only")), nrow = 1)+
  scale_color_manual(values = c("firebrick4","red")) + geom_text( x = 15, y = 1.5,color = "grey10",aes(label = paste0(comma(TargCount))), size = 10)  + geom_text( x = 12, y = 1.5,color = "grey10",aes(label = paste0("\nPrecursors")), size = 6)
ggsave(paste0(FigOut,"Rebuttal_coverage_MQliveStats_Legend.pdf"), width = 7, height = 5, units = "in")

detach("package:scales", unload=TRUE)

#CollectedContrast_sum <- CollectedContrast %>% group_by(Type,legendTitle,Title) %>% summarize(medVal = median(value, na.rm = TRUE))


```
# Figure 2: benchmarking prioritization against shotgun acquisition

```{r Fig2a}
evi.cov <- read.delim(paste0(datIn,"Fig2/Fig2_a_Coverage/Coverage_evidence_Shotgun_fullFASTA.txt"))
evi.cov_lim <- read.delim(paste0(datIn,"Fig2/Fig2_a_Coverage/Coverage_evidence_pSCoPE_fullFASTA.txt"))
msmsS.cov <- read.delim(paste0(datIn,"Fig2/Fig2_a_Coverage/Coverage_msmsScans_Shotgun_fullFASTA.txt"))
msmsS.cov_lim <- read.delim(paste0(datIn,"Fig2/Fig2_a_Coverage/Coverage_msmsScans_pSCoPE_fullFASTA.txt"))
msms.cov <- read.delim(paste0(datIn,"Fig2/Fig2_a_Coverage/Coverage_msms_Shotgun_fullFASTA.txt"))
msms.cov_lim <- read.delim(paste0(datIn,"Fig2/Fig2_a_Coverage/Coverage_msms_pSCoPE_fullFASTA.txt"))

Rebuttal_cov_ShotgunSet <- c("wGH0643","wGH0644","wGH0645","wGH0646","wGH0647","wGH0648","wGH0649","wGH0650")
Rebuttal_cov_PrioriSet <- c("wGH0672","wGH0679","wGH0680","wGH0678","wGH0675","wGH0681","wGH0671","wGH0673")

evi.cov <- evi.cov %>% dplyr::filter(Raw.file %in% Rebuttal_cov_ShotgunSet)
msmsS.cov <- msmsS.cov %>% dplyr::filter(Raw.file %in% Rebuttal_cov_ShotgunSet)
msms.cov <- msms.cov %>% dplyr::filter(Raw.file %in% Rebuttal_cov_ShotgunSet)
evi.cov_lim <- evi.cov_lim %>% dplyr::filter(Raw.file %in% Rebuttal_cov_PrioriSet)
msmsS.cov_lim <- msmsS.cov_lim %>% dplyr::filter(Raw.file %in% Rebuttal_cov_PrioriSet)
msms.cov_lim <- msms.cov_lim %>% dplyr::filter(Raw.file %in% Rebuttal_cov_PrioriSet)

evi.cov <- evi.cov %>% select(-Acetyl..Protein.N.term.)
msms.cov <- msms.cov %>% select(-Acetyl..Protein.N.term.)

evi.cov <- rbind(evi.cov,evi.cov_lim)
msmsS.cov <- rbind(msmsS.cov,msmsS.cov_lim)
msms.cov <- rbind(msms.cov,msms.cov_lim)

# data/sc data frames

CG.cov <- read.delim(paste0(datIn,"Fig2/Fig2_a_Coverage/ChanKey_Rebuttal_Coverage_PIF50_mrri10.txt"))
CG.cov$Raw.file <- str_sub(CG.cov$Raw.file, start = 2)

MDp.cov <- read.delim(paste0(datIn,"Fig2/Fig2_a_Coverage/Prot_missing_Rebuttal_Coverage_PIF50_mrri10.txt"))

#Productive Scans defined as non-CON/REV identifications
evi.cov.f <- Empirical_FDR_PEP(evi.cov)
evi.cov_PSMs <- evi.cov.f %>% dplyr::filter((!grepl("CON",Proteins))&(qVal <= .01)&(!grepl("REV",Proteins))&(Potential.contaminant != "+")&(Reverse != "+")) %>% group_by(Raw.file,Modified.sequence) %>% top_n(-1,PEP) %>% slice(1)


msmsS.cov_sum <- msmsS.cov %>% dplyr::group_by(Raw.file) %>% dplyr::summarize(numObs = n()) %>% mutate(PSM.filt = "MS/MS\nScans")
evi.cov_PSMs_sum <- evi.cov_PSMs %>% ungroup() %>% dplyr::group_by(Raw.file) %>% dplyr::summarize(numObs = n()) %>% mutate(PSM.filt = "Peptides")

msmsS.cov_Productive_sum <- msmsS.cov %>% dplyr::group_by(Raw.file) %>% dplyr::filter((!grepl("REV",Proteins))&(!grepl("CON",Proteins))&(Identified == "+")) %>% dplyr::summarize(numObs = n()) %>% mutate(PSM.filt = "Productive MS/MS\nScans")

msmsS.cov_Productive_sum_v2 <- msmsS.cov_Productive_sum %>% left_join(msmsS.cov_sum, by = "Raw.file")
msmsS.cov_Productive_sum_v2$numObs <- msmsS.cov_Productive_sum_v2$numObs.x/msmsS.cov_Productive_sum_v2$numObs.y
msmsS.cov_Productive_sum_v3 <- msmsS.cov_Productive_sum_v2 %>% select(Raw.file,numObs,PSM.filt.x)

msmsS.cov_Productive_sum_v3$Method <- ifelse(msmsS.cov_Productive_sum_v3$Raw.file %in% Rebuttal_cov_ShotgunSet,"Shotgun",ifelse(msmsS.cov_Productive_sum_v3$Raw.file %in% Rebuttal_cov_PrioriSet,"Prioritized","Omit"))
msmsS.cov_sum$Method <- ifelse(msmsS.cov_sum$Raw.file %in% Rebuttal_cov_ShotgunSet,"Shotgun",ifelse(msmsS.cov_sum$Raw.file %in% Rebuttal_cov_PrioriSet,"Prioritized","Omit"))
evi.cov_PSMs_sum$Method <- ifelse(evi.cov_PSMs_sum$Raw.file %in% Rebuttal_cov_ShotgunSet,"Shotgun",ifelse(evi.cov_PSMs_sum$Raw.file %in% Rebuttal_cov_PrioriSet,"Prioritized","Omit"))

msmsS.cov_Productive_sum_v3.f <- msmsS.cov_Productive_sum_v3 %>% dplyr::filter(Method != "Omit") 
msmsS.cov_sum.f <- msmsS.cov_sum %>% dplyr::filter(Method != "Omit") 
evi.cov_PSMs_sum.f <- evi.cov_PSMs_sum %>% dplyr::filter(Method != "Omit")

comb_MS2_stats <- rbind(msmsS.cov_sum.f,evi.cov_PSMs_sum.f)
Cov_PSMs_box <- comb_MS2_stats %>% ggplot(aes(x = PSM.filt, y = numObs,color = factor(Method, levels = c("Shotgun","Prioritized")))) + geom_boxplot(alpha=0.5,lwd=1) + coord_cartesian(ylim = c(0,8200)) + theme_box_MSMS() + labs(y = "Total/experiment") + scale_color_manual(values = c("blue","red")) + theme(axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0)))
#ggsave(paste0(FigOut,"MSMS_and_Evi_Stats.pdf"), width = 7 , height = 5, units = "in")


ProdScans1 <- msmsS.cov_Productive_sum_v3.f %>% ggplot(aes(x = PSM.filt.x, y = numObs*100,color = factor(Method, levels = c("Shotgun","Prioritized")))) + geom_boxplot(alpha=0.5,lwd=1) + coord_cartesian(ylim = c(0,75)) + theme_box_MSMS() + labs(y = "Productive MS2 Scans, %") + scale_color_manual(values = c("blue","red")) + theme(axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)), axis.text.x = element_blank())

PepsPerRun <- evi.cov_PSMs_sum.f %>% ggplot(aes(x = PSM.filt, y = numObs,color = factor(Method, levels = c("Shotgun","Prioritized")))) + geom_boxplot(alpha=0.5,lwd=1) + coord_cartesian(ylim = c(0,5000)) + theme_box_MSMS() + labs(y = "Peptides/run") + scale_color_manual(values = c("blue","red")) + theme(axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0)), axis.text.x = element_blank())

evi.cov.f <- Empirical_FDR_PEP(evi.cov)
msms.cov.f <- Empirical_FDR_msms(msms.cov) 
evi.cov_PSMs <- evi.cov.f %>% dplyr::filter((!grepl("CON",Proteins))&(qVal <= .01)&(!grepl("REV",Proteins))&(Potential.contaminant != "+")&(Reverse != "+")) %>% group_by(Raw.file,Modified.sequence) %>% top_n(-1,PEP) %>% slice(1)

msms.cov_1FDR <- msms.cov.f %>% dplyr::filter((!grepl("CON",Proteins))&(qVal <= .01)&(!grepl("REV",Proteins)))

msmsS.cov_sum <- msmsS.cov %>% dplyr::group_by(Raw.file) %>% dplyr::summarize(numObs = n()) %>% mutate(PSM.filt = "MS/MS\nScans")
msms.cov_sum <- msms.cov_1FDR %>% dplyr::group_by(Raw.file) %>% dplyr::summarize(numObs = n()) %>% mutate(PSM.filt = "Productive MS/MS\nScans")
evi.cov_PSMs_sum <- evi.cov_PSMs %>% ungroup() %>% dplyr::group_by(Raw.file) %>% dplyr::summarize(numObs = n()) %>% mutate(PSM.filt = "Peptides")


msmsS.cov_Productive_sum_v2 <- msmsS.cov_sum %>% left_join(msms.cov_sum, by = "Raw.file")
msmsS.cov_Productive_sum_v2$numObs <- msmsS.cov_Productive_sum_v2$numObs.y/msmsS.cov_Productive_sum_v2$numObs.x
msmsS.cov_Productive_sum_v3 <- msmsS.cov_Productive_sum_v2 %>% select(Raw.file,numObs,PSM.filt.x)

msmsS.cov_Productive_sum_v3$Method <- ifelse(msmsS.cov_Productive_sum_v3$Raw.file %in% Rebuttal_cov_ShotgunSet,"Shotgun",ifelse(msmsS.cov_Productive_sum_v3$Raw.file %in% Rebuttal_cov_PrioriSet,"Prioritized","Omit"))
msmsS.cov_sum$Method <- ifelse(msmsS.cov_sum$Raw.file %in% Rebuttal_cov_ShotgunSet,"Shotgun",ifelse(msmsS.cov_sum$Raw.file %in% Rebuttal_cov_PrioriSet,"Prioritized","Omit"))
evi.cov_PSMs_sum$Method <- ifelse(evi.cov_PSMs_sum$Raw.file %in% Rebuttal_cov_ShotgunSet,"Shotgun",ifelse(evi.cov_PSMs_sum$Raw.file %in% Rebuttal_cov_PrioriSet,"Prioritized","Omit"))

msmsS.cov_Productive_sum_v3.f <- msmsS.cov_Productive_sum_v3 %>% dplyr::filter(Method != "Omit") 
msmsS.cov_sum.f <- msmsS.cov_sum %>% dplyr::filter(Method != "Omit") 
evi.cov_PSMs_sum.f <- evi.cov_PSMs_sum %>% dplyr::filter(Method != "Omit")

comb_MS2_stats <- rbind(msmsS.cov_sum.f,evi.cov_PSMs_sum.f)
Cov_PSMs_box <- comb_MS2_stats %>% ggplot(aes(x = PSM.filt, y = numObs,color = factor(Method, levels = c("Shotgun","Prioritized")))) + geom_boxplot(alpha=0.5,lwd=1) + coord_cartesian(ylim = c(0,8200)) + theme_box_MSMS() + labs(y = "Total/experiment") + scale_color_manual(values = c("blue","red")) + theme(axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0)))
#ggsave(paste0(FigOut,"MSMS_and_Evi_Stats.pdf"), width = 7 , height = 5, units = "in")


#With Jitter: Nature version
ProdScans1 <- msmsS.cov_Productive_sum_v3.f %>% ggplot(aes(x = PSM.filt.x, y = numObs*100,color = factor(Method, levels = c("Shotgun","Prioritized"))))  + geom_boxplot(alpha=0.5,lwd=1) + geom_point(position=position_jitterdodge(), size = 2, shape = 21, fill = "grey95") + coord_cartesian(ylim = c(30,90)) + theme_box_MSMS() + labs(y = "Productive MS2 Scans, %") + scale_color_manual(values = c("blue","red")) + theme(axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)), axis.text.x = element_blank(),axis.text = element_text(color = "grey10",hjust = 1))  + scale_y_continuous(limits = c(30, 90), breaks = seq(30, 90, by = 20))

#With Jitter: Nature version
PepsPerRun <- evi.cov_PSMs_sum.f %>% ggplot(aes(x = PSM.filt, y = numObs,color = factor(Method, levels = c("Shotgun","Prioritized")))) + geom_boxplot(alpha=0.5,lwd=1)  + geom_point(position=position_jitterdodge(), size = 2, shape = 21, fill = "grey95") + coord_cartesian(ylim = c(2000,6500)) + theme_box_MSMS() + labs(y = "Peptides/run") + scale_color_manual(values = c("blue","red")) + theme(axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0)), axis.text.x = element_blank(),axis.text = element_text(color = "grey10",hjust = 1)) + scale_y_continuous(limits = c(2000, 6500), breaks = seq(2000, 6500, by = 1500))

CG.s.cov.f <- CG.cov %>% dplyr::filter(celltype != "control",cvm <.4, Raw.file %in% Rebuttal_cov_ShotgunSet)
CG.p.cov.f <- CG.cov %>% dplyr::filter(celltype != "control",cvm <.4, Raw.file %in% Rebuttal_cov_PrioriSet)

MDp.cov.m <- melt(MDp.cov)

MDp.s.cov.m.f <- MDp.cov.m %>% dplyr::filter(variable %in% CG.s.cov.f$id) 
MDp.p.cov.m.f <- MDp.cov.m %>% dplyr::filter(variable %in% CG.p.cov.f$id)

MDp.s.cov.m.f$Method <- "Shotgun" 
MDp.p.cov.m.f$Method <- "Prioritized"

MDp_join <- rbind(MDp.s.cov.m.f,MDp.p.cov.m.f)
MDp_join_sum <- MDp_join %>% group_by(variable, Method) %>% dplyr::summarize(numProts = sum(!is.na(value), na.rm = TRUE))
MDp_join_sum$PSM.filt = "1% FDR"

# Nature scaling
Cov_Prot_box <- MDp_join_sum %>% ggplot(aes( numProts, color = factor(Method, levels = c("Shotgun","Prioritized")))) + geom_boxplot(alpha=0.5,lwd=1)  + coord_flip(xlim = c(600,1500)) + theme_light() + labs(color = "", x = "Proteins/cell", y = "") + theme_box() + scale_color_manual(values = c("blue","red"))  + theme(strip.text = element_text(size = 20), axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0)),axis.text = element_text(color = "grey10",hjust = 1),panel.grid.minor.x = element_blank(),panel.grid.major.x = element_blank()) + scale_x_continuous(limits = c(600, 1500), breaks = seq(600, 1500, by = 300)) + geom_hline(aes(yintercept = 0), color = "grey90")

MDp_join_sum_tally <- MDp_join_sum %>% group_by(Method) %>% summarize(count = n())

ProdScans1 + PepsPerRun + Cov_Prot_box  + plot_layout(widths = c(1,1,1), guides = "collect") & theme(legend.position = 'null')
ggsave(paste0(FigOut,"Coverage_Rebuttal_rev2_1pFDRMS2.pdf"), width = 7.2 , height = 5.1, units = "in")



```

```{r Fig2b: Precursor Intensity Comparison}

AllExps <- read.delim(paste0(datIn,"Fig2/Fig2_bcde_Consistency/ev_Rebuttal_consistency_filtered_PIF50_mrri10_1pFDR.txt"))

ShotgunSet <- c("XwGH0643","XwGH0644","XwGH0645","XwGH0646","XwGH0647","XwGH0648","XwGH0649","XwGH0650")
pSCoPESet <- c("XwGH0660","XwGH0666","XwGH0663","XwGH0668","XwGH0665","XwGH0669","XwGH0661","XwGH0667")

evi.cov <- AllExps %>% dplyr::filter(Raw.file %in% ShotgunSet)

evi.cov_lim <- AllExps %>% dplyr::filter(Raw.file %in% pSCoPESet)

ShotInt <- evi.cov
PriInt <- evi.cov_lim

ShotInt$Platform <- "Shotgun"
PriInt$Platform <- "Prioritized"

IntComb <- rbind(ShotInt,PriInt)


IntComb.lim.sum.plat <- IntComb %>% group_by(Platform) %>% summarize(medInt = median((Intensity), na.rm = TRUE))

#paste0("The difference in medians: ",round(IntComb.lim.sum.plat[2,2]/IntComb.lim.sum.plat[1,2],2))
#paste0("The difference in medians: ",round(IntComb.lim.sum.plat[1,2]/IntComb.lim.sum.plat[2,2],2))

IntComb.lim.sum.raw <- IntComb %>% group_by(Platform,Raw.file) %>% summarize(Count = n())
IntComb.lim.sum.raw.plat <- IntComb.lim.sum.raw %>% ungroup() %>% group_by(Platform) %>% summarize(medCount = median(Count, na.rm = TRUE))


IntComb.lim.2 <- IntComb
IntComb.lim.2 <- IntComb.lim.2 %>% left_join(IntComb.lim.sum.plat, by = "Platform")
IntComb.lim.2 <- IntComb.lim.2 %>% left_join(IntComb.lim.sum.raw.plat, by = "Platform")


ContingencyTable <- as.data.frame(table(cut(log10(IntComb.lim.2$Intensity),30),IntComb.lim.2$Raw.file))
ContingencyTable$Platform <- ifelse(ContingencyTable$Var2 %in% ShotgunSet, "Shotgun", "Prioritized")
ContingencyTable$Start <- stringr::str_match(ContingencyTable$Var1, "\\(([^,]+)")[,2]
#ContingencyTable %>% filter(Var2 == "XwGH0643") %>% ggplot(aes(x = Start, y = Freq)) + geom_col()

ContingencyTable_sum <- ContingencyTable %>% group_by(Platform, Start) %>% summarize(meanFreq = mean(Freq, na.rm = TRUE), sdFreq = sd(Freq, na.rm = TRUE))

ContingencyTable_sum <- ContingencyTable_sum %>% left_join(IntComb.lim.sum.raw.plat, by = "Platform")
ContingencyTable_sum <- ContingencyTable_sum %>% left_join(IntComb.lim.sum.plat, by = "Platform")

ContingencyTable_sum %>% ggplot(aes(x = as.numeric(Start), y = meanFreq, color = Platform, group = Platform)) + geom_line(size = 2) + geom_point(size = 3, shape = 21, fill = "white", color = "grey30") + theme_light() + theme(legend.position = "null", strip.text = element_text(size = 18, color = "black"), strip.background = element_rect(fill = "white", color = "grey70"), axis.text = element_text(size =20), axis.title = element_text(size = 22),plot.margin = margin(0,0,0,0.5, "cm")) + scale_color_manual(values = c("red","blue")) + labs(x =bquote('Log'[10]~ 'Precursor Intensity'), y = "Precursors") +
 geom_errorbar(aes(ymin = (meanFreq - sdFreq), ymax = (meanFreq + sdFreq)), color = "grey40") + scale_x_continuous(limits = c(5, 9.5), breaks = seq(5, 9.5, by = 1)) 
ggsave(paste0(FigOut,"IntensityCurve_Rebuttal.pdf"), width = 7, height = 5, units = "in")
```

```{r Fig 2c}

ShotgunSet <- c("wGH0643","wGH0644","wGH0645","wGH0646","wGH0647","wGH0648","wGH0649","wGH0650")
pSCoPESet <- c("wGH0660","wGH0666","wGH0663","wGH0668","wGH0665","wGH0669","wGH0661","wGH0667")

# Missing data at single-cell level

Rebuttal_t0_both <- read.delim(paste0(datIn,"Fig2/Fig2_bcde_Consistency/t0_Rebuttal_Consistency_PIF50_mrri10.txt"))

Rebuttal_MP_both <- read.delim(paste0(datIn,"Fig2/Fig2_bcde_Consistency/Prot_missing_Rebuttal_Consistency_PIF50_mrri10.txt"))

Rebuttal_CG_both <- read.delim(paste0(datIn,"Fig2/Fig2_bcde_Consistency/ChanKey_Rebuttal_Consistency_PIF50_mrri10.txt"))

AllExps <- read.delim(paste0(datIn,"Fig2/Fig2_bcde_Consistency/ev_Rebuttal_Consistency_filtered_PIF50_mrri10_1pFDR.txt"))

AllExps$SeqCharge <- paste0(AllExps$Modified.sequence,AllExps$Charge)
AllExps$Raw.file <- str_sub(AllExps$Raw.file, start = 2)

# Full Evidence files for each experiment
evi.tier.s <- read.delim(paste0(datIn,"Fig2/Fig2_bcde_Consistency/Consistency_evidence_Shotgun_fullFASTA.txt"))
evi.tier.p <- read.delim(paste0(datIn,"Fig2/Fig2_bcde_Consistency/Consistency_evidence_pSCoPE_fullFASTA.txt"))

evi.tier.s <- evi.tier.s %>% select(-Acetyl..Protein.N.term.)

evi.tier <- rbind(evi.tier.s,evi.tier.p)
evi.tier.pep.prot <- evi.tier %>% dplyr::select(Modified.sequence,Leading.razor.protein) %>% group_by(Modified.sequence) %>% slice(1)
  
RefList <- read.delim(paste0(datIn,"Fig2/Fig2_bcde_Consistency/pSCoPE_FinalTarg_Consistency.txt"))

RefList_Top <- RefList %>% dplyr::filter(Priority == 3) %>% dplyr::select(SeqCharge)
RefList_Mid <- RefList %>% dplyr::filter(Priority == 2) %>% dplyr::select(SeqCharge)
RefList_Both <- RefList %>% dplyr::filter(Priority %in% c(2,3)) %>% dplyr::select(SeqCharge)
RefList_all <- RefList %>% dplyr::filter(Priority = TRUE) %>% dplyr::select(SeqCharge)

TargList <- read.delim(paste0(datIn,"Fig2/Fig2_bcde_Consistency/pSCoPE_FinalTarg_Consistency.txt"))


TargList_Top <- TargList %>% dplyr::filter(Priority %in% c(3)) %>% dplyr::select(SeqCharge)
TargList_Mid <- TargList %>% dplyr::filter(Priority %in% c(2)) %>% dplyr::select(SeqCharge)
TargList_Both <- TargList %>% dplyr::filter(Priority %in% c(3)) %>% dplyr::select(SeqCharge)
TargList_All <- TargList %>% dplyr::filter(TargBool = TRUE) %>% select(SeqCharge)

TargList_Both$pep <- str_sub(TargList_Both$SeqCharge, end = -2)
TargList_Both.pep <- as.data.frame(unique(TargList_Both$pep))
colnames(TargList_Both.pep) <- "Modified.sequence"

TargList_All$pep <- str_sub(TargList_All$SeqCharge, end = -2)
TargList_All.pep <- as.data.frame(unique(TargList_All$pep))
colnames(TargList_All.pep) <- "Modified.sequence"

# Getting cell identities
Rebuttal_CG_both.f <- Rebuttal_CG_both %>% dplyr::filter(celltype != "control", cvm < .4)
Rebuttal_CG_both.f$Raw.file <- str_sub(Rebuttal_CG_both.f$Raw.file, start = 2)
Rebuttal_CG_s.f <- Rebuttal_CG_both.f %>% dplyr::filter(Raw.file %in% ShotgunSet)
Rebuttal_CG_p.f <- Rebuttal_CG_both.f %>% dplyr::filter((Raw.file %in% pSCoPESet))

Rebuttal_Consistency_cellOrder <- Rebuttal_CG_s.f$id
Rebuttal_Consistency_cellOrder.p <- Rebuttal_CG_p.f$id

# Extracting single cells and high priority peptides from the t0 matrix
Rebuttal_t0_both.m <- melt(Rebuttal_t0_both)
Rebuttal_t0_both.m.f <- Rebuttal_t0_both.m %>% dplyr::filter(variable %in% Rebuttal_CG_both.f$id, SeqCharge %in% TargList_Both$SeqCharge)
Rebuttal_t0_both.m.f$pep <- str_sub(Rebuttal_t0_both.m.f$SeqCharge, end = -2) 

# Collapsing multiple charge states for the same peptide
Rebuttal_t0_both.m.f.pep <- Rebuttal_t0_both.m.f %>% dplyr::group_by(variable,pep) %>% dplyr::summarize(ValCollected = sum(value,na.rm = TRUE))
Rebuttal_t0_both.m.f.pep$ValCollected_na <- ifelse(Rebuttal_t0_both.m.f.pep$ValCollected == 0, NA,Rebuttal_t0_both.m.f.pep$ValCollected)
Rebuttal_t0_both.m.f.pep$Method <- ifelse(Rebuttal_t0_both.m.f.pep$variable %in% Rebuttal_CG_s.f$id, "Shotgun",ifelse(Rebuttal_t0_both.m.f.pep$variable %in% Rebuttal_CG_p.f$id,"Prioritized",NA))


TargList_Both.pep_in <- as.data.frame(TargList_Both.pep)
colnames(TargList_Both.pep_in) <- "pep"

for(i in 1:length(unique(Rebuttal_t0_both.m.f.pep$variable))){
  SingleCell <- Rebuttal_t0_both.m.f.pep %>% ungroup() %>% dplyr::filter(variable == unique(Rebuttal_t0_both.m.f.pep$variable)[i], pep %in% TargList_Both.pep$Modified.sequence) %>% dplyr::select(pep,ValCollected_na)
  colnames(SingleCell) <- c("pep",as.character(unique(Rebuttal_t0_both.m.f.pep$variable)[[i]]))
  #SingleExp_2 <- SingleExp %>% group_by(SeqCharge) %>% slice(1)
  TargList_Both.pep_in <- TargList_Both.pep_in %>% left_join(SingleCell, by = "pep")
}

TargList_Both.pep_in.m <- melt(TargList_Both.pep_in)
TargList_Both.pep_in.m$Valbin <- ifelse(is.na(TargList_Both.pep_in.m$value), 0, 1)

TargList_Both.m.pep_sums <- TargList_Both.pep_in.m %>% dplyr::group_by(pep) %>% dplyr::summarize(numCells = sum(Valbin, na.rm = TRUE)) %>% arrange(-numCells) %>% filter(numCells != 0)


TargList_Both.pep_in.m$Method <- ifelse(TargList_Both.pep_in.m$variable %in% Rebuttal_CG_s.f$id,"Shotgun",ifelse(TargList_Both.pep_in.m$variable %in% Rebuttal_CG_p.f$id,"Prioritized",NA)) 


RefList_Both.m3 <- TargList_Both.pep_in.m %>% left_join(TargList_Both.m.pep_sums, by = "pep")
RefList_Both.m3 <- RefList_Both.m3 %>% filter(!is.na(numCells))

Both_Tier_fig <- RefList_Both.m3 %>% ggplot(aes(x = factor(variable, levels = Rebuttal_Consistency_cellOrder), y = reorder(pep,-numCells), fill = factor(Valbin))) + geom_tile()  + facet_wrap(~factor(Method,levels = c("Shotgun","Prioritized"))) + theme(axis.text.y = element_blank(), axis.text.x = element_blank(),legend.position = "null", strip.text = element_text(size = 16),axis.text = element_text(size = 14), axis.title = element_text(size = 16), title = element_text(size =12)) + scale_fill_manual(values = c("white","red")) + labs(x = "Cells", y = "Peptides")


####
Rebuttal_Consistency_cellOrder_df <- as.data.frame(Rebuttal_Consistency_cellOrder)
Rebuttal_Consistency_cellOrder_df$index <- as.numeric(str_sub(Rebuttal_Consistency_cellOrder_df$Rebuttal_Consistency_cellOrder,start= 2))
Rebuttal_Consistency_cellOrder_df_arr <- Rebuttal_Consistency_cellOrder_df %>% arrange(index)
ConsistencyShotgunCellOrder <- Rebuttal_Consistency_cellOrder_df_arr$Rebuttal_Consistency_cellOrder
#integrating new data
ConsistencyPrioritizedCellOrder <- Rebuttal_Consistency_cellOrder.p

CellOrder <- c(ConsistencyShotgunCellOrder,ConsistencyPrioritizedCellOrder)
CellOrder.df <- as.data.frame(CellOrder)
indVec <- c(c(1:length(ConsistencyShotgunCellOrder)),c(1:length(ConsistencyPrioritizedCellOrder)))
#indVec <- ConsistencyShotgunCellOrder
CellOrder.df$Inds <- indVec


RefList_Both.m3$ValBin2 <- ifelse((RefList_Both.m3$Method == "Shotgun" & RefList_Both.m3$Valbin == 1),2,RefList_Both.m3$Valbin)
RefList_Both.m4 <- RefList_Both.m3 %>% left_join(CellOrder.df, by = c("variable"="CellOrder"))

cellHeatMap2 <- RefList_Both.m4 %>% ggplot(aes(x = Inds, y = reorder(pep,-numCells), fill = factor(ValBin2))) + geom_rect(aes(xmin = -Inf, xmax = Inf, ymin = -Inf, ymax = Inf), fill="white")+ geom_tile() + facet_wrap(~factor(Method,levels = c("Shotgun","Prioritized")), scales = "free_x")  + theme_missingData_tickChart() + theme(plot.title = element_text(vjust = 2), axis.title.y = element_text(margin = margin(t = 0, r = 30, b = 0, l = 0)), axis.text.x = element_text(size = 18, angle = 45, hjust = 1, vjust = 1)) + scale_fill_manual(values = c("white","red","blue")) + labs(x = "Single Cells", y = "Peptides") + scale_x_continuous(breaks = seq(20, 100, by = 20))
#+ ggtitle("High Priority Peptides")

cellHeatMap2
ggsave(paste0(FigOut,"TierHeatMap_Rebuttal_patchwork_fig_v4_noBlank.pdf"), width = 6.2 , height = 5, units = "in")
```

```{r Fig 2d}
scp_filt.ev <- read.delim(paste0(datIn,"Fig2/Fig2_bcde_Consistency/ev_Rebuttal_Consistency_filtered_PIF50_mrri10_1pFDR.txt"))

# Follow up on this later
scp_filt.ev_shotgun <- scp_filt.ev %>% dplyr::filter(Raw.file %in% paste0("X",ShotgunSet)) %>% dplyr::group_by(modseq) %>% dplyr::slice(1)
scp_filt.ev_pSCoPE <- scp_filt.ev %>% filter(Raw.file %in% paste0("X",pSCoPESet))  %>% group_by(modseq) %>% slice(1)
interPep_forConsistency <- intersect(scp_filt.ev_shotgun$modseq, scp_filt.ev_pSCoPE$modseq)

Rebuttal_CG_both$Raw.file <- str_sub(Rebuttal_CG_both$Raw.file, start = 2)
Rebuttal_CG_s <- Rebuttal_CG_both %>% filter(Raw.file %in% ShotgunSet, cvm < .4, celltype != "control")
Rebuttal_CG_p <- Rebuttal_CG_both %>% filter((Raw.file %in% pSCoPESet), cvm < .4, celltype != "control")

# Constraining comparison just to those peptides which passed the scp filtration metrics

TargList_Both_lim <- TargList_Both %>% dplyr::filter(SeqCharge %in% interPep_forConsistency)
TargList_All_lim <- TargList_All %>% dplyr::filter(SeqCharge %in% interPep_forConsistency)

Rebuttal_t0_both.m <- melt(Rebuttal_t0_both)
Rebuttal_t0_both.m$pep <- str_sub(Rebuttal_t0_both.m$SeqCharge, end = -2)

Rebuttal_t0_both.m_Top <- Rebuttal_t0_both.m %>% dplyr::filter(SeqCharge %in% TargList_Both_lim$SeqCharge)
Rebuttal_t0_both.m_All <- Rebuttal_t0_both.m %>% dplyr::filter(SeqCharge %in% TargList_All_lim$SeqCharge)

Rebuttal_t0_both.m_Top_sum <- Rebuttal_t0_both.m_Top %>% dplyr::group_by(variable,pep) %>% dplyr::summarize(valSum = sum(value, na.rm = TRUE))
Rebuttal_t0_both.m_All_sum <- Rebuttal_t0_both.m_All %>% dplyr::group_by(variable,pep) %>% dplyr::summarize(valSum = sum(value, na.rm = TRUE))
Rebuttal_t0_both.m_Top_sum$valSumNa <- ifelse(Rebuttal_t0_both.m_Top_sum$valSum == 0, NA, Rebuttal_t0_both.m_Top_sum$valSum)
Rebuttal_t0_both.m_All_sum$valSumNa <- ifelse(Rebuttal_t0_both.m_All_sum$valSum == 0, NA, Rebuttal_t0_both.m_All_sum$valSum)

Rebuttal_t0_both.m_Top_sum_MD <- Rebuttal_t0_both.m_Top_sum %>% dplyr::group_by(variable) %>% dplyr::summarize(FractionMD = sum(is.na(valSumNa))/n())
Rebuttal_t0_both.m_All_sum_MD <- Rebuttal_t0_both.m_All_sum %>% dplyr::group_by(variable) %>% dplyr::summarize(FractionMD = sum(is.na(valSumNa))/n())

Rebuttal_t0_both.m_Top_sum_MD$Method <- ifelse(Rebuttal_t0_both.m_Top_sum_MD$variable %in% Rebuttal_CG_s.f$id,"Shotgun",ifelse(Rebuttal_t0_both.m_Top_sum_MD$variable %in% Rebuttal_CG_p.f$id,"Prioritized","Omit"))
Rebuttal_t0_both.m_All_sum_MD$Method <- ifelse(Rebuttal_t0_both.m_All_sum_MD$variable %in% Rebuttal_CG_s.f$id,"Shotgun",ifelse(Rebuttal_t0_both.m_All_sum_MD$variable %in% Rebuttal_CG_p.f$id,"Prioritized","Omit"))

Rebuttal_t0_both.m_Top_sum_MD$Tier <- "High Priority"
Rebuttal_t0_both.m_All_sum_MD$Tier <- "All Levels"

Rebuttal_t0_both.m_join <- rbind(Rebuttal_t0_both.m_Top_sum_MD,Rebuttal_t0_both.m_All_sum_MD)
Rebuttal_t0_both.m_join$percent <- Rebuttal_t0_both.m_join$FractionMD*100

Rebuttal_t0_both.m_join$percentComplete <- 100 - Rebuttal_t0_both.m_join$percent
  
PeptideMD <- Rebuttal_t0_both.m_join %>% ggplot(aes(percentComplete, color = factor(Method, levels = c("Shotgun","Prioritized")))) + geom_boxplot(alpha = 0.5,lwd=1) + coord_flip(xlim = c(0,100)) + labs(color = "", x = "") + facet_wrap(~factor(Tier, levels = c("High Priority","All Levels"))) + theme_box() + theme(plot.title = element_text(size = 24,hjust = 0.5, vjust = 2) , axis.title.y = element_text(size = 22,margin = margin(t = 0, r = 20, b = 0, l = 0)), strip.text = element_text(size = 20),panel.grid.minor.x = element_blank(),panel.grid.major.x = element_blank(),plot.margin = unit(c(0,0,0,0), "cm"),axis.text = element_text(color = "grey10",hjust = 1))  + scale_color_manual(values = c("blue","red")) + labs(x="% Data Completeness",y = "")  + ggtitle("Peptides") + scale_x_continuous(limits = c(0, 100), breaks = seq(0, 100, by = 20)) + geom_hline(aes(yintercept = 0), color = "grey90")

Rebuttal_t0_both.m_join_sum <- Rebuttal_t0_both.m_join %>% group_by(Method) %>% summarize(Count = n())

Fig1d_PeptideConsistency_stats <- Rebuttal_t0_both.m_join %>% group_by(Method,Tier) %>% summarize(medianVal = median(percent,na.rm = TRUE), medianComplete = median(percentComplete, na.rm = TRUE))


Rebuttal_CG_s <- Rebuttal_CG_both %>% filter(Raw.file %in% ShotgunSet, cvm < .4, celltype != "control")
Rebuttal_CG_p <- Rebuttal_CG_both %>% filter((Raw.file %in% pSCoPESet), cvm < .4, celltype != "control")

# Combining targeting list with precursor-to-protein dictionary
TargList_Both.pepProt <- TargList_Both.pep %>% left_join(evi.tier.pep.prot, by = "Modified.sequence")
TargList_All.pepProt <- TargList_All.pep %>% left_join(evi.tier.pep.prot, by = "Modified.sequence")

# Same intersected Precursors as previous
interPep_forConsistency.df <- as.data.frame(interPep_forConsistency)
colnames(interPep_forConsistency.df) <- "SeqCharge"
interPep_forConsistency.df$Pep <- str_sub(interPep_forConsistency.df$SeqCharge, end = -2)

#-------------------
#
#-------------------

scp_filt.ev_shotgunProt <- scp_filt.ev %>% filter(Raw.file %in% paste0("X",ShotgunSet)) %>% group_by(Leading.razor.protein) %>% slice(1)
scp_filt.ev_pSCoPEProt <- scp_filt.ev %>% filter(Raw.file %in% paste0("X",pSCoPESet))  %>% group_by(Leading.razor.protein) %>% slice(1)
interProt_forConsistency <- intersect(scp_filt.ev_shotgunProt$Leading.razor.protein, scp_filt.ev_pSCoPEProt$Leading.razor.protein)

#-------------------
ProtMetric <- "Proteins"
#ProtMetric <- "Sequence"

if(ProtMetric == "Sequence"){  
TargList_Both.pepProt.lim <- TargList_Both.pepProt %>% filter(Modified.sequence %in% interPep_forConsistency.df$Pep)
TargList_All.pepProt.lim <- TargList_All.pepProt %>% filter(Modified.sequence %in% interPep_forConsistency.df$Pep)
}

if(ProtMetric == "Proteins"){  
TargList_Both.pepProt.lim <- TargList_Both.pepProt %>% filter(Leading.razor.protein %in% interProt_forConsistency)
TargList_All.pepProt.lim <- TargList_All.pepProt %>% filter(Leading.razor.protein %in% interProt_forConsistency)
}

#Left join dataframe of single cells to this with Prot-level info
Rebuttal_MP_both.m <- melt(Rebuttal_MP_both)
Rebuttal_MP_both.m$Method <- ifelse(Rebuttal_MP_both.m$variable %in% Rebuttal_CG_s$id, "Shotgun",ifelse(Rebuttal_MP_both.m$variable %in% Rebuttal_CG_p$id,"Prioritization",NA))

Rebuttal_MP_both.m_Top <- Rebuttal_MP_both.m %>% dplyr::filter(Proteins %in% TargList_Both.pepProt.lim$Leading.razor.protein)
Rebuttal_MP_both.m_All <- Rebuttal_MP_both.m %>% dplyr::filter(Proteins %in% TargList_All.pepProt.lim$Leading.razor.protein)

Rebuttal_MP_both.m_Top_sum <- Rebuttal_MP_both.m_Top %>% group_by(variable,Proteins) %>% summarize(ProtQuant = sum(value, na.rm = TRUE))
Rebuttal_MP_both.m_Top_sum$ProtQuantna <- ifelse(Rebuttal_MP_both.m_Top_sum$ProtQuant==0,NA,Rebuttal_MP_both.m_Top_sum$ProtQuant)

Rebuttal_MP_both.m_All_sum <- Rebuttal_MP_both.m_All %>% group_by(variable,Proteins) %>% summarize(ProtQuant = sum(value, na.rm = TRUE))
Rebuttal_MP_both.m_All_sum$ProtQuantna <- ifelse(Rebuttal_MP_both.m_All_sum$ProtQuant==0,NA,Rebuttal_MP_both.m_All_sum$ProtQuant)


Rebuttal_MP_both.m_Top_sum_missing <- Rebuttal_MP_both.m_Top_sum %>% dplyr::group_by(variable) %>% dplyr::summarize(fracMissing = sum(is.na(ProtQuantna))/n())
Rebuttal_MP_both.m_Top_sum_missing$Method <- ifelse(Rebuttal_MP_both.m_Top_sum_missing$variable %in% Rebuttal_CG_s$id, "Shotgun","Prioritized")
Rebuttal_MP_both.m_Top_sum_missing$Tiers <- "High Priority"

Rebuttal_MP_both.m_All_sum_missing <- Rebuttal_MP_both.m_All_sum %>% dplyr::group_by(variable) %>% dplyr::summarize(fracMissing = sum(is.na(ProtQuantna))/n())
Rebuttal_MP_both.m_All_sum_missing$Method <- ifelse(Rebuttal_MP_both.m_All_sum_missing$variable %in% Rebuttal_CG_s$id, "Shotgun",ifelse(Rebuttal_MP_both.m_All_sum_missing$variable %in% Rebuttal_CG_p$id,"Prioritized",NA))
Rebuttal_MP_both.m_All_sum_missing$Tiers <- "All Levels"

Rebuttal_MP_both.m_combTiers <- rbind(Rebuttal_MP_both.m_Top_sum_missing,Rebuttal_MP_both.m_All_sum_missing)
Rebuttal_MP_both.m_combTiers$percent <- 100*Rebuttal_MP_both.m_combTiers$fracMissing

Rebuttal_MP_both.m_combTiers$percentComplete <- 100 - Rebuttal_MP_both.m_combTiers$percent


ProtMD_Box <- Rebuttal_MP_both.m_combTiers %>% filter(Tiers == "All Levels") %>% ggplot(aes(percentComplete, color = factor(Method, levels = c("Shotgun","Prioritized")))) + geom_boxplot(lwd=1) + coord_flip(xlim = c(0, 100)) + theme_light() + labs(color = "", x = "", y = "") + theme_box() + theme(axis.text.y = element_blank(), strip.text = element_text(size = 20),panel.grid.minor.x = element_blank(),panel.grid.major.x = element_blank(),plot.margin = unit(c(0,0,0,0), "cm"),plot.title = element_text(size = 24,hjust = 0.5, vjust = 2)) + facet_wrap(~factor(Tiers, levels = c("High Priority","All Levels")), scales = "free_x")+ scale_color_manual(values = c("blue","red")) + ggtitle("Proteins")  + scale_x_continuous(limits = c(0, 100), breaks = seq(0, 100, by = 20)) + geom_hline(aes(yintercept = 0), color = "grey90")
#ggsave("FractionMD.png", width = 7, height = 5, units = "in")

Rebuttal_MP_both.m_combTiers_sum <- Rebuttal_MP_both.m_combTiers %>% group_by(Method) %>% summarize(count = n())

PeptideMD + ProtMD_Box  + plot_layout(widths = c(2, 1))
#ggsave("C:/My_Drive/MS/SCoPE/t-SCoPE/code/pSCoPE_pubFigs/Figs/TechnicalFig1/Pep_ProtMD_consistency_Rebuttal_fig_v3.pdf", width = 7 , height = 5, units = "in")
ggsave(paste0(FigOut,"Pep_ProtMD_consistency_Rebuttal_fig_v4_completeness.pdf"), width = 7.5 , height = 5.1, units = "in")
```

```{r Fig 2e: HEK and Melanoma PCA}
BC_data <- read.csv(paste0(datIn,"Fig2/Fig2_bcde_consist_pSCoPEOnly/limmaCorrected_normed_prePCA_Rebuttal_Consistency_pSCoPE_PIF50_mrri10.csv"))
CG <- read.delim(paste0(datIn,"Fig2/Fig2_bcde_consist_pSCoPEOnly/ChanKey_Rebuttal_Consistency_pSCoPE_PIF50_mrri10.txt"))

CG$cellLab <- ifelse(CG$celltype == "M","Melanoma",ifelse(CG$celltype == "H","HEK","control"))
pSCoPE_consist <- c("XwGH0660","XwGH0666","XwGH0663","XwGH0668","XwGH0665","XwGH0669","XwGH0661","XwGH0667")
CG_prioritized <- CG %>% filter(Raw.file %in% pSCoPE_consist)


row.names(BC_data) <- BC_data$X
BC_data_preMat <- BC_data %>% select(-X)
BC_data_preMat <- BC_data %>% select(any_of(CG_prioritized$id))

BC_data.mat <- as.matrix(BC_data_preMat)
BC_data.mat <- cr_norm_log(BC_data.mat)
melHEK_PC_out <- prcomp(t(BC_data.mat), scale = TRUE, center = TRUE)
melHEK_PC_coord <- as.data.frame(melHEK_PC_out$x[,1:6])
melHEK_PC_weights <- as.data.frame(melHEK_PC_out$rotation[,1:2])
melHEK_PC_weights$Proteins <- row.names(melHEK_PC_weights)

PCAsummary<- summary(melHEK_PC_out)
PC1_fracVar <- PCAsummary$importance[2,1]
PC2_fracVar <- PCAsummary$importance[2,2]
melHEK_PC_coord$id <- row.names(melHEK_PC_coord)
melHEK_PC_coord.anno <- melHEK_PC_coord %>% left_join(CG, by = "id")
Okabe_Ito <- c("#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7", "#000000")

cellType_melHEK_PCA <- melHEK_PC_coord.anno %>% ggplot(aes(x = PC1, y = PC2, color = cellLab)) + geom_point(shape =16,size = 5, alpha = 0.50) + theme_light() +labs(color = "", x = paste0("PC1 (",round(PC1_fracVar,2)*100,"%)"), y = paste0("PC2 (",round(PC2_fracVar,2)*100,"%)")) + scale_color_manual(values = c(Okabe_Ito[1],Okabe_Ito[2],Okabe_Ito[3])) + 
  geom_text(aes(x = 18, y = 25),size = 8,color = Okabe_Ito[2],label = "Melanoma") + 
  geom_text(aes(x = -14, y = 25),size = 8,color = Okabe_Ito[1],label = "HEK") + theme(legend.position = "null",axis.title = element_text(size = 22), axis.text = element_text(size = 20)) 

GOPath <- paste0(datIn,"BMDM_figs_4_5_6/")
GO_DB_Path <- paste0(GOPath,"GOA_db.txt")
GO_db <- read.delim(GO_DB_Path, sep = "", header = TRUE)
GO_db.2 <- GO_db %>% filter(Letter != "C")
GO_db.2 <- GO_db 

ProtMissing <- read.delim(paste0(datIn,"Fig2/Fig2_bcde_consist_pSCoPEOnly/Prot_missing_Rebuttal_Consistency_pSCoPE_PIF50_mrri10.txt"))
BC_dat <- read.csv(paste0(datIn,"Fig2/Fig2_bcde_consist_pSCoPEOnly/limmaCorrected_normed_prePCA_Rebuttal_Consistency_pSCoPE_PIF50_mrri10.csv"))

row.names(BC_data) <- BC_data$X
BC_data_preMat <- BC_data %>% select(-X)
BC_data_preMat <- BC_data %>% select(any_of(CG_prioritized$id))

BC_data.mat <- as.matrix(BC_data_preMat)
BC_data.mat <- cr_norm_log(BC_data.mat)
BC_data.df <- as.data.frame(BC_data.mat)
BC_data.df$X <- row.names(BC_data.df)
ProtMissing.m <- melt(ProtMissing)
BC_dat.m <- melt(BC_data.df, by = "X")
colnames(BC_dat.m) <- c("Proteins","variable","value")
BC_dat.m <- BC_dat.m %>% left_join(ProtMissing.m, by = c("Proteins", "variable"))
BC_dat.m$newVar <- ifelse(is.na(BC_dat.m$value.y),NA,BC_dat.m$value.x)
BC_dat.m2 <- BC_dat.m %>% select(Proteins,variable,newVar)


library(tidyr)



#test <- PCA_PSEA_PDAC(PDAC_PC_weights[,c(1,7)],PDAC_ProtGeneMap,GO_db.2)
melHEK_PSEA <- PCA_PSEA_Rebuttal_prot(melHEK_PC_weights,GO_db.2)
#PDAC_PSEA <- test
melHEK_PSEA.f <- melHEK_PSEA %>% filter(!is.na(pVal),PC <= 2)
melHEK_PSEA.f$qVal <- p.adjust(melHEK_PSEA.f$pVal, method = "fdr")
melHEK_PSEA.f_fdr <- melHEK_PSEA.f %>% filter((qVal <= .05))
melHEK_PSEA.f_fdr$neglogq <- -log10(melHEK_PSEA.f_fdr$qVal)
melHEK_PSEA.f_fdr$neglogq_bin <- ifelse(melHEK_PSEA.f_fdr$neglogq >=3,3,melHEK_PSEA.f_fdr$neglogq)

#Save DF here
#write.csv(melHEK_PSEA,paste0(datOut,"HEK_Mel_GO_test.txt"), row.names = FALSE, quote = FALSE) 
#melHEK_PSEA <- read.csv(paste0(datOut,"HEK_Mel_GO_test.txt"))


# Dot plot for data exploration
#melHEK_PSEA.f_fdr %>% ggplot(aes(x = PC, y = str_wrap(Process, width = 50),fill = round(Association,1), size = neglogq_bin)) + geom_point(shape = 21, stroke = 1) + scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0, breaks = c(-1,0,1)) + theme_light() + labs(fill = "Loading\nz-score",x = "Principal component",y = "", size = bquote("-log"[10]~"(q val)")) + theme(axis.text.y = element_text(size = 16),axis.text.x = element_text(size = 26),axis.title = element_text(size = 30), legend.title = element_text(size = 20), legend.text = element_text(size = 15), legend.position = "bottom", legend.box="vertical", legend.margin=margin()) +scale_size_continuous(breaks = seq(2,3,1),range=c(1,15),limits = c(0, 4), labels = c("2",">3")) +  scale_x_discrete(limits = c("1","2"), expand=c(0.1, 0)) 


PSEA_Terms_to_keep <- c("gluconeogenesis","glycolysis", "G2/M transition of mitotic cell cycle","respiratory electron transport chain", "regulation of ubiquitin-protein ligase activity involved in mitotic cell cycle", "protein N-linked glycosylation via asparagine","de novo' posttranslational protein folding")


for (i in 1:length(PSEA_Terms_to_keep)){
  GO_db_lim <- GO_db %>% filter(GO_term_name %in% PSEA_Terms_to_keep[i])
  BC_dat.m2_lim <- BC_dat.m2 %>% filter(Proteins %in% GO_db_lim$Uniprot)
  BC_dat.m2_lim_sum <- BC_dat.m2_lim %>% dplyr::group_by(variable) %>% dplyr::summarize(medAbund = median(newVar, na.rm = TRUE))
  BC_dat.m2_lim_sum$medAbund <- (BC_dat.m2_lim_sum$medAbund-mean(BC_dat.m2_lim_sum$medAbund, na.rm=TRUE))/sd(BC_dat.m2_lim_sum$medAbund, na.rm=TRUE)
  BC_dat.m2_lim_sum$medAbund <- ifelse(BC_dat.m2_lim_sum$medAbund >= 2,2,ifelse(BC_dat.m2_lim_sum$medAbund <= -2,-2,BC_dat.m2_lim_sum$medAbund))
  colnames(BC_dat.m2_lim_sum) <- c("id",as.character(PSEA_Terms_to_keep[[i]]))
  if(i == 1){
    GSEA_PC_out <- BC_dat.m2_lim_sum
  }
  if(i > 1){
    GSEA_PC_out <- GSEA_PC_out %>% left_join(BC_dat.m2_lim_sum, by = "id")
  }
}

melHEK_PC_coord2.pSCoPE <- melHEK_PC_coord.anno %>% left_join(GSEA_PC_out, by = "id")

Glycolysis_melHEK <- melHEK_PC_coord2.pSCoPE %>% ggplot(aes(x = PC1,y = PC2, color = `glycolysis`)) + geom_point(shape =16,size = 5, alpha = 0.75) + theme_light() +labs(color = "") + scale_color_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0) + labs(color = "") + theme(axis.title = element_blank(),axis.text = element_blank(), legend.position = "null") + ggtitle("Glycolysis")

G2_melHEK <- melHEK_PC_coord2.pSCoPE %>% ggplot(aes(x = PC1,y = PC2, color = `G2/M transition of mitotic cell cycle`)) + geom_point(shape =16,size = 5, alpha = 0.75) + theme_light() +labs(color = "") + scale_color_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0) + labs(color = "") + theme(axis.title = element_blank(),axis.text = element_blank(), legend.position = "null") + ggtitle("G2/M transition of mitotic cell cycle")

Glycolysis_melHEK_legend <- melHEK_PC_coord2.pSCoPE %>% ggplot(aes(x = PC1,y = PC2, color = `glycolysis`)) + geom_point(shape =16,size = 5, alpha = 0.75) + theme_light() +labs(color = "") + scale_color_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0, breaks=c(-2,0,2)) + labs(color = "") + theme(axis.title = element_blank(),axis.text = element_blank(), legend.position = "bottom", legend.text = element_text(size = 14)) + ggtitle("Glycolysis") 

G2_melHEK_legend <- melHEK_PC_coord2.pSCoPE %>% ggplot(aes(x = PC1,y = PC2, color = `G2/M transition of mitotic cell cycle`)) + geom_point(shape =16,size = 5, alpha = 0.75) + theme_light() +labs(color = "") + scale_color_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0) + labs(color = "") + theme(axis.title = element_blank(),axis.text = element_blank(), legend.position = "bottom") + ggtitle("G2/M transition of mitotic cell cycle")


cellType_melHEK_PCA +  Glycolysis_melHEK + G2_melHEK + plot_layout(nrow = 1)
ggsave(paste0(FigOut,"HEK_Mel_PCA_composite_v2.pdf"), width = 16, height = 5.33, units = "in") 

Glycolysis_melHEK_legend
ggsave(paste0(FigOut,"HEK_Mel_PCA_forLegend.pdf"), width = 5.33, height = 5.33, units = "in") 
G2_melHEK_legend

```

# Figure 3: Spike-in quantitation evaluation

```{r Fig 3a: Reporter-ion distribution plot}
# Spike-in peptide sets
SI <- read.delim(paste0(datIn,"Fig3/evidence.txt"))

sc_chan<- c("Reporter.intensity.corrected.5","Reporter.intensity.corrected.6","Reporter.intensity.corrected.7","Reporter.intensity.corrected.8","Reporter.intensity.corrected.9","Reporter.intensity.corrected.10","Reporter.intensity.corrected.11","Reporter.intensity.corrected.12","Reporter.intensity.corrected.13","Reporter.intensity.corrected.14","Reporter.intensity.corrected.15","Reporter.intensity.corrected.16","Reporter.intensity.corrected.17","Reporter.intensity.corrected.18")

StandardFill <- c("XwGH0760","XwGH0763","XwGH0766","XwGH0769","XwGH0772","XwGH0775","XwGH0778","XwGH0781")

SI$SeqCharge <- paste0(SI$Modified.sequence,SI$Charge)
SI <- Empirical_FDR_PEP(SI)
SI.seqCharge <- SI %>% filter(qVal <= .01,Type != "MSMS") %>% group_by(SeqCharge) %>% top_n(-1, PEP) %>% slice(1)
length(unique(SI.seqCharge$SeqCharge))
SI.seqCharge.SI <- SI.seqCharge %>% filter(grepl("Spike",Leading.razor.protein)) %>% arrange(-Intensity)
SI.seqCharge.SI$SeqChargelim <- str_sub(SI.seqCharge.SI$SeqCharge, start = 2)

colorDict <- data.frame(Sequence = character(), color = character(), stringsAsFactors = FALSE)
colorDict[1:4,1] <- c(unique(SI.seqCharge.SI$Sequence))
colorDict[1:4,2] <- c(palette.colors(palette = "Okabe-Ito")[2],palette.colors(palette = "Okabe-Ito")[3],palette.colors(palette = "Okabe-Ito")[4],palette.colors(palette = "Okabe-Ito")[7])

SI.seqCharge.SI <- SI.seqCharge.SI %>% left_join(colorDict, by = "Sequence")


evi.tier.p.spikePre <- SI %>% filter((Type != "MSMS"), PIF >= .5, PEP <= .02) %>% mutate(SeqCharge = paste0(Modified.sequence, Charge)) %>% mutate(Type = ifelse(grepl("Spike",Leading.razor.protein),"Spike-in","Endogenous")) %>% select(Raw.file,SeqCharge,sc_chan,Type)

evi.tier.p.spikePre$Raw.file <- paste0("X",evi.tier.p.spikePre$Raw.file)

evi.tier.p.spikePre.m <- melt(evi.tier.p.spikePre, id.vars = c("Raw.file","SeqCharge","Type"))
#evi.tier.p.spikePre.m$Type <-ifelse(grepl("Spike",evi.tier.p.spikePre.m$Leading.razor.protein),"Spike","Endogenous")

evi.tier.p.spikePre.m$ExpGroup <- ifelse(evi.tier.p.spikePre.m$Raw.file %in% StandardFill, "Standard","other")

RImed <- evi.tier.p.spikePre.m %>% group_by(Type,ExpGroup) %>% summarize(medianVal = median(log10(value),na.rm =TRUE))

RImed.standard <- RImed %>% filter(ExpGroup == "Standard")


evi.tier.p.spikePre.m_Standard <- evi.tier.p.spikePre.m %>% filter(ExpGroup == "Standard")

evi.tier.p.spikePre.m_Standard$log10RI <- log10(evi.tier.p.spikePre.m_Standard$value)
evi.tier.p.spikePre.m_Standard$log10RIadj <- ifelse(!is.finite(evi.tier.p.spikePre.m_Standard$log10RI),NA,evi.tier.p.spikePre.m_Standard$log10RI)
evi.tier.p.spikePre.m_Standard$log10RIadj_missing <- ifelse(!is.finite(evi.tier.p.spikePre.m_Standard$log10RI),-1,evi.tier.p.spikePre.m_Standard$log10RI)

ContingencyTable.RI <- as.data.frame(table(cut((evi.tier.p.spikePre.m_Standard$log10RIadj),30),evi.tier.p.spikePre.m_Standard$Type))
#ContingencyTable.RI$Platform <- ifelse(RImed.standard$ %in% ShotgunSet, "Shotgun", "Prioritized")
ContingencyTable.RI$Start <- stringr::str_match(ContingencyTable.RI$Var1, "\\(([^,]+)")[,2]

#ContingencyTable.RI %>% filter(Var2 == "XwGH0643") %>% ggplot(aes(x = Start, y = Freq)) + geom_col() + coord_cartesian(ylim = c(1,6))

#out <- table(cut((evi.tier.p.spikePre.m$log10RIadj,30)))

ContingencyTable.RIneg <- ContingencyTable.RI
ContingencyTable.RIneg$Freq <- -ContingencyTable.RIneg$Freq

ContingencyFull <- rbind(ContingencyTable.RI,ContingencyTable.RIneg)
ContingencyFull$StartNum <- as.numeric(ContingencyFull$Start)
ContingencyFull$FacetLab <- ifelse(ContingencyFull$Var2 == "Endogenous","Single cell","Spike in")

ContingencyFull %>% ggplot(aes(y = Freq,x = StartNum)) + geom_col() + coord_flip() + facet_wrap(~FacetLab, scales="free_x") + theme_light() + labs(x = expression("Log"[10]~"(Reporter-Ion Intensity)"), y = "Fraction of Identified Peptides") + scale_x_continuous(limits = c(1, 6), breaks = seq(1, 6, by = 1)) + theme(axis.title.x = element_text(size =22),axis.title.y = element_text(size =23), axis.text = element_text(size = 20), axis.text.x = element_blank(), strip.background = element_rect( fill = "grey95", color = "grey70"), strip.text = element_text(color= "black", size = 22,face = "bold"))
ggsave(paste0(FigOut,"SpikeInRII_violin_v2.pdf"), width = 7, height = 5, units = "in")
```

```{r Fig 3b: Spike-in peptide quantitation}
sc_chan<- c("Reporter.intensity.corrected.5","Reporter.intensity.corrected.6","Reporter.intensity.corrected.7","Reporter.intensity.corrected.8","Reporter.intensity.corrected.9","Reporter.intensity.corrected.10","Reporter.intensity.corrected.11","Reporter.intensity.corrected.12","Reporter.intensity.corrected.13","Reporter.intensity.corrected.14","Reporter.intensity.corrected.15","Reporter.intensity.corrected.16","Reporter.intensity.corrected.17","Reporter.intensity.corrected.18")

sc_chanLoop <- c("Reporter.intensity.corrected.5","Reporter.intensity.corrected.6","Reporter.intensity.corrected.7","Reporter.intensity.corrected.8","Reporter.intensity.corrected.9","Reporter.intensity.corrected.10","Reporter.intensity.corrected.11","Reporter.intensity.corrected.12","Reporter.intensity.corrected.13","Reporter.intensity.corrected.14","Reporter.intensity.corrected.15","Reporter.intensity.corrected.16","Reporter.intensity.corrected.17","Reporter.intensity.corrected.18","Reporter.intensity.corrected.5","Reporter.intensity.corrected.6","Reporter.intensity.corrected.7","Reporter.intensity.corrected.8","Reporter.intensity.corrected.9","Reporter.intensity.corrected.10","Reporter.intensity.corrected.11","Reporter.intensity.corrected.12","Reporter.intensity.corrected.13","Reporter.intensity.corrected.14","Reporter.intensity.corrected.15","Reporter.intensity.corrected.16","Reporter.intensity.corrected.17","Reporter.intensity.corrected.18")


#Rebuttal_CG_both <- read.csv(paste0(datIn,"Fig3/Annotation_for_scp.csv"))
#Rebuttal_CG_both.m <- melt(Rebuttal_CG_both, id.vars = "Set")
#Rebuttal_CG_both.m$Set <- paste0("X",Rebuttal_CG_both.m$Set)
#Rebuttal_CG_both.m$RI <- gsub("RI","Reporter.intensity.corrected.",Rebuttal_CG_both.m$variable)
#Rebuttal_CG_both.m.lim <- Rebuttal_CG_both.m %>% filter(value %in% c("H","M"))

evi.tier.p.spike <- SI %>% filter(grepl("Spike",Leading.razor.protein)&(Type != "MSMS"), PIF >= .5, PEP <= .02) %>% mutate(SeqCharge = paste0(Modified.sequence, Charge)) %>% select(Raw.file,SeqCharge,sc_chan)
evi.tier.p.spike$Raw.file <- paste0("X",evi.tier.p.spike$Raw.file)

evi.tier.p.spike.m <- melt(evi.tier.p.spike)

#New search pattern
patternIndStartRef <- c("Reporter.intensity.corrected.13","Reporter.intensity.corrected.5","Reporter.intensity.corrected.16","Reporter.intensity.corrected.13","Reporter.intensity.corrected.5","Reporter.intensity.corrected.16","Reporter.intensity.corrected.13","Reporter.intensity.corrected.5")

SpikeInLabs <- c("SeqCharge","4x.1","4x.2","4x.3","2x.1","2x.2","2x.3","1x.1","1x.2","1x.3",".5x.1",".5x.2",".5x.3",".25x.1",".25x.2")


model_outputs <- data.frame(Raw.file = character(), intercept = numeric(), slope = numeric(), rSq = numeric(), stringsAsFactors = FALSE)

#Rebuttal_CG_both

for(i in 1:length(unique(evi.tier.p.spike$Raw.file))){
  
  startInd <- which(sc_chan %>% match(patternIndStartRef[i])==1)
  
  labelOrder <- sc_chanLoop[startInd:(startInd+13)]
  
  evi_consist.single <- evi.tier.p.spike %>% filter(Raw.file ==unique(evi.tier.p.spike$Raw.file)[i]) %>% select(SeqCharge,labelOrder)
  

  RI_input_dictionary <- as.data.frame(colnames(evi_consist.single))
  colnames(RI_input_dictionary) <- "Channel"
  RI_input_dictionary$input <- SpikeInLabs
  RI_input_dictionary.lim <- RI_input_dictionary %>% filter(Channel != "SeqCharge")
  colnames(evi_consist.single) <- SpikeInLabs
  evi_consist.single.m <- melt(evi_consist.single)
  
  evi_consist.single.m$SpikeInQuantity <- ifelse(grepl("4x",evi_consist.single.m$variable),4,ifelse(grepl("2x",evi_consist.single.m$variable),2,ifelse(grepl("1x",evi_consist.single.m$variable),1,ifelse(grepl(".25x",evi_consist.single.m$variable),0.25,.5))))
  evi_consist.single.m$logConcentration <- log2(evi_consist.single.m$SpikeInQuantity)
  evi_consist.single.m$logInt <- log2(evi_consist.single.m$value)
  evi_consist.single.m <- evi_consist.single.m %>% left_join(RI_input_dictionary.lim, by = c("variable" = "input"))
  
  pepMed <- evi_consist.single.m %>% group_by(SeqCharge, SpikeInQuantity) %>% summarize(medQuant = median(logInt, na.rm = TRUE)) %>% filter(SpikeInQuantity == 1) %>% ungroup() %>% select(SeqCharge, medQuant)
  
  evi_consist.single.m <- evi_consist.single.m %>% left_join(pepMed, by = "SeqCharge")
  evi_consist.single.m$normInt <- evi_consist.single.m$logInt - evi_consist.single.m$medQuant
  
  evi_consist.single.m$Raw.file <- unique(evi.tier.p.spike$Raw.file)[i]
  
  if(i == 1){
    evi_out <- evi_consist.single.m
  }
  if(i > 1){
    evi_out <- rbind(evi_out,evi_consist.single.m)
  }
  
}


StandardFill <- c("XwGH0760","XwGH0763","XwGH0766","XwGH0769","XwGH0772","XwGH0775","XwGH0778","XwGH0781")

evi_out$logInt.cor <- ifelse(!is.finite(evi_out$normInt),NA,evi_out$normInt)

evi_out.StandardFill <- evi_out %>% filter(Raw.file %in% StandardFill)

  
  modelFull.StandardFill = lm(logInt.cor ~ logConcentration + 0, dat = evi_out.StandardFill)

  #interceptFull <- round(summary(modelFull)[[4]][1],2)
  interceptFull.StandardFill <- 0
  slopeFull.StandardFill <- round(summary(modelFull.StandardFill)[[4]][1],2)
  rSqFull.StandardFill <- round(summary(modelFull.StandardFill)[[8]],2)
  
  
evi_out_sum <- evi_out %>% filter(Raw.file %in% StandardFill) %>% mutate(Sequence = str_sub(SeqCharge, -2)) %>% group_by(Sequence, Raw.file, variable) %>% summarize(medInt = median(logInt.cor,na.rm = TRUE), medConc = median(logConcentration, na.rm = TRUE))

#Setting the 0th amount to the lowest spike-in concentration, rather than using the midpoint as the reference value 
evi_out_sum$medInt_rev <- evi_out_sum$medInt + 2
evi_out_sum$medConc_rev <- evi_out_sum$medConc + 2

modelFull.StandardFill = lm(medInt_rev ~ medConc_rev + 0, dat = evi_out_sum)
  interceptFull.StandardFill <- 0
  slopeFull.StandardFill <- round(summary(modelFull.StandardFill)[[4]][1],2)
  rSqFull.StandardFill <- round(summary(modelFull.StandardFill)[[8]],2)

  
evi_out_sumsum <- evi_out_sum %>% group_by(medConc_rev) %>% summarize(medVal = median(medInt_rev, na.rm=TRUE), sdVal = sd(medInt_rev,na.rm = TRUE))   
  
evi_out_sum_count <- evi_out_sum %>% group_by(medConc_rev) %>% summarize(countVal = n())

evi_out_sumsum %>% ggplot(aes(x = medConc_rev , y = medVal)) + geom_point(size = 6, shape = 21, fill = "grey50") + geom_errorbar(aes(x =medConc_rev,ymin = medVal - sdVal,ymax = medVal + sdVal), height = 0, width = 0.3)  + geom_abline(intercept = interceptFull.StandardFill, slope =slopeFull.StandardFill) + geom_text(x = 4, y = 0, label = paste0("Slope = ",round(slopeFull.StandardFill,2)), size = 8) + 
  geom_text(x = 3.24, y = -.7, label = bquote("R^2"),size = 8, parse = TRUE) + geom_text(x = 3.95, y = -.75, label = paste0("= ",rSqFull.StandardFill), size = 8) + 
  theme_light() + 
  labs(x = expression("Log"[2]~"(spike-in amount)"), y = expression("Log"[2]~"(measured amount)")) + theme(axis.title = element_text(size = 30), axis.text = element_text(size = 27))+ 
  scale_x_continuous(limits = c(-1, 5), breaks = seq(0, 4, by = 1)) +
  scale_y_continuous(limits = c(-1, 5), breaks = seq(0, 4, by = 1)) +
  geom_text(data = evi_out_sum_count, aes(x = medConc_rev, y = 5, label = paste0("n=",countVal)), size = 7)
  #coord_cartesian(ylim = c(0,6))
ggsave(paste0(FigOut,"RegressionAllData_RISeqComb_StandardFill_MedandPoints_v3_adjScale.pdf"), width = 7, 
       height = 7, units = "in")


```

#------------------------------------
# Biological analyses of BMDM samples
#------------------------------------

#----------------
# Fig. 4: BMDM PCA and Protein-set-enrichment analysis
#----------------

```{r data import for BMDM analyses}
#Base directory where you unzipped the Figure_gen_files folder

datPath <- paste0(datIn,"BMDM_figs_4_5_6/")

#Precursor-level
t0_mat <- read.delim(paste0(datPath, "t0_BMDM_priori_mrri02_PIF50_DART_1pFDR.txt"))
prePCA <- read.csv(paste0(datPath,"limmaCorrected_normed_prePCA_Priori_mrri02_PIF50_DART_1pFDR.csv"),row.names = "X")
CG <- read.delim(paste0(datPath, "ChanKey_BMDM_Priori_mrri02_PIF50_DART_FDR1p.txt"))
GeneMappings_single <- read.csv(paste0(datPath,"GeneMappings.csv"))
ProtMissing <- read.delim(paste0(datPath,"Prot_missing_BMDM_priori_mrri02_PIF50_DART_1pFDR.txt"))
GO_DB_Path <- paste0(datPath,"GOA_db.txt")
GO_db <- read.delim(GO_DB_Path, sep = "", header = TRUE)
#GO_db.2 <- GO_db %>% filter(Letter != "C")

# Gene mappings processing
library(tidyr)
GeneMappings_single2 <- separate_rows(GeneMappings_single,GENES, sep = " ", convert = FALSE)
GeneMappings_single2$GENES <- toupper(GeneMappings_single2$GENES)
GeneMappings_single2 <- GeneMappings_single2 %>% select(-X)
detach("package:tidyr", unload=TRUE)

# Dextran-Uptake Data
Dextran_DIA_LPS <- read.delim(paste0(datPath,"DexLPS_LowOverHigh__eGH_358_eGH_360_Data_for_volcanoplot.txt"))
Dextran_DIA_NT <- read.delim(paste0(datPath,"DexNT_LowOverHigh__eGH_362_eGH_364_Data_for_volcanoplot.txt"))

# Marker Databases for MEROPS
MarkerDB_SCoPE2 <- read.delim(paste0(datPath,"SCoPE2_markers.txt"))
SCoPE2_Human2Mouse <- read.delim(paste0(datPath,"SCoPE2_Markers_Dict.tab"))
BulkDDARatioSet <- read.delim(paste0(datPath,"BulkRatioEvidence.txt"))

# pre-digestion labeled bulk sample for MEROPS analysis
DIA_Bulk <- read.delim(paste0(datPath,"GHRT_eGH692_Report.xls"))

```

```{r PCA coordinate and protein weight extraction}
# Xcondition comparison
PCAout_both <- prcomp(t(prePCA))
PCAout_both_coords <- PCAout_both$x
PCAout_both_weights <- as.data.frame(PCAout_both$rotation[,1:5])
PCAout_both_weights$Proteins <- row.names(PCAout_both_weights)

# LPS 
CG_LPS <- CG %>% filter(celltype == "LPS", cvm < .4)
prePCA_LPS <- prePCA %>% select(CG_LPS$id) 
prePCA_LPS_norm <- cr_norm_log(as.matrix(prePCA_LPS))
PCAout_LPS <- prcomp(t(prePCA_LPS_norm))
PCAout_LPS_coords <- PCAout_LPS$x
PCAout_LPS_weights <- as.data.frame(PCAout_LPS$rotation[,1:5])
PCAout_LPS_weights$Proteins <- row.names(PCAout_LPS_weights)

# Untreated
CG_NT <- CG %>% filter(celltype != "LPS", cvm < .4)
prePCA_NT <- prePCA %>% select(CG_NT$id) 
prePCA_NT_norm <- cr_norm_log(as.matrix(prePCA_NT))
PCAout_NT <- prcomp(t(prePCA_NT_norm))
PCAout_NT_coords <- PCAout_NT$x
PCAout_NT_weights <- as.data.frame(PCAout_NT$rotation[,1:5])
PCAout_NT_weights$Proteins <- row.names(PCAout_NT_weights)
```

```{r Gene Ontology DB prep for BMDM data}

# Adding mus musculus UNIPROT mappings to Gene Ontology DB
GO_db.2 <- GO_db %>% left_join(GeneMappings_single2, by = c("Gene"="GENES"))
GO_db.2 <- GO_db.2[,c(12,2,3,4,5,6,7,8,9,10,11,1)]
GO_db.2 <- GO_db.2 %>% select(-Uniprot)
colnames(GO_db.2)[1] <- "Uniprot"
# Removing 'Cellular component' annotations from GO DB
GO_db.3 <- GO_db.2 %>% filter(Letter  != "C")


# Generating a more specific GO-term featuring only proton-transport-annotated V-type ATPase components
GO.db.3.ProtonTransportLysosomal <- GO_db.3 %>% filter((GO_term_name == "proton transport")&(grepl("ATP6V",Gene)))
GO.db.3.ProtonTransportLysosomal$GO_term_name <- "proton transport, V-type ATPase"
GO_db.3 <- rbind(GO_db.3, GO.db.3.ProtonTransportLysosomal)

```


```{r BMDM PSEA test}

# Run PC-weight-based PSEA (PC weights, reference database, number of PCs to evaluate)
#     Note: this function takes a number of minutes to evaluate


xcond_PSEA <- PCA_PSEA_Rebuttal_BMDM_prot(PCAout_both_weights,GO_db.3,5)
LPS_PSEA <- PCA_PSEA_Rebuttal_BMDM_prot(PCAout_LPS_weights,GO_db.3,4)
NT_PSEA <- PCA_PSEA_Rebuttal_BMDM_prot(PCAout_NT_weights,GO_db.3,4)

#************
# Uncomment to save the results of the re-run analyses
#************
#write.table(xcond_PSEA,paste0(datOut,"xCond2.txt"))
#write.table(LPS_PSEA,paste0(datOut,"LPS_only_PCAPSEA.txt"))
#write.table(NT_PSEA,paste0(datOut,"NT_only_PCAPSEA.txt"))

#************
# Uncomment to load your previously saved PC-weight-based PSEA results
#************
#xcond_PSEA <- read.table(paste0(datOut,"xCond2.txt"))
#LPS_PSEA <- read.table(paste0(datOut,"LPS_only_PCAPSEA.txt"))
#NT_PSEA <- read.table(paste0(datOut,"NT_only_PCAPSEA.txt"))

```


```{r PC-weight-based PCA result filtration}
# Removing redundant Protein Sets

XCondition_excludeList <- c("viral infectious cycle","viral transcription","melanosome","proton-transporting ATPase activity, rotational mechanism","proton transport","ribosomal small subunit biogenesis","ATP hydrolysis coupled proton transport","cellular iron ion homeostasis")

SingleCondition_excludeList <- c("viral infectious cycle","viral transcription","melanosome","proton-transporting ATPase activity, rotational mechanism","translation initiation factor activity","mRNA binding","SRP-dependent protein targeting to membrane","proton transport")

# q value correction and FDR filtration of PSEA results
xcond.f <- xcond_PSEA %>% filter(!is.na(pVal))
xcond.f$qVal <- p.adjust(xcond.f$pVal, method = "fdr")
xcond.f_fdr <- xcond.f %>% filter((qVal <= .05)& !(Process %in% XCondition_excludeList))

# LPS
LPS_PSEA.f <- LPS_PSEA %>% filter(!is.na(pVal),PC < 5)
LPS_PSEA.f$qVal <- p.adjust(LPS_PSEA.f$pVal, method = "fdr")
LPS_PSEA.f_fdr <- LPS_PSEA.f %>% filter(qVal <= .05 & !(Process %in% SingleCondition_excludeList)) %>% mutate(Condition = "LPS")
LPS_PSEA.f_fdr2 <- LPS_PSEA.f_fdr
#LPS_PSEA.f_fdr2$Association <- ifelse(LPS_PSEA.f_fdr2$PC == 1,LPS_PSEA.f_fdr2$Association*-1,ifelse(LPS_PSEA.f_fdr2$PC == 2,LPS_PSEA.f_fdr2$Association*-1,LPS_PSEA.f_fdr2$Association))

# NT
NT_PSEA.f <- NT_PSEA %>% filter(!is.na(pVal),PC < 5)
NT_PSEA.f$qVal <- p.adjust(NT_PSEA.f$pVal, method = "fdr")
NT_PSEA.f_fdr <- NT_PSEA.f %>% filter(qVal <= .05 & !(Process %in% SingleCondition_excludeList)) %>% mutate(Condition = "Untreated")

# Note: Association multiplied by negative 1, since eigenvector sign is arbitrary 
LPS_PSEA.f_fdr2$Association <- ifelse(LPS_PSEA.f_fdr2$PC == 2, LPS_PSEA.f_fdr2$Association*-1, LPS_PSEA.f_fdr2$Association)

NT_PSEA.f_fdr$Association <- ifelse(NT_PSEA.f_fdr$PC == 1, NT_PSEA.f_fdr$Association*-1, NT_PSEA.f_fdr$Association)


```


```{r Fig4a PCA accompaniment data wrangling}
# Batch corrected, imputed cells x proteins matrix
BC_imp_PxC2 <- prePCA
BC_imp_PxC2$Proteins <- row.names(BC_imp_PxC2)
BC_imp_PxC2.m <- melt(BC_imp_PxC2, id.vars = "Proteins")

# Removing imputed protein measurements, by cross-referencing pre-imputation cells x proteins matrix
ProtMissing.m <- melt(ProtMissing)
BC_imp_PxC3.m <- BC_imp_PxC2.m %>% left_join(ProtMissing.m, by = c("Proteins", "variable"))
BC_imp_PxC3.m$newVar <- ifelse(is.na(BC_imp_PxC3.m$value.y),NA,BC_imp_PxC3.m$value.x)
BC_unImp_PxC3.m2 <- BC_imp_PxC3.m %>% select(Proteins,variable,newVar)
BC_unImp_PxC3 <- dcast(BC_unImp_PxC3.m2, Proteins~variable)
row.names(BC_unImp_PxC3) <- BC_unImp_PxC3$Proteins
BC_unImp_PxC4 <- BC_unImp_PxC3 %>% select(-Proteins)
BC_unImp_CxP <- as.data.frame(t(BC_unImp_PxC4))
BC_unImp_CxP$id <- row.names(BC_unImp_CxP)
BC_unImp_PxC3.m3 <- BC_unImp_PxC3.m2 %>% left_join(GeneMappings_single2, by = c("Proteins"="UNIPROTKB"))

GO_Set_for_xCond <- c("type I interferon-mediated signaling pathway","phagosome maturation")
for (i in 1:length(GO_Set_for_xCond)){
  GO_db_lim <- GO_db %>% filter(GO_term_name %in% GO_Set_for_xCond[i])
  BC_unImp_PxC3.m3_lim <- BC_unImp_PxC3.m3 %>% filter(GENES %in% GO_db_lim$Gene)
  BC_unImp_PxC3.m3_lim_sum <- BC_unImp_PxC3.m3_lim %>% dplyr::group_by(variable) %>% dplyr::summarize(medAbund = median(newVar, na.rm = TRUE)) 
  colnames(BC_unImp_PxC3.m3_lim_sum) <- c("id",as.character(GO_Set_for_xCond[[i]]))
  if(i == 1){
    GSEA_PC_out <- BC_unImp_PxC3.m3_lim_sum
  }
  if(i > 1){
    GSEA_PC_out <- GSEA_PC_out %>% left_join(BC_unImp_PxC3.m3_lim_sum, by = "id")
  }
}
PCAout_both_coords2 <- as.data.frame(PCAout_both_coords[,1:2])
PCAout_both_coords2$id <- row.names(PCAout_both_coords2)
PCAout_both_coords2 <- PCAout_both_coords2 %>% left_join(GSEA_PC_out, by = "id")
PCAout_both_coords2 <- PCAout_both_coords2 %>% left_join(CG, by = "id")

# treatment labeled PCA without accompanying bulk measurements (unused in publication)
celltype_PCA <- PCAout_both_coords2 %>% ggplot(aes(x = PC1, y = PC2, color = celltype)) + geom_point(size = 3, alpha = 0.7, shape = 16) + theme_light() + theme(axis.title = element_text(size = 24), axis.text = element_text(size = 22), legend.position = "null") + ggtitle("") + annotate("text", x = 5.1, y = 6,size = 7, label = "LPS-treated", color = "#F8766D") + annotate("text", x = -5, y = 6,size = 7, label = "Untreated", color = "#00BFC4")

PCAout_both_coords2$PhagosomeBin2 <- ((PCAout_both_coords2$`phagosome maturation`)-mean(PCAout_both_coords2$`phagosome maturation`,na.rm = TRUE))/sd(PCAout_both_coords2$`phagosome maturation`, na.rm = TRUE)

PCAout_both_coords2$PhagosomeBin3 <- ifelse(PCAout_both_coords2$PhagosomeBin2 >= 2,2,ifelse(PCAout_both_coords2$PhagosomeBin2 <= -2,-2,PCAout_both_coords2$PhagosomeBin2))

PCAout_both_coords2$IFBin2 <- ((PCAout_both_coords2$`type I interferon-mediated signaling pathway`)-mean(PCAout_both_coords2$`type I interferon-mediated signaling pathway`,na.rm = TRUE))/sd(PCAout_both_coords2$`type I interferon-mediated signaling pathway`, na.rm = TRUE)

PCAout_both_coords2$IFBin3 <- ifelse(PCAout_both_coords2$IFBin2 >= 2,2,ifelse(PCAout_both_coords2$IFBin2 <= -2,-2,PCAout_both_coords2$IFBin2))

# IFN signaling PCA (Fig 2A)
IFN_PCA <- PCAout_both_coords2 %>% ggplot(aes(x = PC1, y = PC2, color = IFBin3)) + geom_point(size = 2, alpha = 0.7, shape = 16) + theme_light() + theme(axis.title = element_text(size = 16), axis.text = element_blank(), title = element_text(size = 16),legend.title = element_text(size = 22), legend.text = element_text(size = 22), plot.margin = unit(c(0,0.3,0,0),"cm")) + scale_color_gradient2(limits = c(-2,2),low = "blue", mid = "white", high = "red", midpoint =0, breaks = c(-2,0,2), labels = c("-2","0","2")) + ggtitle("Type I IFN\nsignaling") + labs(x ="", y = "", color = "Abundance\nz-score")

# Phagosome maturation PCA (Fig 2A)
PM_PCA <- PCAout_both_coords2 %>% ggplot(aes(x = PC1, y = PC2, color = PhagosomeBin3)) + geom_point(size = 2, alpha = 0.7, shape = 16) + theme_light() + theme(axis.title = element_text(size = 16), axis.text = element_blank(), title = element_text(size = 16),legend.title = element_text(size = 22), legend.text = element_text(size = 22), plot.margin = unit(c(0,0,0,0.3),"cm")) + scale_color_gradient2(limits = c(-2,2),low = "blue", mid = "white", high = "red", midpoint =0, breaks = c(-2,0,2), labels = c("-2","0","2")) + ggtitle("Phagosome\nmaturation") + labs(x ="", y = "", color = "Abundance\nz-score") 

AbundancePCAs <- (IFN_PCA + PM_PCA) + plot_layout(guides = "collect") & theme(legend.position = "bottom")

AbundancePCAs & theme(plot.margin =  margin(0, 0, 0, 0, "cm"))
ggsave(paste0(FigOut,"xCond_PSEA_PCs_horiz_v3.pdf"), width = 6, height = 4.5, units = "in")
```

```{r Fig 4a: treatment-condition PCA with bulk}
# PCA of original BMDM single-cell data
PCAout_both <- prcomp(t(prePCA))
PCAout_both_coords_for_bulkPCA <- as.data.frame(PCAout_both$x[,1:2])
PCAout_both_coords_for_bulkPCA$id <- row.names(PCAout_both_coords_for_bulkPCA)
PCAout_both_coords_for_bulkPCA <- PCAout_both_coords_for_bulkPCA %>% left_join(CG, by = "id")

# Projection of bulk samples into single-cell low-dimensional space
# Note: bulk samples had previously been incorporated into the scp pipeline for normalization,
#       imputation, and batch correction with the original single-cell BMDM measurements

bulkBC <- read.delim(paste0(datPath,"imputed_BC_BMDM_priori_mrri02_PIF50_DART1pFDR_withBulk.txt"))
bulkBC_imp.mat <- bulkBC[,which(colnames(bulkBC) %in% c("i700","i701"))]
bulkProjection <- predict((PCAout_both),t(bulkBC_imp.mat))
bulkProjection_coord <- as.data.frame(bulkProjection[,1:2])
bulkProjection_coord$id <- row.names(bulkProjection_coord)
bulkProjection_coord$Raw.file <- c("XwGH215","XwGH215")  
bulkProjection_coord$celltype <- c("LPS","untreated")  
bulkProjection_coord$cvm <- c(0.2,0.2) 
bulkProjection_coord$RI <- c("Reporter.intensity.corrected.3","Reporter.intensity.corrected.4")
sc_and_bulkprojection_comb <- rbind(PCAout_both_coords_for_bulkPCA,bulkProjection_coord)
sc_and_bulkprojection_comb$Source <- ifelse(sc_and_bulkprojection_comb$Raw.file %in% c("XwGH215","XwGH215","XwGH216","XwGH216"),"Bulk","Single cell")
sc_and_bulkprojection_comb %>% ggplot(aes(x = PC1, y = PC2, color = celltype, shape = Source, size = Source,alpha = Source))  + geom_point(stroke = 2) + scale_size_discrete(range = c(15,3)) + scale_shape_manual(values = c(23,16)) +   
  geom_point(data = subset(sc_and_bulkprojection_comb, Source == 'Bulk'),aes(x = PC1, y = PC2),size = 14, color = "grey50", shape = 23,stroke = 2)+   geom_point(data = subset(sc_and_bulkprojection_comb, Source == 'Bulk'),aes(x = PC1, y = PC2),size = 16, color = "grey50", shape = 23,stroke = 2)+
  geom_point(data = subset(sc_and_bulkprojection_comb, Source == 'Bulk'),aes(x = PC1, y = PC2, color = celltype), shape = 23,stroke = 2) +   
  theme_light() + scale_alpha_discrete(range = c(1,.6)) +
theme(axis.title = element_text(size = 24), axis.text = element_text(size = 22),legend.position = c(.24,.15),legend.background = element_rect(color = "transparent", fill = "transparent"),legend.text = element_text(size = 17)) + ggtitle("") + 
  annotate("text", x = 5.1, y = 6,size = 7, label = "LPS-treated", color = "#F8766D") + 
  annotate("text", x = -5, y = 6,size = 7, label = "Untreated", color = "#00BFC4") + 
  guides(color = "none",fill = "none", size = "none",alpha = "none") + labs(shape = "",x="PC1",y = "PC2") 
ggsave(paste0(FigOut,"celltypePlusBulk.pdf"), width = 5, height = 5, units = "in")
```

```{r Fig4b: PC-weight-based PCA result clustering and fig-gen}
#Shortening Process Names
xcond.f_fdr$Process <- ifelse(xcond.f_fdr$Process=="nuclear-transcribed mRNA catabolic process, nonsense-mediated decay","nuclear-transcribed mRNA catabolic process, NMD",ifelse(xcond.f_fdr$Process == "SRP-dependent cotranslational protein targeting to membrane","SRP-dependent protein targeting to membrane",ifelse(xcond.f_fdr$Process == "positive regulation of NF-kappaB transcription factor activity","positive regulation of NF-kappaB TF activity",xcond.f_fdr$Process))) 

#clustering 
xcond.f_fdr_sel <- xcond.f_fdr %>% select(Process,PC,Association)
#xcond.f_fdr_sel <-  xcond.f_fdr_sel %>% filter(!(Process %in% Excl_x)) 
xcond.f_fdr_clustIn <- (dcast(xcond.f_fdr_sel,Process~PC))
row.names(xcond.f_fdr_clustIn) <- xcond.f_fdr_clustIn$Process
xcond.f_fdr_clustIn <- xcond.f_fdr_clustIn %>% select(-Process)
xcond.f_fdr_clustIn <- as.matrix(xcond.f_fdr_clustIn)
xcond.f_fdr_clustIn[is.na(xcond.f_fdr_clustIn)] <- 0
xcond_dist1 <- dist(xcond.f_fdr_clustIn)
xcond_dist2 <- dist(t(xcond.f_fdr_clustIn))
xcond_clust1 <- hclust(t(xcond_dist1), method = "ward.D")
xcond_clust2 <- hclust(t(xcond_dist2), method = "ward.D")
ProcOrder <- xcond_clust1$labels[xcond_clust1$order]
xcond.f_fdr$negLogQ <- -log10(xcond.f_fdr$qVal)
xcond.f_fdr$binnedNegLogQ <- ifelse(xcond.f_fdr$negLogQ >= 3, 3,xcond.f_fdr$negLogQ)

Xc_breaks <- c(-2, 0, 2)

xcond.f_fdr %>% ggplot(aes(x = PC, y = factor(str_wrap(Process, width = 30), levels = str_wrap(ProcOrder, width = 30)),fill = Association, size = binnedNegLogQ)) + geom_point(shape = 21, stroke = 1) + 
  scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0, breaks = Xc_breaks, limits = c(-2.5,2.5)) + 
  theme_light() + 
  labs(fill = "Loading\nz-score",x = "Principal component",y = "", size = bquote("-log"[10]~"(q val)")) + theme(axis.text.y = element_text(size = 18),axis.text.x = element_text(size = 26),axis.title = element_text(size = 30), legend.title = element_text(size = 20), legend.text = element_text(size = 20), legend.position = "bottom", legend.box="vertical", legend.margin=margin()) +
  scale_size_continuous(breaks = seq(2,3,1),range=c(1,15),
                  limits = c(0, 4), labels = c("2",">3")) +  
  scale_x_discrete(limits = c("1",
     "2","3","4","5"), expand=c(0.1, 0)) 
ggsave(paste0(FigOut,"xCond_PC_PSEA_legendBottom.pdf"), width = 7.75, height = 10, units = "in")
```
```{r figure 4c and associated supplemental figure}
prePCA_BMDM <- read.csv(paste0(datPath,"limmaCorrected_normed_prePCA_Priori_mrri02_PIF50_DART_1pFDR.csv"),row.names = "X")
CG_BMDM <- read.delim(paste0(datPath, "ChanKey_BMDM_Priori_mrri02_PIF50_DART_FDR1p.txt"))
GeneMappings_single <- read.csv(paste0(datPath,"GeneMappings.csv"))
ProtMissing_BMDM <- read.delim(paste0(datPath,"Prot_missing_BMDM_priori_mrri02_PIF50_DART_1pFDR.txt"))

prePCA_BMDM$prots <- row.names(prePCA_BMDM) 
prePCA_BMDM.m <- melt(prePCA_BMDM)
ProtMissing_BMDM.m <- melt(ProtMissing_BMDM)

GOin <- prePCA_BMDM.m %>% left_join(ProtMissing_BMDM.m, by = c("prots"="Proteins","variable"))
GOin$unimp <- as.numeric(ifelse(is.na(GOin$value.y),NA,GOin$value.x))
CG_BMDM.lim <- CG_BMDM %>% select(id,celltype)
GOin <- GOin %>% left_join(CG_BMDM.lim, by = c("variable"="id"))
GOin <- GOin %>% left_join(GeneMappings_single, by = c("prots"="UNIPROTKB"))
GOin$Gene <- toupper(gsub(" .*$","",GOin$GENES))

GO_db_diffTest_df <- data.frame(GOTerm = character(), pVal = numeric(), foldChange = numeric(), 
                            Proteins = numeric(), fractionOfProteins = numeric(), stringsAsFactors = FALSE)


#Identifying GO terms with differential abundance between conditions for single-cell samples

for (i in 1:length(unique(GO_db.3$GO_term_name))){
  GOTerm.single.name <- unique(GO_db.3$GO_term_name)[i]
  GOTerm.single <- GO_db.3 %>% filter(GO_term_name == GOTerm.single.name)
  GOin.singleGOterm <- GOin %>% filter(Gene %in% GOTerm.single$Gene) %>% ungroup()
  GO_db_diffTest_df[i,1] <- GOTerm.single.name
  GOin.singleGOterm_sum <- GOin.singleGOterm %>% filter(!is.na(unimp)) %>% group_by(celltype) %>% dplyr::summarize(count = n())
  
  if((nrow(GOin.singleGOterm)>1)&(nrow(GOin.singleGOterm_sum)==2)){
    kt.out  <- kruskal.test(unimp~celltype,GOin.singleGOterm)
    ForProtCalc <- GOin.singleGOterm %>% filter(is.na(unimp))
    FC <- GOin.singleGOterm %>% group_by(celltype) %>% dplyr::summarize(medVal = median(unimp, na.rm=TRUE))
    GO_db_diffTest_df[i,2] <- kt.out$p.value
    GO_db_diffTest_df[i,3] <- FC[1,2] - FC[2,2]
    GO_db_diffTest_df[i,4] <- length(unique(ForProtCalc$Gene))
    GO_db_diffTest_df[i,5] <- length(unique(ForProtCalc$Gene))/length(unique(GOTerm.single$Gene))
    
  }
  if((nrow(GOin.singleGOterm)<2)&(nrow(GOin.singleGOterm_sum)>1)){
    GO_db_diffTest_df[i,2] <- NA
    GO_db_diffTest_df[i,3] <- NA
    GO_db_diffTest_df[i,4] <- NA
    GO_db_diffTest_df[i,5] <- NA
  }
} 

GO_db_diffTest_df.f <- GO_db_diffTest_df %>% filter(!is.na(Proteins)&Proteins > 2)
GO_db_diffTest_df.f$qVal <- p.adjust(GO_db_diffTest_df.f$pVal)

GO_db_diffTest_df.f.q <- GO_db_diffTest_df.f %>% filter(qVal <= .05)

#

bulkBC <- read.delim(paste0(datPath,"imputed_BC_BMDM_priori_mrri02_PIF50_DART1pFDR_withBulk.txt"))
bulkBC_unImp <- read.delim(paste0(datPath,"Prot_missing_BMDM_priori_mrri02_PIF50_DART_1pFDR_withBulk.txt"))

bulkBC_imp.mat <- bulkBC[,which(colnames(bulkBC) %in% c("i700","i701"))]
bulkBC_imp.mat_sc <- bulkBC[,(which(!(colnames(bulkBC) %in% c("i700","i701"))))]
bulkBC_imp.mat_sc$protein <- row.names(bulkBC_imp.mat_sc)
bulkBC$Protein <- row.names(bulkBC) 

GeneMappings_single$GENES <- gsub(" .*","",GeneMappings_single$GENES)
GeneMappings_single$GENES <- toupper(GeneMappings_single$GENES)
GeneMappings_single2 <- GeneMappings_single %>% select(-X)

CG <- read.delim(paste0(datPath,"ChanKey_BMDM_Priori_mrri02_PIF50_DART1pFDR_withBULK.txt"))
CG.lim <- CG %>% select(celltype, id)
CG.lim$celltype <- ifelse(CG.lim$id == "i700","LPS_bulk",ifelse(CG.lim$id == "i701","Untreated_bulk",CG.lim$celltype))

bulkBC_imp.mat.m <- melt(bulkBC)
bulkBC_imp.mat.m <- bulkBC_imp.mat.m %>% filter(!(variable %in% c("i702","i703")))
bulkBC_imp.mat.m <- bulkBC_imp.mat.m %>% left_join(CG.lim, by = c("variable"="id"))

bulkBC_unimp.mat.m <- melt(bulkBC_unImp)
bulkBC_unimp.mat.m <- bulkBC_unimp.mat.m %>% filter(!(variable %in% c("i702","i703")))

# Removing imputed values
bulkBC_imp.mat.m <- bulkBC_imp.mat.m %>% left_join(bulkBC_unimp.mat.m, by = c("Protein" = "Proteins","variable"))
bulkBC_imp.mat.m$unimp <- ifelse(is.na(bulkBC_imp.mat.m$value.y),NA,bulkBC_imp.mat.m$value.x)

bulkBC_imp.mat.m_sum <- bulkBC_imp.mat.m %>% group_by(Protein,celltype) %>% dplyr::summarize(medianVal = median(unimp,na.rm = TRUE))

bulkBC_imp.mat.m.dcast <- dcast(bulkBC_imp.mat.m_sum,Protein~celltype, value.var = "medianVal")

bulkBC_imp.mat.m.dcast$scRat <- bulkBC_imp.mat.m.dcast$LPS-bulkBC_imp.mat.m.dcast$untreated
bulkBC_imp.mat.m.dcast$bulkRat <- bulkBC_imp.mat.m.dcast$LPS_bulk-bulkBC_imp.mat.m.dcast$Untreated_bulk

bulkBC_imp.mat.m.dcast <- bulkBC_imp.mat.m.dcast %>% left_join(GeneMappings_single2, by = c("Protein"="UNIPROTKB"))

GOtermsForCor <- c(unique(GO_db_diffTest_df.f.q$GOTerm),"proton transport","phagosome maturation","hydrogen peroxide catabolic process")

for (i in 1:length(GOtermsForCor)){
  DB.lim <- GO_db.3 %>% filter(GO_term_name %in% unique(GOtermsForCor)[i])
  
  singleProcessHits <- bulkBC_imp.mat.m.dcast %>% filter(GENES %in% DB.lim$Gene)
  
  corPear <- cor(as.matrix(singleProcessHits[,c(6:7)]), method = "pearson", use = "pairwise.complete.obs")[1,2]
  corSpear <- cor(as.matrix(singleProcessHits[,c(6:7)]), method = "spearman", use = "pairwise.complete.obs")[1,2]
  
  singleProcessHits$pearson <- corPear
  singleProcessHits$spearman <- corSpear
  singleProcessHits$Process <- unique(GOtermsForCor)[i]
  
  if(i == 1){
    ProcessOut <- singleProcessHits
  }
  if(i > 1){
    ProcessOut <- rbind(ProcessOut,singleProcessHits)
  }
  
} 

selectedProcessForPlot <- c("proton transport, V-type ATPase","type I interferon-mediated signaling pathway","hydrogen peroxide catabolic process","mitochondrial electron transport, NADH to ubiquinone","S100 protein binding","hydrolase activity, hydrolyzing O-glycosyl compounds","phagocytosis","positive regulation of nitric oxide biosynthetic process")

ProcessOut.sel <- ProcessOut %>% filter(Process %in% selectedProcessForPlot)

SpearmanPSEA <- cor(ProcessOut.sel$scRat,ProcessOut.sel$bulkRat, method = "spearman", use = "pairwise.complete.obs")
PearsonPSEA <- cor(ProcessOut.sel$scRat,ProcessOut.sel$bulkRat, method = "pearson", use = "pairwise.complete.obs")

#palette <- c("#999999","#E69F00","#000000","#009E73","#F0E442","#0072B2","#D55E00","#CC79A7","#56B4E9")
paletteRev <- c("#56B4E9","#0072B2","#D55E00","#CC79A7","#F0E442","#E69F00","#000000","#009E73")
#Fig4Palette <- c("#56B4E9","#CC79A7","#999999")
Fig4Palette <- c("#ffc71e","#009E73","#999999")

# Supplemental Plot
titleFormatted <- expression(paste(Log[2],"(LPS/Untreated)"))
Scatter <- ProcessOut %>% filter(Process %in% selectedProcessForPlot)  %>% arrange(rev(Process)) %>%
  ggplot(aes(x=scRat,y =bulkRat, fill = Process)) + 
  geom_point(shape = 21, size = 5, stroke = 1.5, alpha =0.5) + 
  theme_light() + labs(x = "Single cells", y = "Bulk") +
  theme(legend.position = c(0.43,0.82), 
        legend.title = element_blank(),
        legend.background = element_rect(fill = "transparent"),
        legend.text = element_text(size = 24),
        axis.title = element_text(size = 30),
        axis.text = element_text(size = 22),
        plot.title = element_text(size = 40)) +
  annotate("text",x = 1.3, y = -0.55, label = paste0("\u03C1: ",round(SpearmanPSEA,2)), size = 14) +
coord_cartesian(xlim = c(-.75,1.5), ylim = c(-.7,3.5)) +
  scale_fill_manual(values = paletteRev) + 
  ggtitle(titleFormatted)
Scatter
ggsave(paste0(FigOut,"ScatterRevised_supplemental_v2.png"), width =11, height = 11, units = "in")

ProcessOut$MinimalLabels <- ifelse(ProcessOut$Process == "type I interferon-mediated signaling pathway","Type I IFN signaling",ifelse(ProcessOut$Process == "proton transport, V-type ATPase","Proton transport, V-type ATPase", "Other GO terms"))

# Figure 4c

ProcessOut$Size <- ifelse(ProcessOut$MinimalLabels == "Other GO terms",4,8)
ProcessOut$Size <- as.factor(ProcessOut$Size)

ScatterFig4 <- ProcessOut %>% filter(Process %in% selectedProcessForPlot)  %>% arrange(rev(Process)) %>%
  ggplot(aes(x=scRat,y =bulkRat, fill = factor(MinimalLabels, levels = c("Proton transport, V-type ATPase","Type I IFN signaling","Other GO terms")), size = Size)) + 
  geom_point(shape = 21, stroke = 1.5, alpha =0.5) + 
  theme_light() + labs(x = expression("Single Cells"), y = expression("Bulk")) +
  theme(legend.position = c(0.5,0.87), 
        legend.title = element_blank(),
        legend.text = element_text(size = 18),
        legend.background = element_rect(fill = "transparent"),
        axis.title = element_text(size = 25),
        axis.text = element_text(size = 26)) +
  annotate("text",x = 1, y = -.7, label = paste0("\u03C1: ",round(SpearmanPSEA,2)), size = 10) +
coord_cartesian(xlim = c(-.55,1.5), ylim = c(-1,4)) +
  scale_fill_manual(values = Fig4Palette) + 
  scale_size_manual("Size",values = c(3,8)) +
  guides(size = "none", fill = guide_legend(override.aes = list(size = 8)))
ScatterFig4
ggsave(paste0(FigOut,"ScatterRevised_Fig4c.pdf"), width =5, height = 5, units = "in", device = cairo_pdf)

```

```{r Fig4d}
#ATPase <- c("ATP6V1H","ATP6V1D","ATP6V1F","ATP6V1C1","ATP6V0D1","ATP6V1E1","ATP6V1B2","ATP6V1G1","ATP6V1A1")

GOin.sel <- GOin %>% select(unimp,Gene,variable) %>% filter(!is.na(Gene))

GOin.sel.dcast <- dcast(GOin.sel, variable~Gene, value.var = "unimp")
GOin.sel.dcast_proton <- GOin.sel.dcast %>% select(variable,contains("ATP6V"))

row.names(GOin.sel.dcast_proton) <- GOin.sel.dcast_proton$variable
GOin.sel.dcast_proton.join <- GOin.sel.dcast_proton %>% left_join(CG.lim, by = c("variable" = "id"))
GOin.sel.dcast_proton.preMat <- GOin.sel.dcast_proton %>% select(-variable)
GOin.sel.dcast_proton.mat <- as.matrix(GOin.sel.dcast_proton.preMat)

GOin.sel.dcast_proton.join$celltype.format <- ifelse(GOin.sel.dcast_proton.join$celltype == "LPS","LPS treated","Untreated")

flattenCorrMatrix <- function(cormat, pmat) {
  ut <- upper.tri(cormat)

    testOut <- data.frame(
    row = rownames(cormat)[row(cormat)[ut]],
    column = rownames(cormat)[col(cormat)[ut]],
    cor  =(cormat)[ut],
    p = pmat[ut]
    )
    return(testOut)
}

GOin.sel.dcast_proton.join.LPS <- GOin.sel.dcast_proton.join %>% filter(celltype == "LPS")
GOin.sel.dcast_proton.join.Un <- GOin.sel.dcast_proton.join %>% filter(celltype != "LPS")

library(Hmisc)

TopHalf.LPS.cor <- rcorr(GOin.sel.dcast_proton.join.LPS$ATP6V1B2,GOin.sel.dcast_proton.join.LPS$ATP6V1A, type ="spearman")
BottomHalf.LPS.cor <- rcorr(GOin.sel.dcast_proton.join.LPS$ATP6V1B2,GOin.sel.dcast_proton.join.LPS$ATP6V1E1,type ="spearman")
TopHalf.LPS.cor.df <- flattenCorrMatrix(TopHalf.LPS.cor$r,TopHalf.LPS.cor$P)
BottomHalf.LPS.cor.df <- flattenCorrMatrix(BottomHalf.LPS.cor$r,BottomHalf.LPS.cor$P)
TopHalf.Un.cor <- rcorr(GOin.sel.dcast_proton.join.Un$ATP6V1B2,GOin.sel.dcast_proton.join.Un$ATP6V1A,type ="spearman")
BottomHalf.Un.cor <- rcorr(GOin.sel.dcast_proton.join.Un$ATP6V1B2,GOin.sel.dcast_proton.join.Un$ATP6V1E1,type ="spearman")
TopHalf.Un.cor.df <- flattenCorrMatrix(TopHalf.Un.cor$r,TopHalf.Un.cor$P)
BottomHalf.Un.cor.df <- flattenCorrMatrix(BottomHalf.Un.cor$r,BottomHalf.Un.cor$P)

detach("package:Hmisc", unload=TRUE)

CorDF <- rbind(TopHalf.Un.cor.df,BottomHalf.Un.cor.df,TopHalf.LPS.cor.df,BottomHalf.LPS.cor.df)
CorDF$celltype.format <- c("Untreated","Untreated","LPS treated","LPS treated")
CorDF$panel <- c("Top","Bottom","Top","Bottom")

corDF.top <- CorDF %>% filter(panel == "Top")
corDF.bottom <- CorDF %>% filter(panel == "Bottom")

corDF.bottom$ATP6V1B2 <- c(2,2)
corDF.bottom$ATP6V1E1 <- c(2,2)

ProtonPumpPair <- GOin.sel.dcast_proton.join %>% ggplot(aes(x = ATP6V1B2, y = ATP6V1E1)) + 
  geom_point(alpha = 0.6, size = 4, color = "grey70",fill = "#ffc71e", shape=21) +
  theme_light() +
  theme(axis.title = element_text(size = 25),
        axis.text = element_text(size = 25),
        strip.background = element_rect(fill = "grey40", color = "grey70"),
        strip.text = element_text(color = "white", size = 25)) + 
  coord_cartesian(xlim = c(-2,2), ylim = c(-2,2)) +
  facet_wrap(~factor(celltype.format, levels = c("Untreated","LPS treated"))) + 
  labs(color = "", x = "Vacuolar ATPase ATP6V1B2") + guides(color = "none") + 
  geom_text(
  data    = corDF.bottom,
  mapping = aes(x = 0, y = 3.5, label = paste0("\u03C1:  ", round(cor,2))),
  hjust   = 1,
  vjust   = 1,
  color = "grey20"
) +
geom_text(data = corDF.bottom, aes(label = paste0("\u03C1: ",round(cor,2)), x = 1.1, y = -1.8), size = 10)  
ProtonPumpPair
ggsave(paste0(FigOut,"ProtonPumpPair.pdf"), width = 7, height = 5, units = "in", device = cairo_pdf)

#
# Supplemental Accompaniment: ATPase correlation heatmap
#

CG_BMDM.lim.LPS <- CG_BMDM.lim %>% filter(celltype == "LPS")

GOin.sel.dcast_proton.mat <- as.matrix(GOin.sel.dcast_proton.preMat)
#corLPS <- GOin.sel.dcast_proton.mat[CG.lim.LPS$id,]
corLPS <- subset(GOin.sel.dcast_proton.mat, row.names(GOin.sel.dcast_proton.mat) %in% CG_BMDM.lim.LPS$id)
corUn <- subset(GOin.sel.dcast_proton.mat, !(row.names(GOin.sel.dcast_proton.mat) %in% CG_BMDM.lim.LPS$id))

library(Hmisc)

corLPS.mat <- rcorr(corLPS, type = "spearman")
corUn.mat <- rcorr(corUn,type = "spearman")

flattenCorrMatrix.full <- function(cormat, pmat, nmat) {
  ut <- upper.tri(cormat)

    testOut <- data.frame(
    row = rownames(cormat)[row(cormat)[ut]],
    column = rownames(cormat)[col(cormat)[ut]],
    cor  =(cormat)[ut],
    p = pmat[ut],
    n = nmat[ut]
    )
    return(testOut)
}

detach("package:Hmisc", unload=TRUE)

corLPS.mat.df_rho <- as.data.frame(corLPS.mat$r)
corLPS.mat.df_n <- as.data.frame(corLPS.mat$n)

corUn.mat.df_rho <- as.data.frame(corUn.mat$r)
corUn.mat.df_n <- as.data.frame(corUn.mat$n)
corUn.mat.df_P <- as.data.frame(corUn.mat$P)

corLPS.mat.df_rho$var <- row.names(corLPS.mat.df_rho)
corLPS.mat.df_n$var <- row.names(corLPS.mat.df_n)
corUn.mat.df_rho$var <- row.names(corUn.mat.df_rho)
corUn.mat.df_n$var <- row.names(corUn.mat.df_n)

corLPS.mat.df_rho.m <- melt(corLPS.mat.df_rho)
corLPS.mat.df_n.m <- melt(corLPS.mat.df_n)
corUn.mat.df_rho.m <- melt(corUn.mat.df_rho)
corUn.mat.df_n.m <- melt(corUn.mat.df_n)

corLPS.forHeat <- corLPS.mat.df_rho.m %>% left_join(corLPS.mat.df_n.m, by = c("var","variable"))
corUn.forHeat <- corUn.mat.df_rho.m %>% left_join(corUn.mat.df_n.m, by = c("var","variable"))

#CorLev <- unique(corLPS.forHeat$var)

#clustOut <- hclust( dist(corUn.mat.df_rho, method = "euclidean"), method = "ward.D" )
#CorLev<- clustOut$labels[clustOut$order]

#groups<-cutree(clustOut, k=3)
#x<-cbind(corUn.mat.df_rho,groups)

CorLev <- c("ATP6V1F","ATP6V1D","ATP6V1A","ATP6V1B2","ATP6V1G1","ATP6V1E1","ATP6V1C1","ATP6V1H","ATP6V0D1")

CorLPSHeat <- corLPS.forHeat %>% ggplot(aes(x = factor(var, levels = CorLev), y = factor(variable, levels = CorLev), fill = value.x)) + 
  geom_tile() +
  geom_text(aes(label = value.y), size = 6) + 
  theme(axis.title.y = element_blank(), axis.text.y = element_blank(),
        axis.text.x = element_text(angle = 45, hjust = 1), axis.text = element_text(size = 12),
        legend.text = element_text(size = 10),
        legend.title = element_text(size = 12),
        plot.title = element_text(size = 20)) +
  scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0, limits = c(-1,1)) + 
  labs(x = "", fill = "\u03C1") + 
  ggtitle("LPS Treated")+ guides( fill=guide_colourbar(title.vjust=1))

CorUnHeat <- corUn.forHeat %>% ggplot(aes(x = factor(var, levels = CorLev), y = factor(variable, levels = CorLev), fill = value.x)) + 
  geom_tile() +
  geom_text(aes(label = value.y), size = 6) + 
  scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0, limits = c(-1,1))  + 
  labs(x = "", y = "", fill = "\u03C1")+
  theme(axis.text.x = element_text(angle = 45, hjust = 1), axis.text = element_text(size = 12),
  legend.text = element_text(size = 10),
  legend.title = element_text(size = 12),
  plot.title = element_text(size = 20)) + ggtitle("Untreated") + guides( fill=guide_colourbar(title.vjust=1))


CorUnHeat + CorLPSHeat  + plot_layout(widths = c(4,4), guides = "collect") & theme(legend.position = 'bottom', legend.text = element_text(size =22, angle = 45, hjust = 1),                  legend.title = element_text(size = 40),                
legend.key.size = unit(0.5, "in")) 
ggsave(paste0(FigOut,"ProtonPump_CorHeat_Supplemental.pdf"), width = 11, height = 7.33, units = "in", device = cairo_pdf)

```

#---------------
# Fig. 5: Single-condition PSEA and endocytosis figure components
#---------------
```{r Fig5a: Joint LPS and untreated PSEA fig}
Joint_PC_PSEA <- rbind(LPS_PSEA.f_fdr2,NT_PSEA.f_fdr)

Joint_PC_PSEA$Process <- ifelse(Joint_PC_PSEA$Process == "antigen processing and presentation of exogenous peptide antigen via MHC class II","antigen processing and presentation via MHC class II", ifelse(Joint_PC_PSEA$Process=="nuclear-transcribed mRNA catabolic process, nonsense-mediated decay","nuclear-transcribed mRNA catabolic process, NMD",ifelse(Joint_PC_PSEA$Process=="SRP-dependent cotranslational protein targeting to membrane","SRP-dependent protein targeting to membrane",Joint_PC_PSEA$Process))) 

Ccond.f_fdr_sel <- NT_PSEA.f_fdr %>% select(Process,PC,Association)
Ccond.f_fdr_clustIn <- (dcast(Ccond.f_fdr_sel,Process~PC))
row.names(Ccond.f_fdr_clustIn) <- Ccond.f_fdr_clustIn$Process
Ccond.f_fdr_clustIn <- Ccond.f_fdr_clustIn %>% select(-Process)
Ccond.f_fdr_clustIn <- as.matrix(Ccond.f_fdr_clustIn)
Ccond.f_fdr_clustIn[is.na(Ccond.f_fdr_clustIn)] <- 0
Ccond_dist1 <- dist(Ccond.f_fdr_clustIn)
Ccond_dist2 <- dist(t(Ccond.f_fdr_clustIn))
Ccond_clust1 <- hclust(t(Ccond_dist1), method = "ward.D")
Ccond_clust2 <- hclust(t(Ccond_dist2), method = "ward.D")
Joint_PC_PSEA_sum <- Joint_PC_PSEA %>% dplyr::group_by(Process) %>% dplyr::summarize(numCond = length(unique(Condition))) %>% dplyr::arrange(numCond)
Joint_PC_PSEA_sum1 <- Joint_PC_PSEA_sum %>% dplyr::filter(numCond ==1)
ProcOrderCc <- (Ccond_clust1$labels[Ccond_clust1$order])
ProcOrderCc_filt <- ProcOrderCc[!(ProcOrderCc)%in%Joint_PC_PSEA_sum1$Process]
interProts <- intersect(levels(factor(Joint_PC_PSEA$Process)),ProcOrderCc_filt) 
ProcOrderCc2 <- c(ProcOrderCc_filt,levels(factor(Joint_PC_PSEA$Process))[(!levels(factor(Joint_PC_PSEA$Process)) %in% interProts)])             
Joint_PC_PSEA$negLog10q <- -log10(Joint_PC_PSEA$qVal)              
Joint_PC_PSEA$nLog_bin <- ifelse(Joint_PC_PSEA$negLog10q >= 3,3,Joint_PC_PSEA$negLog10q)
Joint_PC_PSEA_termSum <- Joint_PC_PSEA %>% dplyr::group_by(Process) %>% dplyr::summarize(Process_order = n(), sumPC = sum(PC)) %>% dplyr::arrange(Process_order, sumPC)
Joint_PC_PSEA_termSum$Ind <- seq(1,nrow(Joint_PC_PSEA_termSum)) 
Joint_PC_PSEA <- Joint_PC_PSEA %>% dplyr::left_join(Joint_PC_PSEA_termSum, by = "Process")
Joint_PC_PSEA$PC_lab <- paste0("PC",Joint_PC_PSEA$PC)
Joint_PC_PSEA$Condition2 <- ifelse(Joint_PC_PSEA$Condition == "LPS","LPS-treated","Untreated")


Joint_PC_PSEA %>% dplyr::filter(PC != 6) %>% ggplot(aes(x = factor(Condition2, levels = c("Untreated","LPS-treated")), y =reorder(str_wrap(Process, width = 30), -Ind),fill = (Association), size = nLog_bin)) + 
  geom_point(shape = 21, stroke = 1) + 
  scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0, breaks = Xc_breaks, limits = c(-2.5,2.5)) + 
  theme_light() + 
  labs(fill = "Loading\nz-score", y = "",x = "",size = bquote("-log"[10]~"(q val)")) + 
  theme(axis.text.y = element_text(size = 16),
        axis.text.x = element_text(angle = 45, hjust = 1, size = 16), 
        legend.title =element_text(size = 28), 
        legend.text = element_text(size = 20, hjust = 0.4),
        strip.background = element_rect(fill = "white", color = "grey70"),
        strip.text = element_text(size = 22, color = "grey20", face = "bold"), 
        axis.title.x = element_text(size = 18), 
        legend.position = "bottom", legend.box="vertical", legend.margin=margin(),plot.margin=unit(c(0,1,0,0),"cm")) + 
  facet_wrap(~PC_lab,nrow = 1) + 
  scale_color_manual(values = c("red","blue")) +
  scale_size_continuous(breaks = seq(2,3,1),range=c(3,15),
                  limits = c(1, 4), labels = c("2","> 3"))
ggsave(paste0(FigOut,"CondComp_PC_PSEA_v3.pdf"), width = 9, height = 9, units = "in")
```


```{r Fig 5a: color-coded PCA (Proton transport)}
PCAout_NT_coords.df <- as.data.frame(PCAout_NT_coords[,1:5])
PCAout_LPS_coords.df <- as.data.frame(PCAout_LPS_coords[,1:5])
PCAout_NT_coords.df$id <- row.names(PCAout_NT_coords.df)
PCAout_LPS_coords.df$id <- row.names(PCAout_LPS_coords.df)
prePCA_LPS_norm.m <- melt(prePCA_LPS_norm)
prePCA_NT_norm.m <- melt(prePCA_NT_norm)

#LPS cells (unimputed)
BC_imp_LPS_PxC3.m <- prePCA_LPS_norm.m %>% left_join(ProtMissing.m, by = c("Var1"="Proteins", "Var2"="variable"))
BC_imp_LPS_PxC3.m$newVar <- ifelse(is.na(BC_imp_LPS_PxC3.m$value.y),NA,BC_imp_LPS_PxC3.m$value.x)
BC_unimp_LPS_PxC3.m <- BC_imp_LPS_PxC3.m %>% select(Var1,Var2,newVar)

#Untreated cells (unimputed)
BC_imp_NT_PxC3.m <- prePCA_NT_norm.m %>% left_join(ProtMissing.m, by = c("Var1"="Proteins", "Var2"="variable"))
BC_imp_NT_PxC3.m$newVar <- ifelse(is.na(BC_imp_NT_PxC3.m$value.y),NA,BC_imp_NT_PxC3.m$value.x)
BC_unimp_NT_PxC3.m <- BC_imp_NT_PxC3.m %>% select(Var1,Var2,newVar)

#untreated sets
NT_sets <- c("proton transport, V-type ATPase")

#LPS sets
LPS_sets <- c("proton transport, V-type ATPase")

BC_unimp_NT_GOinput <- BC_unimp_NT_PxC3.m %>% left_join(GeneMappings_single2, by = c("Var1"="UNIPROTKB"))
BC_unimp_LPS_GOinput <- BC_unimp_LPS_PxC3.m %>% left_join(GeneMappings_single2, by = c("Var1"="UNIPROTKB"))
GOout_NT.df <- GO_for_PCA_BMDM(NT_sets,BC_unimp_NT_GOinput)
GOout_LPS.df<- GO_for_PCA_BMDM(LPS_sets,BC_unimp_LPS_GOinput)
PCAout_NT_coords.df2 <- GOout_NT.df %>% left_join(PCAout_NT_coords.df, by = "id")
PCAout_NT_coords.df2$Condition <- "Untreated"
PCAout_LPS_coords.df2 <- GOout_LPS.df %>% left_join(PCAout_LPS_coords.df, by = "id")
PCAout_LPS_coords.df2$Condition <- "LPS-treated"

ProtonNT <- PCAout_NT_coords.df2 %>% ggplot(aes(x = -PC1, y = PC2, color =`proton transport, V-type ATPase`)) + geom_point(size = 3, alpha = 0.7, shape = 16) + scale_color_gradient2(low = "blue", mid = "white", high = "red", midpoint=0) + theme_light() + theme(plot.title = element_text(size = 28), legend.position = "null", axis.text = element_blank(), axis.title = element_text(size = 28), strip.text = element_text(size = 24),strip.background = element_rect(color = "grey40",fill = "grey40"))  + labs(x = "PC1", color = "")+ theme(legend.position = "null") + ggtitle("Proton Transport") + facet_wrap(~Condition, strip.position="top")
ProtonLPS <- PCAout_LPS_coords.df2 %>% ggplot(aes(x = PC1, y = PC2, color =`proton transport, V-type ATPase`)) + geom_point(size = 3, alpha = 0.7, shape = 16) + scale_color_gradient2(low = "blue", mid = "white", high = "red", midpoint=0) + theme_light() +theme(plot.title = element_text(size = 28), legend.position = "null", axis.text = element_blank(), axis.title = element_text(size = 28), strip.text = element_text(size = 24),strip.background = element_rect(color = "grey40",fill = "grey40")) + labs(color = "")+ theme(legend.position = "null") + ggtitle("Proton Transport")+ facet_wrap(~Condition, strip.position="top")
(ProtonNT + ProtonLPS) + plot_layout(guides = 'collect') & theme(legend.position = "null")
ggsave(paste0(FigOut,"Proton_v2.pdf"), width = 8, height = 4, units = "in")
```

```{r Protein names from original FASTA file}
library(seqinr)
myfasta<- read.fasta(file = paste0(datPath,"musmusculus_SPonly_MEROPS_012221.fasta"), seqtype = "AA",as.string = TRUE, set.attributes = FALSE, whole.header = TRUE)
FASTA_in_proteins <- as.data.frame(names(myfasta))
FASTA_in_proteins$Prot <- gsub("sp\\|","",FASTA_in_proteins$`names(myfasta)`)
FASTA_in_proteins$Prot <- gsub("\\|.*","",FASTA_in_proteins$Prot)
FASTA_in_proteins$names <- gsub(".*MOUSE","",FASTA_in_proteins$`names(myfasta)`)
FASTA_in_proteins$names <- gsub("OS.*","",FASTA_in_proteins$names)
FASTA_in_proteins$genes <- gsub("sp\\|","",FASTA_in_proteins$`names(myfasta)`)
FASTA_in_proteins$genes <- gsub(".*\\|","",FASTA_in_proteins$genes)
FASTA_in_proteins$genes <- gsub("_MOUSE.*","",FASTA_in_proteins$genes)
ProtRef_df <- as.data.frame(FASTA_in_proteins[,c(2,3)])
ProtRef_df_gene <- as.data.frame(FASTA_in_proteins[,c(2,4)])
detach("package:seqinr", unload = TRUE)
```


```{r Fig5b: Dextran data import}
Dextran_DIA_LPS_filt <- Dextran_DIA_LPS %>% filter(direction != "Not Significant",abs(median_prot_FC) >= 1,!grepl("keratin",PG.ProteinDescriptions))
Dextran_DIA_NT_filt <- Dextran_DIA_NT %>% filter(direction != "Not Significant",abs(median_prot_FC) >= 1,!grepl("keratin",PG.ProteinDescriptions))
# Extracting sets of antiendocytic and proendocytic proteins
Dextran_DIA_LPS_filt_AntiEndocytic <- Dextran_DIA_LPS %>% filter(direction != "Not Significant",median_prot_FC >= 1,n>1,FC_qval <= .01)
Dextran_DIA_LPS_filt_Endocytic <- Dextran_DIA_LPS %>% filter(direction != "Not Significant",median_prot_FC <= -1,n>1,FC_qval <= .01)
Dextran_DIA_NT_filt_AntiEndocytic <- Dextran_DIA_NT %>% filter(direction != "Not Significant",median_prot_FC >= 1,n>1,FC_qval <= .01)
Dextran_DIA_NT_filt_Endocytic <- Dextran_DIA_NT %>% filter(direction != "Not Significant",median_prot_FC <= -1,n>1,FC_qval <= .01)
Dextran_DIA_LPS_filt_AntiEndocytic_lim <- Dextran_DIA_LPS_filt_AntiEndocytic %>% select(PG.ProteinAccessions)
Dextran_DIA_LPS_filt_Endocytic_lim <- Dextran_DIA_LPS_filt_Endocytic %>% select(PG.ProteinAccessions)
Dextran_DIA_NT_filt_AntiEndocytic_lim <- Dextran_DIA_NT_filt_AntiEndocytic %>% select(PG.ProteinAccessions)
Dextran_DIA_NT_filt_Endocytic_lim <- Dextran_DIA_NT_filt_Endocytic %>% select(PG.ProteinAccessions)
Dextran_dict_LPS <- Dextran_DIA_LPS %>% select(PG.ProteinAccessions,PG.ProteinDescriptions) %>% group_by(PG.ProteinAccessions) %>% slice(1)
Endocytosis_union <- as.data.frame(intersect(Dextran_DIA_LPS_filt_Endocytic_lim$PG.ProteinAccessions,Dextran_DIA_NT_filt_Endocytic_lim$PG.ProteinAccessions))
colnames(Endocytosis_union) <- "PG.ProteinAccessions"
Endocytosis_LPS_outer <- Dextran_DIA_LPS_filt_Endocytic_lim %>% filter(!(PG.ProteinAccessions %in% Endocytosis_union$PG.ProteinAccessions))
Endocytosis_NT_outer <- Dextran_DIA_NT_filt_Endocytic_lim %>% filter(!(PG.ProteinAccessions %in% Endocytosis_union$PG.ProteinAccessions))
Endocytosis_union <- Endocytosis_union %>% left_join(Dextran_dict_LPS, by = "PG.ProteinAccessions")
```

```{r Fig5b (middle panel) LPS Dextran-Uptake Volcano Plot for differentially abundant proteins}
require("ggrepel")
Dextran_DIA_LPS2 <- Dextran_DIA_LPS
Dextran_DIA_LPS2$Labs <- ifelse(Dextran_DIA_LPS2$direction == "Decreased","High uptake",ifelse(Dextran_DIA_LPS2$direction == "Increased","Low uptake","Not significant"))
Dextran_DIA_LPS2$GeneLab2 <- ifelse(Dextran_DIA_LPS2$median_prot_FC < -3,Dextran_DIA_LPS2$Gene.names...primary..,ifelse(Dextran_DIA_LPS2$median_prot_FC > 3,Dextran_DIA_LPS2$Gene.names...primary..,""))
Dextran_DIA_LPS2$GeneLab3 <- ifelse(Dextran_DIA_LPS2$Gene.names...primary.. %in% c("Mrc1","Stab1","Snx17"),Dextran_DIA_LPS2$Gene.names...primary..,"")
Dextran_DIA_LPS2$negLog_q <- -log10(Dextran_DIA_LPS2$FC_qval)
Dextran_DIA_LPS2$negLog_q_bin <- ifelse(Dextran_DIA_LPS2$negLog_q >= 30, 30,Dextran_DIA_LPS2$negLog_q)

Dextran_DIA_LPS2 %>% ggplot(aes(x = -median_prot_FC, y = negLog_q_bin, color = factor(Labs, levels = c("High uptake","Low uptake","Not significant")))) + geom_point(shape = 16,size = 5,alpha = 0.7) + theme_light() + labs(x = bquote("High / low uptake, log"[2]), y = bquote("-log"[10]~"(q val)"), color = "") + scale_color_manual(values = c("#009E73", "#CC79A7","#888888"))+ theme(axis.text = element_text(size = 24), axis.title = element_text(size = 32), axis.title.y = element_text(margin = margin(t = 0, r = 15, b = 0, l = 0)), legend.text = element_text(size = 28), legend.title = element_blank(), legend.position="null",plot.margin = unit(c(1,0,0,0),"cm")) + geom_text_repel(aes(label = GeneLab2),size = 8, color = "black",min.segment.length = unit(0.33, "lines")) + coord_cartesian(xlim = c(-6.5,6.5), ylim = c(0,33))
ggsave(paste0(FigOut,"LPS_DexVolcano_wide_v5_SelectLabs2.pdf"), width =8, height = 4, units = "in")
```

```{r Fig5b (Top Panel) Dextran Uptake Sorting histogram}
library(tidyr)
library(scales)

LPS_uptake1 <- read.delim(paste0(datPath,"dextran_graph_LPS1.txt"))
LPS_uptake2 <- read.delim(paste0(datPath,"dextran_graph_LPS2.txt"))

LPS_uptake1_obs <- LPS_uptake1 %>% uncount(weights = Events)
LPS_uptake1_obs.f <- LPS_uptake1_obs %>% filter((LPS_uptake1_obs$MFI > 1000)&(LPS_uptake1_obs$MFI < 50000))


LPS_uptake1_obs.f$log10MFI <- log10(LPS_uptake1_obs.f$MFI)

LPS_quants <- (quantile((LPS_uptake1_obs.f$MFI),c(.1,.5,.9)))

#ggplot(LPS_uptake1_obs.f, aes(x = (MFI), y = ..count.., fill = ..x..)) +
#  geom_histogram(binwidth = .05, pad = TRUE, color = "grey70") +
#  scale_x_log10(breaks = trans_breaks("log10", function(x) 10^x),
#              labels = trans_format("log10", math_format(.x))) + 
#  scale_fill_gradient2(low = "#ad0060",high = "#008f67", midpoint = log10(LPS_quants[2]))  + theme_light() +labs(y = "Cells", x = bquote("Dextran Intensity, log"[10])) + theme(axis.text = element_text(size = 24), axis.title = element_text(size = 32), legend.text = element_text(size = 26), legend.position="null",plot.margin =unit(c(1,0,0,0),"cm")) + geom_vline(xintercept = LPS_quants[[1]],linetype = "dashed", color = "grey60", size = 2) + geom_vline(xintercept = LPS_quants[[3]],linetype = "dashed",color = "grey60", size = 2) + 
#  annotate("text",x = (LPS_quants[[1]] - 3750) , y = 1550,size = 12, label = "Low", color = "#D1509D")+
#  annotate("text",x = (LPS_quants[[1]] - 3750) , y = 1275,size = 12, label = "Uptake", color = "#D1509D")+
#  annotate("text",x = (LPS_quants[[1]] - 3750) , y = 1000,size = 8.5, label = "10th Percentile", color = "#D1509D")+
#  annotate("text",x = (LPS_quants[[3]] + 44000) , y = 1550,size = 12 ,label = "High", color = "#02875E") +
#  annotate("text",x = (LPS_quants[[3]] + 44000) , y = 1275,size = 12 ,label = "Uptake", color = "#02875E") +
#  annotate("text",x = (LPS_quants[[3]] + 44000) , y = 1000,size = 8.5 ,label = "90th Percentile", color = "#02875E") +coord_cartesian(xlim = c(10^3,11^5))
#ggsave(paste0(FigOut,"LPS_DextranSort_wide.pdf"), width = 8.15, height = 4.2, units = "in")


ggplot(LPS_uptake1_obs.f, aes(x = (MFI), y = ..count../sum(..count..), fill = ..x..)) +
  geom_histogram(binwidth = .05, color = "grey70") +
  scale_x_log10(breaks = trans_breaks("log10", function(x) 10^x),
              labels = trans_format("log10", math_format(.x))) + scale_y_continuous(labels = percent_format()) +
  scale_fill_gradient2(low = "#ad0060",high = "#008f67", midpoint = log10(LPS_quants[2]))  + theme_light() +labs(y = "Cells, %", x = bquote("Dextran: Alexa Fluor 568, log"[10])) + theme(axis.text = element_text(size = 24), axis.title = element_text(size = 32), legend.text = element_text(size = 26), legend.position="null",plot.margin =unit(c(1,0,0,0),"cm")) + geom_vline(xintercept = LPS_quants[[1]],linetype = "dashed", color = "grey60", size = 2) + geom_vline(xintercept = LPS_quants[[3]],linetype = "dashed",color = "grey60", size = 2) + 
  annotate("text",x = (LPS_quants[[1]] - 3750) , y = .085,size = 12, label = "Low", color = "#D1509D")+
  annotate("text",x = (LPS_quants[[1]] - 3750) , y = .072,size = 12, label = "Uptake", color = "#D1509D")+
  annotate("text",x = (LPS_quants[[1]] - 3750) , y = .060,size = 8.5, label = "10th Percentile", color = "#D1509D")+
  annotate("text",x = (LPS_quants[[3]] + 44000) , y = .085,size = 12 ,label = "High", color = "#02875E") +
  annotate("text",x = (LPS_quants[[3]] + 44000) , y = .072,size = 12 ,label = "Uptake", color = "#02875E") +
  annotate("text",x = (LPS_quants[[3]] + 44000) , y = .060,size = 8.5 ,label = "90th Percentile", color = "#02875E") +coord_cartesian(xlim = c(10^3,11^5))
ggsave(paste0(FigOut,"LPS_DextranSort_wide_LPS_percent.pdf"), width = 8.15, height = 4.2, units = "in")

```

```{r Fig5b (bottom panel) Dextran Uptake Annotated PCA}
#PCAout_NT_coords.df2 <- GOout_NT.df %>% left_join(PCAout_NT_coords.df, by = "id")
#PCAout_NT_coords.df2$Condition <- "Untreated"

Dextran_DIA_LPS_filt_Endocytic_lim$PG.ProteinAccessions
Dextran_DIA_LPS_filt_AntiEndocytic_lim$PG.ProteinAccessions

BC_unimp_LPS_PxC3.m_ProEndo <- BC_unimp_LPS_PxC3.m %>% 
    dplyr::filter(Var1 %in% Dextran_DIA_LPS_filt_Endocytic_lim$PG.ProteinAccessions) %>%
    dplyr::group_by(Var2) %>% dplyr::summarize(ProEndocytic = median(newVar,na.rm = TRUE))
BC_unimp_LPS_PxC3.m_AntiEndo <- BC_unimp_LPS_PxC3.m %>% 
    dplyr::filter(Var1 %in% Dextran_DIA_LPS_filt_AntiEndocytic_lim$PG.ProteinAccessions) %>%
    dplyr::group_by(Var2) %>% dplyr::summarize(AntiEndocytic = median(newVar,na.rm = TRUE))
#z-score abundances
BC_unimp_LPS_PxC3.m_ProEndo$ProEndocytic <- (BC_unimp_LPS_PxC3.m_ProEndo$ProEndocytic - mean(BC_unimp_LPS_PxC3.m_ProEndo$ProEndocytic, na.rm = TRUE))/sd(BC_unimp_LPS_PxC3.m_ProEndo$ProEndocytic, na.rm = TRUE)
BC_unimp_LPS_PxC3.m_AntiEndo$AntiEndocytic <- (BC_unimp_LPS_PxC3.m_AntiEndo$AntiEndocytic - mean(BC_unimp_LPS_PxC3.m_AntiEndo$AntiEndocytic, na.rm = TRUE))/sd(BC_unimp_LPS_PxC3.m_AntiEndo$AntiEndocytic, na.rm = TRUE)
BC_unimp_LPS_PxC3.m_ProEndo$ProEndocytic <- ifelse(BC_unimp_LPS_PxC3.m_ProEndo$ProEndocytic >= 2, 2, ifelse(BC_unimp_LPS_PxC3.m_ProEndo$ProEndocytic <= -2,-2,BC_unimp_LPS_PxC3.m_ProEndo$ProEndocytic))
BC_unimp_LPS_PxC3.m_AntiEndo$AntiEndocytic <- ifelse(BC_unimp_LPS_PxC3.m_AntiEndo$AntiEndocytic >= 2, 2,ifelse(BC_unimp_LPS_PxC3.m_AntiEndo$AntiEndocytic <= -2,-2,BC_unimp_LPS_PxC3.m_AntiEndo$AntiEndocytic))
#PCAout_LPS_coords.df2 <- GOout_LPS.df %>% left_join(PCAout_LPS_coords.df, by = "id")
#PCAout_LPS_coords.df2$Condition <- "LPS-treated"
PCAout_LPS_coords.df_endo <- PCAout_LPS_coords.df
PCAout_LPS_coords.df_endo <- PCAout_LPS_coords.df_endo %>% left_join(BC_unimp_LPS_PxC3.m_ProEndo, by = c("id"="Var2"))
PCAout_LPS_coords.df_endo <- PCAout_LPS_coords.df_endo %>% left_join(BC_unimp_LPS_PxC3.m_AntiEndo, by = c("id"="Var2"))

LPS_Anti_scatter <- PCAout_LPS_coords.df_endo %>% ggplot(aes(x = PC1,y = PC2, color = AntiEndocytic)) + geom_point(shape = 16, size = 5, alpha = 0.75) + scale_color_gradient2(low = "blue",mid = "white", high = "red", midpoint = 0) + theme_light() + labs(x = "PC1", y = "PC2")  + theme(plot.title = element_text(size = 32, color = "#D1509D"), legend.position = "null", axis.text = element_blank(), axis.title = element_text(size = 32)) + ggtitle("Low Uptake\nProteins")

LPS_Pro_scatter <- PCAout_LPS_coords.df_endo %>% ggplot(aes(x = PC1,y = PC2, color = ProEndocytic)) + geom_point(shape = 16, size = 5, alpha = 0.75) + scale_color_gradient2(low = "blue",mid = "white", high = "red", midpoint = 0) + theme_light() + labs(x = "PC1", y = "PC2")  + theme(plot.title = element_text(size = 32, color = "#02875E"), legend.position = "null", axis.text = element_blank(), axis.title = element_text(size = 32)) + ggtitle("High Uptake\nProteins")

LPS_Anti_scatter + plot_spacer() + LPS_Pro_scatter + plot_layout(widths = c(5,0.75,5))
ggsave(paste0(FigOut,"EndoPCA_v3_forSCoPE_BluetoRed_wide_colorCoded.pdf"), width = 9, height = 5, units = "in")
```

#---------------
# Fig 6: MEROPS Figure Components
#---------------

```{r MEROPS analysis section}
#LPS Channel = Reporter.intensity.corrected.3
#Untreated Channel = Reporter.intensity.corrected.4

BulkDDARatioSet_1 <- BulkDDARatioSet %>% filter(Raw.file == "wGH215", PEP <= .02, PIF > 0.8, !grepl("REV",Leading.razor.protein),!grepl("CON",Leading.razor.protein)) %>% mutate(SeqCharge = paste0(Modified.sequence,Charge)) %>% group_by(SeqCharge) %>% top_n(-1,PEP) %>% slice(1) %>% select(Leading.razor.protein,SeqCharge,Reporter.intensity.corrected.3,Reporter.intensity.corrected.4)
BulkDDARatioSet_1_Seq_Prot <- BulkDDARatioSet_1 %>% select(SeqCharge,Leading.razor.protein)

# Column and row normalize intensities
# Then compare the distribution of intensities for precursors that map to the same Protein
bulk_int <- BulkDDARatioSet_1 %>% select(SeqCharge,Reporter.intensity.corrected.3,Reporter.intensity.corrected.4)                           
bulk_int <- as.data.frame(bulk_int)
row.names(bulk_int) <- bulk_int$SeqCharge                                         
bulk_int_preMat <- bulk_int %>% select(Reporter.intensity.corrected.3,Reporter.intensity.corrected.4)  
colnames(bulk_int_preMat) <- c("LPS","NT")
bulk_int.mat <- as.matrix(bulk_int_preMat)
bulk_int.mat_colMean <- matrixStats::colMeans2(bulk_int.mat)
bim_diag <- diag(1/bulk_int.mat_colMean)
bim_CN <- bulk_int.mat %*% bim_diag
bim_RowMean <- matrixStats::rowMeans2(bim_CN)
bim_RMdiag <- diag(1/bim_RowMean)
bim_CNRN <- bim_RMdiag %*% bim_CN
bim_CNRN.df <- as.data.frame(bim_CNRN)
colnames(bim_CNRN.df) <-  c("LPS","NT")
row.names(bim_CNRN.df) <- row.names(bulk_int)
bim_CNRN.df$SeqCharge <- row.names(bim_CNRN.df)
bim_CNRN.df <- bim_CNRN.df %>% left_join(BulkDDARatioSet_1_Seq_Prot, by = "SeqCharge")
Prot_Comp_out <- data.frame(Protein = character(), Ratio = numeric(), p.value = numeric(),numSeq = numeric(), stringsAsFactors = FALSE)

for(i in 1:length(unique(bim_CNRN.df$Leading.razor.protein))){
  Prot_single <- bim_CNRN.df %>% filter(Leading.razor.protein == unique(bim_CNRN.df$Leading.razor.protein)[i])
  Prot_Comp_out[i,1] <- unique(bim_CNRN.df$Leading.razor.protein)[i]
  if(nrow(Prot_single) > 2){
    Prot_Comp_out[i,2] <- median(Prot_single$LPS,na.rm = TRUE)/median(Prot_single$NT,na.rm = TRUE)
    wt <- wilcox.test(Prot_single$LPS,Prot_single$NT, alternative = "two.sided")
    Prot_Comp_out[i,3] <- wt$p.value
    Prot_Comp_out[i,4] <- nrow(Prot_single)
  }
}

Prot_Comp_out.f <- Prot_Comp_out %>% filter(!is.na(p.value))
Prot_Comp_out.f$qVal <- p.adjust(Prot_Comp_out.f$p.value)
Prot_Comp_out.f2 <- Prot_Comp_out.f %>% filter(qVal <= .01)
Prot_Comp_out.f2$Type <- ifelse(Prot_Comp_out.f2$Ratio > 1, "Inflammatory","Control")
Prot_Comp_out.f3_Control <- Prot_Comp_out.f2 %>% top_n(-20,Ratio)
Prot_Comp_out.f3_Inf <- Prot_Comp_out.f2 %>% top_n(20,Ratio)
MarkerDB_DDA <- rbind(Prot_Comp_out.f3_Control,Prot_Comp_out.f3_Inf)
MarkerDB_DDA.f <- MarkerDB_DDA %>% select(Protein,Type)
colnames(MarkerDB_DDA.f) <- c("Entry","Type") 
```

```{r External marker db prep}
MarkerDB_SCoPE2$Gene.Name <- str_sub(MarkerDB_SCoPE2$Gene.Name, start = 2)
MarkerDB_SCoPE2 <- MarkerDB_SCoPE2 %>% select(Gene.Name,Gene.Symbol,M1,M2,M1.M2.Ratio) %>% arrange(-M1.M2.Ratio)
MarkerDB_SCoPE2$Type <- ifelse(MarkerDB_SCoPE2$M1.M2.Ratio > 0, "Inflammatory","AntiInflammatory")
MarkerDB_SCoPE2 <- MarkerDB_SCoPE2 %>% filter(!is.na(Type))
MarkerDB_SCoPE2.join <- MarkerDB_SCoPE2 %>% left_join(SCoPE2_Human2Mouse,by = c("Gene.Symbol"="Gene.Entry"))
MarkerDB_SCoPE2.2 <- MarkerDB_SCoPE2.join %>% select(Entry, Type)
```

```{r Fig 6b: correlation analysis for MEROPS peptides, message = FALSE, echo = FALSE, warning = FALSE}
# MEROPS peptides
MEROPSvec <- c("P60710_DIG_10362102",	"Q9CZU6_DIG_55521","P60710_DIG_1036223","P60710_DIG_1036222")
#************
# Uncomment these lines to perform correlation analysis
#************
# Condition-specific db
#Ratio_markOut <- CorPermutation(BC_unImp_CxP,MarkerDB_DDA.f,"Internal")
# External db
#SCoPE2_markOut <- CorPermutation(BC_unImp_CxP,MarkerDB_SCoPE2.2,"SCoPE2")
#************
#Uncomment these lines to store your results
#write.table(Ratio_markOut,paste0(FigOut,"Ratio_markOut.txt"))
#write.table(SCoPE2_markOut,paste0(FigOut,"SCoPE2_markOut.txt"))

#Read in previously generated correlation results
Ratio_markOut <- read.table(paste0(datPath,"Ratio_markOut.txt"))
SCoPE2_markOut <- read.table(paste0(datPath,"SCoPE2_markOut.txt"))

# combining correlation analysis results
db_bind <- rbind(Ratio_markOut,SCoPE2_markOut)
db_bind2 <- db_bind %>% select(-CorDist)
db_bind.m <- melt(db_bind2, id.vars = c("MEROPS","fraction","qVal","qVal2","lab"))
db_bind.m$xLab <- ifelse((db_bind.m$variable == "corPro")&(db_bind.m$lab == "Internal"),"LPS-treated",ifelse((db_bind.m$variable == "corAnti")&(db_bind.m$lab == "Internal"),"Untreated",ifelse((db_bind.m$variable == "corPro")&(db_bind.m$lab == "SCoPE2"),"M1","M2")))

db_bind.m <- db_bind.m %>% left_join(ProtRef_df, by = c("MEROPS"="Prot"))
db_bind.m$EditedName <- ifelse(grepl("_25_",db_bind.m$names),"Citrate synthase: H25, F2, cathepsin E",ifelse(grepl("_104_",db_bind.m$names),"Actin: L104, F1, cathepsin D",ifelse(grepl("_299__Half1",db_bind.m$names),"Actin: L299, F1, cathepsin D","Actin: L299, F2, cathepsin D")))
levelVec <- c("Actin: L104, F1, cathepsin D","Citrate synthase: H25, F2, cathepsin E","Actin: L299, F2, cathepsin D","Actin: L299, F1, cathepsin D")

db_bind.m$refName <- ifelse(db_bind.m$lab == "Internal","Bulk","Martinez, et al.")
db_bind.m$nLogQVal <- -log10(db_bind.m$qVal2)
db_bind.m$nLogQVal_bin <- ifelse(db_bind.m$nLogQVal >= 3, 3,db_bind.m$nLogQVal)

# Figure 6b
db_bind.m %>% ggplot(aes(x = factor(xLab,levels = c("Untreated","LPS-treated","M2","M1")), y = factor(EditedName,levels = rev(levelVec)), size = nLogQVal_bin, fill = value)) + geom_point(shape = 21, stroke = 1) + theme_light() + facet_wrap(~lab) + scale_fill_gradient2(low = "blue", mid = "white", high = "red",midpoint = 0,breaks=c(-.3,0,0.2)) + theme(axis.text.y = element_text(size = 18),axis.text.x = element_text(angle = 45, hjust = 1, size = 22), legend.title =element_text(size = 23.5), legend.text = element_text(size = 20,hjust =1),strip.background = element_rect(fill = "white", color = "grey70"), strip.text = element_text(size = 24, color = "grey20", face = "bold"), axis.title.x = element_text(size = 34)) + labs(x = "Protein Sets",y="",fill = "Correlation",size = expression("-log"[10]~"(q val)")) + facet_wrap(~refName, scales = "free_x") +
  scale_size_continuous(breaks = seq(2,3,1),range=c(1,15),
                  limits = c(0, 3), labels = c("2",">3")) + scale_y_discrete(labels = c("Actin: L104, F1, cathepsin D" = "Fragment 1", "Citrate synthase: H25, F2, cathepsin E" =  "Fragment 2", "Actin: L299, F2, cathepsin D" = "Fragment 2", "Actin: L299, F1, cathepsin D" = "Fragment 1") )
ggsave(paste0(FigOut,"Internal_and_SCoPE2_DB_MEROPS_v8.pdf"), width = 9, height = 5.25)
```

```{r Fig 6a: Agreement between bulk and single-cell measurements}
#-------------------------------
# Bulk data preprocessing
#-------------------------------
DIA_Bulk$SeqCharge <- paste0(DIA_Bulk$EG.ModifiedSequence,DIA_Bulk$FG.Charge)
DIA_Bulk_LPS <- DIA_Bulk %>% filter(R.FileName == "eGH694", EG.PEP <= .01)
DIA_Bulk_NT <- DIA_Bulk %>% filter(R.FileName == "eGH693", EG.PEP <= .01) 
DIA_bulk_LPS.sel <- DIA_Bulk_LPS %>% select(SeqCharge,EG.TargetQuantity..Settings.) %>% group_by(SeqCharge) %>% slice(1)
DIA_bulk_NT.sel <- DIA_Bulk_NT %>% select(SeqCharge,EG.TargetQuantity..Settings.) %>% group_by(SeqCharge) %>% slice(1)
DIA.j <- DIA_bulk_LPS.sel %>% inner_join(DIA_bulk_NT.sel, by = "SeqCharge")
DIA.j.df <- as.data.frame(DIA.j)
row.names(DIA.j.df) <- DIA.j.df$SeqCharge

# Column and Row normalizing Precursors
# Taking condition-specific ratios
DIA.j.df.sel <- DIA.j.df %>% select(-SeqCharge) 
DIA.j.mat <- as.matrix(DIA.j.df.sel)
colMed_dex <- matrixStats::colMedians(DIA.j.mat)
colMed_diag <- diag(1/colMed_dex)
DIA.j.mat.CN <- DIA.j.mat %*% colMed_diag
rowMean_dex <- matrixStats::rowMeans2(DIA.j.mat.CN)
DIA.j.mat.CN.df <- as.data.frame(DIA.j.mat.CN)
DIA.j.mat.CN.df$RowmeanFrac <- (1/rowMean_dex)
DIA.j.mat.CN.df[,c(1,2)] <- DIA.j.mat.CN.df[,c(1,2)]*DIA.j.mat.CN.df[,3]
DIA.j.mat.CNRN.df <- DIA.j.mat.CN.df %>% select(V1,V2)
colnames(DIA.j.mat.CNRN.df) <- c("LPS","NT")
SeqProtLib <- DIA_Bulk %>% select(SeqCharge,PG.ProteinAccessions,PG.ProteinDescriptions) %>% group_by(SeqCharge) %>% slice(1)
DIA.j.mat.CNRN.df$SeqCharge <- row.names(DIA.j.mat.CNRN.df)
DIA.j.mat.CNRN.df <- DIA.j.mat.CNRN.df %>% left_join(SeqProtLib, by = "SeqCharge")
DIA.j.mat.CNRN.df$Ratio <- DIA.j.mat.CNRN.df$LPS/DIA.j.mat.CNRN.df$NT

#-------------------------------
# Single-cell data preprocessing
#-------------------------------
CxP.m <- melt(BC_unImp_CxP)
CxP.m <- CxP.m %>% left_join(CG, by = "id")
CxP.m_sum <- CxP.m %>% dplyr::group_by(celltype,variable) %>% dplyr::summarize(medianInt = median(value, na.rm = TRUE))
CxP.m_sum2 <- CxP.m_sum %>% dplyr::group_by(variable) %>% dplyr::summarize(ratio = medianInt[celltype=="LPS"]-medianInt[celltype == "untreated"])
CxP.m_sum3 <- CxP.m_sum2 %>% left_join(ProtRef_df, by = c("variable" = "Prot"))
CxP.m_sum3$MatchID <- ifelse(CxP.m_sum3$variable == "P60710_DIG_10362102","P60710_DIG_10362103",CxP.m_sum3$variable)
CxP.m_sum3.lim <- CxP.m_sum3 %>% filter(variable %in% c("P60710_DIG_10362102",	
"P60710_DIG_1036222","P60710_DIG_1036223","Q9CZU6_DIG_55521"))
CxP.m_sum3.lim2 <- CxP.m_sum3.lim %>% dplyr::select(variable,names,ratio)
colnames(CxP.m_sum3.lim2) <- c("UniProt","Desc","log2Ratio")
CxP.m_sum3.lim2$Lab <- "Single Cell"
CxP.m_sum3.lim2$Desc <- str_sub(CxP.m_sum3.lim2$Desc, start =2)

# Generating plot 6a: agreement between bulk and single-cell measurements

DIA.j.mat.CNRN.df_sel <- DIA.j.mat.CNRN.df %>% filter(SeqCharge %in% c("_[+C15+H25+N3+O3]LTEAPLNPK_2","_[+C15+H25+N3+O3]ASASSTNLK_2","_[+C15+H25+N3+O3]SGGTTM[Oxidation (M)]YPGIADR_3"))
DIA.j.mat.CNRN.df_sel$joinDef <- gsub(";.*","",DIA.j.mat.CNRN.df_sel$PG.ProteinDescriptions)
DIA.j.mat.CNRN.df_sel$joinDef <- gsub("Half.*","",DIA.j.mat.CNRN.df_sel$joinDef)
DIA.j.mat.CNRN.df_sel$logRat <- log2(DIA.j.mat.CNRN.df_sel$Ratio)
DIA.j.mat.CNRN.df_sel.lim <- DIA.j.mat.CNRN.df_sel %>% dplyr::select(PG.ProteinAccessions,PG.ProteinDescriptions,logRat)
colnames(DIA.j.mat.CNRN.df_sel.lim) <- c("UniProt","Desc","log2Ratio")
DIA.j.mat.CNRN.df_sel.lim$Lab <- "Discovery\nBulk"
Bulk_and_SC <- rbind(DIA.j.mat.CNRN.df_sel.lim,CxP.m_sum3.lim2)
Bulk_and_SC$Desc2 <- gsub("__Half.*","",Bulk_and_SC$Desc)
Bulk_and_SC$Desc3 <- ifelse(Bulk_and_SC$Desc2 == "Actin, cytoplasmic 1 _cathepsin D_104","Actin: L104\ncathepsin D",ifelse(Bulk_and_SC$Desc2 == "Actin, cytoplasmic 1 _cathepsin D_299","Actin: L299\ncathepsin D","Citrate synthase: H25\ncathepsin E"))
Bulk_and_SC2 <- Bulk_and_SC
Bulk_and_SC2 <- Bulk_and_SC %>% dplyr::group_by(Desc3,Lab) %>% dplyr::summarize(log2Ratio2 = mean(log2Ratio, na.rm = TRUE))
FacetOrder <- c("Actin: L104\ncathepsin D","Citrate synthase: H25\ncathepsin E","Actin: L299\ncathepsin D")
#Bulk_and_SC2 %>% ggplot(aes(y = 1,x = Lab, fill = (log2Ratio2))) + geom_tile() + scale_fill_gradient2(limits = c(-.5,.5),low = "blue", mid = "white", high = "red", midpoint = 0) + labs(y = "", x = "") + facet_wrap(~factor(Desc3, levels = FacetOrder), ncol = 1) + labs(fill = expression("Log"[2]~"(LPS:Untreated)")) + theme_light() + theme(axis.text = element_text(size = 18), axis.text.y = element_blank(), strip.background = element_rect(fill = "white", color = "grey70"), strip.text = element_text(size = 22, color = "grey20", face = "bold"), legend.position = "bottom", legend.text = element_text(size = 14, angle = 45, hjust = 1), legend.title = element_text(size = 16))
#ggsave(paste0(FigOut,"BulkAndSC_Val_v2.png"), width = 6, height = 8)

Top2 <- Bulk_and_SC2 %>% filter(Desc3 %in% c("Actin: L104\ncathepsin D","Citrate synthase: H25\ncathepsin E")) %>% ggplot(aes(y = 1,x = Lab, fill = (log2Ratio2))) + geom_tile() + scale_fill_gradient2(limits = c(-.5,.5),low = "blue", mid = "white", high = "red", midpoint = 0) + labs(y = "", x = "") + facet_wrap(~factor(Desc3, levels = FacetOrder), ncol = 1) + labs(fill = expression("Log"[2]~"(LPS:Untreated)")) + theme_light() + theme( axis.text = element_blank(), strip.background = element_rect(fill = "white", color = "grey70"), strip.text = element_text(size = 18, color = "grey20", face = "bold"), legend.position = "bottom", legend.text = element_text(size = 14, angle = 45, hjust = 1),panel.spacing.y=unit(0, "lines"), legend.title = element_text(size = 16))

BottomFacet <- Bulk_and_SC2 %>% filter(!(Desc3 %in% c("Actin: L104\ncathepsin D","Citrate synthase: H25\ncathepsin E"))) %>% ggplot(aes(y = 1,x = Lab, fill = (log2Ratio2))) + geom_tile() + scale_fill_gradient2(limits = c(-.5,.5),low = "blue", mid = "white", high = "red", midpoint = 0) + labs(y = "", x = "") + facet_wrap(~factor(Desc3, levels = FacetOrder), ncol = 1) + labs(fill = expression("Log"[2]~"(LPS:Untreated)")) + theme_light() + theme(axis.text = element_text(size = 16), axis.text.y = element_blank(), strip.background = element_rect(fill = "white", color = "grey70"), strip.text = element_text(size = 18, color = "grey20", face = "bold"), legend.position = "bottom", legend.text = element_text(size = 14, angle = 45, hjust = 1), legend.title = element_text(size = 16),plot.margin = unit(c(0.0,0,0.3,0),"cm"))
Top2 / BottomFacet + plot_layout(guides = "collect",heights = c(3,1.5)) & theme(legend.position = "bottom",plot.margin =  margin(0, 0, 0, 0, "cm")) 
ggsave(paste0(FigOut,"BulkAndSC_Val_v3_2part_v4.pdf"), width = 4, height = 5.25)
```

```{r fig6c: PCA color-coded by MEROPS peptide abundances}

#PC_both <- prcomp(t(BC_imp_PxC))
PC_both <- prcomp(t(prePCA))
PC_both_coords <- as.data.frame(PC_both$x[,1:2])
PC_both_coords$id <- row.names(PC_both_coords)
BC_unImp_CxP_DIG <- BC_unImp_CxP %>% dplyr::select(contains("_DIG_"))
BC_unImp_CxP_DIG$id <- row.names(BC_unImp_CxP_DIG)
PC_both_coords <- PC_both_coords %>% left_join(BC_unImp_CxP_DIG, by = "id")

#"Actin: L104, F1, cathepsin D","Citrate synthase: H25, F2, cathepsin E"
#"P60710_DIG_10362102",	"P60710_DIG_1036222","P60710_DIG_1036223","Q9CZU6_DIG_55521"
PC_both_coords_Actin_L299_F1 <- PC_both_coords %>% dplyr::select(id,PC1,PC2,P60710_DIG_1036222)
PC_both_coords_Actin_L299_F2 <- PC_both_coords %>% dplyr::select(id,PC1,PC2,P60710_DIG_1036223)
PC_both_coords_Actin_L104_F1 <- PC_both_coords %>% dplyr::select(id,PC1,PC2,P60710_DIG_10362102)
PC_both_coords_CS_H25_F1 <- PC_both_coords %>% dplyr::select(id,PC1,PC2,Q9CZU6_DIG_55521)
PC_both_coords_Actin_L104_F1$AlphVal <- ifelse(is.na(PC_both_coords_Actin_L104_F1$P60710_DIG_10362102),"Low","High")
PC_both_coords_Actin_L299_F1$AlphVal <- ifelse(is.na(PC_both_coords_Actin_L299_F1$P60710_DIG_1036222),"Low","High")
PC_both_coords_Actin_L299_F2$AlphVal <- ifelse(is.na(PC_both_coords_Actin_L299_F2$P60710_DIG_1036223),"Low","High")
PC_both_coords_CS_H25_F1$AlphVal <- ifelse(is.na(PC_both_coords_CS_H25_F1$Q9CZU6_DIG_55521),"Low","High")
# z transformed intensities
PC_both_coords_Actin_L104_F1$zInt <- (PC_both_coords_Actin_L104_F1$P60710_DIG_10362102 - mean(PC_both_coords_Actin_L104_F1$P60710_DIG_10362102, na.rm = TRUE))/sd(PC_both_coords_Actin_L104_F1$P60710_DIG_10362102,na.rm = TRUE)
PC_both_coords_Actin_L299_F1$zInt <- (PC_both_coords_Actin_L299_F1$P60710_DIG_1036222 - mean(PC_both_coords_Actin_L299_F1$P60710_DIG_1036222, na.rm = TRUE))/sd(PC_both_coords_Actin_L299_F1$P60710_DIG_1036222,na.rm = TRUE)
PC_both_coords_Actin_L299_F2$zInt <- (PC_both_coords_Actin_L299_F2$P60710_DIG_1036223 - mean(PC_both_coords_Actin_L299_F2$P60710_DIG_1036223, na.rm = TRUE))/sd(PC_both_coords_Actin_L299_F2$P60710_DIG_1036223,na.rm = TRUE)
PC_both_coords_CS_H25_F1$zInt <- (PC_both_coords_CS_H25_F1$Q9CZU6_DIG_55521 - mean(PC_both_coords_CS_H25_F1$Q9CZU6_DIG_55521, na.rm = TRUE))/sd(PC_both_coords_CS_H25_F1$Q9CZU6_DIG_55521,na.rm = TRUE)
PC_both_coords_Actin_L104_F1$zInt_bin <- ifelse(PC_both_coords_Actin_L104_F1$zInt >= 2, 2, ifelse(PC_both_coords_Actin_L104_F1$zInt <= -2,-2,PC_both_coords_Actin_L104_F1$zInt))
PC_both_coords_Actin_L299_F1$zInt_bin <- ifelse(PC_both_coords_Actin_L299_F1$zInt >= 2, 2, ifelse(PC_both_coords_Actin_L299_F1$zInt <= -2,-2,PC_both_coords_Actin_L299_F1$zInt))
PC_both_coords_Actin_L299_F2$zInt_bin <- ifelse(PC_both_coords_Actin_L299_F2$zInt >= 2, 2, ifelse(PC_both_coords_Actin_L299_F2$zInt <= -2,-2,PC_both_coords_Actin_L299_F2$zInt))
PC_both_coords_CS_H25_F1$zInt_bin <- ifelse(PC_both_coords_CS_H25_F1$zInt >= 2, 2, ifelse(PC_both_coords_CS_H25_F1$zInt <= -2,-2,PC_both_coords_CS_H25_F1$zInt))
sizeVal <- 2

#-----
# Missing data shown as point
#-----
ActL299_F1_v2 <- PC_both_coords_Actin_L299_F1 %>% ggplot(aes(x = PC1, y = PC2, color = zInt_bin,alpha = AlphVal,shape = AlphVal, size = AlphVal)) + 
  geom_point() +
  scale_color_gradient2(low = "blue", mid = "white", high = "red", midpoint = 0, limits = c(-2,2),breaks = c(-2,0,2), labels = c("-2","0","2"),na.value="black") + 
  theme_light() + 
  guides(alpha = "none",shape = "none") + 
  ggtitle("Actin, L299-F1") + 
  theme(axis.text = element_blank(),axis.title = element_text(size = 20),title = element_text(size = 24),legend.title = element_text(size = 20), legend.text = element_text(size = 14), legend.position = "right") +
  scale_alpha_discrete(range = c(0.75, 0.10)) + 
  scale_shape_manual(values = c(16,16))+ 
  scale_size_manual(values = c(6,1), labels = c("Quantified","Missing")) + 
  labs(color = "Abundance\nz-score", size = "")

ActL104_v2 <- PC_both_coords_Actin_L104_F1 %>% ggplot(aes(x = PC1, y = PC2, color = zInt_bin,alpha = AlphVal,shape = AlphVal, size = AlphVal)) + 
  geom_point() +
  scale_color_gradient2(low = "blue", mid = "white", high = "red", midpoint = 0, limits = c(-2,2),breaks = c(-2,0,2), labels = c("-2","0","2"),na.value="black") + 
  theme_light() + guides(alpha = "none",shape = "none") + 
  ggtitle("Actin, L104-F1") + 
  theme(axis.text = element_blank(),axis.title = element_text(size = 20),title = element_text(size = 24),legend.title = element_text(size = 20), legend.text = element_text(size = 14), legend.position = "right") +
  scale_alpha_discrete(range = c(0.75, 0.10)) + 
  scale_shape_manual(values = c(16,16)) + 
  scale_size_manual(values = c(6,1), labels = c("Quantified","Missing")) + 
  labs(color = "Abundance\nz-score", size = "")

((ActL299_F1_v2) + plot_spacer() + (ActL104_v2)) + plot_layout(widths = c(1,0.3,1), guides = "collect") & theme(legend.position = "null")
ggsave(paste0(FigOut,"MEROPS_PCA_Fig2E_v2.pdf"), width = 10, height = 4.25, units = "in")

((ActL299_F1_v2) + plot_spacer() + (ActL104_v2)) + plot_layout(widths = c(1,0.5,1), guides = "collect") & theme(legend.position = "right")
ggsave(paste0(FigOut,"MEROPS_PCA_Fig2E_v2_legend.pdf"), width = 10, height = 4.25, units = "in")
```



# %%%%%%%%%%%%%%%%%%%%%%
# Extended Data figures
# %%%%%%%%%%%%%%%%%%%%%%

# EDF 1: MQ.live with and without prioritization
```{r Precursors detected at MS1 }
Rebuttal_coverage_targList <- paste0(datIn,"Fig1/Rebuttal_aSQC_MQL_contrast_v2.txt")

Rebuttal_log_path <- paste0(datIn,"Fig1/Fig1c_logs/MQL_base/")
Rebuttal_log_path.p <- paste0(datIn,"Fig1/Fig1c_logs/MQL_prioritized/")

#re-coding priority b/c we had both RT-only and targetable peptides assigned priority 0
tList_instance <- read.delim(Rebuttal_coverage_targList)
tList_instance$id <- seq(0,nrow(tList_instance)-1,1)
tList_instance$Orig_Priority <- tList_instance$Priority
tList_instance$Priority <- ifelse(tList_instance$TargBool == FALSE, 0, tList_instance$Priority + 1)

tList_instance$id <- seq(0,nrow(tList_instance)-1,1)

logs <- list.files(Rebuttal_log_path)
log_filt <- logs[!(logs %in% c("desktop.ini"))]
numLogs <- length(log_filt)

logs.p <- list.files(Rebuttal_log_path.p)
log.p_filt <- logs.p[!(logs.p %in% c("desktop.ini"))]
numLogs.p <- length(log.p_filt)

# MaxQuant.live base
for(i in 1:numLogs){
  log_instance <- read.delim(paste0(Rebuttal_log_path,log_filt[i]), header = FALSE)
  colnames(log_instance) <-  c("Date","Time","Scan_protocol","RT","Message")

  MS1 <- getMS1(log_instance)
  MS1_unique <- unique(MS1$id)
  MS2 <- getMS2(log_instance)
  tList_instance$seen <- ifelse(tList_instance$id %in% MS1_unique,TRUE,FALSE)
  tList_instance$MS2d <- ifelse(tList_instance$id %in% MS2$id,TRUE,FALSE)
  tList_sum <- tList_instance %>% dplyr::group_by(Priority) %>% 
    dplyr::summarize(seenSum = sum(seen),seenFrac = seenSum/n(), MS2Sum = sum(MS2d),MS2Frac = MS2Sum/n())
  #tList_sum <- tList_instance %>% dplyr::group_by(PriorityTop) %>% 
    #dplyr::summarize(seenSum = sum(seen),seenFrac = seenSum/n(), MS2Sum = sum(MS2d),MS2Frac = MS2Sum/n())
  #tList_sum3 <- tList_instance %>% dplyr::group_by(PriorityTop3) %>% 
    #dplyr::summarize(seenSum = sum(seen),seenFrac = seenSum/n(), MS2Sum = sum(MS2d),MS2Frac = MS2Sum/n())
  tList_sum$file_ind <- i
  if(i ==1){
    tList_sum_out <-tList_sum
    #tList_sum_out3 <-tList_sum3
  }
  if(i > 1){
    tList_sum_out <- rbind(tList_sum_out,tList_sum)
    #tList_sum_out3 <- rbind(tList_sum_out3,tList_sum3)
  }
}

# MaxQuant.live with prioritization
for(i in 1:numLogs.p){
  log_instance <- read.delim(paste0(Rebuttal_log_path.p,log.p_filt[i]), header = FALSE)
  colnames(log_instance) <-  c("Date","Time","Scan_protocol","RT","Message")

  MS1 <- getMS1(log_instance)
  MS1_unique <- unique(MS1$id)
  MS2 <- getMS2(log_instance)
  tList_instance$seen <- ifelse(tList_instance$id %in% MS1_unique,TRUE,FALSE)
  tList_instance$MS2d <- ifelse(tList_instance$id %in% MS2$id,TRUE,FALSE)
  tList_sum <- tList_instance %>% dplyr::group_by(Priority) %>% 
    dplyr::summarize(seenSum = sum(seen),seenFrac = seenSum/n(), MS2Sum = sum(MS2d),MS2Frac = MS2Sum/n())
  #tList_sum <- tList_instance %>% dplyr::group_by(PriorityTop) %>% 
    #dplyr::summarize(seenSum = sum(seen),seenFrac = seenSum/n(), MS2Sum = sum(MS2d),MS2Frac = MS2Sum/n())
  #tList_sum3 <- tList_instance %>% dplyr::group_by(PriorityTop3) %>% 
    #dplyr::summarize(seenSum = sum(seen),seenFrac = seenSum/n(), MS2Sum = sum(MS2d),MS2Frac = MS2Sum/n())
  tList_sum$file_ind <- i
  if(i ==1){
    tList_sum_out.p <-tList_sum
    #tList_sum_out3 <-tList_sum3
  }
  if(i > 1){
    tList_sum_out.p <- rbind(tList_sum_out.p,tList_sum)
    #tList_sum_out3 <- rbind(tList_sum_out3,tList_sum3)
  }
}

targList_Contrast_sum <- tList_instance %>% dplyr::group_by(Priority) %>% dplyr::summarize(TargCount = n())

Priority_dict.contrast <- data.frame(Priority = numeric(), Title = character(), stringsAsFactors = FALSE)
Priority_dict.contrast[1:4,1] <- c(4,3,2,0)
Priority_dict.contrast[1:4,2] <- c("High","Medium","Low","RT only")

# Rebuttal consistency
#Priority_dict.consist <- data.frame(Priority = numeric(), Title = character(), stringsAsFactors = FALSE)
#Priority_dict.consist[1:5,1] <- c(4,3,2,1,0)
#Priority_dict.consist[1:5,2] <- c("High 1","High 2","Medium","Low","RT only")

tList_sum_out.contrast.j <- tList_sum_out %>% left_join(Priority_dict.contrast, by = "Priority")
tList_sum_out.contrast.j.f <- tList_sum_out.contrast.j %>% select(seenFrac,MS2Frac,Priority,file_ind,Title)
tList_sum_out.contrast.j.f.m <- melt(tList_sum_out.contrast.j.f, id.vars = c("file_ind","Priority","Title"))

tList_sum_out.contrast.p.j <- tList_sum_out.p %>% left_join(Priority_dict.contrast, by = "Priority")
tList_sum_out.contrast.p.j.f <- tList_sum_out.contrast.p.j %>% select(seenFrac,MS2Frac,Priority,file_ind,Title)
tList_sum_out.contrast.p.j.f.m <- melt(tList_sum_out.contrast.p.j.f, id.vars = c("file_ind","Priority","Title"))

tList_sum_out.contrast.j.f.m <- tList_sum_out.contrast.j.f.m %>% left_join(targList_Contrast_sum, by = "Priority")
tList_sum_out.contrast.p.j.f.m <- tList_sum_out.contrast.p.j.f.m %>% left_join(targList_Contrast_sum, by = "Priority")

tList_sum_out.contrast.j.f.m$Type <- "NoPrioritization"
tList_sum_out.contrast.p.j.f.m$Type <- "Prioritization"

CollectedContrast <- rbind(tList_sum_out.contrast.j.f.m,tList_sum_out.contrast.p.j.f.m)

CollectedContrast$legendTitle <- ifelse(CollectedContrast$variable == "seenFrac","Detected at MS1","Sent for MS2")

#MQBase_logPlot <- CollectedContrast %>% filter(Type == "NoPrioritization") %>% ggplot(aes(value, color = legendTitle)) + geom_boxplot(lwd = 1) + 
#  ggtitle("MaxQuant.live without Prioritization") + 
#  coord_flip() + 
#  theme_light() + 
#  theme(axis.text.x =  element_blank(), 
#        strip.background = element_rect(color = "grey70", fill = "transparent"), 
#        strip.text = element_text(color= "black", size = 18),legend.position = "bottom",
#        axis.text.y = element_text(size =14),
#        axis.title.y = element_text(size =16),
#        plot.title = element_text(size = 18),
#        legend.text = element_text(size = 14)) + 
#  labs(color = "", x = "Fraction of Prioritized Peptides") + 
#  facet_wrap(~factor(Title,levels = c("High","Medium","Low","RT only")), nrow = 1)+
#  scale_color_manual(values = c("black","grey60")) + geom_text( x = .1, y = 0,color = "grey10",aes(label = paste0(TargCount,"\n Precursors")))
#ggsave(paste0(FigOut,"Rebuttal_coverage_MQliveStats.png"), width = 6, height = 4, units = "in")

#MQP_logPlot <- CollectedContrast %>% filter(Type == "Prioritization") %>% ggplot(aes(value, color = legendTitle)) + geom_boxplot(lwd = 1) + 
#  ggtitle("MaxQuant.live with Prioritization") + 
#  coord_flip() + 
#  theme_light() + 
#  theme(axis.text.x =  element_blank(), 
#        strip.background = element_rect(color = "grey70", fill = "transparent"), 
#        strip.text = element_text(color= "black", size = 18),legend.position = "bottom",
#        axis.text.y = element_blank(),
#        axis.title.y = element_blank(),
#        plot.title = element_text(size = 18),
#        legend.text = element_text(size = 14)) + 
#  labs(color = "", x = "") + 
#  facet_wrap(~factor(Title,levels = c("High","Medium","Low","RT only")), nrow = 1)+
#  scale_color_manual(values = c("firebrick4","red")) + geom_text( x = .1, y = 0,color = "grey10",aes(label = paste0(TargCount,"\n Precursors")))
#ggsave(paste0(FigOut,"Rebuttal_consist_MQliveStats.png"), width = 6, height = 4, units = "in")

#MQBase_logPlot + MQP_logPlot

CollectedContrast$Platform <- ifelse(CollectedContrast$Type == "NoPrioritization","Not Prioritized","Prioritized")

library(scales)
#comma(int_vals)

ContrastMS1 <- CollectedContrast %>% filter(variable == "seenFrac") %>% ggplot(aes(x =(value*100),y = Platform, color = Platform)) + geom_boxplot(lwd = 1) + 
  geom_point(position=position_jitterdodge(), size = 2, shape = 21, fill = "grey95") +
  #ggtitle("Precursors Detected at MS1") + 
  coord_flip(xlim = c(0,100)) + 
  theme_light() + 
  theme(axis.text.x =  element_blank(), 
        strip.background = element_rect( fill = "grey95", color = "grey70"), 
        strip.text = element_text(color= "black", size = 18,face = "bold"),
        legend.position = "bottom",
        axis.text.y = element_text(size =14),
        axis.title.y = element_text(size =16),
        plot.title = element_text(size = 18),
        legend.text = element_text(size = 14)) + 
  labs(color = "",x="Precursors detected at MS1, %", y = "") + 
  facet_wrap(~factor(Title,levels = c("High","Medium","Low","RT only")), nrow = 1)+
  scale_color_manual(values = c("black","red")) + geom_text( x = 13.5, y = 1.5,color = "grey10",aes(label = paste0(comma(TargCount))), size = 8)  + geom_text( x = 10, y = 1.5,color = "grey10",aes(label = paste0("\nPrecursors")), size = 5)

ContrastMS2 <- CollectedContrast %>% filter(variable == "MS2Frac") %>% ggplot(aes(value*100, color = Platform, y = Platform)) + geom_boxplot(lwd = 1) +
 geom_point(position=position_jitterdodge(), size = 2, shape = 21, fill = "grey95")+ 
  #ggtitle("Precursors Sent for MS2") + 
  coord_flip(xlim = c(0,100)) + 
  theme_light() + 
  theme(axis.text.x =  element_blank(), 
        strip.background = element_rect( fill = "grey95", color = "grey70"), 
        strip.text = element_text(color= "black", size = 18,face = "bold"),
        legend.position = "bottom",
        axis.text.y = element_text(size = 14),
        axis.title.y = element_text(size = 16),
        plot.title = element_text(size = 18),
        legend.text = element_text(size = 14)) + 
  labs(color = "", x = "Precursors sent for MS2, %", y = "") + 
  facet_wrap(~factor(Title,levels = c("High","Medium","Low","RT only")), nrow = 1)+
  scale_color_manual(values = c("black","red"))

ContrastMS1 /plot_spacer()/ ContrastMS2 + plot_layout(heights = c(4,.5,4), guides = "collect") & theme(legend.position = 'bottom')
ggsave(paste0(FigOut,"Rebuttal_contrast_MQliveStats.pdf"), width = 7, height = 8, units = "in")
```

# EDF 2: CVs from SCP pipeline
```{r}
cov_ShotgunSet <- c("XwGH0643","XwGH0644","XwGH0645","XwGH0646","XwGH0647","XwGH0648","XwGH0649","XwGH0650")
cov_PrioriSet <- c("XwGH0672","XwGH0679","XwGH0680","XwGH0678","XwGH0675","XwGH0681","XwGH0671","XwGH0673")


pSCoPECVs_fig2a <- read.delim(paste0(datIn,"Fig2/Fig2_a_Coverage/ChanKey_Rebuttal_Coverage_PIF50_mrri10.txt"))
pSCoPECVs_fig2bcde <- read.delim(paste0(datIn,"Fig2/Fig2_bcde_consist_pSCoPEOnly/ChanKey_Rebuttal_Consistency_pSCoPE_PIF50_mrri10.txt"))
pSCoPECVs_fig4_5_6 <- read.delim(paste0(datIn,"BMDM_figs_4_5_6/ChanKey_BMDM_Priori_mrri02_PIF50_DART_FDR1p.txt"))

ShotgunCVs_fig2 <- pSCoPECVs_fig2a %>% filter(Raw.file %in% cov_ShotgunSet)
pSCoPECVs_fig2a <- pSCoPECVs_fig2a %>% filter(Raw.file %in% cov_PrioriSet)

ShotgunCVs_fig2$colorCode <- ifelse(ShotgunCVs_fig2$celltype == "neg","Control","Cell")
pSCoPECVs_fig2a$colorCode <- ifelse(pSCoPECVs_fig2a$celltype == "neg","Control","Cell")
pSCoPECVs_fig2bcde$colorCode <- ifelse(pSCoPECVs_fig2bcde$celltype == "neg","Control","Cell")
pSCoPECVs_fig4_5_6$colorCode <- ifelse(pSCoPECVs_fig4_5_6$celltype == "control","Control","Cell")

library(ggpubr)

ShotgunCVs_fig2 %>% ggplot(aes(x = cvm, fill = colorCode)) + geom_density(alpha=0.5, adjust=.8) + theme_light() + coord_cartesian(xlim = c(0,1))  + scale_x_continuous(limits = c(0,1)) +
  geom_vline(xintercept = .4, linetype = "dashed", size = 1, color = "grey50") + 
  scale_fill_manual(values=c("purple","grey10")) + theme_pubr() + labs(x = "Quantification Variability, CV", y = "Density") + theme(legend.title = element_blank(), axis.text = element_text(size = 14), axis.title = element_text(size = 16))
ggsave(paste0(FigOut,"Fig2_ShotgunCVs.png"), width = 7, height = 5, units = "in")

pSCoPECVs_fig2a %>% ggplot(aes(x = cvm, fill = colorCode)) + geom_density(alpha=0.5, adjust=.7) + theme_light() + coord_cartesian(xlim = c(0,1))  + scale_x_continuous(limits = c(0,1)) +
  geom_vline(xintercept = .4, linetype = "dashed", size = 1, color = "grey50") +
  scale_fill_manual(values=c("purple","grey10")) + theme_pubr() + labs(x = "Quantification Variability, CV", y = "Density") + theme(legend.title = element_blank(), axis.text = element_text(size = 14), axis.title = element_text(size = 16))
ggsave(paste0(FigOut,"Fig2s_pSCoPECVs.png"), width = 7, height = 5, units = "in")

pSCoPECVs_fig2bcde %>% ggplot(aes(x = cvm, fill = colorCode)) + geom_density(alpha=0.5, adjust=.7) + theme_light() + coord_cartesian(xlim = c(0,1))  + scale_x_continuous(limits = c(0,1)) +
  geom_vline(xintercept = .4, linetype = "dashed", size = 1, color = "grey50") +
  scale_fill_manual(values=c("purple","grey10")) + theme_pubr() + labs(x = "Quantification Variability, CV", y = "Density") + theme(legend.title = element_blank(), axis.text = element_text(size = 14), axis.title = element_text(size = 16))
ggsave(paste0(FigOut,"Fig2bcde_pSCoPECVs.png"), width = 7, height = 5, units = "in")

pSCoPECVs_fig4_5_6 %>% ggplot(aes(x = cvm, fill = colorCode)) + geom_density(alpha=0.5, adjust=.7) + theme_light() + coord_cartesian(xlim = c(0,1))  + scale_x_continuous(limits = c(0,1)) +
  geom_vline(xintercept = .4, linetype = "dashed", size = 1, color = "grey50") +
  scale_fill_manual(values=c("purple","grey10")) + theme_pubr() + labs(x = "Quantification Variability, CV", y = "Density") + theme(legend.title = element_blank(), axis.text = element_text(size = 14), axis.title = element_text(size = 16))
ggsave(paste0(FigOut,"Fig4_5_6_pSCoPECVs.png"), width = 7, height = 5, units = "in")
```

# EDF 3: Properties of peptides not confidently identified
```{r}
ShotgunSet <- c("wGH0643","wGH0644","wGH0645","wGH0646","wGH0647","wGH0648","wGH0649","wGH0650")
pSCoPESet <- c("wGH0671","wGH0672","wGH0673","wGH0675","wGH0678","wGH0679","wGH0680","wGH0681")

# Missing data at single-cell level

Rebuttal_t0_both <- read.delim(paste0(datIn,"Fig2/Fig2_a_Coverage/t0_Rebuttal_Coverage_PIF50_mrri10.txt"))

Rebuttal_MP_both <- read.delim(paste0(datIn,"Fig2/Fig2_a_Coverage/Prot_missing_Rebuttal_Coverage_PIF50_mrri10.txt"))

Rebuttal_CG_both <- read.delim(paste0(datIn,"Fig2/Fig2_a_Coverage/ChanKey_Rebuttal_Coverage_PIF50_mrri10.txt"))

AllExps <- read.delim(paste0(datIn,"Fig2/Fig2_a_Coverage/ev_Rebuttal_Coverage_filtered_PIF50_mrri10_1pFDR.txt"))

AllExps$SeqCharge <- paste0(AllExps$Modified.sequence,AllExps$Charge)
AllExps$Raw.file <- str_sub(AllExps$Raw.file, start = 2)

# Full Evidence files for each experiment
evi.tier.s <- read.delim(paste0(datIn,"Fig2/Fig2_a_Coverage/Coverage_evidence_Shotgun_fullFASTA.txt"))
evi.tier.p <- read.delim(paste0(datIn,"Fig2/Fig2_a_Coverage/Coverage_evidence_pSCoPE_fullFASTA.txt"))

evi.tier.s <- evi.tier.s %>% select(-Acetyl..Protein.N.term.)

evi.tier <- rbind(evi.tier.s,evi.tier.p)
evi.tier.pep.prot <- evi.tier %>% dplyr::select(Modified.sequence,Leading.razor.protein) %>% group_by(Modified.sequence) %>% slice(1)
  
RefList <- read.delim(paste0(datIn,"Fig2/Fig2_a_Coverage/pSCoPE_FinalTarg_Coverage.txt"))

RefList_Top <- RefList %>% dplyr::filter(Priority == 3) %>% dplyr::select(SeqCharge)
RefList_Mid <- RefList %>% dplyr::filter(Priority == 2) %>% dplyr::select(SeqCharge)
RefList_Both <- RefList %>% dplyr::filter(Priority %in% c(2,3)) %>% dplyr::select(SeqCharge)
RefList_all <- RefList %>% dplyr::filter(Priority = TRUE) %>% dplyr::select(SeqCharge)

TargList <- read.delim(paste0(datIn,"Fig2/Fig2_a_Coverage/pSCoPE_FinalTarg_Coverage.txt"))

TargList_Top <- TargList %>% dplyr::filter(Priority %in% c(3)) %>% dplyr::select(SeqCharge)
TargList_Mid <- TargList %>% dplyr::filter(Priority %in% c(2)) %>% dplyr::select(SeqCharge)
TargList_Both <- TargList %>% dplyr::filter(Priority %in% c(3)) %>% dplyr::select(SeqCharge)
TargList_All <- TargList %>% dplyr::filter(TargBool = TRUE) %>% select(SeqCharge)

TargList_Both$pep <- str_sub(TargList_Both$SeqCharge, end = -2)
TargList_Both.pep <- as.data.frame(unique(TargList_Both$pep))
colnames(TargList_Both.pep) <- "Modified.sequence"

TargList_All$pep <- str_sub(TargList_All$SeqCharge, end = -2)
TargList_All.pep <- as.data.frame(unique(TargList_All$pep))
colnames(TargList_All.pep) <- "Modified.sequence"
#%%%%%%%%%%%%%%%


Priority_dict_coverage <- data.frame(Priority = numeric(), Title = character(), stringsAsFactors = FALSE)
Priority_dict_coverage[1:4,1] <- c(4,3,2,0)
Priority_dict_coverage[1:4,2] <- c("High","Medium","Low","RT only")

#------------------------
# Reading search results
#------------------------

Coverage_priori <- read.delim(paste0(datIn,"Fig2/Fig2_a_Coverage/MQSearchResults_pSCoPE/msms.txt"))
Coverage_shotgun <- read.delim(paste0(datIn,"Fig2/Fig2_a_Coverage/MQSearchResults_Shotgun/msms.txt"))

Coverage_shotgun$SeqCharge <- paste0(Coverage_shotgun$Modified.sequence, Coverage_shotgun$Charge)
Coverage_priori$SeqCharge <- paste0(Coverage_priori$Modified.sequence, Coverage_priori$Charge)

Coverage_shotgun_evi <- read.delim(paste0(datIn,"Fig2/Fig2_a_Coverage/MQSearchResults_Shotgun/evidence.txt"))
Coverage_priori_evi <- read.delim(paste0(datIn,"Fig2/Fig2_a_Coverage/MQSearchResults_pSCoPE/evidence.txt"))

Coverage_shotgun_evi$SeqCharge <- paste0(Coverage_shotgun_evi$Modified.sequence, Coverage_shotgun_evi$Charge)
Coverage_priori_evi$SeqCharge <- paste0(Coverage_priori_evi$Modified.sequence, Coverage_priori_evi$Charge)

#-------------------------
# Reading targeting list and reformating tiers for visualizing RT-only peptides separately
#-------------------------

targList_instance_Coverage <- read.delim(paste0(datIn,"Fig2/Fig2_a_Coverage/pSCoPE_FinalTarg_Coverage.txt"))
targList_instance_Coverage$id <- seq(0,nrow(targList_instance_Coverage)-1,1)
targList_instance_Coverage$Orig_Priority <- targList_instance_Coverage$Priority
targList_instance_Coverage$Priority <- ifelse(targList_instance_Coverage$TargBool == FALSE, 0,targList_instance_Coverage$Priority + 1)

#------------------------
# Filtering search results
#------------------------

ShotgunSet <- c("wGH0643","wGH0644","wGH0645","wGH0646","wGH0647","wGH0648","wGH0649","wGH0650")
PrioriSet <- c("wGH0671","wGH0672","wGH0673","wGH0675","wGH0678","wGH0679","wGH0680","wGH0681")

Coverage_shotgun.f <- Coverage_shotgun %>% dplyr::filter(Raw.file %in% ShotgunSet) %>% group_by(Raw.file, SeqCharge) %>% top_n(-1, PEP) %>% slice(1)
Coverage_priori.f <- Coverage_priori %>% dplyr::filter(Raw.file %in% PrioriSet) %>% group_by(Raw.file, SeqCharge) %>% top_n(-1, PEP) %>% slice(1)

Coverage_shotgun_evi.f <- Coverage_shotgun_evi %>% dplyr::filter(Raw.file %in% ShotgunSet)
Coverage_priori_evi.f <- Coverage_priori_evi %>% dplyr::filter(Raw.file %in% PrioriSet)

Coverage_priori_evi.fFDR <- Empirical_FDR_PEP(Coverage_priori_evi.f)
Coverage_priori_evi.fFDR <- Coverage_priori_evi.fFDR %>% dplyr::filter(qVal <= .01)

Coverage_shotgun_evi.f.t <- Coverage_shotgun_evi.f %>% filter(SeqCharge %in% targList_instance_Coverage$SeqCharge)

#------------------------
# Extracting fragment info and joining with evidence
#------------------------
Coverage_shotgun.f_frag <- Coverage_shotgun.f %>% dplyr::select(Raw.file,SeqCharge,Number.of.matches)
Coverage_shotgun_evi.f.t2 <- Coverage_shotgun_evi.f.t %>% left_join(Coverage_shotgun.f_frag, by = c("Raw.file","SeqCharge"))

Coverage_shotgun_evi.f.t2_sum <- Coverage_shotgun_evi.f.t2 %>% group_by(SeqCharge) %>% summarize(medianFrag = median(Number.of.matches, na.rm = TRUE),medianPEP = median(PEP,na.rm = TRUE))

interSeq<- intersect(Coverage_shotgun_evi.f.t2_sum$SeqCharge, Coverage_priori_evi.fFDR$SeqCharge)

Coverage_shotgun_evi.f.t2_sum$IDd <- ifelse(Coverage_shotgun_evi.f.t2_sum$SeqCharge %in% interSeq,"PSM, 1% FDR", "No PSM, 1% FDR")
Coverage_shotgun_evi.f.t2_sum$Targ <- ifelse(Coverage_shotgun_evi.f.t2_sum$SeqCharge %in% Coverage_shotgun_evi.f.t2_sum$SeqCharge,TRUE, FALSE)
Coverage_shotgun_evi.f.t2_sum <- Coverage_shotgun_evi.f.t2_sum %>% dplyr::filter(Targ == TRUE) 
Coverage_shotgun_evi.f.t2_sum$negLogPEP <- -log10(Coverage_shotgun_evi.f.t2_sum$medianPEP)
Coverage_shotgun_evi.f.t2_sum$binPEP <- ifelse(Coverage_shotgun_evi.f.t2_sum$negLogPEP >= 5,5,Coverage_shotgun_evi.f.t2_sum$negLogPEP)


Coverage_shotgun_evi.f.t2_sum3 <- Coverage_shotgun_evi.f.t2_sum %>% left_join(targList_instance_Coverage, by = "SeqCharge")

Coverage_shotgun_evi.f.t2_sum3 <- Coverage_shotgun_evi.f.t2_sum3 %>% left_join(Priority_dict_coverage, by = "Priority")

Coverage_shotgun_evi.f.t2_sum3_RTOnly <- Coverage_shotgun_evi.f.t2_sum3 %>% filter(Priority == 0)
Coverage_shotgun_evi.f.t2_sum3_targ <- Coverage_shotgun_evi.f.t2_sum3 %>% filter(Priority != 0)
Coverage_shotgun_evi.f.t2_sum3_RTOnly.f <- Coverage_shotgun_evi.f.t2_sum3_RTOnly %>% filter(!(SeqCharge %in% Coverage_shotgun_evi.f.t2_sum3_targ$SeqCharge),IDd == "PSM")


Coverage_shotgun_evi.f.t2_sum3_PEP <- Coverage_shotgun_evi.f.t2_sum3 %>% group_by(IDd,Priority) %>% summarize(medPEP = median(negLogPEP),medFrag = median(medianFrag))

Coverage_shotgun_evi.f.t2_sum3 <- Coverage_shotgun_evi.f.t2_sum3 %>% left_join(Coverage_shotgun_evi.f.t2_sum3_PEP, by = c("IDd","Priority"))

Coverage_shotgun_evi.f.t2_sum3 %>% ggplot(aes(binPEP, fill = IDd)) + geom_histogram(position = position_identity(), alpha = 0.5) + theme_light()   + facet_wrap(~factor(Title, levels = c("High","Medium","Low","RT only")),nrow = 1, scales = "free_x") + geom_vline(aes(xintercept = medPEP, color = IDd)) + labs(x = bquote('-Log'[10]~ 'PEP'), y = "Prioritized Precursors", color = "", fill = "") + coord_flip() + theme(axis.text.x = element_text(angle = 45, hjust = 1),
  axis.text = element_text(size = 20),
  axis.title = element_text(size=22),
  legend.text = element_text(size = 20),
  strip.background = element_rect( fill = "grey95", color = "grey70"), 
  strip.text = element_text(color= "black", size = 22,face = "bold"),
  legend.position = "bottom",
  plot.title = element_text(size = 18)
)
  #ggtitle("Shotgun PSM Properties:  PEP")
ggsave(paste0(FigOut,"PSMs1pFDR_PEPHistogram_forPub_Rebut.pdf"), width = 8, height = 6, units = "in")

Coverage_shotgun_evi.f.t2_sum3 %>% ggplot(aes(medianFrag, fill = IDd)) + geom_histogram(position = position_identity(), alpha = 0.5) + theme_light()   + facet_wrap(~factor(Title, levels = c("High","Medium","Low","RT only")),nrow = 1, scales = "free_x") + geom_vline(aes(xintercept = medFrag, color = IDd)) + labs(x = "Number of Matching Fragments", y = "Prioritized Precursors", color = "", fill = "") + coord_flip()+ theme(axis.text.x = element_text(angle = 45, hjust = 1),
  axis.text = element_text(size = 20),
  axis.title = element_text(size=22),
  legend.text = element_text(size = 20),
  strip.background = element_rect( fill = "grey95", color = "grey70"), 
  strip.text = element_text(color= "black", size = 22,face = "bold"),
  legend.position = "bottom",
  plot.title = element_text(size = 18)
)   
  #ggtitle("Shotgun PSM Properties: Matching Fragments") 
ggsave(paste0(FigOut,"PSMs1pFDR_fragHistogram_forPub_Rebut.pdf"), width = 8, height = 6, units = "in")  
  
Coverage_shotgun_evi.f.t2_sum3 %>% filter(Priority %in% c(4,3)) %>% ggplot(aes(binPEP, color = IDd)) + geom_density(position = position_identity(), alpha = 0.5, lwd = 2) + theme_light()  + labs(x = bquote('-Log'[10]~ 'PEP'), y = "Prioritized Precursors, Density", color = "", fill = "") + coord_flip() + theme(axis.text.x = element_text(angle = 45, hjust = 1),
  axis.text = element_text(size = 20),
  axis.title = element_text(size=22),
  legend.text = element_text(size = 20),
  strip.background = element_rect( fill = "grey95", color = "grey70"), 
  strip.text = element_text(color= "black", size = 22,face = "bold"),
  legend.position = "bottom",
  plot.title = element_text(size = 18)
)  +
  #ggtitle("Shotgun PSM Properties:  PEP") + 
facet_wrap(~"Top Two Priority Levels")
ggsave(paste0(FigOut,"PSMs1pFDR_PEPDensity_forPub_Rebut.pdf"), width = 6, height = 6, units = "in")

Coverage_shotgun_evi.f.t2_sum3 %>% filter(Priority %in% c(4,3)) %>% ggplot(aes(medianFrag, color = IDd)) + geom_density(position = position_identity(), alpha = 0.5, lwd = 2) + theme_light()  + labs(x = "Number of Matching Fragments", y = "Prioritized Precursors, Density", color = "", fill = "") + coord_flip() + theme(axis.text.x = element_text(angle = 45, hjust = 1),
  axis.text = element_text(size = 20),
  axis.title = element_text(size=22),
  legend.text = element_text(size = 20),
  strip.background = element_rect( fill = "grey95", color = "grey70"), 
  strip.text = element_text(color= "black", size = 22,face = "bold"),
  legend.position = "bottom",
  plot.title = element_text(size = 18)
) +
  #ggtitle("Shotgun PSM Properties: Matching Fragments") + 
  facet_wrap(~"Top Two Priority Levels")
ggsave(paste0(FigOut,"PSMs1pFDR_FragDensity_forPub_Rebut.pdf"), width = 6, height = 6, units = "in")

```

# EDF 4: IMBR
```{r}
IMBR <- read.delim(paste0(datIn,"EDF4_IMBR/IMBR_shotgunSC_evidence.txt"))
ShotgunComp <- read.delim(paste0(datIn,"EDF4_IMBR/shotgunSC_noIMBR_evidence.txt"))
CovExps <- read.delim(paste0(datIn,"EDF4_IMBR/pSCoPE_SC_contrast_Fig2a_dataset_evidence.txt"))

CovExps <- Empirical_FDR_PEP(CovExps)
CovExps$SeqCharge <- paste0(CovExps$Modified.sequence, CovExps$Charge)
PrioriSet <- c("wGH0671","wGH0672","wGH0673","wGH0675","wGH0678","wGH0679","wGH0680","wGH0681")
CovExps.f <-CovExps %>% filter(qVal <= .01, Raw.file %in% PrioriSet)
  
IMBR.match <- IMBR %>% filter(grepl("MATCH",Type)&!grepl("CON",Leading.razor.protein))
IMBR.match_sum <- IMBR.match %>% group_by(Type,Raw.file) %>% summarize(eventCount = n(),Rev = sum(grepl("REV",Leading.razor.protein)),fracRev = Rev/eventCount, sumForward = eventCount - Rev)

IMBR.match.sum.m <- melt(IMBR.match_sum)

# What fraction of all MBR events are accompanied by an MSMS
IMBR.match_sum.MSMS.MBR <- IMBR.match_sum %>% select(Raw.file,eventCount,Type)
IMBR.match.dcast <- dcast(IMBR.match_sum.MSMS.MBR,Raw.file~Type, value.var = "eventCount")
IMBR.match.dcast$MBRsum <- IMBR.match.dcast$`MULTI-MATCH` + IMBR.match.dcast$`MULTI-MATCH-MSMS`
IMBR.match.dcast$fracIMBR <- IMBR.match.dcast$`MULTI-MATCH-MSMS`/IMBR.match.dcast$MBRsum

IMBR.lim <- IMBR.match.dcast %>% select(Raw.file,fracIMBR)
IMBR.match_sum.lim <- IMBR.match_sum %>% ungroup() %>% filter(Type == "MULTI-MATCH-MSMS") %>% select(Raw.file, fracRev)

IMBR.lim <- IMBR.lim %>% left_join(IMBR.match_sum.lim, by = "Raw.file")
IMBR.lim$fracForward <- 1-IMBR.lim$fracRev
IMBR.lim <- IMBR.lim %>% select(Raw.file,fracIMBR,fracForward)
IMBR.lim.m <- melt(IMBR.lim)
IMBR.lim.m_sum <- IMBR.lim.m %>% group_by(variable) %>% summarize(medVal = 100*median(value, na.rm = TRUE), sdVal = 100*sd(value, na.rm = TRUE))
IMBR.lim.m_sum$plotLabel <- ifelse(IMBR.lim.m_sum$variable == "fracIMBR","Matches with an associated MS2","iMBR forward matches")
IMBR.lim.m_sum$facetLabel <- ifelse(IMBR.lim.m_sum$variable == "fracIMBR","MBR PSMs","iMBR PSMs")

# Are these sequences typically identified when targeted in a pSCoPE run? How frequently?
IMBR.match$SeqCharge <- paste0(IMBR.match$Modified.sequence, IMBR.match$Charge)
targList_instance_Coverage <- read.delim(paste0(datIn,"Fig2/Fig2_a_Coverage/pSCoPE_FinalTarg_Coverage.txt"))
tList_Cov <- targList_instance_Coverage %>% filter(TargBoolean == TRUE)

tList_Cov_dict <- tList_Cov %>% select(SeqCharge, Priority)

IMBR.match.targ <- IMBR.match %>% filter(SeqCharge %in% tList_Cov$SeqCharge)
IMBR.match.files <- length(unique(IMBR.match.targ$Raw.file))

#New data filtration
IMBR.PreQFilt <- Empirical_FDR_PEP(IMBR)
IMBR.qFilt <- IMBR.PreQFilt %>% filter(qVal <= .01) %>% select(-c(qVal, Score_naOmit,Ind,revCount,RevHits))
IMBR.targListInter <- IMBR %>% filter((Type == "MULTI-MATCH-MSMS"),(!grepl("CON", Leading.razor.protein)),(!grepl("REV", Leading.razor.protein)))

IMBR.targListInter$SeqCharge <- paste0(IMBR.targListInter$Modified.sequence,IMBR.targListInter$Charge)
IMBR.qFilt$SeqCharge <- paste0(IMBR.qFilt$Modified.sequence,IMBR.qFilt$Charge)

interSeqChargeList <- intersect(IMBR.targListInter$SeqCharge, tList_Cov_dict$SeqCharge)

IMBR.qFilt.comb <- rbind(IMBR.qFilt, IMBR.targListInter)
IMBR.qFilt.comb$SeqCharge <- paste0(IMBR.qFilt.comb$Modified.sequence,IMBR.qFilt.comb$Charge)
IMBR.qFilt.comb.inter <- IMBR.qFilt.comb %>% filter(SeqCharge %in% interSeqChargeList) 
#IMBR.match.targ_sum <- IMBR.match.targ %>% filter(Type != "MULTI-MATCH") %>% group_by(SeqCharge,Type) %>% summarize(numExps = length(unique(Raw.file)), fractionIDd = numExps/IMBR.match.files, Platform = "IMBR") 
IMBR.match.targ_sum <- IMBR.qFilt.comb.inter %>% group_by(SeqCharge) %>% summarize(numExps = length(unique(Raw.file)), fractionIDd = numExps/IMBR.match.files, Platform = "IMBR", Type = "Mixed")

CovExps.f2 <- CovExps.f %>% filter(SeqCharge %in% interSeqChargeList)
CovExps.f2.files <- length(unique(CovExps.f2$Raw.file))
CovExps.f2_sum <- CovExps.f2 %>% group_by(SeqCharge) %>% summarize(numExps = length(unique(Raw.file)), fractionIDd = numExps/CovExps.f2.files, Platform = "pSCoPE",Type = "MSMS") 

IMBR_contrast <- rbind(IMBR.match.targ_sum, CovExps.f2_sum)

IMBR_contrast <- IMBR_contrast %>% left_join(tList_Cov_dict, by = "SeqCharge")

IMBR_contrast$LevelDesc <- ifelse(IMBR_contrast$Priority == 3, "High",ifelse(IMBR_contrast$Priority == 2, "Medium","Low"))

IMBR_contrast.inter <- IMBR_contrast %>% filter(SeqCharge %in% intersect(IMBR.match.targ_sum$SeqCharge,CovExps.f2_sum$SeqCharge))

IMBR_contrast.inter %>% filter(Type != "MULTI-MATCH") %>% ggplot(aes(fractionIDd*100, fill = Platform)) + geom_histogram(position = position_identity(), alpha = 0.5) + facet_wrap(~factor(LevelDesc, levels = c("High","Medium","Low"))) + coord_flip(xlim = c(0,100)) + labs(x = "Runs Identified In, %", y = "Precursors") + theme_light()

IMBR_contrast.inter_sum <- IMBR_contrast.inter %>% group_by(Platform, Priority) %>% summarize(count = n())
IMBR_contrast.inter <- IMBR_contrast.inter %>% left_join(IMBR_contrast.inter_sum, by = c("Platform","Priority"))

IMBR_contrast_box <- IMBR_contrast.inter %>% filter(Type != "MULTI-MATCH") %>% ggplot(aes(x=fractionIDd*100, color = Platform,y = Platform)) + geom_boxplot(lwd = 1, outlier.shape = NA) + facet_wrap(~factor(LevelDesc, levels = c("High","Medium","Low"))) + geom_point(position=position_jitterdodge(), size = 1.5, shape = 21, fill = "grey95") + coord_flip(xlim = c(0,100)) + labs(x = "Runs Identified In, %", color = "") + theme_light() + theme(axis.title.x = element_blank(),axis.text.x = element_blank(), axis.title.y = element_text(size = 22), axis.text.y = element_text(size =20), legend.position = "bottom",   legend.text = element_text(size = 20),
  strip.background = element_rect( fill = "grey95", color = "grey70"), 
  strip.text = element_text(color= "black", size = 22,face = "bold")) + scale_color_manual(values = c("grey20","red")) + geom_text( x =6, y = 1.5,color = "grey10",aes(label = paste0((count))), size = 6)  + geom_text( x = 4, y = 1.5,color = "grey10",aes(label = paste0("\nPrecursors")), size = 5)

IMBR.match.sum.m$TypeLab <- ifelse(IMBR.match.sum.m$Type == "MULTI-MATCH","MBR","MBR associated with MS2 scan")
IMBR.match.sum.m$variable2 <- ifelse(IMBR.match.sum.m$variable == "eventCount","All matches", ifelse(IMBR.match.sum.m$variable == "sumForward","Forward matches","Unused"))

MBR <- IMBR.match.sum.m %>% filter(TypeLab == "MBR",variable2 != "Unused") %>% ggplot(aes(y=value,x = variable2, color = variable2)) + geom_boxplot(lwd = 1) + coord_cartesian(ylim = c(0,3000)) + theme_light() + labs(y = "MBR Matches", x = "", color = "") + geom_point(position=position_jitterdodge(), size = 2, shape = 21, fill = "grey95") +
  theme(axis.text.x = element_blank(), 
        axis.title = element_text(size = 22), 
        axis.text.y = element_text(size = 20), 
        strip.background = element_rect( fill = "grey95", color = "grey70"), 
        strip.text = element_text(color= "black", size = 18,face = "bold"),
        legend.position = "bottom") + 
  scale_color_manual(values = c("grey70","grey20")) + facet_wrap(~"All\nPrecursors")   + scale_y_continuous(limits = c(0, 3000), breaks = seq(0, 3000, by = 1500))
#+ geom_errorbar(aes(ymin = (medVal - sdVal), ymax = (medVal + sdVal)), color = "grey40", width = 0.5)

MBR2 <- IMBR.match.sum.m %>% filter(TypeLab != "MBR",variable2 != "Unused") %>% ggplot(aes(y=value, x = variable2, color = variable2)) + geom_boxplot(lwd = 1) + coord_cartesian(ylim = c(0,1000)) + theme_light() + labs(y = "", x = "", color = "") + geom_point(position=position_jitterdodge(), size = 2, shape = 21, fill = "grey95") +
  theme(axis.text.x = element_blank(), 
        axis.title = element_text(size = 22), 
        axis.text.y = element_text(size = 20), 
        strip.background = element_rect( fill = "grey95", color = "grey70"), 
        strip.text = element_text(color= "black", size = 18,face = "bold"),
        legend.position = "bottom")  + scale_color_manual(values = c("grey70","grey20")) + facet_wrap(~"Precursors with\nMS2 Scans")  + scale_y_continuous(limits = c(0, 1000), breaks = seq(0, 1000, by = 500))

MBR + MBR2 + plot_layout(guides = "collect")  & theme(legend.position = 'bottom', legend.text = element_text(size = 14))
ggsave(paste0(FigOut,"IMBR_contrast_Top.pdf"), width = 7, height = 5, units = "in")

IMBR_contrast_box 
ggsave(paste0(FigOut,"IMBR_contrast_bottom.pdf"), width = 7, height = 5, units = "in")


IMBR.match.sum.m.sum <- IMBR.match.sum.m %>% group_by(variable,Type) %>% summarize(medVal = median(value, na.rm = TRUE))
```

# EDF 5: MQ.live log stats for Fig2 experiments
```{r}

#-----
# For fig2a data
#-----

Rebuttal_coverage_targList <- paste0(datIn,"Fig2/Fig2_a_Coverage/pSCoPE_FinalTarg_Coverage.txt")
#Rebuttal_coverage_log_path <- paste0(DataPath,"Rebuttal_technical_Fig_1bcd/Fig1_b_Coverage/MQL_logs_Rebuttal_coverage/")
Rebuttal_coverage_log_path <- paste0(datIn,"Fig2/Fig2_a_Coverage/MQL_logs_Rebuttal_coverage/")

#re-coding priority b/c we had both RT-only and targetable peptides assigned priority 0
tList_instance <- read.delim(Rebuttal_coverage_targList)
tList_instance$id <- seq(0,nrow(tList_instance)-1,1)
tList_instance$Orig_Priority <- tList_instance$Priority
tList_instance$Priority <- ifelse(tList_instance$TargBool == FALSE, 0, tList_instance$Priority + 1)

tList_instance$id <- seq(0,nrow(tList_instance)-1,1)
logs <- list.files(Rebuttal_coverage_log_path)
log_filt <- logs[!(logs %in% c("desktop.ini","1016_20210927-1118.txt"))]
numLogs <- length(log_filt)

for(i in 1:numLogs){
  log_instance <- read.delim(paste0(Rebuttal_coverage_log_path,log_filt[i]), header = FALSE)
  colnames(log_instance) <-  c("Date","Time","Scan_protocol","RT","Message")

  MS1 <- getMS1(log_instance)
  MS1_unique <- unique(MS1$id)
  MS2 <- getMS2(log_instance)
  tList_instance$seen <- ifelse(tList_instance$id %in% MS1_unique,TRUE,FALSE)
  tList_instance$MS2d <- ifelse(tList_instance$id %in% MS2$id,TRUE,FALSE)
  tList_sum <- tList_instance %>% dplyr::group_by(Priority) %>% 
    dplyr::summarize(seenSum = sum(seen),seenFrac = seenSum/n(), MS2Sum = sum(MS2d),MS2Frac = MS2Sum/n())
  #tList_sum <- tList_instance %>% dplyr::group_by(PriorityTop) %>% 
    #dplyr::summarize(seenSum = sum(seen),seenFrac = seenSum/n(), MS2Sum = sum(MS2d),MS2Frac = MS2Sum/n())
  #tList_sum3 <- tList_instance %>% dplyr::group_by(PriorityTop3) %>% 
    #dplyr::summarize(seenSum = sum(seen),seenFrac = seenSum/n(), MS2Sum = sum(MS2d),MS2Frac = MS2Sum/n())
  tList_sum$file_ind <- i
  if(i ==1){
    tList_sum_out <-tList_sum
    #tList_sum_out3 <-tList_sum3
  }
  if(i > 1){
    tList_sum_out <- rbind(tList_sum_out,tList_sum)
    #tList_sum_out3 <- rbind(tList_sum_out3,tList_sum3)
  }
}

#-----
# For Fig2bcde data
#-----
# Rebuttal consistency targList and path
Rebuttal_consistency_targList <- paste0(datIn,"Fig2/Fig2_bcde_Consistency/pSCoPE_FinalTarg_Consistency.txt")
Rebuttal_consistency_log_path <- paste0(datIn,"Fig2/Fig2_bcde_Consistency/MQL_logs_Rebuttal_consistency/")

#re-coding priority b/c we had both RT-only and targetable peptides assigned priority 0
tList_instance.consist <- read.delim(Rebuttal_consistency_targList)
tList_instance.consist$id <- seq(0,nrow(tList_instance.consist)-1,1)
tList_instance.consist$Orig_Priority <- tList_instance.consist$Priority
tList_instance.consist$Priority <- ifelse(tList_instance.consist$TargBool == FALSE, 0, tList_instance.consist$Priority + 1)
tList_instance.consist$PriorityTop <- ifelse(tList_instance.consist$Priority %in% c(4,3),"Top","Bottom")
tList_instance.consist$PriorityTop3 <- ifelse(tList_instance.consist$Priority %in% c(4,3,2),"Top","Bottom")

tList_instance.consist$id <- seq(0,nrow(tList_instance.consist)-1,1)

log_Match_DF.consist <- data.frame(log = character(), stringsAsFactors = FALSE)

logs.consistency <- list.files(Rebuttal_consistency_log_path)
log.consistency_filt <- logs.consistency[!(logs.consistency %in% c("desktop.ini","1029_20211105-2240.txt","1029_20211106-0841.txt"))]
numLogs.consistency <- length(log.consistency_filt)

for(i in 1:numLogs.consistency){
  log_instance <- read.delim(paste0(Rebuttal_consistency_log_path,log.consistency_filt[i]), header = FALSE)
  colnames(log_instance) <-  c("Date","Time","Scan_protocol","RT","Message")

  MS1 <- getMS1(log_instance)
  MS1_unique <- unique(MS1$id)
  MS2 <- getMS2(log_instance)
  tList_instance.consist$seen <- ifelse(tList_instance.consist$id %in% MS1_unique,TRUE,FALSE)
  tList_instance.consist$MS2d <- ifelse(tList_instance.consist$id %in% MS2$id,TRUE,FALSE)
  tList_sum <- tList_instance.consist %>% dplyr::group_by(Priority) %>% 
    dplyr::summarize(seenSum = sum(seen),seenFrac = seenSum/n(), MS2Sum = sum(MS2d),MS2Frac = MS2Sum/n())
  #tList_sum3 <- tList_instance.consist %>% dplyr::group_by(PriorityTop3) %>% 
  # dplyr::summarize(seenSum = sum(seen),seenFrac = seenSum/n(), MS2Sum = sum(MS2d),MS2Frac = MS2Sum/n())
  tList_sum$file_ind <- i
  if(i ==1){
    tList_sum_out.consistency <-tList_sum
  #  tList_sum_out3.consistency <-tList_sum3
  }
  if(i > 1){
    tList_sum_out.consistency <- rbind(tList_sum_out.consistency,tList_sum)
  #  tList_sum_out3.consistency <- rbind(tList_sum_out3.consistency,tList_sum3)
  }
}

targList_cov_sum <- tList_instance %>% dplyr::group_by(Priority) %>% dplyr::summarize(TargCount = n())
targList_consist_sum <- tList_instance.consist %>% dplyr::group_by(Priority) %>% dplyr::summarize(TargCount = n())

library(scales)

# BMDM and Rebuttal coverage
Priority_dict.cov <- data.frame(Priority = numeric(), Title = character(), stringsAsFactors = FALSE)
Priority_dict.cov[1:4,1] <- c(4,3,2,0)
Priority_dict.cov[1:4,2] <- c("High","Medium","Low","RT only")

# Rebuttal consistency
Priority_dict.consist <- data.frame(Priority = numeric(), Title = character(), stringsAsFactors = FALSE)
Priority_dict.consist[1:5,1] <- c(4,3,2,1,0)
Priority_dict.consist[1:5,2] <- c("High","Medium High","Medium Low","Low","RT only")

tList_sum_out.cov.j <- tList_sum_out %>% left_join(Priority_dict.cov, by = "Priority")
tList_sum_out.cov.j.f <- tList_sum_out.cov.j %>% select(seenFrac,MS2Frac,Priority,file_ind,Title)
tList_sum_out.cov.j.f.m <- melt(tList_sum_out.cov.j.f, id.vars = c("file_ind","Priority","Title"))

tList_sum_out.consistency.j <- tList_sum_out.consistency %>% left_join(Priority_dict.consist, by = "Priority")
tList_sum_out.consistency.j.f <- tList_sum_out.consistency.j %>% select(seenFrac,MS2Frac,Priority,file_ind,Title)
tList_sum_out.consistency.j.f.m <- melt(tList_sum_out.consistency.j.f, id.vars = c("file_ind","Priority","Title"))

tList_sum_out.cov.j.f.m <- tList_sum_out.cov.j.f.m %>% left_join(targList_cov_sum, by = "Priority")
tList_sum_out.consist.j.f.m <- tList_sum_out.consistency.j.f.m %>% left_join(targList_consist_sum, by = "Priority")

tList_sum_out.cov.j.f.m$legendTitle <- ifelse(tList_sum_out.cov.j.f.m$variable == "seenFrac","Detected at MS1","Sent for MS2")
tList_sum_out.cov.j.f.m$Percent <- 100*tList_sum_out.cov.j.f.m$value

tList_sum_out.consist.j.f.m$legendTitle <- ifelse(tList_sum_out.consist.j.f.m$variable == "seenFrac","Detected at MS1","Sent for MS2")
tList_sum_out.consist.j.f.m$Percent <- 100*tList_sum_out.consist.j.f.m$value

#%%%%%%%%%%%%%%%%%%
# Fig2a-data: MaxQuant.Live log file plot 
#%%%%%%%%%%%%%%%%%%

CoverageMQL.Log <- tList_sum_out.cov.j.f.m %>% ggplot(aes(Percent, color = legendTitle, y = legendTitle)) + geom_boxplot(lwd = 1) + 
  #ggtitle("Technical Performance: Coverage") + 
  geom_point(position=position_jitterdodge(), size = 2, shape = 21, fill = "grey95") +
  coord_flip(xlim = c(0,100)) + 
  theme_light() + 
  theme(axis.text.x =  element_blank(), 
        strip.background = element_rect( fill = "grey95", color = "grey70"), 
        strip.text = element_text(color= "black", size = 18,face = "bold"),
        axis.text.y = element_text(size =14),
        axis.title.y = element_text(size =16),
        plot.title = element_text(size = 18),
        legend.position = "bottom") + 
  labs(color = "", x = "Prioritized Peptides, %", y ="") + 
  facet_wrap(~factor(Title,levels = c("High","Medium","Low","RT only")), labeller = label_wrap_gen(width=10), nrow = 1)+
  scale_color_manual(values = c("firebrick4","red"))  + geom_text( x = 13, y = 1.5,color = "grey10",aes(label = paste0(comma(TargCount))), size = 7)  + geom_text( x = 10, y = 1.5,color = "grey10",aes(label = paste0("\nPrecursors")), size = 4)
#%%%%%%%%
# Fig2bcde-data: MaxQuant.Live log file plot
#%%%%%%%%

ConsistencyMQL.Log <- tList_sum_out.consist.j.f.m %>% ggplot(aes(Percent, color = legendTitle, y = legendTitle)) + geom_boxplot(lwd = 1) + 
  geom_point(position=position_jitterdodge(), size = 2, shape = 21, fill = "grey95") +
  #ggtitle("Technical Performance: Coverage") + 
  coord_flip(xlim = c(0,100)) + 
  theme_light() + 
  theme(axis.text.x =  element_blank(), 
        strip.background = element_rect( fill = "grey95", color = "grey70"), 
        strip.text = element_text(color= "black", size = 18,face = "bold"),
        axis.text.y = element_text(size =14),
        axis.title.y = element_text(size =16),
        plot.title = element_text(size = 18),
        legend.position = "bottom") + 
  labs(color = "", x = "Prioritized Peptides, %", y = "") + 
  facet_wrap(~factor(Title,levels = c("High","Medium High","Medium Low","Low","RT only")), labeller = label_wrap_gen(width=10), nrow = 1)+
  scale_color_manual(values = c("firebrick4","red"))  + geom_text( x = 13, y = 1.5,color = "grey10",aes(label = paste0(comma(TargCount))), size = 7)  + geom_text( x = 10, y = 1.5,color = "grey10",aes(label = paste0("\nPrecursors")), size = 4)



CoverageMQL.Log /plot_spacer()/ ConsistencyMQL.Log + plot_layout(heights = c(4,.5,4), guides = "collect") & theme(legend.position = 'bottom')
ggsave(paste0(FigOut,"Rebuttal_SC_MQliveStats.pdf"), width = 7, height = 8, units = "in")
```

# EDF 6: pSCoPE benchmarking against shotgun (BMDM data)
```{r}
BMDM_ShotgunSet <- c("wGH0456","wGH0457","wGH0458","wGH0459","wGH0460","wGH0461","wGH0462","wGH0463","wGH0464","wGH0465","wGH0466","wGH0467","wGH0468","wGH0469","wGH0470","wGH0471","wGH0472","wGH0473","wGH0474","wGH0475")
BMDM_PrioriSet <- c("wGH0500","wGH0501","wGH0502","wGH0503","wGH0504","wGH0505","wGH0506","wGH0507","wGH0508","wGH0509","wGH0510","wGH0511","wGH0512","wGH0513","wGH0514","wGH0515","wGH0516","wGH0517","wGH0518","wGH0519")
#BMDM_CG.s <- read.delim(paste0(BMDM_basePath,"ChanKey_Bio_Shot_20_PIF50.txt"))
#BMDM_CG.p <- read.delim(paste0(BMDM_basePath,"ChanKey_Bio_pSCoPE_20_PIF50.txt"))

BMDM_CG <- read.delim(paste0(datIn,"BMDM_technical_supplemental/ChanKey_BMDM_20Shotgun20pSCoPE_mrri02_PIF50_DART_FDR1p.txt"))
#BMDM_t0.s <- read.delim(paste0(BMDM_basePath,"t0Bio_BMDM_PIF50_limFASTA_ShotgunOnly.txt"))
#BMDM_t0.p <- read.delim(paste0(BMDM_basePath,"t0Bio_BMDM_PIF50_limFASTA_pSCoPEOnly.txt"))
#BMDM_t0.s <- read.delim(paste0(BMDM_basePath,"t0Bio_Shot_20_PIF50.txt"))
#BMDM_t0.p <- read.delim(paste0(BMDM_basePath,"t0Bio_pSCoPE_20_PIF50.txt"))

BMDM_t0 <- read.delim(paste0(datIn,"BMDM_technical_supplemental/t0_BMDM_20Shotgun20pSCoPE_mrri02_PIF50_DART_1pFDR.txt")) 
#BMDM_MDp.s <- read.delim(paste0(BMDM_basePath,"Prot_missing_BMDM_PIF50_limFASTA_ShotgunOnly_Bio.txt"))
#BMDM_MDp.p <- read.delim(paste0(BMDM_basePath,"Prot_missing_BMDM_PIF50_limFASTA_pSCoPEOnly_Bio.txt"))
#BMDM_MDp.s <- read.delim(paste0(BMDM_basePath,"Prot_missing_Shot_20_PIF50_Bio.txt"))
#BMDM_MDp.p <- read.delim(paste0(BMDM_basePath,"Prot_missing_pSCoPE_20_PIF50_Bio.txt"))
BMDM_MDp <- read.delim(paste0(datIn,"BMDM_technical_supplemental/Prot_missing_BMDM_20Shotgun20pSCoPE_mrri02_PIF50_DART_1pFDR.txt"))
  
#BMDM_tList <- read.delim("C:/My_Drive/MS/SCoPE/t-SCoPE/TLists/08_10_2021/CEO_TargList_081021_2.txt")
BMDM_tList <- read.delim(paste0(datIn,"BMDM_technical_supplemental/CEO_TargList_081021_2.txt"))
tList_priorityDictionary <- BMDM_tList %>% dplyr::filter(TargBool == TRUE) %>% dplyr::select(SeqCharge,Priority)
#BMDM_CG.s$Raw.file <- str_sub(BMDM_CG.s$Raw.file, start = 2)
#BMDM_CG.p$Raw.file <- str_sub(BMDM_CG.p$Raw.file, start = 2)
BMDM_CG$Raw.file <- str_sub(BMDM_CG$Raw.file, start = 2)
#BMDM_CG.s <- BMDM_CG.s %>% filter(Raw.file %in% BMDM_ShotgunSet, celltype != "control", cvm < .4)
#BMDM_CG.p <- BMDM_CG.p %>% filter(Raw.file %in% BMDM_PrioriSet, celltype != "control", cvm < .4)
BMDM_CG.s <- BMDM_CG %>% filter(Raw.file %in% BMDM_ShotgunSet, celltype != "control", cvm < .4)
BMDM_CG.p <- BMDM_CG %>% filter(Raw.file %in% BMDM_PrioriSet, celltype != "control", cvm < .4)
#BMDM_t0.s.m <- melt(BMDM_t0.s)
#BMDM_t0.p.m <- melt(BMDM_t0.p)
BMDM_t0.m <- melt(BMDM_t0)
BMDM_t0.s.m <- BMDM_t0.m %>% dplyr::filter(variable %in% BMDM_CG.s$id)
BMDM_t0.p.m <- BMDM_t0.m %>% dplyr::filter(variable %in% BMDM_CG.p$id)
BMDM_t0.s.m_seqSum <- BMDM_t0.s.m %>% dplyr::group_by(SeqCharge) %>% dplyr::summarize(MD = sum(is.na(value))) %>% filter(MD > 0)
BMDM_t0.p.m_seqSum <- BMDM_t0.p.m %>% dplyr::group_by(SeqCharge) %>% dplyr::summarize(MD = sum(is.na(value))) %>% filter(MD > 0)
intersectBMDM <- intersect(BMDM_t0.s.m_seqSum$SeqCharge,BMDM_t0.p.m_seqSum$SeqCharge) 
BMDM_t0.s.m_inter <- BMDM_t0.s.m %>% filter(SeqCharge %in% intersectBMDM) 
BMDM_t0.p.m_inter <- BMDM_t0.p.m %>% filter(SeqCharge %in% intersectBMDM)
BMDM_t0.s.m_inter2 <- BMDM_t0.s.m_inter %>% left_join(tList_priorityDictionary, by = "SeqCharge")
BMDM_t0.p.m_inter2 <- BMDM_t0.p.m_inter %>% left_join(tList_priorityDictionary, by = "SeqCharge")
BMDM_t0.s.m_inter2_groupCount <- BMDM_t0.s.m_inter2 %>% group_by(SeqCharge,Priority) %>% slice(1)
BMDM_t0.p.m_inter2_groupCount <- BMDM_t0.p.m_inter2 %>% group_by(SeqCharge,Priority) %>% slice(1)
BMDM_t0.s.m_inter2_groupCount_sum <- BMDM_t0.s.m_inter2_groupCount %>% dplyr::group_by(Priority) %>% dplyr::summarize(num = n()) %>% mutate(Method = "Shotgun")
BMDM_t0.p.m_inter2_groupCount_sum <- BMDM_t0.p.m_inter2_groupCount %>% dplyr::group_by(Priority) %>% dplyr::summarize(num = n()) %>% mutate(Method = "Prioritization") 
PriorityCounts <- rbind(BMDM_t0.s.m_inter2_groupCount_sum,BMDM_t0.p.m_inter2_groupCount_sum)
BMDM_t0.s.m_inter_sum <- BMDM_t0.s.m_inter2 %>% dplyr::group_by(variable,Priority) %>% dplyr::summarize(fracMD = sum(is.na(value))/n()) %>% mutate(Method = "Shotgun")
BMDM_t0.p.m_inter_sum <- BMDM_t0.p.m_inter2 %>% dplyr::group_by(variable, Priority) %>% dplyr::summarize(fracMD = sum(is.na(value))/n()) %>% mutate(Method = "Prioritization")
BMDM_inter_bind <- rbind(BMDM_t0.s.m_inter_sum,BMDM_t0.p.m_inter_sum)
#BMDM_inter_bind2 <- BMDM_inter_bind %>% left_join(BMDM_tList, by = "SeqCharge")
BMDM_inter_bind <- BMDM_inter_bind %>% left_join(PriorityCounts, by = c("Method","Priority"))
precursor_MD <- BMDM_inter_bind %>% ggplot(aes(fracMD*100, fill = Method)) + geom_boxplot() + coord_flip(xlim = c(0,100)) + labs(x = "% Missing Data") + theme_box() + theme(legend.position = "bottom") + ggtitle("Precursor-Level Missing Data","by priority tier") + facet_wrap(~factor(Priority, levels = c("3","2","1",NA)), nrow = 1) + geom_text(aes(x = 15 , y = 0 , label = paste(num, "precursors")))

BMDM_inter_bind_sum <- BMDM_inter_bind %>% group_by(Method) %>% summarize(count = n())
#--------
# peptide-level
#--------
# How to handle when peptides could be on multiple tiers due to charge states
BMDM_t0.s.m_pep <- BMDM_t0.s.m
BMDM_t0.p.m_pep <- BMDM_t0.p.m
BMDM_t0.s.m_pep$Peptide <- str_sub(BMDM_t0.s.m_pep$SeqCharge, end = -2)
BMDM_t0.p.m_pep$Peptide <- str_sub(BMDM_t0.p.m_pep$SeqCharge, end = -2)
BMDM_t0.s.m_pepSum <- BMDM_t0.s.m_pep %>% dplyr::group_by(Peptide, variable) %>% dplyr::summarize(Val = sum(value, na.rm = TRUE))
BMDM_t0.p.m_pepSum <- BMDM_t0.p.m_pep %>% dplyr::group_by(Peptide, variable) %>% dplyr::summarize(Val = sum(value, na.rm = TRUE))
BMDM_t0.s.m_pepSum$Val2 <- ifelse(BMDM_t0.s.m_pepSum$Val == 0, NA,BMDM_t0.s.m_pepSum$Val)
BMDM_t0.p.m_pepSum$Val2 <- ifelse(BMDM_t0.p.m_pepSum$Val == 0, NA,BMDM_t0.p.m_pepSum$Val)
interPepBMDM <- intersect(BMDM_t0.s.m_pepSum$Peptide,BMDM_t0.p.m_pepSum$Peptide) 
BMDM_t0.s.m_pepSum_inter <- BMDM_t0.s.m_pepSum %>% filter(Peptide %in% interPepBMDM) 
BMDM_t0.p.m_pepSum_inter <- BMDM_t0.p.m_pepSum %>% filter(Peptide %in% interPepBMDM)
tList_priorityDictionary$Peptide <- str_sub(tList_priorityDictionary$SeqCharge, end = -2) 
tList_priorityDictionary_lim <- tList_priorityDictionary %>% group_by(Peptide) %>% top_n(1, Priority) %>% slice(1)
BMDM_t0.s.m_pepSum_inter2 <- BMDM_t0.s.m_pepSum_inter %>% left_join(tList_priorityDictionary_lim, by = "Peptide")
BMDM_t0.p.m_pepSum_inter2 <- BMDM_t0.p.m_pepSum_inter %>% left_join(tList_priorityDictionary_lim, by = "Peptide")
BMDM_t0.s.m_pepSum_inter2_groupCount <- BMDM_t0.s.m_pepSum_inter2 %>% group_by(Peptide,Priority) %>% slice(1)
BMDM_t0.p.m_pepSum_inter2_groupCount <- BMDM_t0.p.m_pepSum_inter2 %>% group_by(Peptide,Priority) %>% slice(1)
BMDM_t0.s.m_pepSum_inter2_groupCount_sum <- BMDM_t0.s.m_pepSum_inter2_groupCount %>% dplyr::group_by(Priority) %>% dplyr::summarize(num = n()) %>% mutate(Method = "Shotgun")
BMDM_t0.p.m_pepSum_inter2_groupCount_sum <- BMDM_t0.p.m_pepSum_inter2_groupCount %>% dplyr::group_by(Priority) %>% dplyr::summarize(num = n()) %>% mutate(Method = "Prioritization") 
PriorityCountsPep <- rbind(BMDM_t0.s.m_pepSum_inter2_groupCount_sum,BMDM_t0.p.m_pepSum_inter2_groupCount_sum)
BMDM_t0.s.m_pepSum_inter_sum <- BMDM_t0.s.m_pepSum_inter2 %>% dplyr::group_by(variable,Priority) %>% dplyr::summarize(fracMD = sum(is.na(Val2))/n()) %>% mutate(Method = "Shotgun")
BMDM_t0.p.m_pepSum_inter_sum <- BMDM_t0.p.m_pepSum_inter2 %>% dplyr::group_by(variable, Priority) %>% dplyr::summarize(fracMD = sum(is.na(Val2))/n()) %>% mutate(Method = "Prioritization")
BMDM_interPep_bind <- rbind(BMDM_t0.s.m_pepSum_inter_sum,BMDM_t0.p.m_pepSum_inter_sum)
#BMDM_inter_bind2 <- BMDM_inter_bind %>% left_join(BMDM_tList, by = "SeqCharge")
BMDM_interPep_bind <- BMDM_interPep_bind %>% left_join(PriorityCountsPep, by = c("Method","Priority"))
BMDM_interPep_bind$DataCompleteness <- 1 - BMDM_interPep_bind$fracMD
peptide_MD <- BMDM_interPep_bind %>% ggplot(aes(fracMD*100, fill = Method)) + geom_boxplot() + coord_flip(xlim = c(0,100)) + labs(x = "% Missing Data") + theme_box() + theme(legend.position = "bottom") + ggtitle("Peptide-Level Missing Data","by priority tier") + facet_wrap(~factor(Priority, levels = c("3","2","1",NA)), nrow = 1) + geom_text(aes(x = 15 , y = 0 , label = paste(num, "peptides")))
#precursor_MD 
#peptide_MD
#------
# Protein level
#------
#BMDM_ev.s <- read.delim(paste0(BMDM_basePath,"evidence_filtBio_BMDM_PIF50_limFASTA_ShotgunOnly.txt"))
#BMDM_ev.p <- read.delim(paste0(BMDM_basePath,"evidence_filtBio_BMDM_PIF50_limFASTA_pSCoPEOnly.txt"))
#BMDM_ev.s <- read.delim(paste0(BMDM_basePath,"evidence_filtBio_Shot_20_PIF50.txt"))
#BMDM_ev.p <- read.delim(paste0(BMDM_basePath,"evidence_filtBio_pSCoPE_20_PIF50.txt"))
BMDM_ev <- read.delim(paste0(datIn,"BMDM_technical_supplemental/ev_20Shotgun20pSCoPE_02182022_mrri02_PIF50_DART_1pFDR.txt"))
#BMDM_ev <- BMDM_ev %>% select(-Acetyl..Protein.N.term.)
##BMDM_ev <- rbind(BMDM_ev.s,BMDM_ev.p)
BMDM_ev$SeqCharge <- BMDM_ev$modseq
BMDM_ev_lim <- BMDM_ev %>% dplyr::select(SeqCharge, Leading.razor.protein) %>% group_by(SeqCharge) %>% slice(1)
# Perhaps better is to go from Sequence to Protein
BMDM_t0.s.m_4Prot <- BMDM_t0.s.m
BMDM_t0.p.m_4Prot <- BMDM_t0.p.m
BMDM_t0.s.m_4Prot <- BMDM_t0.s.m_4Prot %>% left_join(BMDM_ev_lim, by = "SeqCharge") 
BMDM_t0.p.m_4Prot <- BMDM_t0.p.m_4Prot %>% left_join(BMDM_ev_lim, by = "SeqCharge")
BMDM_t0.s.m_4Protsum <- BMDM_t0.s.m_4Prot %>% dplyr::group_by(variable, Leading.razor.protein) %>% dplyr::summarize(ProtVal = sum(value, na.rm = TRUE))
BMDM_t0.p.m_4Protsum <- BMDM_t0.p.m_4Prot %>% dplyr::group_by(variable, Leading.razor.protein) %>% dplyr::summarize(ProtVal = sum(value, na.rm = TRUE))
prot_inter <- intersect(BMDM_t0.s.m_4Protsum$Leading.razor.protein,BMDM_t0.p.m_4Protsum$Leading.razor.protein)
BMDM_t0.s.m_4Protsum_inter <- BMDM_t0.s.m_4Protsum %>% dplyr::filter(Leading.razor.protein %in% prot_inter) %>% dplyr::mutate(Method = "Shotgun")
BMDM_t0.p.m_4Protsum_inter <- BMDM_t0.p.m_4Protsum %>% dplyr::filter(Leading.razor.protein %in% prot_inter) %>% dplyr::mutate(Method = "Prioritization")
BMDM_prot <- rbind(BMDM_t0.s.m_4Protsum_inter,BMDM_t0.p.m_4Protsum_inter)
tList_priorityDictionary_Protlim <- BMDM_tList %>% dplyr::filter(TargBool == TRUE) %>% dplyr::select(Leading.razor.protein,Priority) %>% group_by(Leading.razor.protein) %>% top_n(1, Priority) %>% slice(1)
BMDM_prot <- BMDM_prot %>% left_join(tList_priorityDictionary_Protlim, by = "Leading.razor.protein")
BMDM_prot$Val2 <- ifelse(BMDM_prot$ProtVal == 0, NA, BMDM_prot$ProtVal)
BMDM_prot_sum <- BMDM_prot %>% dplyr::group_by(variable, Priority, Method) %>% dplyr::summarize(fracMD = sum(is.na(Val2))/n())
BMDM_prot_sumPriority <- BMDM_prot %>% dplyr::group_by(Priority, Leading.razor.protein) %>% slice(1)
BMDM_prot_sumPriority2 <- BMDM_prot_sumPriority %>% dplyr::ungroup() %>% dplyr::group_by(Priority) %>% dplyr::summarize(numProts = n())
BMDM_prot_sum <- BMDM_prot_sum %>% dplyr::left_join(BMDM_prot_sumPriority2, by = "Priority")
BMDM_prot_sum$FacetLab <- ifelse(BMDM_prot_sum$Priority == 3, "High",ifelse(BMDM_prot_sum$Priority == 2, "Medium",ifelse(BMDM_prot_sum$Priority == 1, "Low","NA")))
BMDM_prot_sum$DataCompleteness <- 1 - BMDM_prot_sum$fracMD
BMDM_interPep_bind$FacetLab <- ifelse(BMDM_interPep_bind$Priority == 3, "High",ifelse(BMDM_interPep_bind$Priority == 2, "Medium",ifelse(BMDM_interPep_bind$Priority == 1, "Low","NA")))

BMDM_prot_sum_sum <- BMDM_prot_sum %>% group_by(Method, Priority) %>% summarize(count = n())

BMDM_interPep_bind_sum <- BMDM_interPep_bind %>% group_by(Method) %>% summarize(count = n())

protMD <- BMDM_prot_sum %>% filter(FacetLab != "NA") %>% ggplot(aes(DataCompleteness*100, color = factor(Method, levels = c("Shotgun","Prioritization")))) + geom_boxplot() + coord_flip(xlim = c(0,100)) + labs(x = "", color = "", y = "") + theme_box() + theme(legend.position = "null", axis.text.y = element_blank(), axis.title = element_text(size = 26), strip.background = element_rect(fill = "grey95", color = "grey70"), strip.text = element_text(color = "black",size = 20), plot.title = element_text(size = 28)) + ggtitle("Proteins") + facet_wrap(~factor(FacetLab, levels = c("High","Medium","Low","NA")), nrow = 1) + geom_text(aes(x = 2 , y = 0 , label = paste(numProts, "\nproteins")), color = "black",size = 6)  + scale_color_manual(values = c("blue","red"))

peptide_MD <- BMDM_interPep_bind %>% filter(FacetLab != "NA") %>% ggplot(aes(DataCompleteness*100, color = factor(Method, levels = c("Shotgun","Prioritization")))) + geom_boxplot() + coord_flip(xlim = c(0,100)) + labs(x = "% Data Completeness", color = "", y = "") + theme_box() + theme(legend.position = "null", axis.title.y = element_text(size = 30), strip.background = element_rect(fill = "grey95", color = "grey70"), strip.text = element_text(color = "black", size = 20), axis.text.y = element_text(size = 28), plot.title = element_text(size = 28)) + ggtitle("Peptides") + facet_wrap(~factor(FacetLab, levels = c("High","Medium","Low","NA")), nrow = 1) + geom_text(aes(x = 2 , y = 0 , label = paste(num, "\npeptides")), color = "black",size = 6) + scale_color_manual(values = c("blue","red"))
#precursor_MD 
#peptide_MD
#protMD
fig1Supp_pepMD <- BMDM_interPep_bind %>% group_by(Method,FacetLab) %>% summarize(medVal = median(fracMD, na.rm = TRUE))
fig1Supp_protMD <- BMDM_prot_sum %>% group_by(Method,FacetLab) %>% summarize(medVal = median(fracMD, na.rm = TRUE))

#%%%%%%%%%%%%%
# EDF6a: peptide and protein-level contrast
#%%%%%%%%%%%%%

peptide_MD + protMD + plot_annotation(title = 'BMDMs: Missing Data per Cell',theme = theme(plot.title = element_text(size = 28))) + plot_layout(guides = "collect",widths = c(1,1)) & theme(legend.position = 'null')
ggsave(paste0(FigOut,"BMDM_missingData_v2_mrri02.pdf"), width = 9 , height = 7, units = "in")

#----------------

BMDM_MDp.s <- BMDM_MDp %>% dplyr::select(any_of(BMDM_CG.s$id))
BMDM_MDp.p <- BMDM_MDp %>% dplyr::select(any_of(BMDM_CG.p$id))
BMDM_MDp.s.m <- melt(BMDM_MDp.s)
BMDM_MDp.p.m <- melt(BMDM_MDp.p)
BMDM_MDp.s.m$Method <- "Shotgun"
BMDM_MDp.p.m$Method <- "Prioritization"
BMDM_MDp.s.m <- BMDM_MDp.s.m %>% dplyr::filter(variable %in% BMDM_CG.s$id)
BMDM_MDp.p.m <- BMDM_MDp.p.m %>% dplyr::filter(variable %in% BMDM_CG.p$id)
BMDM_MDp.join <- rbind(BMDM_MDp.s.m,BMDM_MDp.p.m)
BMDM_MDp.join_sum <- BMDM_MDp.join %>% dplyr::group_by(variable, Method) %>% dplyr::summarize("Proteins/cell"=sum(!is.na(value))) 

BMDM_MDp.join_sum_sum <- BMDM_MDp.join_sum %>% group_by(Method) %>% summarize(count = n())

protsPerCell <- BMDM_MDp.join_sum %>% ggplot(aes(x = `Proteins/cell`, color = factor(Method, levels = c("Shotgun","Prioritization")))) + geom_boxplot() + coord_flip(xlim = c(0,800)) + theme_box() + theme(legend.position = "bottom", axis.title.y = element_text(size = 30), axis.text.y = element_text(size = 28)) + labs(color = "") + scale_color_manual(values = c("blue","red"))


BMDM_t0.s.m_pepSum$Method <- "Shotgun"
BMDM_t0.p.m_pepSum$Method <- "Prioritization"
BMDM_pep.join <- rbind(BMDM_t0.s.m_pepSum,BMDM_t0.p.m_pepSum)
BMDM_pep.join_sum <- BMDM_pep.join %>% dplyr::group_by(variable, Method) %>% dplyr::summarize("Peptides/cell"=sum(!is.na(Val2))) 

pepsPerCell <- BMDM_pep.join_sum %>% ggplot(aes(x = `Peptides/cell`, color = factor(Method, levels = c("Shotgun","Prioritization")))) + geom_boxplot() + coord_flip(xlim = c(0,2200)) + theme_box() + theme(legend.position = "bottom",axis.title.y = element_text(size = 30), axis.text.y = element_text(size = 28)) + labs(color = "") + scale_color_manual(values = c("blue","red"))

fig1Supp2_protCov <- BMDM_MDp.join_sum %>% group_by(Method) %>% summarize(medVal = median(`Proteins/cell`, na.rm = TRUE))
fig1Supp2_pepCov <- BMDM_pep.join_sum %>% group_by(Method) %>% summarize(medVal = median(`Peptides/cell`, na.rm = TRUE))

#%%%%%%%%%%%%%
# EDF6b: peptides and proteins per cell
#%%%%%%%%%%%%%

pepsPerCell + protsPerCell + plot_annotation(title = 'BMDMs: Data Points per Cell',theme = theme(plot.title = element_text(size = 28))) + plot_layout(guides = "collect",widths = c(1,1)) & theme(legend.position = 'null')  
ggsave(paste0(FigOut,"BMDM_DataPointsPerCell_mrri02.pdf"), width = 7, height = 7, units = "in")

#-----------------

#For precursors with higher fill times, were these identified in more runs?
BMDM_tList_4FT_1 <- read.delim(paste0(datIn,"BMDM_technical_supplemental/CEO_TargList_081021_2.txt"))
BMDM_tList_4FT_1 <- BMDM_tList_4FT_1 %>% dplyr::filter(TargBool == TRUE)
BMDM_tList_4FT <- BMDM_tList_4FT_1 %>% dplyr::filter(FT_priority > 500)
#BMDM_ev.s <- read.delim(paste0(BMDM_basePath,"evidence_filtBio_BMDM_PIF50_limFASTA_ShotgunOnly.txt"))
#BMDM_ev.p <- read.delim(paste0(BMDM_basePath,"evidence_filtBio_BMDM_PIF50_limFASTA_pSCoPEOnly.txt"))

BMDM_ev <- read.delim(paste0(datIn,"BMDM_technical_supplemental/ev_20Shotgun20pSCoPE_02182022_mrri02_PIF50_DART_1pFDR.txt"))
#BMDM_ev.s <- read.delim(paste0(BMDM_basePath,"evidence_filtBio_Shot_20_PIF50.txt"))
#BMDM_ev.p <- read.delim(paste0(BMDM_basePath,"evidence_filtBio_pSCoPE_20_PIF50.txt"))
BMDM_t0.s.m_FT <- BMDM_t0.s.m %>% dplyr::filter(SeqCharge %in% BMDM_tList_4FT$SeqCharge)
BMDM_t0.p.m_FT <- BMDM_t0.p.m %>% dplyr::filter(SeqCharge %in% BMDM_tList_4FT$SeqCharge)
interFT <- intersect(BMDM_t0.s.m_FT$SeqCharge,BMDM_t0.p.m_FT$SeqCharge)
interFTdf <- as.data.frame(interFT)
interFTdf$peptides <- str_sub(interFTdf$interFT, end = -2)
BMDM_t0.s.m_FT_inter <- BMDM_t0.s.m_FT %>% dplyr::filter(SeqCharge %in% interFT)
BMDM_t0.p.m_FT_inter <- BMDM_t0.p.m_FT %>% dplyr::filter(SeqCharge %in% interFT)
BMDM_t0.s.m_FT_inter_sum <- BMDM_t0.s.m_FT_inter %>% dplyr::group_by(variable) %>% dplyr::summarize(fracMissing = sum(is.na(value))/n()) %>% dplyr::mutate(Method = "Shotgun")
BMDM_t0.p.m_FT_inter_sum <- BMDM_t0.p.m_FT_inter %>% dplyr::group_by(variable) %>% dplyr::summarize(fracMissing = sum(is.na(value))/n()) %>% dplyr::mutate(Method = "Prioritization")
FT_bind <- rbind(BMDM_t0.s.m_FT_inter_sum,BMDM_t0.p.m_FT_inter_sum)
FT_bind$DataCompleteness <- 1 - FT_bind$fracMissing

FT_bind_sum <- FT_bind %>% group_by(Method) %>% summarize(count = n())

MDhist <- FT_bind %>% ggplot(aes(DataCompleteness*100, color = factor(Method, levels = c("Shotgun","Prioritization")))) + geom_boxplot() + coord_flip(xlim = c(0, 100)) + theme_box() + labs(x = "% Data Completeness, Peptides", y = "") + geom_text(aes(y = 0, x = 1, label = paste0(length(unique(interFTdf$peptides)), " peptides")), color = "black",size = 8)+ theme(legend.position ="null",axis.title.y = element_text(size = 30), axis.text.y = element_text(size = 28)) + scale_color_manual(values = c("blue","red"))

# Showing abundance
pSCoPE_evi_4FT <- read.delim(paste0(datIn,"BMDM_technical_supplemental/ev_20Shotgun20pSCoPE_02182022_mrri02_PIF50_DART_1pFDR.txt"))
pSCoPE_evi_4FT$SeqCharge <- paste0(pSCoPE_evi_4FT$Modified.sequence,pSCoPE_evi_4FT$Charge)
pSCoPE_evi_4FT_join <- pSCoPE_evi_4FT %>% left_join(BMDM_tList_4FT_1, by = "SeqCharge")
#MDhist <- pSCoPE_evi_4FT_join %>% dplyr::filter(Priority == 3) %>% ggplot(aes(log10(Intensity.x), fill = factor(FT_priority))) + geom_boxplot(position = position_dodge()) + coord_flip()+ theme(legend.position ="null")
BMDM_tList_4FT_quartiles <- quantile(BMDM_tList_4FT_1$Intensity, c(.33,.66))
BMDM_tList_4FT_1_2 <- BMDM_tList_4FT_1
BMDM_tList_4FT_1_2$TertileColor <- ifelse(BMDM_tList_4FT_1_2$Intensity <= BMDM_tList_4FT_quartiles[1],"1000ms",ifelse((BMDM_tList_4FT_1_2$Intensity > BMDM_tList_4FT_quartiles[1])&(BMDM_tList_4FT_1_2$Intensity <= BMDM_tList_4FT_quartiles[2]),"750ms","500ms"))

intHist <- BMDM_tList_4FT_1_2 %>% dplyr::filter(Priority %in% c(3,2,1)) %>% ggplot(aes(log10(Intensity), fill = factor(TertileColor, levels = c("500ms","750ms","1000ms")))) + geom_histogram(alpha = 0.9, bins = 100) + coord_flip() + theme_light() + geom_vline(aes(xintercept = log10(BMDM_tList_4FT_quartiles[1]))) + geom_vline(aes(xintercept = log10(BMDM_tList_4FT_quartiles[2]))) + scale_fill_manual(values = c("paleturquoise1","paleturquoise3","paleturquoise4")) + theme_box() + 
  theme(legend.position=c(.65,.20),axis.title.y = element_text(size = 30), axis.text.y = element_text(size = 28)) + labs(y = "", x = bquote("Intensity, Log"[10]), fill = "Fill time")

fig1Supp2_FillTimeSet<- FT_bind %>% group_by(Method) %>% summarize(medVal = median(fracMissing, na.rm = TRUE))

library(patchwork)

#%%%%%%%%%
# EDF6c: FT-dependent data completion rates 
#%%%%%%%%%

intHist + MDhist + plot_annotation(title = 'Intensity-dependent MS2 Fill Times',theme = theme(plot.title = element_text(size = 28)))
ggsave(paste0(FigOut,"BMDM_IncreasedFillTimes_mrri02.pdf"), width = 7, height = 7, units = "in")

```

# EDF 7: PCA of pSCoPE BMDM data using only observed data
```{r}

datPath <- paste0(datIn,"BMDM_figs_4_5_6/")
#Precursor-level
t0_mat <- read.delim(paste0(datPath, "t0_BMDM_priori_mrri02_PIF50_DART_1pFDR.txt"))
prePCA <- read.csv(paste0(datPath,"limmaCorrected_normed_prePCA_Priori_mrri02_PIF50_DART_1pFDR.csv"),row.names = "X")
CG <- read.delim(paste0(datPath, "ChanKey_BMDM_Priori_mrri02_PIF50_DART_FDR1p.txt"))
GeneMappings_single <- read.csv(paste0(datPath,"GeneMappings.csv"))
ProtMissing <- read.delim(paste0(datPath,"Prot_missing_BMDM_priori_mrri02_PIF50_DART_1pFDR.txt"))
GO_DB_Path <- paste0(datPath,"GOA_db.txt")
GO_db <- read.delim(GO_DB_Path, sep = "", header = TRUE)
GO_db.2 <- GO_db %>% filter(Letter != "C")

GeneMappings_single$GENES <- gsub(" .*","",GeneMappings_single$GENES)
GeneMappings_single$GENES <- toupper(GeneMappings_single$GENES)
GeneMappings_single2 <- GeneMappings_single %>% select(-X)

BC_imp_PxC2 <- prePCA
BC_imp_PxC2$Proteins <- row.names(BC_imp_PxC2)
BC_imp_PxC2.m <- melt(BC_imp_PxC2, id.vars = "Proteins")
# Removing imputed protein measurements, by cross-referencing pre-imputation cells x proteins matrix
ProtMissing.m <- melt(ProtMissing)
BC_imp_PxC3.m <- BC_imp_PxC2.m %>% left_join(ProtMissing.m, by = c("Proteins", "variable"))
BC_imp_PxC3.m$newVar <- ifelse(is.na(BC_imp_PxC3.m$value.y),NA,BC_imp_PxC3.m$value.x)
BC_unImp_PxC3.m2 <- BC_imp_PxC3.m %>% select(Proteins,variable,newVar)
BC_unImp_PxC3 <- dcast(BC_unImp_PxC3.m2, Proteins~variable)
row.names(BC_unImp_PxC3) <- BC_unImp_PxC3$Proteins
BC_unImp_PxC4 <- BC_unImp_PxC3 %>% select(-Proteins)
BC_unImp_CxP <- as.data.frame(t(BC_unImp_PxC4))
BC_unImp_CxP$id <- row.names(BC_unImp_CxP)
BC_unImp_PxC3.m3 <- BC_unImp_PxC3.m2 %>% left_join(GeneMappings_single2, by = c("Proteins"="UNIPROTKB"))

# column and row normalizing unimputed data for PCA
centered_unimp_data <- scale(BC_unImp_PxC4, center = TRUE, scale = FALSE)
corrCells <- cor(centered_unimp_data, method = "pearson", use = "pairwise.complete.obs")
sampXfeature <- scale(as.matrix(t(BC_unImp_PxC4)),center = TRUE, scale = FALSE)
#sampXfeature <- (as.matrix(t(BC_unImp_PCin)))
corrFeature <- cor(sampXfeature, use = "pairwise.complete.obs")
corrFeature.df <- (as.data.frame(corrFeature))
corrFeature.f <- corrFeature[!is.na(colSums(corrFeature)),!is.na(colSums(corrFeature))]
eigDecomp.feature <- eigen(corrFeature.f)
eigDecomp.feature.vec <- as.data.frame(eigDecomp.feature$vectors[,1:5])
row.names(eigDecomp.feature.vec) <- row.names(corrFeature.f)
colnames(eigDecomp.feature.vec) <- c("PC1","PC2","PC3","PC4","PC5")
eigDecomp.cells <- eigen(corrCells)
eigDecomp.cells.vec <- as.data.frame(eigDecomp.cells$vectors[,1:5])
row.names(eigDecomp.cells.vec) <- row.names(corrCells)
eigDecomp.cells.vec$id <- row.names(eigDecomp.cells.vec)
eigDecomp.cells.vec <- eigDecomp.cells.vec %>% left_join(CG, by = "id")
featureXsamp_lim <- centered_unimp_data[row.names(centered_unimp_data) %in% row.names(eigDecomp.feature.vec),]
featureXsamp_lim.zeroed <- featureXsamp_lim 
featureXsamp_lim.zeroed[is.na(featureXsamp_lim.zeroed)] <- 0
PC_coords <- as.data.frame(t(as.matrix(featureXsamp_lim.zeroed)) %*% as.matrix(eigDecomp.feature.vec))
PC_coords$id <- row.names(PC_coords)
PC_coords <- PC_coords %>% left_join(CG, by = "id")

#BC_unimp_LPS_PxC3.m %>% left_join(GeneMappings_single2, by = c("Var1"="UNIPROTKB"))

unimp_xcond_PS_vec <- c("phagosome maturation","type I interferon-mediated signaling pathway")
BC_unImp_PxC3.m3_GO_in <- BC_unImp_PxC3.m3
BC_unImp_PxC3.m3_GO_in$Var2 <- BC_unImp_PxC3.m3_GO_in$variable
unimp_xcond_PS_abund <- GO_for_PCA(unimp_xcond_PS_vec,BC_unImp_PxC3.m3_GO_in)

PC_coords_PSabund <- PC_coords %>% left_join(unimp_xcond_PS_abund, by = "id")

celltype_unImp <- PC_coords_PSabund %>% ggplot(aes(x = -PC1, y = -PC2, color = celltype)) + geom_point(size = 3,alpha = 0.75, shape = 16) + theme_light()  + labs(color = "") + annotate("text", x = 3, y = 4.5,size = 8, label = "LPS-treated", color = "#F8766D") + annotate("text", x = -2.5, y = 4.5,size = 8, label = "Untreated", color = "#00BFC4") + theme(legend.position = "null", axis.text = element_text(size = 16),axis.title = element_text(size = 20)) + labs(x = "PC1", y = "PC2")

typeIIFN <- PC_coords_PSabund %>% ggplot(aes(x = -PC1, y = -PC2, color = `type I interferon-mediated signaling pathway`)) + geom_point(size = 2) + theme_light() + ggtitle("Type I IFN signaling") + scale_color_gradient2(low = "blue",mid = "white",high = "red",midpoint = 0) + labs(color = "Abundance\nz-score") + theme(legend.title = element_text(size = 12),plot.title = element_text(size = 14)) + labs(x = "PC1", y = "PC2")

PhagosomeMat <- PC_coords_PSabund %>% ggplot(aes(x = -PC1, y = -PC2, color = `phagosome maturation`)) + geom_point(size = 2) + theme_light() + ggtitle("Phagosome Maturation") + scale_color_gradient2(low = "blue",mid = "white",high = "red",midpoint = 0) + labs(color = "Abundance\nz-score") + theme(legend.title = element_text(size = 12),plot.title = element_text(size = 14)) + labs(x = "PC1", y = "PC2")

#%%%%%%%%%%
#EDF7 (left panel)
#%%%%%%%%%%

celltype_unImp 
ggsave(paste0(FigOut,"unimp_PCA_celltype_header.pdf"), width = 5.5, height = 5.5, units = "in")
  
#%%%%%%%%%%
#EDF7 (right panel)
#%%%%%%%%%%

(typeIIFN + PhagosomeMat) + plot_layout(guides = "collect", widths = c(1,1)) & theme(legend.position = "bottom", legend.text = element_text(size = 14),legend.title = element_text(size = 12)) 
ggsave(paste0(FigOut,"unimp_PCA_PSEA_footer_select.pdf"), width = 5.5, height = 3.75, units = "in")
```

# EDF 8: PCA of pSCoPE BMDM data color-coded by missingness
```{r}

# NOTE:
# Use data loaded in EDF7 code chunk, above

#Precursor-level assessment of missingness

t0.m <- melt(t0_mat, id.vars = "SeqCharge")
t0.m_sum <- t0.m %>% group_by(variable) %>% summarize(fracMD = sum(is.na(value))/n())

PCAout_both <- prcomp(t(prePCA))
PCAout_both_coords <- PCAout_both$x


PCAout_both_coords_MDsupp <- as.data.frame(PCAout_both_coords[,1:2])
PCAout_both_coords_MDsupp$id <- row.names(PCAout_both_coords_MDsupp)
t0_pca <- PCAout_both_coords_MDsupp %>% left_join(t0.m_sum, by = c("id" = "variable"))
t0_pca$Completeness <- (1-t0_pca$fracMD)*100


#%%%%%%%%%%%%%%%%%%
# Precursor level plot (not used in publication, but demonstrates similar trend to protein-level plot)
#%%%%%%%%%%%%%%%%%%

MD_pca <- t0_pca %>% ggplot(aes(x = PC1, y = PC2, color = Completeness)) + geom_point( size = 5, alpha = 0.7,shape = 16) + scale_color_gradient2(low = "blue", mid = "white", high = "red", midpoint = 25) + theme_light() + labs(color = "Precursor-level\ndata completeness, %")  + theme(legend.position = "bottom")  + theme(legend.position = "bottom",legend.title = element_text(size = 14),legend.text = element_text(size = 14), axis.text = element_text(size = 18),axis.title = element_text(size = 24)) + labs(x = "PC1", y = "PC2")
MD_pca
#ggsave(paste0(FigOut,"DataCompleteness_PCA_supplemental.pdf"), width = 7, height = 7, units = "in")

#%%%%%%%%%%%%%%%%%%
# EDF8: PCA color-coded by protein-level missingness
#%%%%%%%%%%%%%%%%%%

# Protein-level assessment of missingness
PMD.m <- melt(ProtMissing, id.vars = "Proteins")
PMD.m_sum <- PMD.m %>% group_by(variable) %>% summarize(fracMD = sum(is.na(value))/n())
PCAout_both_coords_MDsupp <- as.data.frame(PCAout_both_coords[,1:2])
PCAout_both_coords_MDsupp$id <- row.names(PCAout_both_coords_MDsupp)
MD_pca <- PCAout_both_coords_MDsupp %>% left_join(PMD.m_sum, by = c("id" = "variable"))
MD_pca$Completeness <- (1-MD_pca$fracMD)*100
medMD <- mean(MD_pca$Completeness)
MD_pca_plot <- MD_pca %>% ggplot(aes(x = PC1, y = PC2, color = Completeness)) + geom_point( size = 5, alpha = 0.7,shape = 16) + scale_color_gradient2(low = "blue", mid = "white", high = "red", midpoint = 40, breaks = c(20,40,60)) + theme_light() + labs(color = "Protein-level\ndata completeness, %")  + theme(legend.position = "bottom")  + theme(legend.position = "bottom",legend.title = element_text(size = 14),legend.text = element_text(size = 14), axis.text = element_text(size = 18),axis.title = element_text(size = 24)) + labs(x = "PC1", y = "PC2")
MD_pca_plot
ggsave(paste0(FigOut,"DataCompleteness_PCA_supplemental.pdf"), width = 7, height = 7, units = "in")
```

# EDF 9: Sorter info
# Note: Plot generated from FACS instrument software

#EDF 10: Dextran uptake for untreated samples
```{r}
# Note:
# Requires data and processed data from EDF7 code chunk


Dextran_DIA_LPS <- read.delim(paste0(datPath,"DexLPS_LowOverHigh__eGH_358_eGH_360_Data_for_volcanoplot.txt"))
Dextran_DIA_NT <- read.delim(paste0(datPath,"DexNT_LowOverHigh__eGH_362_eGH_364_Data_for_volcanoplot.txt"))

Dextran_DIA_LPS_filt <- Dextran_DIA_LPS %>% filter(direction != "Not Significant",abs(median_prot_FC) >= 1,!grepl("keratin",PG.ProteinDescriptions))
Dextran_DIA_NT_filt <- Dextran_DIA_NT %>% filter(direction != "Not Significant",abs(median_prot_FC) >= 1,!grepl("keratin",PG.ProteinDescriptions))

# Extracting sets of antiendocytic and proendocytic proteins
Dextran_DIA_LPS_filt_AntiEndocytic <- Dextran_DIA_LPS %>% filter(direction != "Not Significant",median_prot_FC >= 1,n>1,FC_qval <= .01)
Dextran_DIA_LPS_filt_Endocytic <- Dextran_DIA_LPS %>% filter(direction != "Not Significant",median_prot_FC <= -1,n>1,FC_qval <= .01)
Dextran_DIA_NT_filt_AntiEndocytic <- Dextran_DIA_NT %>% filter(direction != "Not Significant",median_prot_FC >= 1,n>1,FC_qval <= .01)
Dextran_DIA_NT_filt_Endocytic <- Dextran_DIA_NT %>% filter(direction != "Not Significant",median_prot_FC <= -1,n>1,FC_qval <= .01)

Dextran_DIA_LPS_filt_AntiEndocytic_lim <- Dextran_DIA_LPS_filt_AntiEndocytic %>% select(PG.ProteinAccessions)
Dextran_DIA_LPS_filt_Endocytic_lim <- Dextran_DIA_LPS_filt_Endocytic %>% select(PG.ProteinAccessions)
Dextran_DIA_NT_filt_AntiEndocytic_lim <- Dextran_DIA_NT_filt_AntiEndocytic %>% select(PG.ProteinAccessions)
Dextran_DIA_NT_filt_Endocytic_lim <- Dextran_DIA_NT_filt_Endocytic %>% select(PG.ProteinAccessions)

Dextran_dict_LPS <- Dextran_DIA_LPS %>% select(PG.ProteinAccessions,PG.ProteinDescriptions) %>% group_by(PG.ProteinAccessions) %>% slice(1)

Endocytosis_union <- as.data.frame(intersect(Dextran_DIA_LPS_filt_Endocytic_lim$PG.ProteinAccessions,Dextran_DIA_NT_filt_Endocytic_lim$PG.ProteinAccessions))
colnames(Endocytosis_union) <- "PG.ProteinAccessions"

Endocytosis_LPS_outer <- Dextran_DIA_LPS_filt_Endocytic_lim %>% filter(!(PG.ProteinAccessions %in% 
                                                                        Endocytosis_union$PG.ProteinAccessions))
Endocytosis_NT_outer <- Dextran_DIA_NT_filt_Endocytic_lim %>% filter(!(PG.ProteinAccessions %in% 
                                                                         Endocytosis_union$PG.ProteinAccessions))

Endocytosis_union <- Endocytosis_union %>% left_join(Dextran_dict_LPS, by = "PG.ProteinAccessions")


#%%%%%%%%%%%%%%%%%
# EDF10: Sorter Plot (Top Panel)
#%%%%%%%%%%%%%%%%%

library(tidyr)
library(scales)

NT_uptake1 <- read.delim(paste0(datPath,"dextran_graph_NT1.txt"))
NT_uptake2 <- read.delim(paste0(datPath,"dextran_graph_NT2.txt"))

NT_uptake1_obs <- NT_uptake1 %>% uncount(weights = Events)
NT_uptake1_obs.f <- NT_uptake1_obs %>% filter((NT_uptake1_obs$MFI > 1000)&(NT_uptake1_obs$MFI < 50000))


NT_uptake1_obs.f$log10MFI <- log10(NT_uptake1_obs.f$MFI)

NT_quants <- (quantile((NT_uptake1_obs.f$MFI),c(.1,.5,.9)))

ggplot(NT_uptake1_obs.f, aes(x = (MFI), y = ..count../sum(..count..), fill = ..x..)) +
  geom_histogram(binwidth = .05, color = "grey70") +
  scale_x_log10(breaks = trans_breaks("log10", function(x) 10^x),
              labels = trans_format("log10", math_format(.x))) + scale_y_continuous(labels = percent_format()) +
  scale_fill_gradient2(low = "#ad0060",high = "#008f67", midpoint = log10(NT_quants[2]))  + theme_light() +labs(y = "Cells, %", x = bquote("Dextran: Alexa Fluor 568, log"[10])) + theme(axis.text = element_text(size = 24), axis.title = element_text(size = 32), legend.text = element_text(size = 26), legend.position="null",plot.margin =unit(c(1,0,0,0),"cm")) + geom_vline(xintercept = NT_quants[[1]],linetype = "dashed", color = "grey60", size = 2) + geom_vline(xintercept = NT_quants[[3]],linetype = "dashed",color = "grey60", size = 2) + 
  annotate("text",x = (NT_quants[[1]] - 2550) , y = .075,size = 12, label = "Low", color = "#D1509D")+
  annotate("text",x = (NT_quants[[1]] - 2550) , y = .062,size = 12, label = "Uptake", color = "#D1509D")+
  annotate("text",x = (NT_quants[[1]] - 2550) , y = .050,size = 8.5, label = "10th Percentile", color = "#D1509D")+
  annotate("text",x = (NT_quants[[3]] + 46000) , y = .075,size = 12 ,label = "High", color = "#02875E") +
  annotate("text",x = (NT_quants[[3]] + 46000) , y = .062,size = 12 ,label = "Uptake", color = "#02875E") +
  annotate("text",x = (NT_quants[[3]] + 46000) , y = .050,size = 8.5 ,label = "90th Percentile", color = "#02875E") +coord_cartesian(xlim = c(10^2.5,11^5))
ggsave(paste0(FigOut,"DextranSort_wide_NT_percent.pdf"), width = 8.15, height = 4.2, units = "in")
detach("package:tidyr", unload = TRUE)
detach("package:scales", unload = TRUE)


#%%%%%%%%%%%%%%%%%
# EDF10: Volcano Plot (middle panel)
#%%%%%%%%%%%%%%%%%

require("ggrepel")
Dextran_DIA_NT2 <- Dextran_DIA_NT
Dextran_DIA_NT2$Labs <- ifelse(Dextran_DIA_NT2$direction == "Decreased","High uptake",ifelse(Dextran_DIA_NT2$direction == "Increased","Low uptake","Not significant"))
Dextran_DIA_NT2$GeneLab2 <- ifelse(Dextran_DIA_NT2$median_prot_FC < -3,Dextran_DIA_NT2$Gene.names...primary..,ifelse(Dextran_DIA_NT2$median_prot_FC > 3,Dextran_DIA_NT2$Gene.names...primary..,""))
Dextran_DIA_NT2$GeneLab3 <- ifelse(Dextran_DIA_NT2$Gene.names...primary.. %in% c("Mrc1","Stab1","Snx17"),Dextran_DIA_NT2$Gene.names...primary..,"")
Dextran_DIA_NT2$negLog_q <- -log10(Dextran_DIA_NT2$FC_qval)
Dextran_DIA_NT2$negLog_q_bin <- ifelse(Dextran_DIA_NT2$negLog_q >= 30, 30,Dextran_DIA_NT2$negLog_q)
Dextran_DIA_NT2 %>% ggplot(aes(x = -median_prot_FC, y = negLog_q_bin, color = factor(Labs, levels = c("High uptake","Low uptake","Not significant")))) + geom_point(shape = 16,size = 5,alpha = 0.7) + theme_light() + labs(x = bquote("High / low uptake, log"[2]), y = bquote("-log"[10]~"(q val)"), color = "") + scale_color_manual(values = c("#009E73", "#CC79A7","#888888"))+ theme(axis.text = element_text(size = 24), axis.title = element_text(size = 32), axis.title.y = element_text(margin = margin(t = 0, r = 15, b = 0, l = 0)), legend.text = element_text(size = 28), legend.title = element_blank(), legend.position="null",plot.margin = unit(c(1,0,0,0),"cm")) + geom_text_repel(aes(label = GeneLab2),size = 8, color = "black",min.segment.length = unit(0.33, "lines")) + coord_cartesian(xlim = c(-6.5,6.5), ylim = c(0,33))
ggsave(paste0(FigOut,"NT_DexVolcano_wide_v5_SelectLabs2.pdf"), width =8, height = 4, units = "in")



#%%%%%%%%%%%%%
# Untreated PCA, color-coded by endocytosis-annotated proteins
#%%%%%%%%%%%%%
#Untreated cells (unimputed)
CG_NT <- CG %>% filter(celltype != "LPS", cvm < .4)
prePCA_NT <- prePCA %>% select(CG_NT$id) 
prePCA_NT_norm <- cr_norm_log(as.matrix(prePCA_NT))
prePCA_NT_norm.m <- melt(prePCA_NT_norm)
PCAout_NT <- prcomp(t(prePCA_NT_norm))
PCAout_NT_coords <- PCAout_NT$x
PCAout_NT_weights <- as.data.frame(PCAout_NT$rotation[,1:5])
PCAout_NT_weights$Proteins <- row.names(PCAout_NT_weights)

BC_imp_NT_PxC3.m <- prePCA_NT_norm.m %>% left_join(ProtMissing.m, by = c("Var1"="Proteins", "Var2"="variable"))
BC_imp_NT_PxC3.m$newVar <- ifelse(is.na(BC_imp_NT_PxC3.m$value.y),NA,BC_imp_NT_PxC3.m$value.x)
BC_unimp_NT_PxC3.m <- BC_imp_NT_PxC3.m %>% select(Var1,Var2,newVar)

BC_unimp_NT_PxC3.m_ProEndo <- BC_unimp_NT_PxC3.m %>% 
    dplyr::filter(Var1 %in% Dextran_DIA_NT_filt_Endocytic_lim$PG.ProteinAccessions) %>%
    dplyr::group_by(Var2) %>% dplyr::summarize(ProEndocytic = median(newVar,na.rm = TRUE))
BC_unimp_NT_PxC3.m_AntiEndo <- BC_unimp_NT_PxC3.m %>% 
    dplyr::filter(Var1 %in% Dextran_DIA_NT_filt_AntiEndocytic_lim$PG.ProteinAccessions) %>%
    dplyr::group_by(Var2) %>% dplyr::summarize(AntiEndocytic = median(newVar,na.rm = TRUE))

#z-score abundances
BC_unimp_NT_PxC3.m_ProEndo$ProEndocytic <- (BC_unimp_NT_PxC3.m_ProEndo$ProEndocytic - mean(BC_unimp_NT_PxC3.m_ProEndo$ProEndocytic, na.rm = TRUE))/sd(BC_unimp_NT_PxC3.m_ProEndo$ProEndocytic, na.rm = TRUE)
BC_unimp_NT_PxC3.m_AntiEndo$AntiEndocytic <- (BC_unimp_NT_PxC3.m_AntiEndo$AntiEndocytic - mean(BC_unimp_NT_PxC3.m_AntiEndo$AntiEndocytic, na.rm = TRUE))/sd(BC_unimp_NT_PxC3.m_AntiEndo$AntiEndocytic, na.rm = TRUE)

BC_unimp_NT_PxC3.m_ProEndo$ProEndocytic <- ifelse(BC_unimp_NT_PxC3.m_ProEndo$ProEndocytic >= 2, 2, ifelse(BC_unimp_NT_PxC3.m_ProEndo$ProEndocytic <= -2,-2,BC_unimp_NT_PxC3.m_ProEndo$ProEndocytic))
BC_unimp_NT_PxC3.m_AntiEndo$AntiEndocytic <- ifelse(BC_unimp_NT_PxC3.m_AntiEndo$AntiEndocytic >= 2, 2,ifelse(BC_unimp_NT_PxC3.m_AntiEndo$AntiEndocytic <= -2,-2,BC_unimp_NT_PxC3.m_AntiEndo$AntiEndocytic))

PCAout_NT_coords.df <- as.data.frame(PCAout_NT_coords[,1:5])
PCAout_NT_coords.df$id <- row.names(PCAout_NT_coords.df)

PCAout_NT_coords.df_endo <- PCAout_NT_coords.df
PCAout_NT_coords.df_endo <- PCAout_NT_coords.df_endo %>% left_join(BC_unimp_NT_PxC3.m_ProEndo, by = c("id"="Var2"))
PCAout_NT_coords.df_endo <- PCAout_NT_coords.df_endo %>% left_join(BC_unimp_NT_PxC3.m_AntiEndo, by = c("id"="Var2"))

NT_Anti_scatter <- PCAout_NT_coords.df_endo %>% ggplot(aes(x = PC1,y = PC4, color = AntiEndocytic)) + geom_point(shape = 16, size = 5, alpha = 0.75) + scale_color_gradient2(low = "blue",mid = "white", high = "red", midpoint = 0) + theme_light() + labs(x = "PC1", y = "PC4")  + theme(plot.title = element_text(size = 32, color = "#D1509D"), legend.position = "null", axis.text = element_blank(), axis.title = element_text(size = 32)) + ggtitle("Low Uptake\nProteins")

NT_Pro_scatter <- PCAout_NT_coords.df_endo %>% ggplot(aes(x = PC1,y = PC4, color = ProEndocytic)) + geom_point(shape = 16, size = 5, alpha = 0.75) + scale_color_gradient2(low = "blue",mid = "white", high = "red", midpoint = 0) + theme_light() + labs(x = "PC1", y = "PC4")  + theme(plot.title = element_text(size = 32, color = "#02875E"), legend.position = "null", axis.text = element_blank(), axis.title = element_text(size = 32)) + ggtitle("High Uptake\nProteins")

NT_Anti_scatter + plot_spacer() + NT_Pro_scatter + plot_layout(widths = c(5,0.75,5))
ggsave(paste0(FigOut,"EndoPCA_NT_v3_forSCoPE_BluetoRed_wide_colorCoded.pdf"), width = 9, height = 5, units = "in")

```


```{r Supp. Figure : Precursor Abundance by tier }
AllExps <- read.delim(paste0(datIn,"Fig2/Fig2_bcde_Consistency/ev_Rebuttal_consistency_filtered_PIF50_mrri10_1pFDR.txt"))

InclusionListSupp <- read.delim(paste0(datIn,"Fig2/Fig2_bcde_Consistency/pSCoPE_FinalTarg_Consistency.txt"))

ShotgunSet <- c("XwGH0643","XwGH0644","XwGH0645","XwGH0646","XwGH0647","XwGH0648","XwGH0649","XwGH0650")
pSCoPESet <- c("XwGH0660","XwGH0666","XwGH0663","XwGH0668","XwGH0665","XwGH0669","XwGH0661","XwGH0667")

evi.consist <- AllExps %>% dplyr::filter(Raw.file %in% ShotgunSet) %>% mutate(SeqCharge = paste0(Modified.sequence,Charge))

evi.consist_lim <- AllExps %>% dplyr::filter(Raw.file %in% pSCoPESet) %>% mutate(SeqCharge = paste0(Modified.sequence,Charge))

ShotInt <- evi.consist
PriInt <- evi.consist_lim

ShotInt$Platform <- "Shotgun"
PriInt$Platform <- "Prioritized"

InclusionListSupp.f <- InclusionListSupp %>% filter(TargBoolean == TRUE) %>% select(SeqCharge, Priority)

PriInt <- PriInt %>% left_join(InclusionListSupp.f, by = "SeqCharge")
ShotInt$Priority = 1
IntComb <- rbind(ShotInt,PriInt)
IntComb <- IntComb %>% filter(!is.na(Priority))

IntComb.lim.sum.plat <- IntComb %>% group_by(Platform, Priority) %>% summarize(medInt = median((Intensity), na.rm = TRUE))

IntComb.lim.sum.raw <- IntComb %>% group_by(Platform,Priority,Raw.file) %>% summarize(Count = n())

IntComb.lim.sum.raw.plat <- IntComb.lim.sum.raw %>% ungroup() %>% group_by(Platform,Priority) %>% summarize(medCount = median(Count, na.rm = TRUE))


IntComb.lim.2 <- IntComb
IntComb.lim.2 <- IntComb.lim.2 %>% left_join(IntComb.lim.sum.plat, by = c("Platform","Priority"))
IntComb.lim.2 <- IntComb.lim.2 %>% left_join(IntComb.lim.sum.raw.plat, by = c("Platform","Priority"))

for (i in 0:3){
  PriInt.singleLevel <- IntComb.lim.2 %>% filter(Platform == "Prioritized",Priority == i)
  ShotInt.singleLevel <- IntComb.lim.2 %>% filter(Platform == "Shotgun")
  jointSingleLevel <- rbind(PriInt.singleLevel,ShotInt.singleLevel)

  ContingencyTable.single <- as.data.frame(table(cut(log10(jointSingleLevel$Intensity),30),jointSingleLevel$Raw.file))
  ContingencyTable.single$Platform <- ifelse(ContingencyTable.single$Var2 %in% ShotgunSet, "Shotgun", "Prioritized")
  ContingencyTable.single$Start <- stringr::str_match(ContingencyTable.single$Var1, "\\(([^,]+)")[,2]
  ContingencyTable.single$Priority <- i
  ContingencyTable.sum <- ContingencyTable.single %>% group_by(Platform, Start) %>% summarize(meanFreq = mean(Freq, na.rm = TRUE), sdFreq = sd(Freq, na.rm = TRUE)) %>% mutate(Priority = i)
  if(i == 0){
    ContingencyOut <- ContingencyTable.sum
  }
  if(i > 0){
    ContingencyOut <- rbind(ContingencyOut,ContingencyTable.sum)
  }
    
}

#ContingencyTable <- as.data.frame(table(cut(log10(IntComb.lim.2$Intensity),30),IntComb.lim.2$Raw.file))
#ContingencyTable$Platform <- ifelse(ContingencyTable$Var2 %in% ShotgunSet, "Shotgun", "Prioritized")
#ContingencyTable$Start <- stringr::str_match(ContingencyTable$Var1, "\\(([^,]+)")[,2]




#ContingencyTable_sum <- ContingencyTable_sum %>% left_join(IntComb.lim.sum.raw.plat, by = "Platform")
#ContingencyTable_sum <- ContingencyTable_sum %>% left_join(IntComb.lim.sum.plat, by = "Platform")



ContingencyOut$PriorityLevel <- ifelse(ContingencyOut$Priority == 0, "Low",ifelse(ContingencyOut$Priority == 1, "Medium Low",ifelse(ContingencyOut$Priority == 2, "Medium High", "High")))

ContingencyOut %>% filter(Platform != "Shotgun") %>% ggplot(aes(x = as.numeric(Start), y = meanFreq, color = Platform, group = Platform)) +
  geom_line(size = 1) + 
  geom_point(size = 3, shape = 21, fill = "white", color = "grey30") + 
  geom_errorbar(aes(ymin = (meanFreq - sdFreq), ymax = (meanFreq + sdFreq)), color = "grey40") +    
  scale_x_continuous(limits = c(5,9.5), breaks = seq(5,9.5, by = 1)) + 
  facet_wrap(~factor(PriorityLevel, levels = c("High","Medium Low","Medium High","Low")), ncol = 2) + 
  theme_light() + 
  theme(legend.position = "none", strip.text = element_text(size = 18, color = "black"), 
        strip.background = element_rect(fill = "white", color = "grey70"), 
        axis.text = element_text(size =20), 
        axis.title = element_text(size = 22),plot.margin = margin(0,0,0,0.5, "cm"),
        legend.text = element_text(size = 12)) + 
  scale_color_manual(values = c("red","blue")) + 
  labs(x =bquote('Log'[10]~ 'Precursor Intensity'), y = "Precursors", color = "")
ggsave(paste0(FigOut,"PrecursorIntHistogram.pdf"), height = 7, width = 7, units = "in")

#Wide version
ContingencyOut %>% filter(Platform != "Shotgun") %>% ggplot(aes(x = as.numeric(Start), y = meanFreq, color = Platform, group = Platform)) +
  geom_line(size = 1) + 
  geom_point(size = 3, shape = 21, fill = "white", color = "grey30") + 
  geom_errorbar(aes(ymin = (meanFreq - sdFreq), ymax = (meanFreq + sdFreq)), color = "grey40") +    
  scale_x_continuous(limits = c(5,9.5), breaks = seq(5,9.5, by = 1)) + 
  facet_wrap(~factor(PriorityLevel, levels = c("High","Medium High","Medium Low","Low")), nrow = 1) + 
  theme_light() + 
  theme(legend.position = "none", strip.text = element_text(size = 18, color = "black"), 
        strip.background = element_rect(fill = "white", color = "grey70"), 
        axis.text = element_text(size =20), 
        axis.text.x = element_text(angle = 45, hjust = 1),
        axis.title = element_text(size = 22),plot.margin = margin(0,0.5,0,0.5, "cm"),
        legend.text = element_text(size = 12),
        panel.spacing = unit(1, "lines")) + 
  scale_color_manual(values = c("red","blue")) + 
  labs(x =bquote('Log'[10]~ 'Precursor Intensity'), y = "Precursors", color = "") +
  coord_flip()
ggsave(paste0(FigOut,"PrecursorIntHistogram_side.pdf"), height = 6, width = 8, units = "in")

```
