---
title: "PDAC_BMDM_technical_figure_generation_script"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r libraries}
library(dplyr)
library(reshape2)
library(ggplot2)
library(stringr)
library(patchwork)
```

```{r functions}
Empirical_FDR_PEP <- function(ev){
  exps <- levels(factor(ev$Raw.file))
  ev$Score_naOmit <- ifelse(is.na(ev$PEP),100,ev$PEP)
  for(i in 1:length(exps)){
    
    ev_single <- ev %>% filter(Raw.file == levels(factor(ev$Raw.file))[i])
    ev_rank <- ev_single %>% arrange(Score_naOmit) 
    ev_rank$Ind <- seq(1:nrow(ev_rank)) 
    ev_rank$revCount <- ifelse(ev_rank$Reverse=="+",1,0)
    ev_rank <- ev_rank %>% mutate(RevHits = cumsum(revCount),qVal = RevHits/Ind)
                       
    if(i== 1){
      ev_rank_out <- ev_rank
    }
    if(i > 1){
      ev_rank_out <- rbind(ev_rank_out,ev_rank)
    }
  }
  return(ev_rank_out)
}

Empirical_FP_FDR_PEP <- function(ev){
  exps <- levels(factor(ev$Raw.file))
  ev$Score_naOmit <- ifelse(is.na(ev$PEP),100,ev$PEP)
  for(i in 1:length(exps)){
    
    ev_single <- ev %>% filter(Raw.file == levels(factor(ev$Raw.file))[i])
    ev_rank <- ev_single %>% arrange(Score_naOmit) 
    ev_rank$Ind <- seq(1:nrow(ev_rank)) 
    ev_rank$revCount <- ifelse(ev_rank$Reverse=="+",1,0)
    ev_rank <- ev_rank %>% mutate(RevHits = cumsum(revCount),qVal = RevHits/Ind)
                       
    if(i== 1){
      ev_rank_out <- ev_rank
    }
    if(i > 1){
      ev_rank_out <- rbind(ev_rank_out,ev_rank)
    }
  }
  return(ev_rank_out)
}



Empirical_FDR_msms <- function(ev){
  exps <- levels(factor(ev$Raw.file))
  ev <- ev %>% dplyr::filter(Length != 0)
  ev$Score_naOmit <- ifelse(is.na(ev$PEP),1,ev$PEP)
  for(i in 1:length(exps)){
    
    ev_single <- ev %>% filter(Raw.file == levels(factor(ev$Raw.file))[i])
    ev_rank <- ev_single %>% arrange(Score_naOmit) 
    ev_rank$Ind <- seq(1:nrow(ev_rank)) 
    ev_rank$revCount <- ifelse(ev_rank$Reverse=="+",1,0)
    ev_rank <- ev_rank %>% mutate(RevHits = cumsum(revCount),qVal = RevHits/Ind)
                       
    if(i== 1){
      ev_rank_out <- ev_rank
    }
    if(i > 1){
      ev_rank_out <- rbind(ev_rank_out,ev_rank)
    }
  }
  return(ev_rank_out)
}

#########
# MQ.live log parsing functions
#########

getMS1 <- function(log){
  MS1 <- log %>% filter(grepl("PEPTIDE_FOUND_MS1",Message))
  
  MS1$id <- sub(".*id: *(.*?) *DetectedInMs1:.*", "\\1", MS1$Message)
  MS1$DetectedInMs1 <- sub(".*DetectedInMs1: *(.*?) *DetectedInMs2:.*", "\\1", MS1$Message)
  MS1$DetectedInMs2 <- sub(".*DetectedInMs2: *(.*?) *Disabled:.*", "\\1", MS1$Message)
  MS1$Disabled <- sub(".*Disabled: *(.*?) *mz:.*", "\\1", MS1$Message)
  MS1$mz <- as.numeric(sub(".*mz: *(.*?) *mz_deviation:.*", "\\1", MS1$Message))
  MS1$mz_deviation <- as.numeric(sub(".*mz_deviation: *(.*?) *rt_expected:.*", "\\1", MS1$Message))
  MS1$rt_expected <- as.numeric(sub(".*rt_expected: *(.*?) *rt_deviation:.*", "\\1", MS1$Message))
  MS1$rt_deviation <- as.numeric(sub(".*rt_deviation: *(.*?) *intensity:.*", "\\1", MS1$Message))
  MS1$intensity <- as.numeric(sub(".*intensity: *(.*?) *int_ratio:.*", "\\1", MS1$Message))
  MS1$int_ratio <- as.numeric(sub(".*int_ratio: *(.*?) *int_max_ratio:.*", "\\1", MS1$Message))
  MS1$int_max_ratio <- as.numeric(sub(".*int_max_ratio: *(.*?) *mod_seq:.*", "\\1", MS1$Message))
  MS1$mod_seq <- (sub(".*mod_seq: *(.*?) *m:.*", "\\1", MS1$Message))
  MS1$m <- as.numeric(sub(".*m: *(.*?) *z:.*", "\\1", MS1$Message))
  MS1$z <- as.numeric(sub(".*z: *(.*?) *ApexMzDiff:.*", "\\1", MS1$Message))
  MS1$ApexMzDiff <- as.numeric(sub(".*ApexMzDiff: *(.*?) *ApexRtDiff:.*", "\\1", MS1$Message))
  MS1$ApexRtDiff <- as.numeric(sub(".*ApexRtDiff: *(.*?) *ApexRt:.*", "\\1", MS1$Message))
  MS1$ApexRt <- as.numeric(sub(".*ApexRt: *(.*?) *ApexIntRatio:.*", "\\1", MS1$Message))
  MS1$ApexIntRatio <- as.numeric(sub(".*ApexIntRatio: *(.*?) *mz_correction:.*", "\\1", MS1$Message))
  MS1$mz_correction <- as.numeric(sub(".*mz_correction: *(.*?) *mz_tol:.*", "\\1", MS1$Message))
  MS1$mz_tol <- as.numeric(sub(".*mz_tol: *(.*?) *rt_correction:.*", "\\1", MS1$Message))
  MS1$rt_correction <- as.numeric(sub(".*rt_correction: *(.*?) *rt_tol:.*", "\\1", MS1$Message))
  MS1$rt_window <- as.numeric(sub(".*rt_tol: *(.*?) *int_correction:.*", "\\1", MS1$Message))
  MS1$int_correction <- as.numeric(sub(".*int_correction: *(.*?)", "\\1", MS1$Message))

  return(MS1)
  }

# Get MS2
getMS2 <- function(log){
  MS2 <- log %>% filter(grepl("Found_fragment_ions",Message))
  MS2$id <- gsub("Found_fragment_ions_in_scan 2","",MS2$Message)
  MS2$id  <- as.numeric(str_remove(MS2$id, "^0+"))
  return(MS2)
}

# Generating correction plots
getCorrection <- function(log){
  log_corrections <- log %>% filter(grepl("UPDATED_CORRECTION", Message)) %>% dplyr::select(Date, Time, Scan_protocol, RT, Message)
  log_corrections$mz_correction <- as.numeric(sub(".*mz_correction: *(.*?) *mz_window:.*", "\\1", log_corrections$Message))
  log_corrections$mz_window <- as.numeric(sub(".*mz_window: *(.*?) *rt_correction:.*", "\\1", log_corrections$Message))
  log_corrections$rt_correction <- as.numeric(sub(".*rt_correction: *(.*?) *rt_window:.*", "\\1", log_corrections$Message))
  log_corrections$rt_window <- as.numeric(sub(".*rt_window: *(.*?) *int_correction:.*", "\\1", log_corrections$Message))
  log_corrections$int_correction <- as.numeric(sub(".*int_correction: *(.*?) *num_of_corr_peptides:.*", "\\1", log_corrections$Message))
  log_corrections$num_of_corr_peptides <- as.numeric(sub(".*num_of_corr_peptides: *(.*?) *used_corr_peptides:.*", "\\1", log_corrections$Message))
  log_corrections$used_corr_peptides <- as.numeric(sub(".*used_corr_peptides: *(.*?) *first_corr_peptide:.*", "\\1", log_corrections$Message))
  log_corrections$first_corr_peptide <- as.numeric(sub(".*first_corr_peptide: *(.*?) *last_corr_peptide:.*", "\\1", log_corrections$Message))
  log_corrections$last_corr_peptide <- as.numeric(sub(".*last_corr_peptide: *(.*?) *only_ms2_detected:.*", "\\1", log_corrections$Message))
  log_corrections$only_ms2_detected <- (sub(".*only_ms2_detected: *(.*?)", "\\1", log_corrections$Message))
  return(log_corrections)
}

PCA_PSEA_PDAC <- function(PCA_weights, GeneProteinMappings, GeneSetDB){
GeneSetDB <- GeneSetDB %>% filter(!is.na(GO_term_name))
GO_terms <- unique(GeneSetDB$GO_term_name)

WRS_test_out <- data.frame(Process = character(),numDBentries = numeric(),numberMatchingEntries = numeric(),fracDB = numeric(),pVal = numeric(),EffectSize = numeric(),PC = numeric(),Association = numeric(), stringsAsFactors = FALSE)
WRS_test_out_full <- data.frame(Process = character(),numDBentries = numeric(),numberMatchingEntries = character(),fracDB = numeric(),pVal = numeric(),EffectSize = numeric(),PC = numeric(),Association = numeric(), stringsAsFactors = FALSE)

ReqNumEntry <- 3
fracDB <- .1
maxNum <- 100

GO_in <- PCA_weights %>% left_join(GeneProteinMappings, by = c("Proteins" = "UNIPROTKB"))

for(i in 1:(ncol(GO_in)-2)){
  GeneInd <- which(colnames(GO_in) == "GENES")
  GO_in_lim <- GO_in[,c(i,GeneInd)]
  
for(j in 1:(length(GO_terms))){

    GO_term <- GeneSetDB %>% dplyr::filter(GO_term_name == GO_terms[[j]])
    GO_term <- GO_term %>% group_by(Gene) %>% slice(1)
    
    GO_in_lim$Lab <- ifelse(GO_in_lim$GENES %in% GO_term$Gene,"Target","Background")
    colnames(GO_in_lim) <- c("Loadings","GENES","Lab")
    WRS_test_out[j,1] <- GO_terms[[j]]
    WRS_test_out[j,2] <- length(unique(GO_term$Gene))
    WRS_test_out[j,3] <- sum(GO_in_lim$Lab == "Target")
    WRS_test_out[j,4] <- WRS_test_out[j,3]/WRS_test_out[j,2]
    WRS_test_out[j,7] <- i
 
    
    if((sum(GO_in_lim$Lab == "Target") >= ReqNumEntry)&(WRS_test_out[j,4] >= fracDB)&(WRS_test_out[j,2] <= maxNum)){
      GO_in_lim2 <- GO_in_lim
      GO_in_lim2$zScore <- (GO_in_lim2$Loadings-mean(GO_in_lim2$Loadings, na.rm=TRUE))/sd(GO_in_lim2$Loadings, na.rm=TRUE)
      
      TargGroup <- GO_in_lim2 %>% dplyr::filter(Lab == "Target")
      BackgroundGroup <- GO_in_lim2 %>% dplyr::filter(Lab != "Target")
      
      WRSTest <- wilcox.test(TargGroup$Loadings, BackgroundGroup$Loadings, alternative = "two.sided", paired = FALSE, exact = FALSE)
      WRS_test_out[j,5] <- WRSTest$p.value
      WRS_test_out[j,6] <- median(GO_in_lim2[GO_in_lim2$Lab == "Target",4], na.rm = TRUE)/median(GO_in_lim2[GO_in_lim2$Lab != "Target",4], na.rm = TRUE)
      WRS_test_out[j,8] <- median(GO_in_lim2[GO_in_lim2$Lab == "Target",4], na.rm = TRUE)

    }
    if((sum(GO_in_lim$Lab == "Target") < ReqNumEntry)|(WRS_test_out[j,4] < fracDB)|(WRS_test_out[j,2] > maxNum)){
      WRS_test_out[j,5] <- NA
      WRS_test_out[j,6] <- NA
      WRS_test_out[j,8] <- NA
    }
    }
  WRS_test_out_full <- rbind(WRS_test_out_full,WRS_test_out)
}
return(WRS_test_out_full)
}




PCA_PSEA_PDAC_prot <- function(PCA_weights, GeneSetDB){
GeneSetDB <- GeneSetDB %>% filter(!is.na(GO_term_name))
GO_terms <- unique(GeneSetDB$GO_term_name)

WRS_test_out <- data.frame(Process = character(),numDBentries = numeric(),numberMatchingEntries = numeric(),fracDB = numeric(),pVal = numeric(),EffectSize = numeric(),PC = numeric(),Association = numeric(), stringsAsFactors = FALSE)
WRS_test_out_full <- data.frame(Process = character(),numDBentries = numeric(),numberMatchingEntries = character(),fracDB = numeric(),pVal = numeric(),EffectSize = numeric(),PC = numeric(),Association = numeric(), stringsAsFactors = FALSE)

ReqNumEntry <- 15
fracDB <- .1
maxNum <- 200

GO_in <- PCA_weights 

for(i in 1:(ncol(GO_in)-1)){
  GeneInd <- which(colnames(GO_in) == "Proteins")
  GO_in_lim <- GO_in[,c(i,GeneInd)]
  
for(j in 1:(length(GO_terms))){

    GO_term <- GeneSetDB %>% dplyr::filter(GO_term_name == GO_terms[[j]])
    GO_term <- GO_term %>% group_by(Uniprot) %>% slice(1)
    
    GO_in_lim$Lab <- ifelse(GO_in_lim$Proteins %in% GO_term$Uniprot,"Target","Background")
    colnames(GO_in_lim) <- c("Loadings","Proteins","Lab")
    WRS_test_out[j,1] <- GO_terms[[j]]
    WRS_test_out[j,2] <- length(unique(GO_term$Gene))
    WRS_test_out[j,3] <- sum(GO_in_lim$Lab == "Target")
    WRS_test_out[j,4] <- WRS_test_out[j,3]/WRS_test_out[j,2]
    WRS_test_out[j,7] <- i
 
    
    if((sum(GO_in_lim$Lab == "Target") >= ReqNumEntry)&(WRS_test_out[j,4] >= fracDB)&(WRS_test_out[j,2] <= maxNum)){
      GO_in_lim2 <- GO_in_lim
      GO_in_lim2$zScore <- (GO_in_lim2$Loadings-mean(GO_in_lim2$Loadings, na.rm=TRUE))/sd(GO_in_lim2$Loadings, na.rm=TRUE)
      
      TargGroup <- GO_in_lim2 %>% dplyr::filter(Lab == "Target")
      BackgroundGroup <- GO_in_lim2 %>% dplyr::filter(Lab != "Target")
      
      WRSTest <- wilcox.test(TargGroup$Loadings, BackgroundGroup$Loadings, alternative = "two.sided", paired = FALSE, exact = FALSE)
      WRS_test_out[j,5] <- WRSTest$p.value
      WRS_test_out[j,6] <- median(GO_in_lim2[GO_in_lim2$Lab == "Target",4], na.rm = TRUE)/median(GO_in_lim2[GO_in_lim2$Lab != "Target",4], na.rm = TRUE)
      WRS_test_out[j,8] <- median(GO_in_lim2[GO_in_lim2$Lab == "Target",4], na.rm = TRUE)

    }
    if((sum(GO_in_lim$Lab == "Target") < ReqNumEntry)|(WRS_test_out[j,4] < fracDB)|(WRS_test_out[j,2] > maxNum)){
      WRS_test_out[j,5] <- NA
      WRS_test_out[j,6] <- NA
      WRS_test_out[j,8] <- NA
    }
    }
  WRS_test_out_full <- rbind(WRS_test_out_full,WRS_test_out)
}
return(WRS_test_out_full)
}

cr_norm_log<-function(dat){
  
  for(k in 1:ncol(dat)){
    
    dat[,k]<-dat[,k]-median(dat[,k], na.rm = T)
    
    
  }
  
  
  for(k in 1:nrow(dat)){
    
    dat[k,]<-dat[k,]-mean(dat[k,], na.rm = T)
    
  }
  
  return(dat)
}
```

```{r}
theme_box <- function() { 
    theme_light() %+replace% 
        theme(
          axis.text.x = element_blank(), 
          axis.text.y = element_text(size = 20), 
          axis.title.y = element_text(size = 22, angle = 90), 
          plot.title = element_text(size =24, vjust = 2),
          legend.position = "null", 
          legend.text = element_text(size = 18),
          strip.text = element_text(size = 18, face= "bold", color = "black",margin = margin(0.1,0,0.1,0, "cm")),
          strip.background = element_rect( fill = "grey95", color = "grey70")
        )
}

theme_box_MSMS <- function() { 
    theme_light() %+replace% 
        theme(
          axis.text = element_text(size = 20),
          axis.title.y = element_text(size = 22, angle = 90),
          axis.title.x = element_blank(),
          plot.title = element_text(size =24, vjust = 2),
          legend.position = "null", 
          legend.text = element_text(size = 18),
          strip.text = element_text(size = 18, face= "bold", color = "black",margin = margin(0.1,0,0.1,0, "cm")),
          strip.background = element_rect( fill = "grey95", color = "grey70")
        )
}

theme_heatmap <- function() { 
    theme_light() %+replace% 
        theme(
          axis.text = element_blank(), 
          axis.title = element_text(size = 22), 
          legend.position = "null",
          strip.text = element_text(size = 18, face= "bold", color = "black",margin = margin(0.1,0,0.1,0, "cm")), 
          strip.background = element_rect( fill = "grey95", color = "grey70")
        )
}

theme_missingData_tickChart <- function() {
  theme_light() %+replace% 
    theme(
      axis.text.y = element_blank(), 
      legend.position = "null",
      axis.title = element_text(size = 22), 
      axis.text.x = element_text(size = 20), 
      strip.text = element_text(color ="black", face = "bold",size = 18,margin = margin(0.1,0,0.1,0, "cm")), 
      strip.background = element_rect(fill = "grey95", color = "grey70"),
      plot.title = element_text(size = 24,hjust = 0.5, vjust = 2)
    )
}

```

```{r Tier data}
#Fill in with the path to downloaded directory of figure_gen files
DataPath <- ".../Figure_gen_files/"
ShotgunSet <- c("wAL00295","wAL00296","wAL00297","wAL00298","wAL00299","wAL00300")

#Your output paths here
FigOut <- ""
datOut <- ""

# Missing data at single-cell level

PDAC_t0_both <- read.delim(paste0(DataPath,"PDAC_technical_Fig_1bcd/Fig1_cd_Consistency/t0_PDAC_consistency_PIF50_mrri10.txt"))

PDAC_MP_both <- read.delim(paste0(DataPath,"PDAC_technical_Fig_1bcd/Fig1_cd_Consistency/Prot_missing_PDAC_consistency_PIF50_mrri10.txt"))

PDAC_CG_both <- read.delim(paste0(DataPath,"PDAC_technical_Fig_1bcd/Fig1_cd_Consistency/ChanKey_PDAC_consistency_PIF50_mrri10.txt"))

AllExps <- read.delim(paste0(DataPath,"PDAC_technical_Fig_1bcd/Fig1_cd_Consistency/ev_PDAC_consistency_filtered_PIF50_mrri10_1pFDR.txt"))

AllExps$SeqCharge <- paste0(AllExps$Modified.sequence,AllExps$Charge)
AllExps$Raw.file <- str_sub(AllExps$Raw.file, start = 2)

# Full Evidence files for each experiment
evi.tier.s <- read.delim(paste0(DataPath,"PDAC_technical_Fig_1bcd/Fig1_cd_Consistency/Consistency_evidence_Shotgun.txt"))
evi.tier.p <- read.delim(paste0(DataPath,"PDAC_technical_Fig_1bcd/Fig1_cd_Consistency/Consistency_evidence_pSCoPE.txt"))


evi.tier <- rbind(evi.tier.s,evi.tier.p)
evi.tier.pep.prot <- evi.tier %>% dplyr::select(Modified.sequence,Leading.razor.protein) %>% group_by(Modified.sequence) %>% slice(1)
  
RefList <- read.csv(paste0(DataPath,"PDAC_technical_Fig_1bcd/Fig1_cd_Consistency/TierAssignmentOfDDAPeps_PDAC_110521.csv"))

RefList_Top <- RefList %>% dplyr::filter(Tier == 3) %>% dplyr::select(SeqCharge)
RefList_Mid <- RefList %>% dplyr::filter(Tier == 2) %>% dplyr::select(SeqCharge)
RefList_Both <- RefList %>% dplyr::filter(Tier %in% c(2,3)) %>% dplyr::select(SeqCharge)
RefList_all <- RefList %>% dplyr::filter(TargBool = TRUE) %>% dplyr::select(SeqCharge)

TargList <- read.delim(paste0(DataPath,"PDAC_technical_Fig_1bcd/Fig1_cd_Consistency/PDAC_1x_Uniform_1105_3.txt"))

# Fix Spectronaut formatting in inclusion list
TargList$SeqCharge <- gsub("\\[Oxidation \\(M\\)\\]","\\(Oxidation \\(M\\)\\)",TargList$SeqCharge)
TargList$SeqCharge <- gsub("\\[Acetyl \\(Protein N-term\\)\\]","\\(Acetyl \\(Protein N-term\\)\\)",TargList$SeqCharge)

TargList_Top <- TargList %>% dplyr::filter(Priority %in% c(3)) %>% dplyr::select(SeqCharge)
TargList_Mid <- TargList %>% dplyr::filter(Priority %in% c(2)) %>% dplyr::select(SeqCharge)
TargList_Both <- TargList %>% dplyr::filter(Priority %in% c(2,3)) %>% dplyr::select(SeqCharge)
TargList_All <- TargList %>% dplyr::filter(TargBool = TRUE) %>% select(SeqCharge)

TargList_Both$pep <- str_sub(TargList_Both$SeqCharge, end = -2)
TargList_Both.pep <- as.data.frame(unique(TargList_Both$pep))
colnames(TargList_Both.pep) <- "Modified.sequence"

TargList_All$pep <- str_sub(TargList_All$SeqCharge, end = -2)
TargList_All.pep <- as.data.frame(unique(TargList_All$pep))
colnames(TargList_All.pep) <- "Modified.sequence"
```

#-----
# Consistency Figures
#-----

```{r matrix}

PDAC_CG_both.f <- PDAC_CG_both %>% dplyr::filter(celltype != "control", cvm < .4)
PDAC_CG_both.f$Raw.file <- str_sub(PDAC_CG_both.f$Raw.file, start = 2)
PDAC_CG_s.f <- PDAC_CG_both.f %>% dplyr::filter(Raw.file %in% ShotgunSet)
PDAC_CG_p.f <- PDAC_CG_both.f %>% dplyr::filter(!(Raw.file %in% ShotgunSet))

PDAC_Consistency_cellOrder <- PDAC_CG_both.f$id
PDAC_Consistency_cellOrder.p <- PDAC_CG_p.f$id

PDAC_t0_both.m <- melt(PDAC_t0_both)
PDAC_t0_both.m.f <- PDAC_t0_both.m %>% dplyr::filter(variable %in% PDAC_CG_both.f$id, SeqCharge %in% TargList_Both$SeqCharge)
PDAC_t0_both.m.f$pep <- str_sub(PDAC_t0_both.m.f$SeqCharge, end = -2) 
PDAC_t0_both.m.f.pep <- PDAC_t0_both.m.f %>% dplyr::group_by(variable,pep) %>% dplyr::summarize(ValCollected = sum(value,na.rm = TRUE))
PDAC_t0_both.m.f.pep$ValCollected_na <- ifelse(PDAC_t0_both.m.f.pep$ValCollected == 0, NA,PDAC_t0_both.m.f.pep$ValCollected)
PDAC_t0_both.m.f.pep$Method <- ifelse(PDAC_t0_both.m.f.pep$variable %in% PDAC_CG_s.f$id, "Shotgun","Prioritized")


TargList_Both.pep_in <- as.data.frame(TargList_Both.pep)
colnames(TargList_Both.pep_in) <- "pep"

for(i in 1:length(unique(PDAC_t0_both.m.f.pep$variable))){
  SingleCell <- PDAC_t0_both.m.f.pep %>% ungroup() %>% dplyr::filter(variable == unique(PDAC_t0_both.m.f.pep$variable)[i], pep %in% TargList_Both.pep$Modified.sequence) %>% dplyr::select(pep,ValCollected_na)
  colnames(SingleCell) <- c("pep",as.character(unique(PDAC_t0_both.m.f.pep$variable)[[i]]))
  #SingleExp_2 <- SingleExp %>% group_by(SeqCharge) %>% slice(1)
  TargList_Both.pep_in <- TargList_Both.pep_in %>% left_join(SingleCell, by = "pep")
}

TargList_Both.pep_in.m <- melt(TargList_Both.pep_in)
TargList_Both.pep_in.m$Valbin <- ifelse(is.na(TargList_Both.pep_in.m$value), 0, 1)

TargList_Both.m.pep_sums <- TargList_Both.pep_in.m %>% dplyr::group_by(pep) %>% dplyr::summarize(numCells = sum(Valbin, na.rm = TRUE)) %>% arrange(-numCells) %>% filter(numCells != 0)


TargList_Both.pep_in.m$Method <- ifelse(TargList_Both.pep_in.m$variable %in% PDAC_CG_s.f$id,"Shotgun","Prioritized") 


RefList_Both.m3 <- TargList_Both.pep_in.m %>% left_join(TargList_Both.m.pep_sums, by = "pep")
RefList_Both.m3 <- RefList_Both.m3 %>% filter(!is.na(numCells))
Both_Tier_fig <- RefList_Both.m3 %>% ggplot(aes(x = factor(variable, levels = PDAC_Consistency_cellOrder), y = reorder(pep,-numCells), fill = factor(Valbin))) + geom_tile()  + facet_wrap(~factor(Method,levels = c("Shotgun","Prioritized"))) + theme(axis.text.y = element_blank(), axis.text.x = element_blank(),legend.position = "null", strip.text = element_text(size = 16),axis.text = element_text(size = 14), axis.title = element_text(size = 16), title = element_text(size =12)) + scale_fill_manual(values = c("white","red")) + labs(x = "Cells", y = "Peptides")


####

ConsistencyShotgunCellOrder <- c("i5","i6","i7","i8","i9","i10","i11","i12","i13","i14","i15","i21","i22","i23","i24","i25","i26","i27","i28","i29","i30","i37","i38","i39","i40","i41","i42","i43","i45","i46","i48","i53","i54","i55","i56","i57","i58","i60","i61","i63","i64","i69","i70","i71","i72","i73","i74","i75","i76","i78","i79","i80","i85","i86","i87","i88","i89","i90","i91","i93","i94","i95","i96")
#integrating new data
ConsistencyPrioritizedCellOrder <- PDAC_Consistency_cellOrder.p

CellOrder <- c(ConsistencyShotgunCellOrder,ConsistencyPrioritizedCellOrder)
CellOrder.df <- as.data.frame(CellOrder)
indVec <- c(c(1:63),c(1:62))
#indVec <- ConsistencyShotgunCellOrder
CellOrder.df$Inds <- indVec


RefList_Both.m3$ValBin2 <- ifelse((RefList_Both.m3$Method == "Shotgun" & RefList_Both.m3$Valbin == 1),2,RefList_Both.m3$Valbin)
RefList_Both.m4 <- RefList_Both.m3 %>% left_join(CellOrder.df, by = c("variable"="CellOrder"))

cellHeatMap2 <- RefList_Both.m4 %>% ggplot(aes(x = Inds, y = reorder(pep,-numCells), fill = factor(ValBin2))) + geom_rect(aes(xmin = -Inf, xmax = Inf, ymin = -Inf, ymax = Inf), fill="white")+ geom_tile() + facet_wrap(~factor(Method,levels = c("Shotgun","Prioritized")), scales = "free_x")  + theme_missingData_tickChart() + theme(plot.title = element_text(vjust = 2), axis.title.y = element_text(margin = margin(t = 0, r = 30, b = 0, l = 0))) + scale_fill_manual(values = c("white","red","blue")) + labs(x = "Single Cells", y = "Peptides") + scale_x_continuous(breaks = seq(10, 63, by = 10)) + ggtitle("Top Tiers")
cellHeatMap2
ggsave(paste0(FigOut,"TierHeatMap_PDAC_patchwork_fig_v4_noBlank.png"), width = 7.2 , height = 5.1, units = "in")


```

```{r Peptide-level missingness}

PDAC_CG_both$Raw.file <- str_sub(PDAC_CG_both$Raw.file, start = 2)
PDAC_CG_s <- PDAC_CG_both %>% filter(Raw.file %in% ShotgunSet, cvm < .4, celltype != "control")
PDAC_CG_p <- PDAC_CG_both %>% filter(!(Raw.file %in% ShotgunSet), cvm < .4, celltype != "control")

#TargList_Both
#TargList_All

PDAC_t0_both.m <- melt(PDAC_t0_both)
PDAC_t0_both.m$pep <- str_sub(PDAC_t0_both.m$SeqCharge, end = -2)

PDAC_t0_both.m_Top <- PDAC_t0_both.m %>% dplyr::filter(SeqCharge %in% TargList_Both$SeqCharge)
PDAC_t0_both.m_All <- PDAC_t0_both.m %>% dplyr::filter(SeqCharge %in% TargList_All$SeqCharge)

PDAC_t0_both.m_Top_sum <- PDAC_t0_both.m_Top %>% dplyr::group_by(variable,pep) %>% dplyr::summarize(valSum = sum(value, na.rm = TRUE))
PDAC_t0_both.m_All_sum <- PDAC_t0_both.m_All %>% dplyr::group_by(variable,pep) %>% dplyr::summarize(valSum = sum(value, na.rm = TRUE))
PDAC_t0_both.m_Top_sum$valSumNa <- ifelse(PDAC_t0_both.m_Top_sum$valSum == 0, NA, PDAC_t0_both.m_Top_sum$valSum)
PDAC_t0_both.m_All_sum$valSumNa <- ifelse(PDAC_t0_both.m_All_sum$valSum == 0, NA, PDAC_t0_both.m_All_sum$valSum)

PDAC_t0_both.m_Top_sum_MD <- PDAC_t0_both.m_Top_sum %>% dplyr::group_by(variable) %>% dplyr::summarize(FractionMD = sum(is.na(valSumNa))/n())
PDAC_t0_both.m_All_sum_MD <- PDAC_t0_both.m_All_sum %>% dplyr::group_by(variable) %>% dplyr::summarize(FractionMD = sum(is.na(valSumNa))/n())

PDAC_t0_both.m_Top_sum_MD$Method <- ifelse(PDAC_t0_both.m_Top_sum_MD$variable %in% PDAC_CG_s.f$id,"Shotgun",ifelse(PDAC_t0_both.m_Top_sum_MD$variable %in% PDAC_CG_p.f$id,"Prioritized","Omit"))
PDAC_t0_both.m_All_sum_MD$Method <- ifelse(PDAC_t0_both.m_All_sum_MD$variable %in% PDAC_CG_s.f$id,"Shotgun",ifelse(PDAC_t0_both.m_All_sum_MD$variable %in% PDAC_CG_p.f$id,"Prioritized","Omit"))

PDAC_t0_both.m_Top_sum_MD$Tier <- "Top Tiers"
PDAC_t0_both.m_All_sum_MD$Tier <- "All Tiers"

PDAC_t0_both.m_join <- rbind(PDAC_t0_both.m_Top_sum_MD,PDAC_t0_both.m_All_sum_MD)
PDAC_t0_both.m_join$percent <- PDAC_t0_both.m_join$FractionMD*100

PDAC_t0_both.m_join$percentComplete <- 100 - PDAC_t0_both.m_join$percent
  
PeptideMD <- PDAC_t0_both.m_join %>% ggplot(aes(percentComplete, color = factor(Method, levels = c("Shotgun","Prioritized")))) + geom_boxplot(alpha = 0.5,lwd=1) + coord_flip(xlim = c(0,75)) + labs(color = "", x = "") + facet_wrap(~factor(Tier, levels = c("Top Tiers","All Tiers"))) + theme_box() + theme(plot.title = element_text(size = 24,hjust = 0.5, vjust = 2) , axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)), strip.text = element_text(size = 20),panel.grid.minor.x = element_blank(),panel.grid.major.x = element_blank(),plot.margin = unit(c(0,0,0,0), "cm"),axis.text = element_text(color = "grey10",hjust = 1))  + scale_color_manual(values = c("blue","red")) + labs(x="% Data Completeness",y = "")  + ggtitle("Peptides") + scale_x_continuous(limits = c(0, 75), breaks = seq(0, 75, by = 15)) + geom_hline(aes(yintercept = 0), color = "grey90")

Fig1d_PeptideConsistency_stats <- PDAC_t0_both.m_join %>% group_by(Method,Tier) %>% summarize(medianVal = median(percent,na.rm = TRUE), medianComplete = median(percentComplete, na.rm = TRUE))


PeptideMD  + plot_annotation(title = 'Single-Cell Peptide Detection Consistency',theme = theme(plot.title = element_text(size = 18)))
ggsave(paste0(FigOut,"PepMD_consistency_PDAC_fig_v4_completeness.pdf"), width = 5 , height = 5, units = "in")
```

``` {r }
PDAC_CG_s <- PDAC_CG_both %>% filter(Raw.file %in% ShotgunSet, cvm < .4, celltype != "control")
PDAC_CG_p <- PDAC_CG_both %>% filter(!(Raw.file %in% ShotgunSet), cvm < .4, celltype != "control")

TargList_Both.pepProt <- TargList_Both.pep %>% left_join(evi.tier.pep.prot, by = "Modified.sequence")
TargList_All.pepProt <- TargList_All.pep %>% left_join(evi.tier.pep.prot, by = "Modified.sequence")

#Left join dataframe of single cells to this with Prot-level info
PDAC_MP_both.m <- melt(PDAC_MP_both)
PDAC_MP_both.m$Method <- ifelse(PDAC_MP_both.m$variable %in% PDAC_CG_s$id, "Shotgun","Prioritization")

PDAC_MP_both.m_Top <- PDAC_MP_both.m %>% dplyr::filter(Proteins %in% TargList_Both.pepProt$Leading.razor.protein)
PDAC_MP_both.m_All <- PDAC_MP_both.m %>% dplyr::filter(Proteins %in% TargList_All.pepProt$Leading.razor.protein)

PDAC_MP_both.m_Top_sum <- PDAC_MP_both.m_Top %>% group_by(variable,Proteins) %>% summarize(ProtQuant = sum(value, na.rm = TRUE))
PDAC_MP_both.m_Top_sum$ProtQuantna <- ifelse(PDAC_MP_both.m_Top_sum$ProtQuant==0,NA,PDAC_MP_both.m_Top_sum$ProtQuant)

PDAC_MP_both.m_All_sum <- PDAC_MP_both.m_All %>% group_by(variable,Proteins) %>% summarize(ProtQuant = sum(value, na.rm = TRUE))
PDAC_MP_both.m_All_sum$ProtQuantna <- ifelse(PDAC_MP_both.m_All_sum$ProtQuant==0,NA,PDAC_MP_both.m_All_sum$ProtQuant)


PDAC_MP_both.m_Top_sum_missing <- PDAC_MP_both.m_Top_sum %>% dplyr::group_by(variable) %>% dplyr::summarize(fracMissing = sum(is.na(ProtQuantna))/n())
PDAC_MP_both.m_Top_sum_missing$Method <- ifelse(PDAC_MP_both.m_Top_sum_missing$variable %in% PDAC_CG_s$id, "Shotgun","Prioritized")
PDAC_MP_both.m_Top_sum_missing$Tiers <- "Top Tiers"

PDAC_MP_both.m_All_sum_missing <- PDAC_MP_both.m_All_sum %>% dplyr::group_by(variable) %>% dplyr::summarize(fracMissing = sum(is.na(ProtQuantna))/n())
PDAC_MP_both.m_All_sum_missing$Method <- ifelse(PDAC_MP_both.m_All_sum_missing$variable %in% PDAC_CG_s$id, "Shotgun","Prioritized")
PDAC_MP_both.m_All_sum_missing$Tiers <- "All Tiers"

PDAC_MP_both.m_combTiers <- rbind(PDAC_MP_both.m_Top_sum_missing,PDAC_MP_both.m_All_sum_missing)
PDAC_MP_both.m_combTiers$percent <- 100*PDAC_MP_both.m_combTiers$fracMissing

PDAC_MP_both.m_combTiers$percentComplete <- 100 - PDAC_MP_both.m_combTiers$percent

ProtMD_Box <- PDAC_MP_both.m_combTiers %>% ggplot(aes(percentComplete, color = factor(Method, levels = c("Shotgun","Prioritized")))) + geom_boxplot(lwd=1) + coord_flip(xlim = c(0, 100)) + theme_light() + labs(color = "", x = "") + theme_box() + facet_wrap(~factor(Tiers, levels = c("Top Tiers","All Tiers")), scales = "free_x")

ProtMD_Box <- PDAC_MP_both.m_combTiers %>% filter(Tiers == "All Tiers") %>% ggplot(aes(percentComplete, color = factor(Method, levels = c("Shotgun","Prioritized")))) + geom_boxplot(lwd=1) + coord_flip(xlim = c(0, 75)) + theme_light() + labs(color = "", x = "", y = "") + theme_box() + theme(axis.text.y = element_blank(), strip.text = element_text(size = 20),panel.grid.minor.x = element_blank(),panel.grid.major.x = element_blank(),plot.margin = unit(c(0,0,0,0), "cm"),plot.title = element_text(size = 24,hjust = 0.5, vjust = 2)) + facet_wrap(~factor(Tiers, levels = c("Top Tiers","All Tiers")), scales = "free_x")+ scale_color_manual(values = c("blue","red")) + ggtitle("Proteins")  + scale_x_continuous(limits = c(0, 75), breaks = seq(0, 75, by = 15)) + geom_hline(aes(yintercept = 0), color = "grey90")
#ggsave("FractionMD.png", width = 7, height = 5, units = "in")


ProtMD_Box  + plot_annotation(title = 'Single-Cell Protein Detection Consistency',theme = theme(plot.title = element_text(size = 18)))
ggsave(paste0(FigOut,"ProtMD_consistency_PDAC_fig_v3.png"), width = 5 , height = 5, units = "in")

Fig1d_ProteinConsistency_stats <- PDAC_MP_both.m_combTiers %>% group_by(Method,Tiers) %>% summarize(medianVal = median(percent,na.rm = TRUE),medianCompleteness = median(percentComplete, na.rm = TRUE))

PeptideMD + ProtMD_Box 
#ggsave("C:/My_Drive/MS/SCoPE/t-SCoPE/code/pSCoPE_pubFigs/Figs/TechnicalFig1/Pep_ProtMD_consistency_PDAC_fig_v3.pdf", width = 7 , height = 5, units = "in")
ggsave(paste0(FigOut,"Pep_ProtMD_consistency_PDAC_fig_v4_completeness.pdf"), width = 7.2 , height = 5.1, units = "in")



## Stats for Pub
#Prot
Fig1d_ProteinConsistency_stats.pSCoPE <- Fig1d_ProteinConsistency_stats %>% ungroup() %>% dplyr::filter(Method == "Prioritized") %>% dplyr::select(Tiers,medianCompleteness)
Fig1d_ProteinConsistency_stats.Shotgun <- Fig1d_ProteinConsistency_stats %>% ungroup() %>% dplyr::filter(Method == "Shotgun") %>% dplyr::select(Tiers,medianCompleteness)
colnames(Fig1d_ProteinConsistency_stats.pSCoPE) <- c("Tiers","pSCoPE")
colnames(Fig1d_ProteinConsistency_stats.Shotgun) <- c("Tiers","Shotgun")

Fig1d_ProteinConsistency.stats.join <- Fig1d_ProteinConsistency_stats.pSCoPE %>% left_join(Fig1d_ProteinConsistency_stats.Shotgun, by = "Tiers")
Fig1d_ProteinConsistency.stats.join$PercentageChange <- ((Fig1d_ProteinConsistency.stats.join$pSCoPE- Fig1d_ProteinConsistency.stats.join$Shotgun)/Fig1d_ProteinConsistency.stats.join$Shotgun)*100

#Pep
Fig1d_PeptideConsistency_stats.pSCoPE <- Fig1d_PeptideConsistency_stats %>% ungroup() %>% dplyr::filter(Method == "Prioritized") %>% dplyr::select(Tier,medianComplete)
Fig1d_PeptideConsistency_stats.Shotgun <- Fig1d_PeptideConsistency_stats %>% ungroup() %>% dplyr::filter(Method == "Shotgun") %>% dplyr::select(Tier,medianComplete)
colnames(Fig1d_PeptideConsistency_stats.pSCoPE) <- c("Tiers","pSCoPE")
colnames(Fig1d_PeptideConsistency_stats.Shotgun) <- c("Tiers","Shotgun")
Fig1d_PeptideConsistency_stats.join <- Fig1d_PeptideConsistency_stats.pSCoPE %>% left_join(Fig1d_PeptideConsistency_stats.Shotgun, by = "Tiers")
Fig1d_PeptideConsistency_stats.join$PercentageChange <- ((Fig1d_PeptideConsistency_stats.join$pSCoPE- Fig1d_PeptideConsistency_stats.join$Shotgun)/Fig1d_PeptideConsistency_stats.join$Shotgun)*100
```

#-----
# PDAC coverage demo analysis
#-----

```{r}

evi.cov <- read.delim(paste0(DataPath,"PDAC_technical_Fig_1bcd/Fig1_b_Coverage/Coverage_evidence_Shotgun.txt"))
evi.cov_lim <- read.delim(paste0(DataPath,"PDAC_technical_Fig_1bcd/Fig1_b_Coverage/Coverage_evidence_pSCoPE.txt"))
msmsS.cov <- read.delim(paste0(DataPath,"PDAC_technical_Fig_1bcd/Fig1_b_Coverage/Coverage_msmsScans_Shotgun.txt"))
msmsS.cov_lim <- read.delim(paste0(DataPath,"PDAC_technical_Fig_1bcd/Fig1_b_Coverage/Coverage_msmsScans_pSCoPE.txt"))
msms.cov <- read.delim(paste0(DataPath,"PDAC_technical_Fig_1bcd/Fig1_b_Coverage/Coverage_msms_Shotgun.txt"))
msms.cov_lim <- read.delim(paste0(DataPath,"PDAC_technical_Fig_1bcd/Fig1_b_Coverage/Coverage_msms_pSCoPE.txt"))

PDAC_cov_ShotgunSet <- c("wAL00091","wAL00092","wAL00093","wAL00094","wAL00095","wAL00096","wAL00097","wAL00098","wAL00099","wAL00100")
PDAC_cov_PrioriSet <- c("wAL00120","wAL00121","wAL00122","wAL00123","wAL00124","wAL00125","wAL00126","wAL00127","wAL00129","wAL00130")

evi.cov <- evi.cov %>% dplyr::filter(Raw.file %in% PDAC_cov_ShotgunSet)
msmsS.cov <- msmsS.cov %>% dplyr::filter(Raw.file %in% PDAC_cov_ShotgunSet)
msms.cov <- msms.cov %>% dplyr::filter(Raw.file %in% PDAC_cov_ShotgunSet)
evi.cov_lim <- evi.cov_lim %>% dplyr::filter(Raw.file %in% PDAC_cov_PrioriSet)
msmsS.cov_lim <- msmsS.cov_lim %>% dplyr::filter(Raw.file %in% PDAC_cov_PrioriSet)
msms.cov_lim <- msms.cov_lim %>% dplyr::filter(Raw.file %in% PDAC_cov_PrioriSet)

evi.cov <- rbind(evi.cov,evi.cov_lim)
msmsS.cov <- rbind(msmsS.cov,msmsS.cov_lim)
msms.cov <- rbind(msms.cov,msms.cov_lim)

# data/sc data frames

CG.cov <- read.delim(paste0(DataPath,"PDAC_technical_Fig_1bcd/Fig1_b_Coverage/ChanKey_PDAC_cov_PIF50_mrri10.txt"))
CG.cov$Raw.file <- str_sub(CG.cov$Raw.file, start = 2)

MDp.cov <- read.delim(paste0(DataPath, "PDAC_technical_Fig_1bcd/Fig1_b_Coverage/Prot_missing_PDAC_cov_PIF50_mrri10.txt")) 
```

```{r}
#Productive Scans defined as non-CON/REV identifications
evi.cov.f <- Empirical_FDR_PEP(evi.cov)
evi.cov_PSMs <- evi.cov.f %>% dplyr::filter((!grepl("CON",Proteins))&(qVal <= .01)&(!grepl("REV",Proteins))&(Potential.contaminant != "+")&(Reverse != "+")) %>% group_by(Raw.file,Modified.sequence) %>% top_n(-1,PEP) %>% slice(1)


msmsS.cov_sum <- msmsS.cov %>% dplyr::group_by(Raw.file) %>% dplyr::summarize(numObs = n()) %>% mutate(PSM.filt = "MS/MS\nScans")
evi.cov_PSMs_sum <- evi.cov_PSMs %>% ungroup() %>% dplyr::group_by(Raw.file) %>% dplyr::summarize(numObs = n()) %>% mutate(PSM.filt = "Peptides")

msmsS.cov_Productive_sum <- msmsS.cov %>% dplyr::group_by(Raw.file) %>% dplyr::filter((!grepl("REV",Proteins))&(!grepl("CON",Proteins))&(Identified == "+")) %>% dplyr::summarize(numObs = n()) %>% mutate(PSM.filt = "Productive MS/MS\nScans")

msmsS.cov_Productive_sum_v2 <- msmsS.cov_Productive_sum %>% left_join(msmsS.cov_sum, by = "Raw.file")
msmsS.cov_Productive_sum_v2$numObs <- msmsS.cov_Productive_sum_v2$numObs.x/msmsS.cov_Productive_sum_v2$numObs.y
msmsS.cov_Productive_sum_v3 <- msmsS.cov_Productive_sum_v2 %>% select(Raw.file,numObs,PSM.filt.x)

msmsS.cov_Productive_sum_v3$Method <- ifelse(msmsS.cov_Productive_sum_v3$Raw.file %in% PDAC_cov_ShotgunSet,"Shotgun",ifelse(msmsS.cov_Productive_sum_v3$Raw.file %in% PDAC_cov_PrioriSet,"Prioritized","Omit"))
msmsS.cov_sum$Method <- ifelse(msmsS.cov_sum$Raw.file %in% PDAC_cov_ShotgunSet,"Shotgun",ifelse(msmsS.cov_sum$Raw.file %in% PDAC_cov_PrioriSet,"Prioritized","Omit"))
evi.cov_PSMs_sum$Method <- ifelse(evi.cov_PSMs_sum$Raw.file %in% PDAC_cov_ShotgunSet,"Shotgun",ifelse(evi.cov_PSMs_sum$Raw.file %in% PDAC_cov_PrioriSet,"Prioritized","Omit"))

msmsS.cov_Productive_sum_v3.f <- msmsS.cov_Productive_sum_v3 %>% dplyr::filter(Method != "Omit") 
msmsS.cov_sum.f <- msmsS.cov_sum %>% dplyr::filter(Method != "Omit") 
evi.cov_PSMs_sum.f <- evi.cov_PSMs_sum %>% dplyr::filter(Method != "Omit")

comb_MS2_stats <- rbind(msmsS.cov_sum.f,evi.cov_PSMs_sum.f)
Cov_PSMs_box <- comb_MS2_stats %>% ggplot(aes(x = PSM.filt, y = numObs,color = factor(Method, levels = c("Shotgun","Prioritized")))) + geom_boxplot(alpha=0.5,lwd=1) + coord_cartesian(ylim = c(0,8200)) + theme_box_MSMS() + labs(y = "Total/experiment") + scale_color_manual(values = c("blue","red")) + theme(axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0)))
#ggsave(paste0(FigOut,"MSMS_and_Evi_Stats.pdf"), width = 7 , height = 5, units = "in")


ProdScans1 <- msmsS.cov_Productive_sum_v3.f %>% ggplot(aes(x = PSM.filt.x, y = numObs*100,color = factor(Method, levels = c("Shotgun","Prioritized")))) + geom_boxplot(alpha=0.5,lwd=1) + coord_cartesian(ylim = c(0,75)) + theme_box_MSMS() + labs(y = "Productive MS2 Scans, %") + scale_color_manual(values = c("blue","red")) + theme(axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)), axis.text.x = element_blank())

PepsPerRun <- evi.cov_PSMs_sum.f %>% ggplot(aes(x = PSM.filt, y = numObs,color = factor(Method, levels = c("Shotgun","Prioritized")))) + geom_boxplot(alpha=0.5,lwd=1) + coord_cartesian(ylim = c(0,5000)) + theme_box_MSMS() + labs(y = "Peptides/run") + scale_color_manual(values = c("blue","red")) + theme(axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0)), axis.text.x = element_blank())
```

```{r}
evi.cov.f <- Empirical_FDR_PEP(evi.cov)
msms.cov.f <- Empirical_FDR_msms(msms.cov) 
evi.cov_PSMs <- evi.cov.f %>% dplyr::filter((!grepl("CON",Proteins))&(qVal <= .01)&(!grepl("REV",Proteins))&(Potential.contaminant != "+")&(Reverse != "+")) %>% group_by(Raw.file,Modified.sequence) %>% top_n(-1,PEP) %>% slice(1)

msms.cov_1FDR <- msms.cov.f %>% dplyr::filter((!grepl("CON",Proteins))&(qVal <= .01)&(!grepl("REV",Proteins)))

msmsS.cov_sum <- msmsS.cov %>% dplyr::group_by(Raw.file) %>% dplyr::summarize(numObs = n()) %>% mutate(PSM.filt = "MS/MS\nScans")
msms.cov_sum <- msms.cov_1FDR %>% dplyr::group_by(Raw.file) %>% dplyr::summarize(numObs = n()) %>% mutate(PSM.filt = "Productive MS/MS\nScans")
evi.cov_PSMs_sum <- evi.cov_PSMs %>% ungroup() %>% dplyr::group_by(Raw.file) %>% dplyr::summarize(numObs = n()) %>% mutate(PSM.filt = "Peptides")

#msms.cov_Productive_sum <- msms.cov %>% dplyr::group_by(Raw.file) %>% dplyr::filter((!grepl("REV",Proteins))&(!grepl("REV",Proteins))&(Identified == "+")) %>% dplyr::summarize(numObs = n()) %>% mutate(PSM.filt = "Productive MS/MS\nScans")

msmsS.cov_Productive_sum_v2 <- msmsS.cov_sum %>% left_join(msms.cov_sum, by = "Raw.file")
msmsS.cov_Productive_sum_v2$numObs <- msmsS.cov_Productive_sum_v2$numObs.y/msmsS.cov_Productive_sum_v2$numObs.x
msmsS.cov_Productive_sum_v3 <- msmsS.cov_Productive_sum_v2 %>% select(Raw.file,numObs,PSM.filt.x)

msmsS.cov_Productive_sum_v3$Method <- ifelse(msmsS.cov_Productive_sum_v3$Raw.file %in% PDAC_cov_ShotgunSet,"Shotgun",ifelse(msmsS.cov_Productive_sum_v3$Raw.file %in% PDAC_cov_PrioriSet,"Prioritized","Omit"))
msmsS.cov_sum$Method <- ifelse(msmsS.cov_sum$Raw.file %in% PDAC_cov_ShotgunSet,"Shotgun",ifelse(msmsS.cov_sum$Raw.file %in% PDAC_cov_PrioriSet,"Prioritized","Omit"))
evi.cov_PSMs_sum$Method <- ifelse(evi.cov_PSMs_sum$Raw.file %in% PDAC_cov_ShotgunSet,"Shotgun",ifelse(evi.cov_PSMs_sum$Raw.file %in% PDAC_cov_PrioriSet,"Prioritized","Omit"))

msmsS.cov_Productive_sum_v3.f <- msmsS.cov_Productive_sum_v3 %>% dplyr::filter(Method != "Omit") 
msmsS.cov_sum.f <- msmsS.cov_sum %>% dplyr::filter(Method != "Omit") 
evi.cov_PSMs_sum.f <- evi.cov_PSMs_sum %>% dplyr::filter(Method != "Omit")

comb_MS2_stats <- rbind(msmsS.cov_sum.f,evi.cov_PSMs_sum.f)
Cov_PSMs_box <- comb_MS2_stats %>% ggplot(aes(x = PSM.filt, y = numObs,color = factor(Method, levels = c("Shotgun","Prioritized")))) + geom_boxplot(alpha=0.5,lwd=1) + coord_cartesian(ylim = c(0,8200)) + theme_box_MSMS() + labs(y = "Total/experiment") + scale_color_manual(values = c("blue","red")) + theme(axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0)))
#ggsave(paste0(FigOut,"MSMS_and_Evi_Stats.pdf"), width = 7 , height = 5, units = "in")

#Without Jitter: Preprint
#ProdScans1 <- msmsS.cov_Productive_sum_v3.f %>% ggplot(aes(x = PSM.filt.x, y = numObs*100,color = factor(Method, levels = c("Shotgun","Prioritized")))) + geom_boxplot(alpha=0.5,lwd=1) + coord_cartesian(ylim = c(0,75)) + theme_box_MSMS() + labs(y = "Productive MS2 Scans, %") + scale_color_manual(values = c("blue","red")) + theme(axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)), axis.text.x = element_blank(),axis.text = element_text(color = "grey10",hjust = 1)) + scale_y_continuous(limits = c(0, 75), breaks = seq(0, 75, by = 15))

#With Jitter: Nature version
ProdScans1 <- msmsS.cov_Productive_sum_v3.f %>% ggplot(aes(x = PSM.filt.x, y = numObs*100,color = factor(Method, levels = c("Shotgun","Prioritized")))) + geom_boxplot(alpha=0.5,lwd=1) + geom_point(position=position_jitterdodge(), size = 2, shape = 21, fill = "grey95") + coord_cartesian(ylim = c(30,75)) + theme_box_MSMS() + labs(y = "Productive MS2 Scans, %") + scale_color_manual(values = c("blue","red")) + theme(axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)), axis.text.x = element_blank(),axis.text = element_text(color = "grey10",hjust = 1))  + scale_y_continuous(limits = c(30, 75), breaks = seq(30, 75, by = 15))

#msmsS.cov_Productive_sum_v3.f_sum <- msmsS.cov_Productive_sum_v3.f %>% group_by(Method) %>% summarize(medianVal = median(numObs))
#(((msmsS.cov_Productive_sum_v3.f_sum[1,2]*100) - (msmsS.cov_Productive_sum_v3.f_sum[2,2]*100))/(msmsS.cov_Productive_sum_v3.f_sum[2,2]*100))*100

#Without Jitter: Preprint
#PepsPerRun <- evi.cov_PSMs_sum.f %>% ggplot(aes(x = PSM.filt, y = numObs,color = factor(Method, levels = c("Shotgun","Prioritized")))) + geom_boxplot(alpha=0.5,lwd=1) + coord_cartesian(ylim = c(0,5000)) + theme_box_MSMS() + labs(y = "Peptides/run") + scale_color_manual(values = c("blue","red")) + theme(axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0)), axis.text.x = element_blank(),axis.text = element_text(color = "grey10",hjust = 1))

#With Jitter: Nature version
PepsPerRun <- evi.cov_PSMs_sum.f %>% ggplot(aes(x = PSM.filt, y = numObs,color = factor(Method, levels = c("Shotgun","Prioritized")))) + geom_boxplot(alpha=0.5,lwd=1)  + geom_point(position=position_jitterdodge(), size = 2, shape = 21, fill = "grey95") + coord_cartesian(ylim = c(2000,5000)) + theme_box_MSMS() + labs(y = "Peptides/run") + scale_color_manual(values = c("blue","red")) + theme(axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0)), axis.text.x = element_blank(),axis.text = element_text(color = "grey10",hjust = 1))

#evi.cov_PSMs_sum.f_sum <- evi.cov_PSMs_sum.f %>% group_by(Method) %>% summarize(medianVal = median(numObs))
#(((evi.cov_PSMs_sum.f_sum[1,2]*100) - (evi.cov_PSMs_sum.f_sum[2,2]*100))/(evi.cov_PSMs_sum.f_sum[2,2]*100))*100

#ProdScans1 + PepsPerRun
```

```{r}
CG.s.cov.f <- CG.cov %>% dplyr::filter(celltype != "control",cvm <.4, Raw.file %in% PDAC_cov_ShotgunSet)
CG.p.cov.f <- CG.cov %>% dplyr::filter(celltype != "control",cvm <.4, Raw.file %in% PDAC_cov_PrioriSet)

MDp.cov.m <- melt(MDp.cov)

MDp.s.cov.m.f <- MDp.cov.m %>% dplyr::filter(variable %in% CG.s.cov.f$id) 
MDp.p.cov.m.f <- MDp.cov.m %>% dplyr::filter(variable %in% CG.p.cov.f$id)

MDp.s.cov.m.f$Method <- "Shotgun" 
MDp.p.cov.m.f$Method <- "Prioritized"

MDp_join <- rbind(MDp.s.cov.m.f,MDp.p.cov.m.f)
MDp_join_sum <- MDp_join %>% group_by(variable, Method) %>% dplyr::summarize(numProts = sum(!is.na(value), na.rm = TRUE))
MDp_join_sum$PSM.filt = "1% FDR"

# Preprint scaling
#Cov_Prot_box <- MDp_join_sum %>% ggplot(aes(numProts, color = factor(Method, levels = c("Shotgun","Prioritized")))) + geom_boxplot(alpha=0.5,lwd=1) + coord_flip(xlim = c(0,1250)) + theme_light() + labs(color = "", x = "Proteins/cell", y = "") + theme_box() + scale_color_manual(values = c("blue","red"))  + theme(strip.text = element_text(size = 20), axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0)),axis.text = element_text(color = "grey10",hjust = 1),panel.grid.minor.x = element_blank(),panel.grid.major.x = element_blank()) + scale_x_continuous(limits = c(0, 1250), breaks = seq(0, 1250, by = 250)) + geom_hline(aes(yintercept = 0), color = "grey90")

# Nature scaling
Cov_Prot_box <- MDp_join_sum %>% ggplot(aes(numProts, color = factor(Method, levels = c("Shotgun","Prioritized")))) + geom_boxplot(alpha=0.5,lwd=1) + coord_flip(xlim = c(500,1250)) + theme_light() + labs(color = "", x = "Proteins/cell", y = "") + theme_box() + scale_color_manual(values = c("blue","red"))  + theme(strip.text = element_text(size = 20), axis.title.y = element_text(margin = margin(t = 0, r = 10, b = 0, l = 0)),axis.text = element_text(color = "grey10",hjust = 1),panel.grid.minor.x = element_blank(),panel.grid.major.x = element_blank()) + scale_x_continuous(limits = c(500, 1250), breaks = seq(500, 1250, by = 250)) + geom_hline(aes(yintercept = 0), color = "grey90")

#MDp_join_sum_sum <- MDp_join_sum %>% group_by(Method) %>% summarize(medianVal = median(numProts))
#(((MDp_join_sum_sum[1,2]*100) - (MDp_join_sum_sum[2,2]*100))/(MDp_join_sum_sum[2,2]*100))*100

ProdScans1 + PepsPerRun + Cov_Prot_box  + plot_layout(widths = c(1,1,1), guides = "collect") & theme(legend.position = 'null')
ggsave(paste0(FigOut,"Coverage_PDAC_rev2_1pFDRMS2.pdf"), width = 7.2 , height = 5.1, units = "in")

ProdScans1 + PepsPerRun + Cov_Prot_box  + plot_layout(widths = c(1,1,1), guides = "collect") & theme(legend.position = 'bottom')
ggsave(paste0(FigOut,"Coverage_PDAC_rev2_1pFDRMS2_forLegend.pdf"), width = 7.2 , height = 5.1, units = "in")
```

#----------------------------
#----------------------------
# Supplemental technical figs
#----------------------------
#----------------------------

#

```{r}
# PDAC coverage targList and path
PDAC_coverage_targList <- paste0(DataPath,"PDAC_technical_Fig_1bcd/Fig1_b_Coverage/PDAC_Scout_2x_DDA_Final.txt")
PDAC_coverage_log_path <- paste0(DataPath,"PDAC_technical_Fig_1bcd/Fig1_b_Coverage/MQL_logs_PDAC_coverage/")

#re-coding priority b/c we had both RT-only and targetable peptides assigned priority 0
tList_instance <- read.delim(PDAC_coverage_targList)
tList_instance$id <- seq(0,nrow(tList_instance)-1,1)
tList_instance$Orig_Priority <- tList_instance$Priority
tList_instance$Priority <- ifelse(tList_instance$TargBool == FALSE, 0, tList_instance$Priority + 1)

tList_instance$id <- seq(0,nrow(tList_instance)-1,1)
logs <- list.files(PDAC_coverage_log_path)
log_filt <- logs[!(logs %in% c("desktop.ini","1016_20210927-1118.txt"))]
numLogs <- length(log_filt)

for(i in 1:numLogs){
  log_instance <- read.delim(paste0(PDAC_coverage_log_path,log_filt[i]), header = FALSE)
  colnames(log_instance) <-  c("Date","Time","Scan_protocol","RT","Message")

  MS1 <- getMS1(log_instance)
  MS1_unique <- unique(MS1$id)
  MS2 <- getMS2(log_instance)
  tList_instance$seen <- ifelse(tList_instance$id %in% MS1_unique,TRUE,FALSE)
  tList_instance$MS2d <- ifelse(tList_instance$id %in% MS2$id,TRUE,FALSE)
  tList_sum <- tList_instance %>% dplyr::group_by(Priority) %>% 
    dplyr::summarize(seenSum = sum(seen),seenFrac = seenSum/n(), MS2Sum = sum(MS2d),MS2Frac = MS2Sum/n())
  #tList_sum <- tList_instance %>% dplyr::group_by(PriorityTop) %>% 
    #dplyr::summarize(seenSum = sum(seen),seenFrac = seenSum/n(), MS2Sum = sum(MS2d),MS2Frac = MS2Sum/n())
  #tList_sum3 <- tList_instance %>% dplyr::group_by(PriorityTop3) %>% 
    #dplyr::summarize(seenSum = sum(seen),seenFrac = seenSum/n(), MS2Sum = sum(MS2d),MS2Frac = MS2Sum/n())
  tList_sum$file_ind <- i
  if(i ==1){
    tList_sum_out <-tList_sum
    #tList_sum_out3 <-tList_sum3
  }
  if(i > 1){
    tList_sum_out <- rbind(tList_sum_out,tList_sum)
    #tList_sum_out3 <- rbind(tList_sum_out3,tList_sum3)
  }
}
```

```{r}
# PDAC consistency targList and path
PDAC_consistency_targList <- paste0(DataPath,"PDAC_technical_Fig_1bcd/Fig1_cd_Consistency/PDAC_1x_Uniform_1105_3.txt")
PDAC_consistency_log_path <- paste0(DataPath,"PDAC_technical_Fig_1bcd/Fig1_cd_Consistency/MQL_logs_PDAC_consistency/")

#re-coding priority b/c we had both RT-only and targetable peptides assigned priority 0
tList_instance.consist <- read.delim(PDAC_consistency_targList)
tList_instance.consist$id <- seq(0,nrow(tList_instance.consist)-1,1)
tList_instance.consist$Orig_Priority <- tList_instance.consist$Priority
tList_instance.consist$Priority <- ifelse(tList_instance.consist$TargBool == FALSE, 0, tList_instance.consist$Priority + 1)
tList_instance.consist$PriorityTop <- ifelse(tList_instance.consist$Priority %in% c(4,3),"Top","Bottom")
tList_instance.consist$PriorityTop3 <- ifelse(tList_instance.consist$Priority %in% c(4,3,2),"Top","Bottom")

tList_instance.consist$id <- seq(0,nrow(tList_instance.consist)-1,1)

log_Match_DF.consist <- data.frame(log = character(), stringsAsFactors = FALSE)

logs.consistency <- list.files(PDAC_consistency_log_path)
log.consistency_filt <- logs.consistency[!(logs.consistency %in% c("desktop.ini","1029_20211105-2240.txt","1029_20211106-0841.txt"))]
numLogs.consistency <- length(log.consistency_filt)

for(i in 1:numLogs.consistency){
  log_instance <- read.delim(paste0(PDAC_consistency_log_path,log.consistency_filt[i]), header = FALSE)
  colnames(log_instance) <-  c("Date","Time","Scan_protocol","RT","Message")

  MS1 <- getMS1(log_instance)
  MS1_unique <- unique(MS1$id)
  MS2 <- getMS2(log_instance)
  tList_instance.consist$seen <- ifelse(tList_instance.consist$id %in% MS1_unique,TRUE,FALSE)
  tList_instance.consist$MS2d <- ifelse(tList_instance.consist$id %in% MS2$id,TRUE,FALSE)
  tList_sum <- tList_instance.consist %>% dplyr::group_by(Priority) %>% 
    dplyr::summarize(seenSum = sum(seen),seenFrac = seenSum/n(), MS2Sum = sum(MS2d),MS2Frac = MS2Sum/n())
  #tList_sum3 <- tList_instance.consist %>% dplyr::group_by(PriorityTop3) %>% 
  # dplyr::summarize(seenSum = sum(seen),seenFrac = seenSum/n(), MS2Sum = sum(MS2d),MS2Frac = MS2Sum/n())
  tList_sum$file_ind <- i
  if(i ==1){
    tList_sum_out.consistency <-tList_sum
  #  tList_sum_out3.consistency <-tList_sum3
  }
  if(i > 1){
    tList_sum_out.consistency <- rbind(tList_sum_out.consistency,tList_sum)
  #  tList_sum_out3.consistency <- rbind(tList_sum_out3.consistency,tList_sum3)
  }
}

```

```{r}
targList_cov_sum <- tList_instance %>% dplyr::group_by(Priority) %>% dplyr::summarize(TargCount = n())
targList_consist_sum <- tList_instance.consist %>% dplyr::group_by(Priority) %>% dplyr::summarize(TargCount = n())
```


```{r}
# BMDM and PDAC coverage
Priority_dict.cov <- data.frame(Priority = numeric(), Title = character(), stringsAsFactors = FALSE)
Priority_dict.cov[1:4,1] <- c(3,2,1,0)
Priority_dict.cov[1:4,2] <- c("High","Medium","Low","RT only")

# PDAC consistency
Priority_dict.consist <- data.frame(Priority = numeric(), Title = character(), stringsAsFactors = FALSE)
Priority_dict.consist[1:5,1] <- c(4,3,2,1,0)
Priority_dict.consist[1:5,2] <- c("High 1","High 2","Medium","Low","RT only")

tList_sum_out.cov.j <- tList_sum_out %>% left_join(Priority_dict.cov, by = "Priority")
tList_sum_out.cov.j.f <- tList_sum_out.cov.j %>% select(seenFrac,MS2Frac,Priority,file_ind,Title)
tList_sum_out.cov.j.f.m <- melt(tList_sum_out.cov.j.f, id.vars = c("file_ind","Priority","Title"))

tList_sum_out.consistency.j <- tList_sum_out.consistency %>% left_join(Priority_dict.consist, by = "Priority")
tList_sum_out.consistency.j.f <- tList_sum_out.consistency.j %>% select(seenFrac,MS2Frac,Priority,file_ind,Title)
tList_sum_out.consistency.j.f.m <- melt(tList_sum_out.consistency.j.f, id.vars = c("file_ind","Priority","Title"))

tList_sum_out.cov.j.f.m <- tList_sum_out.cov.j.f.m %>% left_join(targList_cov_sum, by = "Priority")
tList_sum_out.consist.j.f.m <- tList_sum_out.consistency.j.f.m %>% left_join(targList_consist_sum, by = "Priority")

tList_sum_out.cov.j.f.m %>% ggplot(aes(value, color = variable)) + geom_boxplot(lwd = 1) + 
  ggtitle("Technical Performance: Coverage Experiments") + 
  coord_flip() + 
  theme_light() + 
  theme(axis.text.x =  element_blank(), 
        strip.background = element_rect(color = "grey70", fill = "transparent"), 
        strip.text = element_text(color= "black", size = 18),legend.position = "bottom",
        axis.text.y = element_text(size =14),
        axis.title.y = element_text(size =16),
        plot.title = element_text(size = 18),
        legend.text = element_text(size = 14)) + 
  labs(color = "", x = "Fraction of Prioritized Peptides") + 
  facet_wrap(~factor(Title,levels = c("High","Medium","Low","RT only")), nrow = 1)+
  scale_color_manual(values = c("black","grey60")) + geom_text( x = .1, y = 0,color = "grey10",aes(label = paste0(TargCount,"\n Precursors")))
ggsave(paste0(FigOut,"PDAC_coverage_MQliveStats.png"), width = 6, height = 4, units = "in")

tList_sum_out.consist.j.f.m %>% ggplot(aes(value, color = variable)) + geom_boxplot(lwd = 1) + 
  ggtitle("Technical Performance: Consistency Experiments") + 
  coord_flip() + 
  theme_light() + 
  theme(axis.text.x =  element_blank(), 
        strip.background = element_rect(color = "grey70", fill = "transparent"), 
        strip.text = element_text(color= "black", size = 18),legend.position = "bottom",
        axis.text.y = element_text(size =14),
        axis.title.y = element_text(size =16),
        plot.title = element_text(size = 18),
        legend.text = element_text(size = 14)) + 
  labs(color = "", x = "Fraction of Prioritized Peptides") + 
  facet_wrap(~factor(Title,levels = c("High 1","High 2","Medium","Low","RT only")), nrow = 1)+
  scale_color_manual(values = c("black","grey60")) + geom_text( x = .1, y = 0,color = "grey10",aes(label = paste0(TargCount,"\n Precursors")))
ggsave(paste0(FigOut,"PDAC_consist_MQliveStats.png"), width = 6, height = 4, units = "in")
```


```{r}
#figOut <- "C:/My_Drive/MS/SCoPE/t-SCoPE/code/pSCoPE_pubFigs/Figs/v2_figs/SupplementalFigs/"

Priority_dict_consist <- data.frame(Priority = numeric(), Title = character(), stringsAsFactors = FALSE)
Priority_dict_consist[1:5,1] <- c(4,3,2,1,0)
Priority_dict_consist[1:5,2] <- c("High 1","High 2","Medium","Low","RT only")

#------------------------
# Reading search results
#------------------------
consist_shotgun <- read.delim(paste0(DataPath,"PDAC_technical_Fig_1bcd/Fig1_cd_Consistency/Consistency_msms_Shotgun.txt"))
consist_priori <- read.delim(paste0(DataPath,"PDAC_technical_Fig_1bcd/Fig1_cd_Consistency/Consistency_msms_pSCoPE.txt"))

consist_shotgun$SeqCharge <- paste0(consist_shotgun$Modified.sequence, consist_shotgun$Charge)
consist_priori$SeqCharge <- paste0(consist_priori$Modified.sequence, consist_priori$Charge)

consist_shotgun_evi <- read.delim(paste0(DataPath,"PDAC_technical_Fig_1bcd/Fig1_cd_Consistency/Consistency_evidence_Shotgun.txt"))
consist_priori_evi <- read.delim(paste0(DataPath,"PDAC_technical_Fig_1bcd/Fig1_cd_Consistency/Consistency_evidence_pSCoPE.txt"))

consist_shotgun_evi$SeqCharge <- paste0(consist_shotgun_evi$Modified.sequence, consist_shotgun_evi$Charge)
consist_priori_evi$SeqCharge <- paste0(consist_priori_evi$Modified.sequence, consist_priori_evi$Charge)

#------------------------
# Reading targeting list and reformating tiers for visualizing RT-only peptides separately
#------------------------
targList_instance_consist <- read.delim(paste0(DataPath,"PDAC_technical_Fig_1bcd/Fig1_cd_Consistency/PDAC_1x_Uniform_1105_3.txt"))
targList_instance_consist$id <- seq(0,nrow(targList_instance_consist)-1,1)
targList_instance_consist$Orig_Priority <- targList_instance_consist$Priority
targList_instance_consist$Priority <- ifelse(targList_instance_consist$TargBool == FALSE, 0,targList_instance_consist$Priority + 1)

#------------------------
# Filtering search results
#------------------------

ShotgunSet <- c("wAL00295","wAL00296","wAL00297","wAL00298","wAL00299","wAL00300")
PrioriSet <- c("wAL00305","wAL00307","wAL00308","wAL00309","wAL00310","wAL00311")

consist_shotgun.f <- consist_shotgun %>% dplyr::filter(Raw.file %in% ShotgunSet) %>% group_by(Raw.file, SeqCharge) %>% top_n(-1, PEP) %>% slice(1)
consist_priori.f <- consist_priori %>% dplyr::filter(Raw.file %in% PrioriSet) %>% group_by(Raw.file, SeqCharge) %>% top_n(-1, PEP) %>% slice(1)

consist_shotgun_evi.f <- consist_shotgun_evi %>% dplyr::filter(Raw.file %in% ShotgunSet)
consist_priori_evi.f <- consist_priori_evi %>% dplyr::filter(Raw.file %in% PrioriSet)

consist_priori_evi.fFDR <- Empirical_FDR_PEP(consist_priori_evi.f)
consist_priori_evi.fFDR <- consist_priori_evi.fFDR %>% dplyr::filter(qVal <= .01)

consist_shotgun_evi.f.t <- consist_shotgun_evi.f %>% filter(SeqCharge %in% targList_instance_consist$SeqCharge)

#------------------------
# Extracting fragment info and joining with evidence
#------------------------
consist_shotgun.f_frag <- consist_shotgun.f %>% dplyr::select(Raw.file,SeqCharge,Number.of.matches)
consist_shotgun_evi.f.t2 <- consist_shotgun_evi.f.t %>% left_join(consist_shotgun.f_frag, by = c("Raw.file","SeqCharge"))

consist_shotgun_evi.f.t2_sum <- consist_shotgun_evi.f.t2 %>% group_by(SeqCharge) %>% summarize(medianFrag = median(Number.of.matches, na.rm = TRUE),medianPEP = median(PEP,na.rm = TRUE))

interSeq<- intersect(consist_shotgun_evi.f.t2_sum$SeqCharge, consist_priori_evi.fFDR$SeqCharge)

consist_shotgun_evi.f.t2_sum$IDd <- ifelse(consist_shotgun_evi.f.t2_sum$SeqCharge %in% interSeq,"PSM, 1% FDR", "No PSM, 1% FDR")
consist_shotgun_evi.f.t2_sum$Targ <- ifelse(consist_shotgun_evi.f.t2_sum$SeqCharge %in% consist_shotgun_evi.f.t2_sum$SeqCharge,TRUE, FALSE)
consist_shotgun_evi.f.t2_sum <- consist_shotgun_evi.f.t2_sum %>% dplyr::filter(Targ == TRUE) 
consist_shotgun_evi.f.t2_sum$negLogPEP <- -log10(consist_shotgun_evi.f.t2_sum$medianPEP)
consist_shotgun_evi.f.t2_sum$binPEP <- ifelse(consist_shotgun_evi.f.t2_sum$negLogPEP >= 5,5,consist_shotgun_evi.f.t2_sum$negLogPEP)


consist_shotgun_evi.f.t2_sum3 <- consist_shotgun_evi.f.t2_sum %>% left_join(targList_instance_consist, by = "SeqCharge")

consist_shotgun_evi.f.t2_sum3 <- consist_shotgun_evi.f.t2_sum3 %>% left_join(Priority_dict_consist, by = "Priority")

consist_shotgun_evi.f.t2_sum3_RTOnly <- consist_shotgun_evi.f.t2_sum3 %>% filter(Priority == 0)
consist_shotgun_evi.f.t2_sum3_targ <- consist_shotgun_evi.f.t2_sum3 %>% filter(Priority != 0)
consist_shotgun_evi.f.t2_sum3_RTOnly.f <- consist_shotgun_evi.f.t2_sum3_RTOnly %>% filter(!(SeqCharge %in% consist_shotgun_evi.f.t2_sum3_targ$SeqCharge),IDd == "PSM")


consist_shotgun_evi.f.t2_sum3_PEP <- consist_shotgun_evi.f.t2_sum3 %>% group_by(IDd,Priority) %>% summarize(medPEP = median(negLogPEP),medFrag = median(medianFrag))

consist_shotgun_evi.f.t2_sum3 <- consist_shotgun_evi.f.t2_sum3 %>% left_join(consist_shotgun_evi.f.t2_sum3_PEP, by = c("IDd","Priority"))
```


```{r Plots}
consist_shotgun_evi.f.t2_sum3 %>% ggplot(aes(binPEP, fill = IDd)) + geom_histogram(position = position_identity(), alpha = 0.5) + theme_light()   + facet_wrap(~factor(Title, levels = c("High 1","High 2","Medium","Low","RT only")),nrow = 1, scales = "free_x") + geom_vline(aes(xintercept = medPEP, color = IDd)) + labs(x = "-log10(PEP)", y = "Prioritized Precursors", color = "", fill = "") + coord_flip() + theme(axis.text.x = element_text(angle = 45, hjust = 1),
  axis.text = element_text(size = 12),
  axis.title = element_text(size=18),
  legend.text = element_text(size = 14),
  strip.text = element_text(color = "black",size = 16),
  strip.background = element_rect(fill = "white", color = "grey70"),
  legend.position = "bottom",
  plot.title = element_text(size = 18)
) + ggtitle("Shotgun PSM Properties:  PEP")
ggsave(paste0(FigOut,"PSMs1pFDR_PEPHistogram_forPub.pdf"), width = 7, height = 5, units = "in")

consist_shotgun_evi.f.t2_sum3 %>% ggplot(aes(medianFrag, fill = IDd)) + geom_histogram(position = position_identity(), alpha = 0.5) + theme_light()   + facet_wrap(~factor(Title, levels = c("High 1","High 2","Medium","Low","RT only")),nrow = 1, scales = "free_x") + geom_vline(aes(xintercept = medFrag, color = IDd)) + labs(x = "Number of Matching Fragments", y = "Prioritized Precursors", color = "", fill = "") + coord_flip()+ theme(axis.text.x = element_text(angle = 45, hjust = 1),
  axis.text = element_text(size = 12),
  axis.title = element_text(size=18),
  legend.text = element_text(size = 14),
  strip.text = element_text(color = "black",size = 16),
  strip.background = element_rect(fill = "white", color = "grey70"),
  legend.position = "bottom",
  plot.title = element_text(size = 18)
)  + ggtitle("Shotgun PSM Properties: Matching Fragments") 
ggsave(paste0(FigOut,"PSMs1pFDR_fragHistogram_forPub.pdf"), width = 7, height = 5, units = "in")  
  
consist_shotgun_evi.f.t2_sum3 %>% filter(Priority != 0) %>% ggplot(aes(binPEP, color = IDd)) + geom_density(position = position_identity(), alpha = 0.5, lwd = 2) + theme_light()  + labs(x = "-log10(PEP)", y = "Prioritized Precursors, Density", color = "", fill = "") + coord_flip() + theme(axis.text.x = element_text(angle = 45, hjust = 1),
  axis.text = element_text(size = 12),
  axis.title = element_text(size=18),
  legend.text = element_text(size = 14),
  strip.text = element_text(color = "black",size = 16),
  strip.background = element_rect(fill = "white", color = "grey70"),
  legend.position = "bottom",
  plot.title = element_text(size = 18)
) + ggtitle("Shotgun PSM Properties:  PEP")
ggsave(paste0(FigOut,"PSMs1pFDR_PEPDensity_forPub.pdf"), width = 6, height = 5, units = "in")

consist_shotgun_evi.f.t2_sum3 %>% filter(Priority != 0) %>% ggplot(aes(medianFrag, color = IDd)) + geom_density(position = position_identity(), alpha = 0.5, lwd = 2) + theme_light()  + labs(x = "Number of Matching Fragments", y = "Prioritized Precursors, Density", color = "", fill = "") + coord_flip() + theme(axis.text.x = element_text(angle = 45, hjust = 1),
  axis.text = element_text(size = 12),
  axis.title = element_text(size=18),
  legend.text = element_text(size = 14),
  strip.text = element_text(color = "black",size = 16),
  strip.background = element_rect(fill = "white", color = "grey70"),
  legend.position = "bottom",
  plot.title = element_text(size = 18)
) + ggtitle("Shotgun PSM Properties: Matching Fragments")
ggsave(paste0(FigOut,"PSMs1pFDR_FragDensity_forPub.pdf"), width = 6, height = 5, units = "in")

consist_shotgun_evi.f.t2_sum3 %>% ggplot(aes(binPEP, color = factor(IDd, levels = c("PSM, 1% FDR","No PSM, 1% FDR")))) + geom_boxplot(alpha = 0.5) + theme_light()  + labs(x = "-log10(PEP)", y = "", color = "", fill = "") + coord_flip() + theme(axis.text.x = element_blank(),
  axis.title = element_text(size=15),
  legend.text = element_text(size = 12),
  strip.text = element_text(size = 16),
  legend.position = "bottom",
  plot.title = element_text(size = 18)
) + ggtitle("Shotgun PSM Properties:  PEP")
ggsave(paste0(FigOut,"PSMs1pFDR_PEPBoxplot_forPub.pdf"), width = 5, height = 5, units = "in")

consist_shotgun_evi.f.t2_sum3 %>% ggplot(aes(medianFrag, fill = factor(IDd, levels = c("PSM, 1% FDR","No PSM, 1% FDR")))) + geom_boxplot(alpha = 0.5) + theme_light()  + labs(x = "Number of Matching Fragments", y = "", color = "", fill = "") + coord_flip() + theme(axis.text.x = element_blank(),
  axis.title = element_text(size=15),
  legend.text = element_text(size = 12),
  strip.text = element_text(size = 16),
  legend.position = "bottom",
  plot.title = element_text(size = 18)
) + ggtitle("Shotgun PSM Properties: Matching Fragments")
ggsave(paste0(FigOut,"PSMs1pFDR_FragBoxplot_forPub.pdf"), width = 5, height = 5, units = "in")
```


#----------------------------
# BMDM technical motivation figure
#----------------------------


```{r}

BMDM_ShotgunSet <- c("wGH0456","wGH0457","wGH0458","wGH0459","wGH0460","wGH0461","wGH0462","wGH0463","wGH0464","wGH0465","wGH0466","wGH0467","wGH0468","wGH0469","wGH0470","wGH0471","wGH0472","wGH0473","wGH0474","wGH0475")
BMDM_PrioriSet <- c("wGH0500","wGH0501","wGH0502","wGH0503","wGH0504","wGH0505","wGH0506","wGH0507","wGH0508","wGH0509","wGH0510","wGH0511","wGH0512","wGH0513","wGH0514","wGH0515","wGH0516","wGH0517","wGH0518","wGH0519")


#BMDM_CG.s <- read.delim(paste0(BMDM_basePath,"ChanKey_Bio_Shot_20_PIF50.txt"))
#BMDM_CG.p <- read.delim(paste0(BMDM_basePath,"ChanKey_Bio_pSCoPE_20_PIF50.txt"))
BMDM_CG <- read.delim(paste0(DataPath,"BMDM_technical_supplemental/ChanKey_BMDM_20Shotgun20pSCoPE_mrri02_PIF50_DART_FDR1p.txt"))

#BMDM_t0.s <- read.delim(paste0(BMDM_basePath,"t0Bio_BMDM_PIF50_limFASTA_ShotgunOnly.txt"))
#BMDM_t0.p <- read.delim(paste0(BMDM_basePath,"t0Bio_BMDM_PIF50_limFASTA_pSCoPEOnly.txt"))
#BMDM_t0.s <- read.delim(paste0(BMDM_basePath,"t0Bio_Shot_20_PIF50.txt"))
#BMDM_t0.p <- read.delim(paste0(BMDM_basePath,"t0Bio_pSCoPE_20_PIF50.txt"))
BMDM_t0 <- read.delim(paste0(DataPath,"BMDM_technical_supplemental/t0_BMDM_20Shotgun20pSCoPE_mrri02_PIF50_DART_1pFDR.txt")) 

#BMDM_MDp.s <- read.delim(paste0(BMDM_basePath,"Prot_missing_BMDM_PIF50_limFASTA_ShotgunOnly_Bio.txt"))
#BMDM_MDp.p <- read.delim(paste0(BMDM_basePath,"Prot_missing_BMDM_PIF50_limFASTA_pSCoPEOnly_Bio.txt"))
#BMDM_MDp.s <- read.delim(paste0(BMDM_basePath,"Prot_missing_Shot_20_PIF50_Bio.txt"))
#BMDM_MDp.p <- read.delim(paste0(BMDM_basePath,"Prot_missing_pSCoPE_20_PIF50_Bio.txt"))
BMDM_MDp <- read.delim(paste0(DataPath,"BMDM_technical_supplemental/Prot_missing_BMDM_20Shotgun20pSCoPE_mrri02_PIF50_DART_1pFDR.txt"))
  
#BMDM_tList <- read.delim("C:/My_Drive/MS/SCoPE/t-SCoPE/TLists/08_10_2021/CEO_TargList_081021_2.txt")
BMDM_tList <- read.delim(paste0(DataPath,"BMDM_technical_supplemental/CEO_TargList_081021_2.txt"))
tList_priorityDictionary <- BMDM_tList %>% dplyr::filter(TargBool == TRUE) %>% dplyr::select(SeqCharge,Priority)



#BMDM_CG.s$Raw.file <- str_sub(BMDM_CG.s$Raw.file, start = 2)
#BMDM_CG.p$Raw.file <- str_sub(BMDM_CG.p$Raw.file, start = 2)
BMDM_CG$Raw.file <- str_sub(BMDM_CG$Raw.file, start = 2)

#BMDM_CG.s <- BMDM_CG.s %>% filter(Raw.file %in% BMDM_ShotgunSet, celltype != "control", cvm < .4)
#BMDM_CG.p <- BMDM_CG.p %>% filter(Raw.file %in% BMDM_PrioriSet, celltype != "control", cvm < .4)
BMDM_CG.s <- BMDM_CG %>% filter(Raw.file %in% BMDM_ShotgunSet, celltype != "control", cvm < .4)
BMDM_CG.p <- BMDM_CG %>% filter(Raw.file %in% BMDM_PrioriSet, celltype != "control", cvm < .4)

#BMDM_t0.s.m <- melt(BMDM_t0.s)
#BMDM_t0.p.m <- melt(BMDM_t0.p)
BMDM_t0.m <- melt(BMDM_t0)


BMDM_t0.s.m <- BMDM_t0.m %>% dplyr::filter(variable %in% BMDM_CG.s$id)
BMDM_t0.p.m <- BMDM_t0.m %>% dplyr::filter(variable %in% BMDM_CG.p$id)

BMDM_t0.s.m_seqSum <- BMDM_t0.s.m %>% dplyr::group_by(SeqCharge) %>% dplyr::summarize(MD = sum(is.na(value))) %>% filter(MD > 0)
BMDM_t0.p.m_seqSum <- BMDM_t0.p.m %>% dplyr::group_by(SeqCharge) %>% dplyr::summarize(MD = sum(is.na(value))) %>% filter(MD > 0)

intersectBMDM <- intersect(BMDM_t0.s.m_seqSum$SeqCharge,BMDM_t0.p.m_seqSum$SeqCharge) 

BMDM_t0.s.m_inter <- BMDM_t0.s.m %>% filter(SeqCharge %in% intersectBMDM) 
BMDM_t0.p.m_inter <- BMDM_t0.p.m %>% filter(SeqCharge %in% intersectBMDM)

BMDM_t0.s.m_inter2 <- BMDM_t0.s.m_inter %>% left_join(tList_priorityDictionary, by = "SeqCharge")
BMDM_t0.p.m_inter2 <- BMDM_t0.p.m_inter %>% left_join(tList_priorityDictionary, by = "SeqCharge")

BMDM_t0.s.m_inter2_groupCount <- BMDM_t0.s.m_inter2 %>% group_by(SeqCharge,Priority) %>% slice(1)
BMDM_t0.p.m_inter2_groupCount <- BMDM_t0.p.m_inter2 %>% group_by(SeqCharge,Priority) %>% slice(1)

BMDM_t0.s.m_inter2_groupCount_sum <- BMDM_t0.s.m_inter2_groupCount %>% dplyr::group_by(Priority) %>% dplyr::summarize(num = n()) %>% mutate(Method = "Shotgun")
BMDM_t0.p.m_inter2_groupCount_sum <- BMDM_t0.p.m_inter2_groupCount %>% dplyr::group_by(Priority) %>% dplyr::summarize(num = n()) %>% mutate(Method = "Prioritization") 

PriorityCounts <- rbind(BMDM_t0.s.m_inter2_groupCount_sum,BMDM_t0.p.m_inter2_groupCount_sum)

BMDM_t0.s.m_inter_sum <- BMDM_t0.s.m_inter2 %>% dplyr::group_by(variable,Priority) %>% dplyr::summarize(fracMD = sum(is.na(value))/n()) %>% mutate(Method = "Shotgun")
BMDM_t0.p.m_inter_sum <- BMDM_t0.p.m_inter2 %>% dplyr::group_by(variable, Priority) %>% dplyr::summarize(fracMD = sum(is.na(value))/n()) %>% mutate(Method = "Prioritization")

BMDM_inter_bind <- rbind(BMDM_t0.s.m_inter_sum,BMDM_t0.p.m_inter_sum)
#BMDM_inter_bind2 <- BMDM_inter_bind %>% left_join(BMDM_tList, by = "SeqCharge")

BMDM_inter_bind <- BMDM_inter_bind %>% left_join(PriorityCounts, by = c("Method","Priority"))

precursor_MD <- BMDM_inter_bind %>% ggplot(aes(fracMD*100, fill = Method)) + geom_boxplot() + coord_flip(xlim = c(0,100)) + labs(x = "% Missing Data") + theme_box() + theme(legend.position = "bottom") + ggtitle("Precursor-Level Missing Data","by priority tier") + facet_wrap(~factor(Priority, levels = c("3","2","1",NA)), nrow = 1) + geom_text(aes(x = 15 , y = 0 , label = paste(num, "precursors")))

#--------
# peptide-level
#--------

# How to handle when peptides could be on multiple tiers due to charge states

BMDM_t0.s.m_pep <- BMDM_t0.s.m
BMDM_t0.p.m_pep <- BMDM_t0.p.m

BMDM_t0.s.m_pep$Peptide <- str_sub(BMDM_t0.s.m_pep$SeqCharge, end = -2)
BMDM_t0.p.m_pep$Peptide <- str_sub(BMDM_t0.p.m_pep$SeqCharge, end = -2)

BMDM_t0.s.m_pepSum <- BMDM_t0.s.m_pep %>% dplyr::group_by(Peptide, variable) %>% dplyr::summarize(Val = sum(value, na.rm = TRUE))
BMDM_t0.p.m_pepSum <- BMDM_t0.p.m_pep %>% dplyr::group_by(Peptide, variable) %>% dplyr::summarize(Val = sum(value, na.rm = TRUE))

BMDM_t0.s.m_pepSum$Val2 <- ifelse(BMDM_t0.s.m_pepSum$Val == 0, NA,BMDM_t0.s.m_pepSum$Val)
BMDM_t0.p.m_pepSum$Val2 <- ifelse(BMDM_t0.p.m_pepSum$Val == 0, NA,BMDM_t0.p.m_pepSum$Val)

interPepBMDM <- intersect(BMDM_t0.s.m_pepSum$Peptide,BMDM_t0.p.m_pepSum$Peptide) 

BMDM_t0.s.m_pepSum_inter <- BMDM_t0.s.m_pepSum %>% filter(Peptide %in% interPepBMDM) 
BMDM_t0.p.m_pepSum_inter <- BMDM_t0.p.m_pepSum %>% filter(Peptide %in% interPepBMDM)

tList_priorityDictionary$Peptide <- str_sub(tList_priorityDictionary$SeqCharge, end = -2) 
tList_priorityDictionary_lim <- tList_priorityDictionary %>% group_by(Peptide) %>% top_n(1, Priority) %>% slice(1)

BMDM_t0.s.m_pepSum_inter2 <- BMDM_t0.s.m_pepSum_inter %>% left_join(tList_priorityDictionary_lim, by = "Peptide")
BMDM_t0.p.m_pepSum_inter2 <- BMDM_t0.p.m_pepSum_inter %>% left_join(tList_priorityDictionary_lim, by = "Peptide")

BMDM_t0.s.m_pepSum_inter2_groupCount <- BMDM_t0.s.m_pepSum_inter2 %>% group_by(Peptide,Priority) %>% slice(1)
BMDM_t0.p.m_pepSum_inter2_groupCount <- BMDM_t0.p.m_pepSum_inter2 %>% group_by(Peptide,Priority) %>% slice(1)

BMDM_t0.s.m_pepSum_inter2_groupCount_sum <- BMDM_t0.s.m_pepSum_inter2_groupCount %>% dplyr::group_by(Priority) %>% dplyr::summarize(num = n()) %>% mutate(Method = "Shotgun")
BMDM_t0.p.m_pepSum_inter2_groupCount_sum <- BMDM_t0.p.m_pepSum_inter2_groupCount %>% dplyr::group_by(Priority) %>% dplyr::summarize(num = n()) %>% mutate(Method = "Prioritization") 

PriorityCountsPep <- rbind(BMDM_t0.s.m_pepSum_inter2_groupCount_sum,BMDM_t0.p.m_pepSum_inter2_groupCount_sum)

BMDM_t0.s.m_pepSum_inter_sum <- BMDM_t0.s.m_pepSum_inter2 %>% dplyr::group_by(variable,Priority) %>% dplyr::summarize(fracMD = sum(is.na(Val2))/n()) %>% mutate(Method = "Shotgun")
BMDM_t0.p.m_pepSum_inter_sum <- BMDM_t0.p.m_pepSum_inter2 %>% dplyr::group_by(variable, Priority) %>% dplyr::summarize(fracMD = sum(is.na(Val2))/n()) %>% mutate(Method = "Prioritization")

BMDM_interPep_bind <- rbind(BMDM_t0.s.m_pepSum_inter_sum,BMDM_t0.p.m_pepSum_inter_sum)
#BMDM_inter_bind2 <- BMDM_inter_bind %>% left_join(BMDM_tList, by = "SeqCharge")

BMDM_interPep_bind <- BMDM_interPep_bind %>% left_join(PriorityCountsPep, by = c("Method","Priority"))
BMDM_interPep_bind$DataCompleteness <- 1 - BMDM_interPep_bind$fracMD

peptide_MD <- BMDM_interPep_bind %>% ggplot(aes(fracMD*100, fill = Method)) + geom_boxplot() + coord_flip(xlim = c(0,100)) + labs(x = "% Missing Data") + theme_box() + theme(legend.position = "bottom") + ggtitle("Peptide-Level Missing Data","by priority tier") + facet_wrap(~factor(Priority, levels = c("3","2","1",NA)), nrow = 1) + geom_text(aes(x = 15 , y = 0 , label = paste(num, "peptides")))

#precursor_MD 
#peptide_MD

#------
# Protein level
#------

#BMDM_ev.s <- read.delim(paste0(BMDM_basePath,"evidence_filtBio_BMDM_PIF50_limFASTA_ShotgunOnly.txt"))
#BMDM_ev.p <- read.delim(paste0(BMDM_basePath,"evidence_filtBio_BMDM_PIF50_limFASTA_pSCoPEOnly.txt"))

#BMDM_ev.s <- read.delim(paste0(BMDM_basePath,"evidence_filtBio_Shot_20_PIF50.txt"))
#BMDM_ev.p <- read.delim(paste0(BMDM_basePath,"evidence_filtBio_pSCoPE_20_PIF50.txt"))
BMDM_ev <- read.delim(paste0(DataPath,"BMDM_technical_supplemental/ev_20Shotgun20pSCoPE_02182022_mrri02_PIF50_DART_1pFDR.txt"))

#BMDM_ev <- BMDM_ev %>% select(-Acetyl..Protein.N.term.)
##BMDM_ev <- rbind(BMDM_ev.s,BMDM_ev.p)
BMDM_ev$SeqCharge <- BMDM_ev$modseq
BMDM_ev_lim <- BMDM_ev %>% dplyr::select(SeqCharge, Leading.razor.protein) %>% group_by(SeqCharge) %>% slice(1)

# Perhaps better is to go from Sequence to Protein

BMDM_t0.s.m_4Prot <- BMDM_t0.s.m
BMDM_t0.p.m_4Prot <- BMDM_t0.p.m

BMDM_t0.s.m_4Prot <- BMDM_t0.s.m_4Prot %>% left_join(BMDM_ev_lim, by = "SeqCharge") 
BMDM_t0.p.m_4Prot <- BMDM_t0.p.m_4Prot %>% left_join(BMDM_ev_lim, by = "SeqCharge")

BMDM_t0.s.m_4Protsum <- BMDM_t0.s.m_4Prot %>% dplyr::group_by(variable, Leading.razor.protein) %>% dplyr::summarize(ProtVal = sum(value, na.rm = TRUE))
BMDM_t0.p.m_4Protsum <- BMDM_t0.p.m_4Prot %>% dplyr::group_by(variable, Leading.razor.protein) %>% dplyr::summarize(ProtVal = sum(value, na.rm = TRUE))

prot_inter <- intersect(BMDM_t0.s.m_4Protsum$Leading.razor.protein,BMDM_t0.p.m_4Protsum$Leading.razor.protein)

BMDM_t0.s.m_4Protsum_inter <- BMDM_t0.s.m_4Protsum %>% dplyr::filter(Leading.razor.protein %in% prot_inter) %>% dplyr::mutate(Method = "Shotgun")
BMDM_t0.p.m_4Protsum_inter <- BMDM_t0.p.m_4Protsum %>% dplyr::filter(Leading.razor.protein %in% prot_inter) %>% dplyr::mutate(Method = "Prioritization")

BMDM_prot <- rbind(BMDM_t0.s.m_4Protsum_inter,BMDM_t0.p.m_4Protsum_inter)

tList_priorityDictionary_Protlim <- BMDM_tList %>% dplyr::filter(TargBool == TRUE) %>% dplyr::select(Leading.razor.protein,Priority) %>% group_by(Leading.razor.protein) %>% top_n(1, Priority) %>% slice(1)

BMDM_prot <- BMDM_prot %>% left_join(tList_priorityDictionary_Protlim, by = "Leading.razor.protein")

BMDM_prot$Val2 <- ifelse(BMDM_prot$ProtVal == 0, NA, BMDM_prot$ProtVal)

BMDM_prot_sum <- BMDM_prot %>% dplyr::group_by(variable, Priority, Method) %>% dplyr::summarize(fracMD = sum(is.na(Val2))/n())

BMDM_prot_sumPriority <- BMDM_prot %>% dplyr::group_by(Priority, Leading.razor.protein) %>% slice(1)
BMDM_prot_sumPriority2 <- BMDM_prot_sumPriority %>% dplyr::ungroup() %>% dplyr::group_by(Priority) %>% dplyr::summarize(numProts = n())


BMDM_prot_sum <- BMDM_prot_sum %>% dplyr::left_join(BMDM_prot_sumPriority2, by = "Priority")
BMDM_prot_sum$FacetLab <- ifelse(BMDM_prot_sum$Priority == 3, "High",ifelse(BMDM_prot_sum$Priority == 2, "Medium",ifelse(BMDM_prot_sum$Priority == 1, "Low","NA")))

BMDM_prot_sum$DataCompleteness <- 1 - BMDM_prot_sum$fracMD

BMDM_interPep_bind$FacetLab <- ifelse(BMDM_interPep_bind$Priority == 3, "High",ifelse(BMDM_interPep_bind$Priority == 2, "Medium",ifelse(BMDM_interPep_bind$Priority == 1, "Low","NA")))

protMD <- BMDM_prot_sum %>% filter(FacetLab != "NA") %>% ggplot(aes(DataCompleteness*100, color = factor(Method, levels = c("Shotgun","Prioritization")))) + geom_boxplot() + coord_flip(xlim = c(0,100)) + labs(x = "", color = "", y = "") + theme_box() + theme(legend.position = "null", axis.text.y = element_blank(), axis.title = element_text(size = 26), strip.background = element_rect(fill = "grey95", color = "grey70"), strip.text = element_text(color = "black",size = 20), plot.title = element_text(size = 28)) + ggtitle("Proteins") + facet_wrap(~factor(FacetLab, levels = c("High","Medium","Low","NA")), nrow = 1) + geom_text(aes(x = 2 , y = 0 , label = paste(numProts, "\nproteins")), color = "black",size = 6)  + scale_color_manual(values = c("blue","red"))

peptide_MD <- BMDM_interPep_bind %>% filter(FacetLab != "NA") %>% ggplot(aes(DataCompleteness*100, color = factor(Method, levels = c("Shotgun","Prioritization")))) + geom_boxplot() + coord_flip(xlim = c(0,100)) + labs(x = "% Data Completeness", color = "", y = "") + theme_box() + theme(legend.position = "null", axis.title.y = element_text(size = 30), strip.background = element_rect(fill = "grey95", color = "grey70"), strip.text = element_text(color = "black", size = 20), axis.text.y = element_text(size = 28), plot.title = element_text(size = 28)) + ggtitle("Peptides") + facet_wrap(~factor(FacetLab, levels = c("High","Medium","Low","NA")), nrow = 1) + geom_text(aes(x = 2 , y = 0 , label = paste(num, "\npeptides")), color = "black",size = 6) + scale_color_manual(values = c("blue","red"))

#precursor_MD 
#peptide_MD
#protMD


fig1Supp_pepMD <- BMDM_interPep_bind %>% group_by(Method,FacetLab) %>% summarize(medVal = median(fracMD, na.rm = TRUE))
fig1Supp_protMD <- BMDM_prot_sum %>% group_by(Method,FacetLab) %>% summarize(medVal = median(fracMD, na.rm = TRUE))

peptide_MD + protMD + plot_annotation(title = 'BMDMs: Missing Data per Cell',theme = theme(plot.title = element_text(size = 28))) + plot_layout(guides = "collect",widths = c(1,1)) & theme(legend.position = 'null')
ggsave(paste0(FigOut,"BMDM_missingData_v2_mrri02.pdf"), width = 9 , height = 7, units = "in")
```

```{r}
BMDM_MDp.s <- BMDM_MDp %>% dplyr::select(any_of(BMDM_CG.s$id))
BMDM_MDp.p <- BMDM_MDp %>% dplyr::select(any_of(BMDM_CG.p$id))

BMDM_MDp.s.m <- melt(BMDM_MDp.s)
BMDM_MDp.p.m <- melt(BMDM_MDp.p)

BMDM_MDp.s.m$Method <- "Shotgun"
BMDM_MDp.p.m$Method <- "Prioritization"

BMDM_MDp.s.m <- BMDM_MDp.s.m %>% dplyr::filter(variable %in% BMDM_CG.s$id)
BMDM_MDp.p.m <- BMDM_MDp.p.m %>% dplyr::filter(variable %in% BMDM_CG.p$id)

BMDM_MDp.join <- rbind(BMDM_MDp.s.m,BMDM_MDp.p.m)
BMDM_MDp.join_sum <- BMDM_MDp.join %>% dplyr::group_by(variable, Method) %>% dplyr::summarize("Proteins/cell"=sum(!is.na(value))) 

protsPerCell <- BMDM_MDp.join_sum %>% ggplot(aes(x = `Proteins/cell`, color = factor(Method, levels = c("Shotgun","Prioritization")))) + geom_boxplot() + coord_flip(xlim = c(0,800)) + theme_box() + theme(legend.position = "bottom", axis.title.y = element_text(size = 30), axis.text.y = element_text(size = 28)) + labs(color = "") + scale_color_manual(values = c("blue","red"))

BMDM_t0.s.m_pepSum$Method <- "Shotgun"
BMDM_t0.p.m_pepSum$Method <- "Prioritization"

BMDM_pep.join <- rbind(BMDM_t0.s.m_pepSum,BMDM_t0.p.m_pepSum)
BMDM_pep.join_sum <- BMDM_pep.join %>% dplyr::group_by(variable, Method) %>% dplyr::summarize("Peptides/cell"=sum(!is.na(Val2))) 

pepsPerCell <- BMDM_pep.join_sum %>% ggplot(aes(x = `Peptides/cell`, color = factor(Method, levels = c("Shotgun","Prioritization")))) + geom_boxplot() + coord_flip(xlim = c(0,2200)) + theme_box() + theme(legend.position = "bottom",axis.title.y = element_text(size = 30), axis.text.y = element_text(size = 28)) + labs(color = "") + scale_color_manual(values = c("blue","red"))

fig1Supp2_protCov <- BMDM_MDp.join_sum %>% group_by(Method) %>% summarize(medVal = median(`Proteins/cell`, na.rm = TRUE))
fig1Supp2_pepCov <- BMDM_pep.join_sum %>% group_by(Method) %>% summarize(medVal = median(`Peptides/cell`, na.rm = TRUE))

pepsPerCell + protsPerCell + plot_annotation(title = 'BMDMs: Data Points per Cell',theme = theme(plot.title = element_text(size = 28))) + plot_layout(guides = "collect",widths = c(1,1)) & theme(legend.position = 'null')  
ggsave(paste0(FigOut,"BMDM_DataPointsPerCell_mrri02.pdf"), width = 7, height = 7, units = "in")

```

```{r}
#For precursors with higher fill times, were these identified in more runs?
BMDM_tList_4FT_1 <- read.delim(paste0(DataPath,"BMDM_technical_supplemental/CEO_TargList_081021_2.txt"))
BMDM_tList_4FT_1 <- BMDM_tList_4FT_1 %>% dplyr::filter(TargBool == TRUE)
BMDM_tList_4FT <- BMDM_tList_4FT_1 %>% dplyr::filter(FT_priority > 500)

#BMDM_ev.s <- read.delim(paste0(BMDM_basePath,"evidence_filtBio_BMDM_PIF50_limFASTA_ShotgunOnly.txt"))
#BMDM_ev.p <- read.delim(paste0(BMDM_basePath,"evidence_filtBio_BMDM_PIF50_limFASTA_pSCoPEOnly.txt"))
BMDM_ev <- read.delim(paste0(DataPath,"BMDM_technical_supplemental/ev_20Shotgun20pSCoPE_02182022_mrri02_PIF50_DART_1pFDR.txt"))

#BMDM_ev.s <- read.delim(paste0(BMDM_basePath,"evidence_filtBio_Shot_20_PIF50.txt"))
#BMDM_ev.p <- read.delim(paste0(BMDM_basePath,"evidence_filtBio_pSCoPE_20_PIF50.txt"))

BMDM_t0.s.m_FT <- BMDM_t0.s.m %>% dplyr::filter(SeqCharge %in% BMDM_tList_4FT$SeqCharge)
BMDM_t0.p.m_FT <- BMDM_t0.p.m %>% dplyr::filter(SeqCharge %in% BMDM_tList_4FT$SeqCharge)

interFT <- intersect(BMDM_t0.s.m_FT$SeqCharge,BMDM_t0.p.m_FT$SeqCharge)
interFTdf <- as.data.frame(interFT)
interFTdf$peptides <- str_sub(interFTdf$interFT, end = -2)

BMDM_t0.s.m_FT_inter <- BMDM_t0.s.m_FT %>% dplyr::filter(SeqCharge %in% interFT)
BMDM_t0.p.m_FT_inter <- BMDM_t0.p.m_FT %>% dplyr::filter(SeqCharge %in% interFT)

BMDM_t0.s.m_FT_inter_sum <- BMDM_t0.s.m_FT_inter %>% dplyr::group_by(variable) %>% dplyr::summarize(fracMissing = sum(is.na(value))/n()) %>% dplyr::mutate(Method = "Shotgun")
BMDM_t0.p.m_FT_inter_sum <- BMDM_t0.p.m_FT_inter %>% dplyr::group_by(variable) %>% dplyr::summarize(fracMissing = sum(is.na(value))/n()) %>% dplyr::mutate(Method = "Prioritization")

FT_bind <- rbind(BMDM_t0.s.m_FT_inter_sum,BMDM_t0.p.m_FT_inter_sum)
FT_bind$DataCompleteness <- 1 - FT_bind$fracMissing
MDhist <- FT_bind %>% ggplot(aes(DataCompleteness*100, color = factor(Method, levels = c("Shotgun","Prioritization")))) + geom_boxplot() + coord_flip(xlim = c(0, 100)) + theme_box() + labs(x = "% Data Completeness, Peptides", y = "") + geom_text(aes(y = 0, x = 1, label = paste0(length(unique(interFTdf$peptides)), " peptides")), color = "black",size = 8)+ theme(legend.position ="null",axis.title.y = element_text(size = 30), axis.text.y = element_text(size = 28)) + scale_color_manual(values = c("blue","red"))

# Showing abundance
pSCoPE_evi_4FT <- read.delim(paste0(DataPath,"BMDM_technical_supplemental/ev_20Shotgun20pSCoPE_02182022_mrri02_PIF50_DART_1pFDR.txt"))
pSCoPE_evi_4FT$SeqCharge <- paste0(pSCoPE_evi_4FT$Modified.sequence,pSCoPE_evi_4FT$Charge)
pSCoPE_evi_4FT_join <- pSCoPE_evi_4FT %>% left_join(BMDM_tList_4FT_1, by = "SeqCharge")
#MDhist <- pSCoPE_evi_4FT_join %>% dplyr::filter(Priority == 3) %>% ggplot(aes(log10(Intensity.x), fill = factor(FT_priority))) + geom_boxplot(position = position_dodge()) + coord_flip()+ theme(legend.position ="null")

BMDM_tList_4FT_quartiles <- quantile(BMDM_tList_4FT_1$Intensity, c(.33,.66))

BMDM_tList_4FT_1_2 <- BMDM_tList_4FT_1
BMDM_tList_4FT_1_2$TertileColor <- ifelse(BMDM_tList_4FT_1_2$Intensity <= BMDM_tList_4FT_quartiles[1],"1000ms",ifelse((BMDM_tList_4FT_1_2$Intensity > BMDM_tList_4FT_quartiles[1])&(BMDM_tList_4FT_1_2$Intensity <= BMDM_tList_4FT_quartiles[2]),"750ms","500ms"))

intHist <- BMDM_tList_4FT_1_2 %>% dplyr::filter(Priority %in% c(3,2,1)) %>% ggplot(aes(log10(Intensity), fill = factor(TertileColor, levels = c("500ms","750ms","1000ms")))) + geom_histogram(alpha = 0.9, bins = 100) + coord_flip() + theme_light() + geom_vline(aes(xintercept = log10(BMDM_tList_4FT_quartiles[1]))) + geom_vline(aes(xintercept = log10(BMDM_tList_4FT_quartiles[2]))) + scale_fill_manual(values = c("paleturquoise1","paleturquoise3","paleturquoise4")) + theme_box() + 
  theme(legend.position=c(.65,.20),axis.title.y = element_text(size = 30), axis.text.y = element_text(size = 28)) + labs(y = "", x = bquote("Intensity, Log"[10]), fill = "Fill time")

fig1Supp2_FillTimeSet<- FT_bind %>% group_by(Method) %>% summarize(medVal = median(fracMissing, na.rm = TRUE))

library(patchwork)
intHist + MDhist + plot_annotation(title = 'Intensity-dependent MS2 Fill Times',theme = theme(plot.title = element_text(size = 28)))
ggsave(paste0(FigOut,"BMDM_IncreasedFillTimes_mrri02.pdf"), width = 7, height = 7, units = "in")
```

# Pub Stats
```{r}
#fig1Supp_pepMD
#fig1Supp_protMD
#fig1Supp2_protCov
#fig1Supp2_pepCov
#fig1Supp2_FillTimeSet


# Missing data stats
fig1Supp_pepMD_Shotgun <- fig1Supp_pepMD %>% filter(Method == "Shotgun") %>% ungroup() %>% mutate(Shotgun = medVal) %>% dplyr::select(FacetLab, Shotgun)
fig1Supp_pepMD_Prioritized <- fig1Supp_pepMD %>% filter(Method == "Prioritization") %>% ungroup() %>% mutate(Prioritization = medVal) %>% dplyr::select(FacetLab, Prioritization)
fig1Supp_pepMD.join <- fig1Supp_pepMD_Shotgun %>% left_join(fig1Supp_pepMD_Prioritized, by = "FacetLab") %>% mutate(Fig = "fig1Supp_PeptideMD")

fig1Supp_protMD_Shotgun <- fig1Supp_protMD %>% filter(Method == "Shotgun") %>% ungroup() %>% mutate(Shotgun = medVal) %>% dplyr::select(FacetLab, Shotgun)
fig1Supp_protMD_Prioritized <- fig1Supp_protMD %>% filter(Method == "Prioritization") %>% ungroup() %>% mutate(Prioritization = medVal) %>% dplyr::select(FacetLab, Prioritization)
fig1Supp_protMD.join <- fig1Supp_protMD_Shotgun %>% left_join(fig1Supp_protMD_Prioritized, by = "FacetLab") %>% mutate(Fig = "fig1Supp_ProtMD")

fig1Supp_MDjoin <- rbind(fig1Supp_pepMD.join,fig1Supp_protMD.join)

# Coverage stats
fig1Supp2_pepCov_Shotgun <- fig1Supp2_pepCov %>% filter(Method == "Shotgun") %>% ungroup() %>% mutate(Shotgun = medVal) %>% dplyr::select(Shotgun) %>% mutate(Fig = "fig1Supp_pepCov")
fig1Supp2_pepCov_Prioritized <- fig1Supp2_pepCov %>% filter(Method == "Prioritization") %>% ungroup() %>% mutate(Prioritization = medVal) %>% dplyr::select(Prioritization)  %>% mutate(Fig = "fig1Supp_pepCov")
fig1Supp2_pepCov.join <- fig1Supp2_pepCov_Shotgun %>% left_join(fig1Supp2_pepCov_Prioritized, by = "Fig")

fig1Supp2_protCov_Shotgun <- fig1Supp2_protCov %>% filter(Method == "Shotgun") %>% ungroup() %>% mutate(Shotgun = medVal) %>% dplyr::select(Shotgun) %>% mutate(Fig = "fig1Supp_protCov")
fig1Supp2_protCov_Prioritized <- fig1Supp2_protCov %>% filter(Method == "Prioritization") %>% ungroup() %>% mutate(Prioritization = medVal) %>% dplyr::select(Prioritization)  %>% mutate(Fig = "fig1Supp_protCov")
fig1Supp2_protCov.join <- fig1Supp2_protCov_Shotgun %>% left_join(fig1Supp2_protCov_Prioritized, by = "Fig")

fig1Supp2_Cov.join <- rbind(fig1Supp2_pepCov.join, fig1Supp2_protCov.join)

# Intensity-dependent-fill-time stats
fig1Supp2_FillTimeSet_Shotgun <- fig1Supp2_FillTimeSet %>% filter(Method == "Shotgun") %>% ungroup() %>% mutate(Shotgun = medVal) %>% dplyr::select(Shotgun) %>% mutate(Fig = "fig1Supp_FT")
fig1Supp2_FillTimeSet_Prioritized <- fig1Supp2_FillTimeSet %>% filter(Method == "Prioritization") %>% ungroup() %>% mutate(Prioritization = medVal) %>% dplyr::select(Prioritization) %>% mutate(Fig = "fig1Supp_FT")
fig1Supp2_FillTimeSet.join <- fig1Supp2_FillTimeSet_Shotgun %>% left_join(fig1Supp2_FillTimeSet_Prioritized, by = "Fig")

fig1Supp_MDjoin$PercentageChange <- ((fig1Supp_MDjoin$Prioritization - fig1Supp_MDjoin$Shotgun)/fig1Supp_MDjoin$Shotgun)*100
fig1Supp2_Cov.join$PercentageChange <- ((fig1Supp2_Cov.join$Prioritization - fig1Supp2_Cov.join$Shotgun)/fig1Supp2_Cov.join$Shotgun)*100 
fig1Supp2_FillTimeSet.join$PercentageChange <- ((fig1Supp2_FillTimeSet.join$Prioritization - fig1Supp2_FillTimeSet.join$Shotgun)/fig1Supp2_FillTimeSet.join$Shotgun)*100 
```


# Fresh coding area for PDAC bio figure

```{r}
#singleCellDir <- "C:/My_Drive/MS/SCoPE/t-SCoPE/Pub/codeBlocks/PDAC_coverage/Coverage_withAnno/"
CG <- read.delim(paste0(DataPath,"PDAC_technical_Fig_1bcd/Fig1_b_Coverage/","ChanKey_PDAC_coverage_PIF50_mrri10.txt"))
BC_data <- read.csv(paste0(DataPath,"PDAC_technical_Fig_1bcd/Fig1_b_Coverage/","limmaCorrected_normed_prePCA_PDAC_cov_PIF50_mrri10.csv"))

CG$cellLab <- ifelse(CG$celltype == "Bx","BxPC-3",ifelse(CG$celltype == "HPAF","HPAF-II",ifelse(CG$celltype == "CPAC","CFPAC-I","control")))
pSCoPE_cov <- c("XwAL00120","XwAL00121","XwAL00122","XwAL00123","XwAL00124","XwAL00125","XwAL00126","XwAL00127","XwAL00129","XwAL00130")
CG_prioritized <- CG %>% filter(Raw.file %in% pSCoPE_cov)


row.names(BC_data) <- BC_data$X
BC_data_preMat <- BC_data %>% select(-X)
BC_data_preMat <- BC_data %>% select(any_of(CG_prioritized$id))

BC_data.mat <- as.matrix(BC_data_preMat)
BC_data.mat <- cr_norm_log(BC_data.mat)
PDAC_PC_out <- prcomp(t(BC_data.mat), scale = TRUE, center = TRUE)
PDAC_PC_coord <- as.data.frame(PDAC_PC_out$x[,1:6])
PDAC_PC_weights <- as.data.frame(PDAC_PC_out$rotation[,1:5])
PDAC_PC_weights$Proteins <- row.names(PDAC_PC_weights)

PCAsummary<- summary(PDAC_PC_out)
PC1_fracVar <- PCAsummary$importance[2,1]
PC2_fracVar <- PCAsummary$importance[2,2]
PDAC_PC_coord$id <- row.names(PDAC_PC_coord)
PDAC_PC_coord.anno <- PDAC_PC_coord %>% left_join(CG, by = "id")
Okabe_Ito <- c("#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7", "#000000")

cellType_PDAC_PCA <- PDAC_PC_coord.anno %>% ggplot(aes(x = PC1, y = PC2, color = cellLab)) + geom_point(shape =16,size = 5, alpha = 0.50) + theme_light() +labs(color = "", x = paste0("PC1 (",round(PC1_fracVar,2)*100,"%)"), y = paste0("PC2 (",round(PC2_fracVar,2)*100,"%)")) + scale_color_manual(values = c(Okabe_Ito[1],Okabe_Ito[2],Okabe_Ito[3])) + 
  geom_text(aes(x = 0, y = 12),size = 8,color = Okabe_Ito[3],label = "HPAF-II") + 
  geom_text(aes(x = 18, y = 2),size = 8,color = Okabe_Ito[2],label = "CFPAC-I") + 
  geom_text(aes(x = -14, y = 2),size = 8,color = Okabe_Ito[1],label = "BxPC-3") + theme(legend.position = "null",axis.title = element_text(size = 24), axis.text = element_text(size = 20)) 

```

```{r}
library(dplyr)
library(ggplot2)
library(stringr)

GOPath <- paste0(DataPath,"/BMDM_figs_2_3_4/")
GO_DB_Path <- paste0(GOPath,"GOA_db.txt")
GO_db <- read.delim(GO_DB_Path, sep = "", header = TRUE)
GO_db.2 <- GO_db %>% filter(Letter != "C")

ProtMissing <- read.delim(paste0(DataPath,"PDAC_technical_Fig_1bcd/Fig1_b_Coverage/Prot_missing_PDAC_cov_PIF50_mrri10.txt"))
BC_dat <- read.csv(paste0(DataPath,"PDAC_technical_Fig_1bcd/Fig1_b_Coverage/limmaCorrected_normed_prePCA_PDAC_cov_PIF50_mrri10.csv"))

row.names(BC_data) <- BC_data$X
BC_data_preMat <- BC_data %>% select(-X)
BC_data_preMat <- BC_data %>% select(any_of(CG_prioritized$id))

BC_data.mat <- as.matrix(BC_data_preMat)
BC_data.mat <- cr_norm_log(BC_data.mat)
BC_data.df <- as.data.frame(BC_data.mat)
BC_data.df$X <- row.names(BC_data.df)
ProtMissing.m <- melt(ProtMissing)
BC_dat.m <- melt(BC_data.df, by = "X")
colnames(BC_dat.m) <- c("Proteins","variable","value")
BC_dat.m <- BC_dat.m %>% left_join(ProtMissing.m, by = c("Proteins", "variable"))
BC_dat.m$newVar <- ifelse(is.na(BC_dat.m$value.y),NA,BC_dat.m$value.x)
BC_dat.m2 <- BC_dat.m %>% select(Proteins,variable,newVar)

PSEA_Terms_to_keep <- c("interferon-gamma-mediated signaling pathway","protein N-linked glycosylation via asparagine","de novo' posttranslational protein folding","protein polyubiquitination","regulation of cellular amino acid metabolic process","DNA damage response, signal transduction by p53 class mediator resulting in cell cycle arrest","cellular nitrogen compound metabolic process","positive regulation of ubiquitin-protein ligase activity involved in mitotic cell cycle","regulation of ubiquitin-protein ligase activity involved in mitotic cell cycle","glycolysis","negative regulation of ubiquitin-protein ligase activity involved in mitotic cell cycle","nucleosome assembly","cellular component movement","double-strand break repair","SRP-dependent cotranslational protein targeting to membrane","chromatin organization","negative regulation of catalytic activity","actin filament binding","regulation of cell shape","respiratory electron transport chain")

for (i in 1:length(PSEA_Terms_to_keep)){
  GO_db_lim <- GO_db %>% filter(GO_term_name %in% PSEA_Terms_to_keep[i])
  BC_dat.m2_lim <- BC_dat.m2 %>% filter(Proteins %in% GO_db_lim$Uniprot)
  BC_dat.m2_lim_sum <- BC_dat.m2_lim %>% dplyr::group_by(variable) %>% dplyr::summarize(medAbund = median(newVar, na.rm = TRUE))
  BC_dat.m2_lim_sum$medAbund <- (BC_dat.m2_lim_sum$medAbund-mean(BC_dat.m2_lim_sum$medAbund, na.rm=TRUE))/sd(BC_dat.m2_lim_sum$medAbund, na.rm=TRUE)
  BC_dat.m2_lim_sum$medAbund <- ifelse(BC_dat.m2_lim_sum$medAbund >= 2,2,ifelse(BC_dat.m2_lim_sum$medAbund <= -2,-2,BC_dat.m2_lim_sum$medAbund))
  colnames(BC_dat.m2_lim_sum) <- c("id",as.character(PSEA_Terms_to_keep[[i]]))
  if(i == 1){
    GSEA_PC_out <- BC_dat.m2_lim_sum
  }
  if(i > 1){
    GSEA_PC_out <- GSEA_PC_out %>% left_join(BC_dat.m2_lim_sum, by = "id")
  }
}

PDAC_PC_coord2.pSCoPE <- PDAC_PC_coord.anno %>% left_join(GSEA_PC_out, by = "id")

IFNg_PDAC <- PDAC_PC_coord2.pSCoPE %>% ggplot(aes(x = PC1,y = PC2, color = `interferon-gamma-mediated signaling pathway`)) + geom_point(shape =16,size = 5, alpha = 0.75) + theme_light() +labs(color = "") + scale_color_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0) + labs(color = "") + theme(axis.title = element_blank(),axis.text = element_blank(), legend.position = "null") + ggtitle("IFNg signaling")


Glycolysis_PDAC <- PDAC_PC_coord2.pSCoPE %>% ggplot(aes(x = PC1,y = PC2, color = `glycolysis`)) + geom_point(shape =16,size = 5, alpha = 0.75) + theme_light() +labs(color = "") + scale_color_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0) + labs(color = "") + theme(axis.title = element_blank(),axis.text = element_blank(), legend.position = "null") + ggtitle("Glycolysis")


deNovo_PDAC <- PDAC_PC_coord2.pSCoPE %>% ggplot(aes(x = PC1,y = PC2, color = `de novo' posttranslational protein folding`)) + geom_point(shape =16,size = 5, alpha = 0.75) + theme_light() +labs(color = "") + scale_color_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0) + labs(color = "") + theme(axis.title = element_blank(),axis.text = element_blank(), legend.position = "null") + ggtitle("De novo post-translational protein folding")

cellType_PDAC_PCA +  IFNg_PDAC + Glycolysis_PDAC + deNovo_PDAC + plot_layout(nrow = 1)
ggsave(paste0(FigOut,"PDAC_PCA_composite_v2.pdf"), width = 16, height = 3, units = "in") 
```

```{r}
library(tidyr)

#ev.in <- read.delim(paste0(DataPath,"PDAC_technical_Fig_1bcd/Fig1_b_Coverage/ev_PDAC_cov_filtered_PIF50_mrri10_1pFDR.txt"))
#ev.in.lim <- ev.in %>% select(Leading.razor.protein,Gene.names)
#ev.in.lim2 <- ev.in.lim 
#ev.in.lim2 <- ev.in.lim %>% 
#    mutate(Gene.names2 = strsplit(as.character(Gene.names), ";")) %>% 
#    unnest(Gene.names2)
#PDAC_ProtGeneMap <- ev.in.lim2 %>% select(-Gene.names)
#colnames(PDAC_ProtGeneMap) <- c("UNIPROTKB","GENES")
  
#PDAC_PC_weights


#test <- PCA_PSEA_PDAC(PDAC_PC_weights[,c(1,7)],PDAC_ProtGeneMap,GO_db.2)
PDAC_PSEA <- PCA_PSEA_PDAC_prot(PDAC_PC_weights,GO_db.2)
#PDAC_PSEA <- test
PDAC_PSEA.f <- PDAC_PSEA %>% filter(!is.na(pVal),PC <= 4)
PDAC_PSEA.f$qVal <- p.adjust(PDAC_PSEA.f$pVal, method = "fdr")
PDAC_PSEA.f_fdr <- PDAC_PSEA.f %>% filter((qVal <= .05))
PDAC_PSEA.f_fdr$neglogq <- -log10(PDAC_PSEA.f_fdr$qVal)
PDAC_PSEA.f_fdr$neglogq_bin <- ifelse(PDAC_PSEA.f_fdr$neglogq >=3,3,PDAC_PSEA.f_fdr$neglogq)

#Terms_to_keep <- c("protein N-linked glycosylation via asparagine","negative regulation of ubiquitin-protein ligase activity involved in mitotic cell cycle","nucleosome assembly","chromatin organization","glucose metabolic process","adherens junction organization", "transmembrane transporter activity","hydrogen ion transporting ATP synthase activity, rotational mechanism","respiratory electron transport chain","oxidative phosphorylation","protein N-linked glycosylation via asparagine","de novo' posttranslational protein folding","mitochondrial ATP synthesis coupled proton transport")                                   
                       
Terms_to_keep2 <- c("interferon-gamma-mediated signaling pathway","protein N-linked glycosylation via asparagine","de novo' posttranslational protein folding","protein polyubiquitination","regulation of cellular amino acid metabolic process","DNA damage response, signal transduction by p53 class mediator resulting in cell cycle arrest","cellular nitrogen compound metabolic process","positive regulation of ubiquitin-protein ligase activity involved in mitotic cell cycle","regulation of ubiquitin-protein ligase activity involved in mitotic cell cycle","glycolysis","negative regulation of ubiquitin-protein ligase activity involved in mitotic cell cycle","nucleosome assembly","cellular component movement","double-strand break repair","SRP-dependent cotranslational protein targeting to membrane","chromatin organization","negative regulation of catalytic activity","actin filament binding","regulation of cell shape","respiratory electron transport chain")
  

PDAC_PSEA.f_fdr %>% filter(Process %in% Terms_to_keep2) %>% ggplot(aes(x = PC, y = str_wrap(Process, width = 50),fill = round(Association,1), size = neglogq_bin)) + geom_point(shape = 21, stroke = 1) + scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0, breaks = c(-1,0,1)) + theme_light() + labs(fill = "Loading\nz-score",x = "Principal component",y = "", size = bquote("-log"[10]~"(q val)")) + theme(axis.text.y = element_text(size = 16),axis.text.x = element_text(size = 26),axis.title = element_text(size = 30), legend.title = element_text(size = 20), legend.text = element_text(size = 15), legend.position = "bottom", legend.box="vertical", legend.margin=margin()) +
  scale_size_continuous(breaks = seq(2,3,1),range=c(1,15),
                  limits = c(0, 4), labels = c("2",">3")) +  scale_x_discrete(limits = c("1",
     "2","3","4"), expand=c(0.1, 0)) 
ggsave("C:/My_Drive/MS/SCoPE/t-SCoPE/Pub/codeBlocks/PDAC_coverage/Coverage_withAnno/PSEA_PDAC_Coverage_pSCoPE_only.png", width = 10, height = 15, units = "in")

GO_db.2_deNov <- GO_db.2 %>% filter(grepl("de novo' posttranslational protein folding",GO_term_name))
PDAC_PC_weights_deNov <- PDAC_PC_weights %>% filter(Proteins %in% GO_db.2_deNov$Uniprot)

GO_db.2_IFN <- GO_db.2 %>% filter(grepl("interferon-gamma-mediated signaling pathway",GO_term_name))
PDAC_PC_weights_IFN <- PDAC_PC_weights %>% filter(Proteins %in% GO_db.2_IFN$Uniprot)
```

